// @polar-sh/sdk/dist/esm/models/components/authorizeorganization.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const AuthorizeOrganization$inboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    avatar_url: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "avatar_url": "avatarUrl",
    });
});
/** @internal */
export const AuthorizeOrganization$outboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    avatarUrl: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        avatarUrl: "avatar_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var AuthorizeOrganization$;
(function (AuthorizeOrganization$) {
    /** @deprecated use `AuthorizeOrganization$inboundSchema` instead. */
    AuthorizeOrganization$.inboundSchema = AuthorizeOrganization$inboundSchema;
    /** @deprecated use `AuthorizeOrganization$outboundSchema` instead. */
    AuthorizeOrganization$.outboundSchema = AuthorizeOrganization$outboundSchema;
})(AuthorizeOrganization$ || (AuthorizeOrganization$ = {}));
export function authorizeOrganizationToJSON(authorizeOrganization) {
    return JSON.stringify(AuthorizeOrganization$outboundSchema.parse(authorizeOrganization));
}
export function authorizeOrganizationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AuthorizeOrganization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AuthorizeOrganization' from JSON`);
}
