// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyssubscriber.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitLicenseKeysSubscriberProperties$inboundSchema, BenefitLicenseKeysSubscriberProperties$outboundSchema, } from "./benefitlicensekeyssubscriberproperties.js";
import { Organization$inboundSchema, Organization$outboundSchema, } from "./organization.js";
/** @internal */
export const BenefitLicenseKeysSubscriberMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitLicenseKeysSubscriberMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeysSubscriberMetadata$;
(function (BenefitLicenseKeysSubscriberMetadata$) {
    /** @deprecated use `BenefitLicenseKeysSubscriberMetadata$inboundSchema` instead. */
    BenefitLicenseKeysSubscriberMetadata$.inboundSchema = BenefitLicenseKeysSubscriberMetadata$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysSubscriberMetadata$outboundSchema` instead. */
    BenefitLicenseKeysSubscriberMetadata$.outboundSchema = BenefitLicenseKeysSubscriberMetadata$outboundSchema;
})(BenefitLicenseKeysSubscriberMetadata$ || (BenefitLicenseKeysSubscriberMetadata$ = {}));
export function benefitLicenseKeysSubscriberMetadataToJSON(benefitLicenseKeysSubscriberMetadata) {
    return JSON.stringify(BenefitLicenseKeysSubscriberMetadata$outboundSchema.parse(benefitLicenseKeysSubscriberMetadata));
}
export function benefitLicenseKeysSubscriberMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysSubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysSubscriberMetadata' from JSON`);
}
/** @internal */
export const BenefitLicenseKeysSubscriber$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: Organization$inboundSchema,
    properties: BenefitLicenseKeysSubscriberProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitLicenseKeysSubscriber$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: Organization$outboundSchema,
    properties: BenefitLicenseKeysSubscriberProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeysSubscriber$;
(function (BenefitLicenseKeysSubscriber$) {
    /** @deprecated use `BenefitLicenseKeysSubscriber$inboundSchema` instead. */
    BenefitLicenseKeysSubscriber$.inboundSchema = BenefitLicenseKeysSubscriber$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysSubscriber$outboundSchema` instead. */
    BenefitLicenseKeysSubscriber$.outboundSchema = BenefitLicenseKeysSubscriber$outboundSchema;
})(BenefitLicenseKeysSubscriber$ || (BenefitLicenseKeysSubscriber$ = {}));
export function benefitLicenseKeysSubscriberToJSON(benefitLicenseKeysSubscriber) {
    return JSON.stringify(BenefitLicenseKeysSubscriber$outboundSchema.parse(benefitLicenseKeysSubscriber));
}
export function benefitLicenseKeysSubscriberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysSubscriber' from JSON`);
}
