// @polar-sh/sdk/dist/esm/models/components/benefitdiscordsubscriberproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitDiscordSubscriberProperties$inboundSchema = z.object({
    guild_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "guild_id": "guildId",
    });
});
/** @internal */
export const BenefitDiscordSubscriberProperties$outboundSchema = z.object({
    guildId: z.string(),
}).transform((v) => {
    return remap$(v, {
        guildId: "guild_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordSubscriberProperties$;
(function (BenefitDiscordSubscriberProperties$) {
    /** @deprecated use `BenefitDiscordSubscriberProperties$inboundSchema` instead. */
    BenefitDiscordSubscriberProperties$.inboundSchema = BenefitDiscordSubscriberProperties$inboundSchema;
    /** @deprecated use `BenefitDiscordSubscriberProperties$outboundSchema` instead. */
    BenefitDiscordSubscriberProperties$.outboundSchema = BenefitDiscordSubscriberProperties$outboundSchema;
})(BenefitDiscordSubscriberProperties$ || (BenefitDiscordSubscriberProperties$ = {}));
export function benefitDiscordSubscriberPropertiesToJSON(benefitDiscordSubscriberProperties) {
    return JSON.stringify(BenefitDiscordSubscriberProperties$outboundSchema.parse(benefitDiscordSubscriberProperties));
}
export function benefitDiscordSubscriberPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordSubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordSubscriberProperties' from JSON`);
}
