// @polar-sh/sdk/dist/esm/models/components/benefitgrantdownloadableswebhook.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDownloadables$inboundSchema, BenefitDownloadables$outboundSchema, } from "./benefitdownloadables.js";
import { BenefitGrantDownloadablesProperties$inboundSchema, BenefitGrantDownloadablesProperties$outboundSchema, } from "./benefitgrantdownloadablesproperties.js";
import { BenefitGrantError$inboundSchema, BenefitGrantError$outboundSchema, } from "./benefitgranterror.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const BenefitGrantDownloadablesWebhook$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    is_granted: z.boolean(),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    is_revoked: z.boolean(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    customer_id: z.string(),
    benefit_id: z.string(),
    error: z.nullable(BenefitGrantError$inboundSchema).optional(),
    customer: Customer$inboundSchema,
    benefit: BenefitDownloadables$inboundSchema,
    properties: BenefitGrantDownloadablesProperties$inboundSchema,
    previous_properties: z.nullable(BenefitGrantDownloadablesProperties$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "is_granted": "isGranted",
        "revoked_at": "revokedAt",
        "is_revoked": "isRevoked",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "previous_properties": "previousProperties",
    });
});
/** @internal */
export const BenefitGrantDownloadablesWebhook$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    isGranted: z.boolean(),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    isRevoked: z.boolean(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    customerId: z.string(),
    benefitId: z.string(),
    error: z.nullable(BenefitGrantError$outboundSchema).optional(),
    customer: Customer$outboundSchema,
    benefit: BenefitDownloadables$outboundSchema,
    properties: BenefitGrantDownloadablesProperties$outboundSchema,
    previousProperties: z.nullable(BenefitGrantDownloadablesProperties$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        isGranted: "is_granted",
        revokedAt: "revoked_at",
        isRevoked: "is_revoked",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        customerId: "customer_id",
        benefitId: "benefit_id",
        previousProperties: "previous_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantDownloadablesWebhook$;
(function (BenefitGrantDownloadablesWebhook$) {
    /** @deprecated use `BenefitGrantDownloadablesWebhook$inboundSchema` instead. */
    BenefitGrantDownloadablesWebhook$.inboundSchema = BenefitGrantDownloadablesWebhook$inboundSchema;
    /** @deprecated use `BenefitGrantDownloadablesWebhook$outboundSchema` instead. */
    BenefitGrantDownloadablesWebhook$.outboundSchema = BenefitGrantDownloadablesWebhook$outboundSchema;
})(BenefitGrantDownloadablesWebhook$ || (BenefitGrantDownloadablesWebhook$ = {}));
export function benefitGrantDownloadablesWebhookToJSON(benefitGrantDownloadablesWebhook) {
    return JSON.stringify(BenefitGrantDownloadablesWebhook$outboundSchema.parse(benefitGrantDownloadablesWebhook));
}
export function benefitGrantDownloadablesWebhookFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantDownloadablesWebhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantDownloadablesWebhook' from JSON`);
}
