// @polar-sh/sdk/dist/esm/models/components/benefitgrantdiscordproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitGrantDiscordProperties$inboundSchema = z.object({
    account_id: z.nullable(z.string()).optional(),
    guild_id: z.string().optional(),
    role_id: z.string().optional(),
    granted_account_id: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "account_id": "accountId",
        "guild_id": "guildId",
        "role_id": "roleId",
        "granted_account_id": "grantedAccountId",
    });
});
/** @internal */
export const BenefitGrantDiscordProperties$outboundSchema = z.object({
    accountId: z.nullable(z.string()).optional(),
    guildId: z.string().optional(),
    roleId: z.string().optional(),
    grantedAccountId: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        accountId: "account_id",
        guildId: "guild_id",
        roleId: "role_id",
        grantedAccountId: "granted_account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantDiscordProperties$;
(function (BenefitGrantDiscordProperties$) {
    /** @deprecated use `BenefitGrantDiscordProperties$inboundSchema` instead. */
    BenefitGrantDiscordProperties$.inboundSchema = BenefitGrantDiscordProperties$inboundSchema;
    /** @deprecated use `BenefitGrantDiscordProperties$outboundSchema` instead. */
    BenefitGrantDiscordProperties$.outboundSchema = BenefitGrantDiscordProperties$outboundSchema;
})(BenefitGrantDiscordProperties$ || (BenefitGrantDiscordProperties$ = {}));
export function benefitGrantDiscordPropertiesToJSON(benefitGrantDiscordProperties) {
    return JSON.stringify(BenefitGrantDiscordProperties$outboundSchema.parse(benefitGrantDiscordProperties));
}
export function benefitGrantDiscordPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantDiscordProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantDiscordProperties' from JSON`);
}
