// @polar-sh/sdk/dist/esm/models/components/benefitgithubrepositorycreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGitHubRepositoryCreateProperties$inboundSchema, BenefitGitHubRepositoryCreateProperties$outboundSchema, } from "./benefitgithubrepositorycreateproperties.js";
/** @internal */
export const BenefitGitHubRepositoryCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitGitHubRepositoryCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGitHubRepositoryCreateMetadata$;
(function (BenefitGitHubRepositoryCreateMetadata$) {
    /** @deprecated use `BenefitGitHubRepositoryCreateMetadata$inboundSchema` instead. */
    BenefitGitHubRepositoryCreateMetadata$.inboundSchema = BenefitGitHubRepositoryCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreateMetadata$outboundSchema` instead. */
    BenefitGitHubRepositoryCreateMetadata$.outboundSchema = BenefitGitHubRepositoryCreateMetadata$outboundSchema;
})(BenefitGitHubRepositoryCreateMetadata$ || (BenefitGitHubRepositoryCreateMetadata$ = {}));
export function benefitGitHubRepositoryCreateMetadataToJSON(benefitGitHubRepositoryCreateMetadata) {
    return JSON.stringify(BenefitGitHubRepositoryCreateMetadata$outboundSchema.parse(benefitGitHubRepositoryCreateMetadata));
}
export function benefitGitHubRepositoryCreateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGitHubRepositoryCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryCreateMetadata' from JSON`);
}
/** @internal */
export const BenefitGitHubRepositoryCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("github_repository"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: BenefitGitHubRepositoryCreateProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitGitHubRepositoryCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("github_repository"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: BenefitGitHubRepositoryCreateProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGitHubRepositoryCreate$;
(function (BenefitGitHubRepositoryCreate$) {
    /** @deprecated use `BenefitGitHubRepositoryCreate$inboundSchema` instead. */
    BenefitGitHubRepositoryCreate$.inboundSchema = BenefitGitHubRepositoryCreate$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreate$outboundSchema` instead. */
    BenefitGitHubRepositoryCreate$.outboundSchema = BenefitGitHubRepositoryCreate$outboundSchema;
})(BenefitGitHubRepositoryCreate$ || (BenefitGitHubRepositoryCreate$ = {}));
export function benefitGitHubRepositoryCreateToJSON(benefitGitHubRepositoryCreate) {
    return JSON.stringify(BenefitGitHubRepositoryCreate$outboundSchema.parse(benefitGitHubRepositoryCreate));
}
export function benefitGitHubRepositoryCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGitHubRepositoryCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryCreate' from JSON`);
}
