// @polar-sh/sdk/dist/esm/models/components/benefitdiscordcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDiscordCreateProperties$inboundSchema, BenefitDiscordCreateProperties$outboundSchema, } from "./benefitdiscordcreateproperties.js";
/** @internal */
export const BenefitDiscordCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitDiscordCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordCreateMetadata$;
(function (BenefitDiscordCreateMetadata$) {
    /** @deprecated use `BenefitDiscordCreateMetadata$inboundSchema` instead. */
    BenefitDiscordCreateMetadata$.inboundSchema = BenefitDiscordCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordCreateMetadata$outboundSchema` instead. */
    BenefitDiscordCreateMetadata$.outboundSchema = BenefitDiscordCreateMetadata$outboundSchema;
})(BenefitDiscordCreateMetadata$ || (BenefitDiscordCreateMetadata$ = {}));
export function benefitDiscordCreateMetadataToJSON(benefitDiscordCreateMetadata) {
    return JSON.stringify(BenefitDiscordCreateMetadata$outboundSchema.parse(benefitDiscordCreateMetadata));
}
export function benefitDiscordCreateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordCreateMetadata' from JSON`);
}
/** @internal */
export const BenefitDiscordCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("discord"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: BenefitDiscordCreateProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitDiscordCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("discord"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: BenefitDiscordCreateProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordCreate$;
(function (BenefitDiscordCreate$) {
    /** @deprecated use `BenefitDiscordCreate$inboundSchema` instead. */
    BenefitDiscordCreate$.inboundSchema = BenefitDiscordCreate$inboundSchema;
    /** @deprecated use `BenefitDiscordCreate$outboundSchema` instead. */
    BenefitDiscordCreate$.outboundSchema = BenefitDiscordCreate$outboundSchema;
})(BenefitDiscordCreate$ || (BenefitDiscordCreate$ = {}));
export function benefitDiscordCreateToJSON(benefitDiscordCreate) {
    return JSON.stringify(BenefitDiscordCreate$outboundSchema.parse(benefitDiscordCreate));
}
export function benefitDiscordCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordCreate' from JSON`);
}
