// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablescreateproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitDownloadablesCreateProperties$inboundSchema = z.object({
    archived: z.record(z.boolean()).optional(),
    files: z.array(z.string()),
});
/** @internal */
export const BenefitDownloadablesCreateProperties$outboundSchema = z.object({
    archived: z.record(z.boolean()).optional(),
    files: z.array(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDownloadablesCreateProperties$;
(function (BenefitDownloadablesCreateProperties$) {
    /** @deprecated use `BenefitDownloadablesCreateProperties$inboundSchema` instead. */
    BenefitDownloadablesCreateProperties$.inboundSchema = BenefitDownloadablesCreateProperties$inboundSchema;
    /** @deprecated use `BenefitDownloadablesCreateProperties$outboundSchema` instead. */
    BenefitDownloadablesCreateProperties$.outboundSchema = BenefitDownloadablesCreateProperties$outboundSchema;
})(BenefitDownloadablesCreateProperties$ || (BenefitDownloadablesCreateProperties$ = {}));
export function benefitDownloadablesCreatePropertiesToJSON(benefitDownloadablesCreateProperties) {
    return JSON.stringify(BenefitDownloadablesCreateProperties$outboundSchema.parse(benefitDownloadablesCreateProperties));
}
export function benefitDownloadablesCreatePropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesCreateProperties' from JSON`);
}
