// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablescreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDownloadablesCreateProperties$inboundSchema, BenefitDownloadablesCreateProperties$outboundSchema, } from "./benefitdownloadablescreateproperties.js";
/** @internal */
export const BenefitDownloadablesCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitDownloadablesCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDownloadablesCreateMetadata$;
(function (BenefitDownloadablesCreateMetadata$) {
    /** @deprecated use `BenefitDownloadablesCreateMetadata$inboundSchema` instead. */
    BenefitDownloadablesCreateMetadata$.inboundSchema = BenefitDownloadablesCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitDownloadablesCreateMetadata$outboundSchema` instead. */
    BenefitDownloadablesCreateMetadata$.outboundSchema = BenefitDownloadablesCreateMetadata$outboundSchema;
})(BenefitDownloadablesCreateMetadata$ || (BenefitDownloadablesCreateMetadata$ = {}));
export function benefitDownloadablesCreateMetadataToJSON(benefitDownloadablesCreateMetadata) {
    return JSON.stringify(BenefitDownloadablesCreateMetadata$outboundSchema.parse(benefitDownloadablesCreateMetadata));
}
export function benefitDownloadablesCreateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesCreateMetadata' from JSON`);
}
/** @internal */
export const BenefitDownloadablesCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("downloadables"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: BenefitDownloadablesCreateProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitDownloadablesCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("downloadables"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: BenefitDownloadablesCreateProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDownloadablesCreate$;
(function (BenefitDownloadablesCreate$) {
    /** @deprecated use `BenefitDownloadablesCreate$inboundSchema` instead. */
    BenefitDownloadablesCreate$.inboundSchema = BenefitDownloadablesCreate$inboundSchema;
    /** @deprecated use `BenefitDownloadablesCreate$outboundSchema` instead. */
    BenefitDownloadablesCreate$.outboundSchema = BenefitDownloadablesCreate$outboundSchema;
})(BenefitDownloadablesCreate$ || (BenefitDownloadablesCreate$ = {}));
export function benefitDownloadablesCreateToJSON(benefitDownloadablesCreate) {
    return JSON.stringify(BenefitDownloadablesCreate$outboundSchema.parse(benefitDownloadablesCreate));
}
export function benefitDownloadablesCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesCreate' from JSON`);
}
