// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyactivationproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitLicenseKeyActivationProperties$inboundSchema = z.object({
    limit: z.number().int(),
    enable_customer_admin: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "enable_customer_admin": "enableCustomerAdmin",
    });
});
/** @internal */
export const BenefitLicenseKeyActivationProperties$outboundSchema = z.object({
    limit: z.number().int(),
    enableCustomerAdmin: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        enableCustomerAdmin: "enable_customer_admin",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeyActivationProperties$;
(function (BenefitLicenseKeyActivationProperties$) {
    /** @deprecated use `BenefitLicenseKeyActivationProperties$inboundSchema` instead. */
    BenefitLicenseKeyActivationProperties$.inboundSchema = BenefitLicenseKeyActivationProperties$inboundSchema;
    /** @deprecated use `BenefitLicenseKeyActivationProperties$outboundSchema` instead. */
    BenefitLicenseKeyActivationProperties$.outboundSchema = BenefitLicenseKeyActivationProperties$outboundSchema;
})(BenefitLicenseKeyActivationProperties$ || (BenefitLicenseKeyActivationProperties$ = {}));
export function benefitLicenseKeyActivationPropertiesToJSON(benefitLicenseKeyActivationProperties) {
    return JSON.stringify(BenefitLicenseKeyActivationProperties$outboundSchema.parse(benefitLicenseKeyActivationProperties));
}
export function benefitLicenseKeyActivationPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeyActivationProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeyActivationProperties' from JSON`);
}
