// @polar-sh/sdk/dist/esm/models/components/benefitgrantdownloadablesproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitGrantDownloadablesProperties$inboundSchema = z.object({
    files: z.array(z.string()).optional(),
});
/** @internal */
export const BenefitGrantDownloadablesProperties$outboundSchema = z.object({
    files: z.array(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantDownloadablesProperties$;
(function (BenefitGrantDownloadablesProperties$) {
    /** @deprecated use `BenefitGrantDownloadablesProperties$inboundSchema` instead. */
    BenefitGrantDownloadablesProperties$.inboundSchema = BenefitGrantDownloadablesProperties$inboundSchema;
    /** @deprecated use `BenefitGrantDownloadablesProperties$outboundSchema` instead. */
    BenefitGrantDownloadablesProperties$.outboundSchema = BenefitGrantDownloadablesProperties$outboundSchema;
})(BenefitGrantDownloadablesProperties$ || (BenefitGrantDownloadablesProperties$ = {}));
export function benefitGrantDownloadablesPropertiesToJSON(benefitGrantDownloadablesProperties) {
    return JSON.stringify(BenefitGrantDownloadablesProperties$outboundSchema.parse(benefitGrantDownloadablesProperties));
}
export function benefitGrantDownloadablesPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantDownloadablesProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantDownloadablesProperties' from JSON`);
}
