// @polar-sh/sdk/dist/esm/models/components/benefitdiscord.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDiscordProperties$inboundSchema, BenefitDiscordProperties$outboundSchema, } from "./benefitdiscordproperties.js";
/** @internal */
export const BenefitDiscordMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitDiscordMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordMetadata$;
(function (BenefitDiscordMetadata$) {
    /** @deprecated use `BenefitDiscordMetadata$inboundSchema` instead. */
    BenefitDiscordMetadata$.inboundSchema = BenefitDiscordMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordMetadata$outboundSchema` instead. */
    BenefitDiscordMetadata$.outboundSchema = BenefitDiscordMetadata$outboundSchema;
})(BenefitDiscordMetadata$ || (BenefitDiscordMetadata$ = {}));
export function benefitDiscordMetadataToJSON(benefitDiscordMetadata) {
    return JSON.stringify(BenefitDiscordMetadata$outboundSchema.parse(benefitDiscordMetadata));
}
export function benefitDiscordMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordMetadata' from JSON`);
}
/** @internal */
export const BenefitDiscord$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: BenefitDiscordProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitDiscord$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: BenefitDiscordProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscord$;
(function (BenefitDiscord$) {
    /** @deprecated use `BenefitDiscord$inboundSchema` instead. */
    BenefitDiscord$.inboundSchema = BenefitDiscord$inboundSchema;
    /** @deprecated use `BenefitDiscord$outboundSchema` instead. */
    BenefitDiscord$.outboundSchema = BenefitDiscord$outboundSchema;
})(BenefitDiscord$ || (BenefitDiscord$ = {}));
export function benefitDiscordToJSON(benefitDiscord) {
    return JSON.stringify(BenefitDiscord$outboundSchema.parse(benefitDiscord));
}
export function benefitDiscordFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscord$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscord' from JSON`);
}
