// @polar-sh/sdk/dist/esm/models/components/benefitpublic.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitType$inboundSchema, BenefitType$outboundSchema, } from "./benefittype.js";
/** @internal */
export const BenefitPublic$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: BenefitType$inboundSchema,
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitPublic$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: BenefitType$outboundSchema,
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitPublic$;
(function (BenefitPublic$) {
    /** @deprecated use `BenefitPublic$inboundSchema` instead. */
    BenefitPublic$.inboundSchema = BenefitPublic$inboundSchema;
    /** @deprecated use `BenefitPublic$outboundSchema` instead. */
    BenefitPublic$.outboundSchema = BenefitPublic$outboundSchema;
})(BenefitPublic$ || (BenefitPublic$ = {}));
export function benefitPublicToJSON(benefitPublic) {
    return JSON.stringify(BenefitPublic$outboundSchema.parse(benefitPublic));
}
export function benefitPublicFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitPublic$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitPublic' from JSON`);
}
