// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyscreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitLicenseKeysCreateProperties$inboundSchema, BenefitLicenseKeysCreateProperties$outboundSchema, } from "./benefitlicensekeyscreateproperties.js";
/** @internal */
export const BenefitLicenseKeysCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitLicenseKeysCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeysCreateMetadata$;
(function (BenefitLicenseKeysCreateMetadata$) {
    /** @deprecated use `BenefitLicenseKeysCreateMetadata$inboundSchema` instead. */
    BenefitLicenseKeysCreateMetadata$.inboundSchema = BenefitLicenseKeysCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreateMetadata$outboundSchema` instead. */
    BenefitLicenseKeysCreateMetadata$.outboundSchema = BenefitLicenseKeysCreateMetadata$outboundSchema;
})(BenefitLicenseKeysCreateMetadata$ || (BenefitLicenseKeysCreateMetadata$ = {}));
export function benefitLicenseKeysCreateMetadataToJSON(benefitLicenseKeysCreateMetadata) {
    return JSON.stringify(BenefitLicenseKeysCreateMetadata$outboundSchema.parse(benefitLicenseKeysCreateMetadata));
}
export function benefitLicenseKeysCreateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysCreateMetadata' from JSON`);
}
/** @internal */
export const BenefitLicenseKeysCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("license_keys"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: BenefitLicenseKeysCreateProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitLicenseKeysCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("license_keys"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: BenefitLicenseKeysCreateProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeysCreate$;
(function (BenefitLicenseKeysCreate$) {
    /** @deprecated use `BenefitLicenseKeysCreate$inboundSchema` instead. */
    BenefitLicenseKeysCreate$.inboundSchema = BenefitLicenseKeysCreate$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreate$outboundSchema` instead. */
    BenefitLicenseKeysCreate$.outboundSchema = BenefitLicenseKeysCreate$outboundSchema;
})(BenefitLicenseKeysCreate$ || (BenefitLicenseKeysCreate$ = {}));
export function benefitLicenseKeysCreateToJSON(benefitLicenseKeysCreate) {
    return JSON.stringify(BenefitLicenseKeysCreate$outboundSchema.parse(benefitLicenseKeysCreate));
}
export function benefitLicenseKeysCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysCreate' from JSON`);
}
