// @polar-sh/sdk/dist/esm/models/components/billingaddressfieldmode.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const BillingAddressFieldMode = {
    Required: "required",
    Optional: "optional",
    Disabled: "disabled",
};
/** @internal */
export const BillingAddressFieldMode$inboundSchema = z.nativeEnum(BillingAddressFieldMode);
/** @internal */
export const BillingAddressFieldMode$outboundSchema = BillingAddressFieldMode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BillingAddressFieldMode$;
(function (BillingAddressFieldMode$) {
    /** @deprecated use `BillingAddressFieldMode$inboundSchema` instead. */
    BillingAddressFieldMode$.inboundSchema = BillingAddressFieldMode$inboundSchema;
    /** @deprecated use `BillingAddressFieldMode$outboundSchema` instead. */
    BillingAddressFieldMode$.outboundSchema = BillingAddressFieldMode$outboundSchema;
})(BillingAddressFieldMode$ || (BillingAddressFieldMode$ = {}));
