// @polar-sh/sdk/dist/esm/models/components/checkoutlinkcreateproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CheckoutLinkCreateProductMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const CheckoutLinkCreateProductMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkCreateProductMetadata$;
(function (CheckoutLinkCreateProductMetadata$) {
    /** @deprecated use `CheckoutLinkCreateProductMetadata$inboundSchema` instead. */
    CheckoutLinkCreateProductMetadata$.inboundSchema = CheckoutLinkCreateProductMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductMetadata$outboundSchema` instead. */
    CheckoutLinkCreateProductMetadata$.outboundSchema = CheckoutLinkCreateProductMetadata$outboundSchema;
})(CheckoutLinkCreateProductMetadata$ || (CheckoutLinkCreateProductMetadata$ = {}));
export function checkoutLinkCreateProductMetadataToJSON(checkoutLinkCreateProductMetadata) {
    return JSON.stringify(CheckoutLinkCreateProductMetadata$outboundSchema.parse(checkoutLinkCreateProductMetadata));
}
export function checkoutLinkCreateProductMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkCreateProductMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductMetadata' from JSON`);
}
/** @internal */
export const CheckoutLinkCreateProduct$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    payment_processor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    require_billing_address: z.boolean().default(false),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    product_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "success_url": "successUrl",
        "product_id": "productId",
    });
});
/** @internal */
export const CheckoutLinkCreateProduct$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    paymentProcessor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    requireBillingAddress: z.boolean().default(false),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    productId: z.string(),
}).transform((v) => {
    return remap$(v, {
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkCreateProduct$;
(function (CheckoutLinkCreateProduct$) {
    /** @deprecated use `CheckoutLinkCreateProduct$inboundSchema` instead. */
    CheckoutLinkCreateProduct$.inboundSchema = CheckoutLinkCreateProduct$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProduct$outboundSchema` instead. */
    CheckoutLinkCreateProduct$.outboundSchema = CheckoutLinkCreateProduct$outboundSchema;
})(CheckoutLinkCreateProduct$ || (CheckoutLinkCreateProduct$ = {}));
export function checkoutLinkCreateProductToJSON(checkoutLinkCreateProduct) {
    return JSON.stringify(CheckoutLinkCreateProduct$outboundSchema.parse(checkoutLinkCreateProduct));
}
export function checkoutLinkCreateProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkCreateProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProduct' from JSON`);
}
