// @polar-sh/sdk/dist/esm/models/components/checkoutlinksortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CheckoutLinkSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Label: "label",
    MinusLabel: "-label",
    SuccessUrl: "success_url",
    MinusSuccessUrl: "-success_url",
    AllowDiscountCodes: "allow_discount_codes",
    MinusAllowDiscountCodes: "-allow_discount_codes",
};
/** @internal */
export const CheckoutLinkSortProperty$inboundSchema = z.nativeEnum(CheckoutLinkSortProperty);
/** @internal */
export const CheckoutLinkSortProperty$outboundSchema = CheckoutLinkSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkSortProperty$;
(function (CheckoutLinkSortProperty$) {
    /** @deprecated use `CheckoutLinkSortProperty$inboundSchema` instead. */
    CheckoutLinkSortProperty$.inboundSchema = CheckoutLinkSortProperty$inboundSchema;
    /** @deprecated use `CheckoutLinkSortProperty$outboundSchema` instead. */
    CheckoutLinkSortProperty$.outboundSchema = CheckoutLinkSortProperty$outboundSchema;
})(CheckoutLinkSortProperty$ || (CheckoutLinkSortProperty$ = {}));
