// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountfixedonceforeverduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountFixedOnceForeverDuration$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
});
/** @internal */
export const CheckoutDiscountFixedOnceForeverDuration$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    type: DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutDiscountFixedOnceForeverDuration$;
(function (CheckoutDiscountFixedOnceForeverDuration$) {
    /** @deprecated use `CheckoutDiscountFixedOnceForeverDuration$inboundSchema` instead. */
    CheckoutDiscountFixedOnceForeverDuration$.inboundSchema = CheckoutDiscountFixedOnceForeverDuration$inboundSchema;
    /** @deprecated use `CheckoutDiscountFixedOnceForeverDuration$outboundSchema` instead. */
    CheckoutDiscountFixedOnceForeverDuration$.outboundSchema = CheckoutDiscountFixedOnceForeverDuration$outboundSchema;
})(CheckoutDiscountFixedOnceForeverDuration$ || (CheckoutDiscountFixedOnceForeverDuration$ = {}));
export function checkoutDiscountFixedOnceForeverDurationToJSON(checkoutDiscountFixedOnceForeverDuration) {
    return JSON.stringify(CheckoutDiscountFixedOnceForeverDuration$outboundSchema.parse(checkoutDiscountFixedOnceForeverDuration));
}
export function checkoutDiscountFixedOnceForeverDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountFixedOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountFixedOnceForeverDuration' from JSON`);
}
