// @polar-sh/sdk/dist/esm/models/components/checkoutlinkproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitPublic$inboundSchema, BenefitPublic$outboundSchema, } from "./benefitpublic.js";
import { LegacyRecurringProductPrice$inboundSchema, LegacyRecurringProductPrice$outboundSchema, } from "./legacyrecurringproductprice.js";
import { ProductMediaFileRead$inboundSchema, ProductMediaFileRead$outboundSchema, } from "./productmediafileread.js";
import { ProductPrice$inboundSchema, ProductPrice$outboundSchema, } from "./productprice.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const CheckoutLinkProductMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const CheckoutLinkProductMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkProductMetadata$;
(function (CheckoutLinkProductMetadata$) {
    /** @deprecated use `CheckoutLinkProductMetadata$inboundSchema` instead. */
    CheckoutLinkProductMetadata$.inboundSchema = CheckoutLinkProductMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkProductMetadata$outboundSchema` instead. */
    CheckoutLinkProductMetadata$.outboundSchema = CheckoutLinkProductMetadata$outboundSchema;
})(CheckoutLinkProductMetadata$ || (CheckoutLinkProductMetadata$ = {}));
export function checkoutLinkProductMetadataToJSON(checkoutLinkProductMetadata) {
    return JSON.stringify(CheckoutLinkProductMetadata$outboundSchema.parse(checkoutLinkProductMetadata));
}
export function checkoutLinkProductMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkProductMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkProductMetadata' from JSON`);
}
/** @internal */
export const CheckoutLinkProductPrices$inboundSchema = z.union([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const CheckoutLinkProductPrices$outboundSchema = z.union([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkProductPrices$;
(function (CheckoutLinkProductPrices$) {
    /** @deprecated use `CheckoutLinkProductPrices$inboundSchema` instead. */
    CheckoutLinkProductPrices$.inboundSchema = CheckoutLinkProductPrices$inboundSchema;
    /** @deprecated use `CheckoutLinkProductPrices$outboundSchema` instead. */
    CheckoutLinkProductPrices$.outboundSchema = CheckoutLinkProductPrices$outboundSchema;
})(CheckoutLinkProductPrices$ || (CheckoutLinkProductPrices$ = {}));
export function checkoutLinkProductPricesToJSON(checkoutLinkProductPrices) {
    return JSON.stringify(CheckoutLinkProductPrices$outboundSchema.parse(checkoutLinkProductPrices));
}
export function checkoutLinkProductPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkProductPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkProductPrices' from JSON`);
}
/** @internal */
export const CheckoutLinkProduct$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    benefits: z.array(BenefitPublic$inboundSchema),
    medias: z.array(ProductMediaFileRead$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "recurring_interval": "recurringInterval",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const CheckoutLinkProduct$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ])),
    benefits: z.array(BenefitPublic$outboundSchema),
    medias: z.array(ProductMediaFileRead$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        recurringInterval: "recurring_interval",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkProduct$;
(function (CheckoutLinkProduct$) {
    /** @deprecated use `CheckoutLinkProduct$inboundSchema` instead. */
    CheckoutLinkProduct$.inboundSchema = CheckoutLinkProduct$inboundSchema;
    /** @deprecated use `CheckoutLinkProduct$outboundSchema` instead. */
    CheckoutLinkProduct$.outboundSchema = CheckoutLinkProduct$outboundSchema;
})(CheckoutLinkProduct$ || (CheckoutLinkProduct$ = {}));
export function checkoutLinkProductToJSON(checkoutLinkProduct) {
    return JSON.stringify(CheckoutLinkProduct$outboundSchema.parse(checkoutLinkProduct));
}
export function checkoutLinkProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkProduct' from JSON`);
}
