// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountpercentagerepeatduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountPercentageRepeatDuration$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    duration_in_months: z.number().int(),
    type: DiscountType$inboundSchema,
    basis_points: z.number().int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "duration_in_months": "durationInMonths",
        "basis_points": "basisPoints",
    });
});
/** @internal */
export const CheckoutDiscountPercentageRepeatDuration$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.number().int(),
    type: DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutDiscountPercentageRepeatDuration$;
(function (CheckoutDiscountPercentageRepeatDuration$) {
    /** @deprecated use `CheckoutDiscountPercentageRepeatDuration$inboundSchema` instead. */
    CheckoutDiscountPercentageRepeatDuration$.inboundSchema = CheckoutDiscountPercentageRepeatDuration$inboundSchema;
    /** @deprecated use `CheckoutDiscountPercentageRepeatDuration$outboundSchema` instead. */
    CheckoutDiscountPercentageRepeatDuration$.outboundSchema = CheckoutDiscountPercentageRepeatDuration$outboundSchema;
})(CheckoutDiscountPercentageRepeatDuration$ || (CheckoutDiscountPercentageRepeatDuration$ = {}));
export function checkoutDiscountPercentageRepeatDurationToJSON(checkoutDiscountPercentageRepeatDuration) {
    return JSON.stringify(CheckoutDiscountPercentageRepeatDuration$outboundSchema.parse(checkoutDiscountPercentageRepeatDuration));
}
export function checkoutDiscountPercentageRepeatDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountPercentageRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountPercentageRepeatDuration' from JSON`);
}
