// @polar-sh/sdk/dist/esm/models/components/checkoutsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CheckoutSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    ExpiresAt: "expires_at",
    MinusExpiresAt: "-expires_at",
    Status: "status",
    MinusStatus: "-status",
};
/** @internal */
export const CheckoutSortProperty$inboundSchema = z.nativeEnum(CheckoutSortProperty);
/** @internal */
export const CheckoutSortProperty$outboundSchema = CheckoutSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutSortProperty$;
(function (CheckoutSortProperty$) {
    /** @deprecated use `CheckoutSortProperty$inboundSchema` instead. */
    CheckoutSortProperty$.inboundSchema = CheckoutSortProperty$inboundSchema;
    /** @deprecated use `CheckoutSortProperty$outboundSchema` instead. */
    CheckoutSortProperty$.outboundSchema = CheckoutSortProperty$outboundSchema;
})(CheckoutSortProperty$ || (CheckoutSortProperty$ = {}));
