// @polar-sh/sdk/dist/esm/models/components/checkoutupdatepublic.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
/** @internal */
export const CheckoutUpdatePublicCustomFieldData$inboundSchema = z.union([
    z.string(),
    z.number().int(),
    z.boolean(),
    z.string().datetime({ offset: true }).transform(v => new Date(v)),
]);
/** @internal */
export const CheckoutUpdatePublicCustomFieldData$outboundSchema = z.union([
    z.string(),
    z.number().int(),
    z.boolean(),
    z.date().transform(v => v.toISOString()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutUpdatePublicCustomFieldData$;
(function (CheckoutUpdatePublicCustomFieldData$) {
    /** @deprecated use `CheckoutUpdatePublicCustomFieldData$inboundSchema` instead. */
    CheckoutUpdatePublicCustomFieldData$.inboundSchema = CheckoutUpdatePublicCustomFieldData$inboundSchema;
    /** @deprecated use `CheckoutUpdatePublicCustomFieldData$outboundSchema` instead. */
    CheckoutUpdatePublicCustomFieldData$.outboundSchema = CheckoutUpdatePublicCustomFieldData$outboundSchema;
})(CheckoutUpdatePublicCustomFieldData$ || (CheckoutUpdatePublicCustomFieldData$ = {}));
export function checkoutUpdatePublicCustomFieldDataToJSON(checkoutUpdatePublicCustomFieldData) {
    return JSON.stringify(CheckoutUpdatePublicCustomFieldData$outboundSchema.parse(checkoutUpdatePublicCustomFieldData));
}
export function checkoutUpdatePublicCustomFieldDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutUpdatePublicCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutUpdatePublicCustomFieldData' from JSON`);
}
/** @internal */
export const CheckoutUpdatePublic$inboundSchema = z.object({
    custom_field_data: z.record(z.nullable(z.union([
        z.string(),
        z.number().int(),
        z.boolean(),
        z.string().datetime({ offset: true }).transform(v => new Date(v)),
    ]))).optional(),
    product_id: z.nullable(z.string()).optional(),
    product_price_id: z.nullable(z.string()).optional(),
    amount: z.nullable(z.number().int()).optional(),
    is_business_customer: z.nullable(z.boolean()).optional(),
    customer_name: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()).optional(),
    customer_billing_name: z.nullable(z.string()).optional(),
    customer_billing_address: z.nullable(Address$inboundSchema).optional(),
    customer_tax_id: z.nullable(z.string()).optional(),
    discount_code: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "custom_field_data": "customFieldData",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "is_business_customer": "isBusinessCustomer",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_billing_name": "customerBillingName",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "discount_code": "discountCode",
    });
});
/** @internal */
export const CheckoutUpdatePublic$outboundSchema = z.object({
    customFieldData: z.record(z.nullable(z.union([
        z.string(),
        z.number().int(),
        z.boolean(),
        z.date().transform(v => v.toISOString()),
    ]))).optional(),
    productId: z.nullable(z.string()).optional(),
    productPriceId: z.nullable(z.string()).optional(),
    amount: z.nullable(z.number().int()).optional(),
    isBusinessCustomer: z.nullable(z.boolean()).optional(),
    customerName: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()).optional(),
    customerBillingName: z.nullable(z.string()).optional(),
    customerBillingAddress: z.nullable(Address$outboundSchema).optional(),
    customerTaxId: z.nullable(z.string()).optional(),
    discountCode: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        customFieldData: "custom_field_data",
        productId: "product_id",
        productPriceId: "product_price_id",
        isBusinessCustomer: "is_business_customer",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerBillingName: "customer_billing_name",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        discountCode: "discount_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutUpdatePublic$;
(function (CheckoutUpdatePublic$) {
    /** @deprecated use `CheckoutUpdatePublic$inboundSchema` instead. */
    CheckoutUpdatePublic$.inboundSchema = CheckoutUpdatePublic$inboundSchema;
    /** @deprecated use `CheckoutUpdatePublic$outboundSchema` instead. */
    CheckoutUpdatePublic$.outboundSchema = CheckoutUpdatePublic$outboundSchema;
})(CheckoutUpdatePublic$ || (CheckoutUpdatePublic$ = {}));
export function checkoutUpdatePublicToJSON(checkoutUpdatePublic) {
    return JSON.stringify(CheckoutUpdatePublic$outboundSchema.parse(checkoutUpdatePublic));
}
export function checkoutUpdatePublicFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutUpdatePublic$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutUpdatePublic' from JSON`);
}
