// @polar-sh/sdk/dist/esm/models/components/customerpaymentmethodcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPaymentMethodCreate$inboundSchema = z.object({
    confirmation_token_id: z.string(),
    set_default: z.boolean(),
    return_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "confirmation_token_id": "confirmationTokenId",
        "set_default": "setDefault",
        "return_url": "returnUrl",
    });
});
/** @internal */
export const CustomerPaymentMethodCreate$outboundSchema = z.object({
    confirmationTokenId: z.string(),
    setDefault: z.boolean(),
    returnUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        confirmationTokenId: "confirmation_token_id",
        setDefault: "set_default",
        returnUrl: "return_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPaymentMethodCreate$;
(function (CustomerPaymentMethodCreate$) {
    /** @deprecated use `CustomerPaymentMethodCreate$inboundSchema` instead. */
    CustomerPaymentMethodCreate$.inboundSchema = CustomerPaymentMethodCreate$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodCreate$outboundSchema` instead. */
    CustomerPaymentMethodCreate$.outboundSchema = CustomerPaymentMethodCreate$outboundSchema;
})(CustomerPaymentMethodCreate$ || (CustomerPaymentMethodCreate$ = {}));
export function customerPaymentMethodCreateToJSON(customerPaymentMethodCreate) {
    return JSON.stringify(CustomerPaymentMethodCreate$outboundSchema.parse(customerPaymentMethodCreate));
}
export function customerPaymentMethodCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPaymentMethodCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodCreate' from JSON`);
}
