// @polar-sh/sdk/dist/esm/models/components/customercustomermeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerCustomerMeterMeter$inboundSchema, CustomerCustomerMeterMeter$outboundSchema, } from "./customercustomermetermeter.js";
/** @internal */
export const CustomerCustomerMeter$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    meter_id: z.string(),
    consumed_units: z.number(),
    credited_units: z.number().int(),
    balance: z.number(),
    meter: CustomerCustomerMeterMeter$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "customer_id": "customerId",
        "meter_id": "meterId",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
    });
});
/** @internal */
export const CustomerCustomerMeter$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    meterId: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.number().int(),
    balance: z.number(),
    meter: CustomerCustomerMeterMeter$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customerId: "customer_id",
        meterId: "meter_id",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerCustomerMeter$;
(function (CustomerCustomerMeter$) {
    /** @deprecated use `CustomerCustomerMeter$inboundSchema` instead. */
    CustomerCustomerMeter$.inboundSchema = CustomerCustomerMeter$inboundSchema;
    /** @deprecated use `CustomerCustomerMeter$outboundSchema` instead. */
    CustomerCustomerMeter$.outboundSchema = CustomerCustomerMeter$outboundSchema;
})(CustomerCustomerMeter$ || (CustomerCustomerMeter$ = {}));
export function customerCustomerMeterToJSON(customerCustomerMeter) {
    return JSON.stringify(CustomerCustomerMeter$outboundSchema.parse(customerCustomerMeter));
}
export function customerCustomerMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerCustomerMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerCustomerMeter' from JSON`);
}
