// @polar-sh/sdk/dist/esm/models/components/customersortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Email: "email",
    MinusEmail: "-email",
    Name: "name",
    MinusName: "-name",
};
/** @internal */
export const CustomerSortProperty$inboundSchema = z.nativeEnum(CustomerSortProperty);
/** @internal */
export const CustomerSortProperty$outboundSchema = CustomerSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSortProperty$;
(function (CustomerSortProperty$) {
    /** @deprecated use `CustomerSortProperty$inboundSchema` instead. */
    CustomerSortProperty$.inboundSchema = CustomerSortProperty$inboundSchema;
    /** @deprecated use `CustomerSortProperty$outboundSchema` instead. */
    CustomerSortProperty$.outboundSchema = CustomerSortProperty$outboundSchema;
})(CustomerSortProperty$ || (CustomerSortProperty$ = {}));
