// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerBenefitGrantSortProperty = {
    GrantedAt: "granted_at",
    MinusGrantedAt: "-granted_at",
    Type: "type",
    MinusType: "-type",
    Organization: "organization",
    MinusOrganization: "-organization",
};
/** @internal */
export const CustomerBenefitGrantSortProperty$inboundSchema = z.nativeEnum(CustomerBenefitGrantSortProperty);
/** @internal */
export const CustomerBenefitGrantSortProperty$outboundSchema = CustomerBenefitGrantSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantSortProperty$;
(function (CustomerBenefitGrantSortProperty$) {
    /** @deprecated use `CustomerBenefitGrantSortProperty$inboundSchema` instead. */
    CustomerBenefitGrantSortProperty$.inboundSchema = CustomerBenefitGrantSortProperty$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantSortProperty$outboundSchema` instead. */
    CustomerBenefitGrantSortProperty$.outboundSchema = CustomerBenefitGrantSortProperty$outboundSchema;
})(CustomerBenefitGrantSortProperty$ || (CustomerBenefitGrantSortProperty$ = {}));
