// @polar-sh/sdk/dist/esm/models/components/customersessioncustomeridcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSessionCustomerIDCreate$inboundSchema = z.object({
    customer_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_id": "customerId",
    });
});
/** @internal */
export const CustomerSessionCustomerIDCreate$outboundSchema = z.object({
    customerId: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerId: "customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSessionCustomerIDCreate$;
(function (CustomerSessionCustomerIDCreate$) {
    /** @deprecated use `CustomerSessionCustomerIDCreate$inboundSchema` instead. */
    CustomerSessionCustomerIDCreate$.inboundSchema = CustomerSessionCustomerIDCreate$inboundSchema;
    /** @deprecated use `CustomerSessionCustomerIDCreate$outboundSchema` instead. */
    CustomerSessionCustomerIDCreate$.outboundSchema = CustomerSessionCustomerIDCreate$outboundSchema;
})(CustomerSessionCustomerIDCreate$ || (CustomerSessionCustomerIDCreate$ = {}));
export function customerSessionCustomerIDCreateToJSON(customerSessionCustomerIDCreate) {
    return JSON.stringify(CustomerSessionCustomerIDCreate$outboundSchema.parse(customerSessionCustomerIDCreate));
}
export function customerSessionCustomerIDCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSessionCustomerIDCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSessionCustomerIDCreate' from JSON`);
}
