// @polar-sh/sdk/dist/esm/models/components/customercancellationreason.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerCancellationReason = {
    CustomerService: "customer_service",
    LowQuality: "low_quality",
    MissingFeatures: "missing_features",
    SwitchedService: "switched_service",
    TooComplex: "too_complex",
    TooExpensive: "too_expensive",
    Unused: "unused",
    Other: "other",
};
/** @internal */
export const CustomerCancellationReason$inboundSchema = z.nativeEnum(CustomerCancellationReason);
/** @internal */
export const CustomerCancellationReason$outboundSchema = CustomerCancellationReason$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerCancellationReason$;
(function (CustomerCancellationReason$) {
    /** @deprecated use `CustomerCancellationReason$inboundSchema` instead. */
    CustomerCancellationReason$.inboundSchema = CustomerCancellationReason$inboundSchema;
    /** @deprecated use `CustomerCancellationReason$outboundSchema` instead. */
    CustomerCancellationReason$.outboundSchema = CustomerCancellationReason$outboundSchema;
})(CustomerCancellationReason$ || (CustomerCancellationReason$ = {}));
