// @polar-sh/sdk/dist/esm/models/components/customerorderconfirmpayment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { PaymentProcessor$inboundSchema, PaymentProcessor$outboundSchema, } from "./paymentprocessor.js";
/** @internal */
export const CustomerOrderConfirmPayment$inboundSchema = z.object({
    confirmation_token_id: z.nullable(z.string()).optional(),
    payment_method_id: z.nullable(z.string()).optional(),
    payment_processor: PaymentProcessor$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "confirmation_token_id": "confirmationTokenId",
        "payment_method_id": "paymentMethodId",
        "payment_processor": "paymentProcessor",
    });
});
/** @internal */
export const CustomerOrderConfirmPayment$outboundSchema = z.object({
    confirmationTokenId: z.nullable(z.string()).optional(),
    paymentMethodId: z.nullable(z.string()).optional(),
    paymentProcessor: PaymentProcessor$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        confirmationTokenId: "confirmation_token_id",
        paymentMethodId: "payment_method_id",
        paymentProcessor: "payment_processor",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerOrderConfirmPayment$;
(function (CustomerOrderConfirmPayment$) {
    /** @deprecated use `CustomerOrderConfirmPayment$inboundSchema` instead. */
    CustomerOrderConfirmPayment$.inboundSchema = CustomerOrderConfirmPayment$inboundSchema;
    /** @deprecated use `CustomerOrderConfirmPayment$outboundSchema` instead. */
    CustomerOrderConfirmPayment$.outboundSchema = CustomerOrderConfirmPayment$outboundSchema;
})(CustomerOrderConfirmPayment$ || (CustomerOrderConfirmPayment$ = {}));
export function customerOrderConfirmPaymentToJSON(customerOrderConfirmPayment) {
    return JSON.stringify(CustomerOrderConfirmPayment$outboundSchema.parse(customerOrderConfirmPayment));
}
export function customerOrderConfirmPaymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerOrderConfirmPayment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrderConfirmPayment' from JSON`);
}
