// @polar-sh/sdk/dist/esm/models/components/customerordersortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerOrderSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Amount: "amount",
    MinusAmount: "-amount",
    NetAmount: "net_amount",
    MinusNetAmount: "-net_amount",
    Product: "product",
    MinusProduct: "-product",
    Subscription: "subscription",
    MinusSubscription: "-subscription",
};
/** @internal */
export const CustomerOrderSortProperty$inboundSchema = z.nativeEnum(CustomerOrderSortProperty);
/** @internal */
export const CustomerOrderSortProperty$outboundSchema = CustomerOrderSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerOrderSortProperty$;
(function (CustomerOrderSortProperty$) {
    /** @deprecated use `CustomerOrderSortProperty$inboundSchema` instead. */
    CustomerOrderSortProperty$.inboundSchema = CustomerOrderSortProperty$inboundSchema;
    /** @deprecated use `CustomerOrderSortProperty$outboundSchema` instead. */
    CustomerOrderSortProperty$.outboundSchema = CustomerOrderSortProperty$outboundSchema;
})(CustomerOrderSortProperty$ || (CustomerOrderSortProperty$ = {}));
