// @polar-sh/sdk/dist/esm/models/components/customerstatemeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerStateMeter$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    meter_id: z.string(),
    consumed_units: z.number(),
    credited_units: z.number().int(),
    balance: z.number(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "meter_id": "meterId",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
    });
});
/** @internal */
export const CustomerStateMeter$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    meterId: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.number().int(),
    balance: z.number(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        meterId: "meter_id",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerStateMeter$;
(function (CustomerStateMeter$) {
    /** @deprecated use `CustomerStateMeter$inboundSchema` instead. */
    CustomerStateMeter$.inboundSchema = CustomerStateMeter$inboundSchema;
    /** @deprecated use `CustomerStateMeter$outboundSchema` instead. */
    CustomerStateMeter$.outboundSchema = CustomerStateMeter$outboundSchema;
})(CustomerStateMeter$ || (CustomerStateMeter$ = {}));
export function customerStateMeterToJSON(customerStateMeter) {
    return JSON.stringify(CustomerStateMeter$outboundSchema.parse(customerStateMeter));
}
export function customerStateMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateMeter' from JSON`);
}
