// @polar-sh/sdk/dist/esm/models/components/customfieldsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomFieldSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Slug: "slug",
    MinusSlug: "-slug",
    Name: "name",
    MinusName: "-name",
    Type: "type",
    MinusType: "-type",
};
/** @internal */
export const CustomFieldSortProperty$inboundSchema = z.nativeEnum(CustomFieldSortProperty);
/** @internal */
export const CustomFieldSortProperty$outboundSchema = CustomFieldSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldSortProperty$;
(function (CustomFieldSortProperty$) {
    /** @deprecated use `CustomFieldSortProperty$inboundSchema` instead. */
    CustomFieldSortProperty$.inboundSchema = CustomFieldSortProperty$inboundSchema;
    /** @deprecated use `CustomFieldSortProperty$outboundSchema` instead. */
    CustomFieldSortProperty$.outboundSchema = CustomFieldSortProperty$outboundSchema;
})(CustomFieldSortProperty$ || (CustomFieldSortProperty$ = {}));
