// @polar-sh/sdk/dist/esm/models/components/customercustomermetersortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerCustomerMeterSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    ModifiedAt: "modified_at",
    MinusModifiedAt: "-modified_at",
    MeterId: "meter_id",
    MinusMeterId: "-meter_id",
    MeterName: "meter_name",
    MinusMeterName: "-meter_name",
    ConsumedUnits: "consumed_units",
    MinusConsumedUnits: "-consumed_units",
    CreditedUnits: "credited_units",
    MinusCreditedUnits: "-credited_units",
    Balance: "balance",
    MinusBalance: "-balance",
};
/** @internal */
export const CustomerCustomerMeterSortProperty$inboundSchema = z.nativeEnum(CustomerCustomerMeterSortProperty);
/** @internal */
export const CustomerCustomerMeterSortProperty$outboundSchema = CustomerCustomerMeterSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerCustomerMeterSortProperty$;
(function (CustomerCustomerMeterSortProperty$) {
    /** @deprecated use `CustomerCustomerMeterSortProperty$inboundSchema` instead. */
    CustomerCustomerMeterSortProperty$.inboundSchema = CustomerCustomerMeterSortProperty$inboundSchema;
    /** @deprecated use `CustomerCustomerMeterSortProperty$outboundSchema` instead. */
    CustomerCustomerMeterSortProperty$.outboundSchema = CustomerCustomerMeterSortProperty$outboundSchema;
})(CustomerCustomerMeterSortProperty$ || (CustomerCustomerMeterSortProperty$ = {}));
