// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantdiscordupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerBenefitGrantDiscordPropertiesUpdate$inboundSchema, CustomerBenefitGrantDiscordPropertiesUpdate$outboundSchema, } from "./customerbenefitgrantdiscordpropertiesupdate.js";
/** @internal */
export const CustomerBenefitGrantDiscordUpdate$inboundSchema = z.object({
    benefit_type: z.literal("discord"),
    properties: CustomerBenefitGrantDiscordPropertiesUpdate$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "benefit_type": "benefitType",
    });
});
/** @internal */
export const CustomerBenefitGrantDiscordUpdate$outboundSchema = z.object({
    benefitType: z.literal("discord"),
    properties: CustomerBenefitGrantDiscordPropertiesUpdate$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        benefitType: "benefit_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantDiscordUpdate$;
(function (CustomerBenefitGrantDiscordUpdate$) {
    /** @deprecated use `CustomerBenefitGrantDiscordUpdate$inboundSchema` instead. */
    CustomerBenefitGrantDiscordUpdate$.inboundSchema = CustomerBenefitGrantDiscordUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantDiscordUpdate$outboundSchema` instead. */
    CustomerBenefitGrantDiscordUpdate$.outboundSchema = CustomerBenefitGrantDiscordUpdate$outboundSchema;
})(CustomerBenefitGrantDiscordUpdate$ || (CustomerBenefitGrantDiscordUpdate$ = {}));
export function customerBenefitGrantDiscordUpdateToJSON(customerBenefitGrantDiscordUpdate) {
    return JSON.stringify(CustomerBenefitGrantDiscordUpdate$outboundSchema.parse(customerBenefitGrantDiscordUpdate));
}
export function customerBenefitGrantDiscordUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantDiscordUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantDiscordUpdate' from JSON`);
}
