// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantdiscord.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDiscordSubscriber$inboundSchema, BenefitDiscordSubscriber$outboundSchema, } from "./benefitdiscordsubscriber.js";
import { BenefitGrantDiscordProperties$inboundSchema, BenefitGrantDiscordProperties$outboundSchema, } from "./benefitgrantdiscordproperties.js";
import { CustomerPortalCustomer$inboundSchema, CustomerPortalCustomer$outboundSchema, } from "./customerportalcustomer.js";
/** @internal */
export const CustomerBenefitGrantDiscord$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    benefit_id: z.string(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    is_granted: z.boolean(),
    is_revoked: z.boolean(),
    customer: CustomerPortalCustomer$inboundSchema,
    benefit: BenefitDiscordSubscriber$inboundSchema,
    properties: BenefitGrantDiscordProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "revoked_at": "revokedAt",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "is_granted": "isGranted",
        "is_revoked": "isRevoked",
    });
});
/** @internal */
export const CustomerBenefitGrantDiscord$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    benefitId: z.string(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    isGranted: z.boolean(),
    isRevoked: z.boolean(),
    customer: CustomerPortalCustomer$outboundSchema,
    benefit: BenefitDiscordSubscriber$outboundSchema,
    properties: BenefitGrantDiscordProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        revokedAt: "revoked_at",
        customerId: "customer_id",
        benefitId: "benefit_id",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        isGranted: "is_granted",
        isRevoked: "is_revoked",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantDiscord$;
(function (CustomerBenefitGrantDiscord$) {
    /** @deprecated use `CustomerBenefitGrantDiscord$inboundSchema` instead. */
    CustomerBenefitGrantDiscord$.inboundSchema = CustomerBenefitGrantDiscord$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantDiscord$outboundSchema` instead. */
    CustomerBenefitGrantDiscord$.outboundSchema = CustomerBenefitGrantDiscord$outboundSchema;
})(CustomerBenefitGrantDiscord$ || (CustomerBenefitGrantDiscord$ = {}));
export function customerBenefitGrantDiscordToJSON(customerBenefitGrantDiscord) {
    return JSON.stringify(CustomerBenefitGrantDiscord$outboundSchema.parse(customerBenefitGrantDiscord));
}
export function customerBenefitGrantDiscordFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantDiscord$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantDiscord' from JSON`);
}
