// @polar-sh/sdk/dist/esm/models/components/customermetersortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerMeterSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    ModifiedAt: "modified_at",
    MinusModifiedAt: "-modified_at",
    CustomerId: "customer_id",
    MinusCustomerId: "-customer_id",
    CustomerName: "customer_name",
    MinusCustomerName: "-customer_name",
    MeterId: "meter_id",
    MinusMeterId: "-meter_id",
    MeterName: "meter_name",
    MinusMeterName: "-meter_name",
    ConsumedUnits: "consumed_units",
    MinusConsumedUnits: "-consumed_units",
    CreditedUnits: "credited_units",
    MinusCreditedUnits: "-credited_units",
    Balance: "balance",
    MinusBalance: "-balance",
};
/** @internal */
export const CustomerMeterSortProperty$inboundSchema = z.nativeEnum(CustomerMeterSortProperty);
/** @internal */
export const CustomerMeterSortProperty$outboundSchema = CustomerMeterSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerMeterSortProperty$;
(function (CustomerMeterSortProperty$) {
    /** @deprecated use `CustomerMeterSortProperty$inboundSchema` instead. */
    CustomerMeterSortProperty$.inboundSchema = CustomerMeterSortProperty$inboundSchema;
    /** @deprecated use `CustomerMeterSortProperty$outboundSchema` instead. */
    CustomerMeterSortProperty$.outboundSchema = CustomerMeterSortProperty$outboundSchema;
})(CustomerMeterSortProperty$ || (CustomerMeterSortProperty$ = {}));
