// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantdiscordpropertiesupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerBenefitGrantDiscordPropertiesUpdate$inboundSchema = z.object({
    account_id: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "account_id": "accountId",
    });
});
/** @internal */
export const CustomerBenefitGrantDiscordPropertiesUpdate$outboundSchema = z.object({
    accountId: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        accountId: "account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantDiscordPropertiesUpdate$;
(function (CustomerBenefitGrantDiscordPropertiesUpdate$) {
    /** @deprecated use `CustomerBenefitGrantDiscordPropertiesUpdate$inboundSchema` instead. */
    CustomerBenefitGrantDiscordPropertiesUpdate$.inboundSchema = CustomerBenefitGrantDiscordPropertiesUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantDiscordPropertiesUpdate$outboundSchema` instead. */
    CustomerBenefitGrantDiscordPropertiesUpdate$.outboundSchema = CustomerBenefitGrantDiscordPropertiesUpdate$outboundSchema;
})(CustomerBenefitGrantDiscordPropertiesUpdate$ || (CustomerBenefitGrantDiscordPropertiesUpdate$ = {}));
export function customerBenefitGrantDiscordPropertiesUpdateToJSON(customerBenefitGrantDiscordPropertiesUpdate) {
    return JSON.stringify(CustomerBenefitGrantDiscordPropertiesUpdate$outboundSchema.parse(customerBenefitGrantDiscordPropertiesUpdate));
}
export function customerBenefitGrantDiscordPropertiesUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantDiscordPropertiesUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantDiscordPropertiesUpdate' from JSON`);
}
