// @polar-sh/sdk/dist/esm/models/components/customersubscriptionmetermeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSubscriptionMeterMeter$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
    });
});
/** @internal */
export const CustomerSubscriptionMeterMeter$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSubscriptionMeterMeter$;
(function (CustomerSubscriptionMeterMeter$) {
    /** @deprecated use `CustomerSubscriptionMeterMeter$inboundSchema` instead. */
    CustomerSubscriptionMeterMeter$.inboundSchema = CustomerSubscriptionMeterMeter$inboundSchema;
    /** @deprecated use `CustomerSubscriptionMeterMeter$outboundSchema` instead. */
    CustomerSubscriptionMeterMeter$.outboundSchema = CustomerSubscriptionMeterMeter$outboundSchema;
})(CustomerSubscriptionMeterMeter$ || (CustomerSubscriptionMeterMeter$ = {}));
export function customerSubscriptionMeterMeterToJSON(customerSubscriptionMeterMeter) {
    return JSON.stringify(CustomerSubscriptionMeterMeter$outboundSchema.parse(customerSubscriptionMeterMeter));
}
export function customerSubscriptionMeterMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionMeterMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionMeterMeter' from JSON`);
}
