// @polar-sh/sdk/dist/esm/models/components/customfieldselectproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomFieldSelectOption$inboundSchema, CustomFieldSelectOption$outboundSchema, } from "./customfieldselectoption.js";
/** @internal */
export const CustomFieldSelectProperties$inboundSchema = z.object({
    form_label: z.string().optional(),
    form_help_text: z.string().optional(),
    form_placeholder: z.string().optional(),
    options: z.array(CustomFieldSelectOption$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
});
/** @internal */
export const CustomFieldSelectProperties$outboundSchema = z.object({
    formLabel: z.string().optional(),
    formHelpText: z.string().optional(),
    formPlaceholder: z.string().optional(),
    options: z.array(CustomFieldSelectOption$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldSelectProperties$;
(function (CustomFieldSelectProperties$) {
    /** @deprecated use `CustomFieldSelectProperties$inboundSchema` instead. */
    CustomFieldSelectProperties$.inboundSchema = CustomFieldSelectProperties$inboundSchema;
    /** @deprecated use `CustomFieldSelectProperties$outboundSchema` instead. */
    CustomFieldSelectProperties$.outboundSchema = CustomFieldSelectProperties$outboundSchema;
})(CustomFieldSelectProperties$ || (CustomFieldSelectProperties$ = {}));
export function customFieldSelectPropertiesToJSON(customFieldSelectProperties) {
    return JSON.stringify(CustomFieldSelectProperties$outboundSchema.parse(customFieldSelectProperties));
}
export function customFieldSelectPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldSelectProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldSelectProperties' from JSON`);
}
