// @polar-sh/sdk/dist/esm/models/components/customersubscriptionproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitPublic$inboundSchema, BenefitPublic$outboundSchema, } from "./benefitpublic.js";
import { LegacyRecurringProductPrice$inboundSchema, LegacyRecurringProductPrice$outboundSchema, } from "./legacyrecurringproductprice.js";
import { Organization$inboundSchema, Organization$outboundSchema, } from "./organization.js";
import { ProductMediaFileRead$inboundSchema, ProductMediaFileRead$outboundSchema, } from "./productmediafileread.js";
import { ProductPrice$inboundSchema, ProductPrice$outboundSchema, } from "./productprice.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const CustomerSubscriptionProductPrices$inboundSchema = z.union([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const CustomerSubscriptionProductPrices$outboundSchema = z.union([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSubscriptionProductPrices$;
(function (CustomerSubscriptionProductPrices$) {
    /** @deprecated use `CustomerSubscriptionProductPrices$inboundSchema` instead. */
    CustomerSubscriptionProductPrices$.inboundSchema = CustomerSubscriptionProductPrices$inboundSchema;
    /** @deprecated use `CustomerSubscriptionProductPrices$outboundSchema` instead. */
    CustomerSubscriptionProductPrices$.outboundSchema = CustomerSubscriptionProductPrices$outboundSchema;
})(CustomerSubscriptionProductPrices$ || (CustomerSubscriptionProductPrices$ = {}));
export function customerSubscriptionProductPricesToJSON(customerSubscriptionProductPrices) {
    return JSON.stringify(CustomerSubscriptionProductPrices$outboundSchema.parse(customerSubscriptionProductPrices));
}
export function customerSubscriptionProductPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionProductPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionProductPrices' from JSON`);
}
/** @internal */
export const CustomerSubscriptionProduct$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    benefits: z.array(BenefitPublic$inboundSchema),
    medias: z.array(ProductMediaFileRead$inboundSchema),
    organization: Organization$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "recurring_interval": "recurringInterval",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const CustomerSubscriptionProduct$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(z.union([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ])),
    benefits: z.array(BenefitPublic$outboundSchema),
    medias: z.array(ProductMediaFileRead$outboundSchema),
    organization: Organization$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        recurringInterval: "recurring_interval",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSubscriptionProduct$;
(function (CustomerSubscriptionProduct$) {
    /** @deprecated use `CustomerSubscriptionProduct$inboundSchema` instead. */
    CustomerSubscriptionProduct$.inboundSchema = CustomerSubscriptionProduct$inboundSchema;
    /** @deprecated use `CustomerSubscriptionProduct$outboundSchema` instead. */
    CustomerSubscriptionProduct$.outboundSchema = CustomerSubscriptionProduct$outboundSchema;
})(CustomerSubscriptionProduct$ || (CustomerSubscriptionProduct$ = {}));
export function customerSubscriptionProductToJSON(customerSubscriptionProduct) {
    return JSON.stringify(CustomerSubscriptionProduct$outboundSchema.parse(customerSubscriptionProduct));
}
export function customerSubscriptionProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionProduct' from JSON`);
}
