// @polar-sh/sdk/dist/esm/models/components/discountfixedonceforeverdurationcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const DiscountFixedOnceForeverDurationCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const DiscountFixedOnceForeverDurationCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountFixedOnceForeverDurationCreateMetadata$;
(function (DiscountFixedOnceForeverDurationCreateMetadata$) {
    /** @deprecated use `DiscountFixedOnceForeverDurationCreateMetadata$inboundSchema` instead. */
    DiscountFixedOnceForeverDurationCreateMetadata$.inboundSchema = DiscountFixedOnceForeverDurationCreateMetadata$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDurationCreateMetadata$outboundSchema` instead. */
    DiscountFixedOnceForeverDurationCreateMetadata$.outboundSchema = DiscountFixedOnceForeverDurationCreateMetadata$outboundSchema;
})(DiscountFixedOnceForeverDurationCreateMetadata$ || (DiscountFixedOnceForeverDurationCreateMetadata$ = {}));
export function discountFixedOnceForeverDurationCreateMetadataToJSON(discountFixedOnceForeverDurationCreateMetadata) {
    return JSON.stringify(DiscountFixedOnceForeverDurationCreateMetadata$outboundSchema.parse(discountFixedOnceForeverDurationCreateMetadata));
}
export function discountFixedOnceForeverDurationCreateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountFixedOnceForeverDurationCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDurationCreateMetadata' from JSON`);
}
/** @internal */
export const DiscountFixedOnceForeverDurationCreate$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string().default("usd"),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    code: z.nullable(z.string()).optional(),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    max_redemptions: z.nullable(z.number().int()).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const DiscountFixedOnceForeverDurationCreate$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    type: DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string().default("usd"),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    code: z.nullable(z.string()).optional(),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    maxRedemptions: z.nullable(z.number().int()).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountFixedOnceForeverDurationCreate$;
(function (DiscountFixedOnceForeverDurationCreate$) {
    /** @deprecated use `DiscountFixedOnceForeverDurationCreate$inboundSchema` instead. */
    DiscountFixedOnceForeverDurationCreate$.inboundSchema = DiscountFixedOnceForeverDurationCreate$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDurationCreate$outboundSchema` instead. */
    DiscountFixedOnceForeverDurationCreate$.outboundSchema = DiscountFixedOnceForeverDurationCreate$outboundSchema;
})(DiscountFixedOnceForeverDurationCreate$ || (DiscountFixedOnceForeverDurationCreate$ = {}));
export function discountFixedOnceForeverDurationCreateToJSON(discountFixedOnceForeverDurationCreate) {
    return JSON.stringify(DiscountFixedOnceForeverDurationCreate$outboundSchema.parse(discountFixedOnceForeverDurationCreate));
}
export function discountFixedOnceForeverDurationCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountFixedOnceForeverDurationCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDurationCreate' from JSON`);
}
