// @polar-sh/sdk/dist/esm/models/components/eventnamessortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const EventNamesSortProperty = {
    Name: "name",
    MinusName: "-name",
    Occurrences: "occurrences",
    MinusOccurrences: "-occurrences",
    FirstSeen: "first_seen",
    MinusFirstSeen: "-first_seen",
    LastSeen: "last_seen",
    MinusLastSeen: "-last_seen",
};
/** @internal */
export const EventNamesSortProperty$inboundSchema = z.nativeEnum(EventNamesSortProperty);
/** @internal */
export const EventNamesSortProperty$outboundSchema = EventNamesSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var EventNamesSortProperty$;
(function (EventNamesSortProperty$) {
    /** @deprecated use `EventNamesSortProperty$inboundSchema` instead. */
    EventNamesSortProperty$.inboundSchema = EventNamesSortProperty$inboundSchema;
    /** @deprecated use `EventNamesSortProperty$outboundSchema` instead. */
    EventNamesSortProperty$.outboundSchema = EventNamesSortProperty$outboundSchema;
})(EventNamesSortProperty$ || (EventNamesSortProperty$ = {}));
