// @polar-sh/sdk/dist/esm/models/components/legacyrecurringproductpricecustom.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const LegacyRecurringProductPriceCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("custom"),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: z.literal("recurring"),
    recurring_interval: SubscriptionRecurringInterval$inboundSchema,
    price_currency: z.string(),
    minimum_amount: z.nullable(z.number().int()),
    maximum_amount: z.nullable(z.number().int()),
    preset_amount: z.nullable(z.number().int()),
    legacy: z.literal(true),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
export const LegacyRecurringProductPriceCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("custom"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: z.literal("recurring"),
    recurringInterval: SubscriptionRecurringInterval$outboundSchema,
    priceCurrency: z.string(),
    minimumAmount: z.nullable(z.number().int()),
    maximumAmount: z.nullable(z.number().int()),
    presetAmount: z.nullable(z.number().int()),
    legacy: z.literal(true),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var LegacyRecurringProductPriceCustom$;
(function (LegacyRecurringProductPriceCustom$) {
    /** @deprecated use `LegacyRecurringProductPriceCustom$inboundSchema` instead. */
    LegacyRecurringProductPriceCustom$.inboundSchema = LegacyRecurringProductPriceCustom$inboundSchema;
    /** @deprecated use `LegacyRecurringProductPriceCustom$outboundSchema` instead. */
    LegacyRecurringProductPriceCustom$.outboundSchema = LegacyRecurringProductPriceCustom$outboundSchema;
})(LegacyRecurringProductPriceCustom$ || (LegacyRecurringProductPriceCustom$ = {}));
export function legacyRecurringProductPriceCustomToJSON(legacyRecurringProductPriceCustom) {
    return JSON.stringify(LegacyRecurringProductPriceCustom$outboundSchema.parse(legacyRecurringProductPriceCustom));
}
export function legacyRecurringProductPriceCustomFromJSON(jsonString) {
    return safeParse(jsonString, (x) => LegacyRecurringProductPriceCustom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LegacyRecurringProductPriceCustom' from JSON`);
}
