// @polar-sh/sdk/dist/esm/models/components/listresourceorder.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceOrder$inboundSchema = z.object({
    items: z.array(Order$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceOrder$outboundSchema = z.object({
    items: z.array(Order$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceOrder$;
(function (ListResourceOrder$) {
    /** @deprecated use `ListResourceOrder$inboundSchema` instead. */
    ListResourceOrder$.inboundSchema = ListResourceOrder$inboundSchema;
    /** @deprecated use `ListResourceOrder$outboundSchema` instead. */
    ListResourceOrder$.outboundSchema = ListResourceOrder$outboundSchema;
})(ListResourceOrder$ || (ListResourceOrder$ = {}));
export function listResourceOrderToJSON(listResourceOrder) {
    return JSON.stringify(ListResourceOrder$outboundSchema.parse(listResourceOrder));
}
export function listResourceOrderFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceOrder$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceOrder' from JSON`);
}
