// @polar-sh/sdk/dist/esm/models/components/listresourcebenefitgrant.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrant$inboundSchema, BenefitGrant$outboundSchema, } from "./benefitgrant.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceBenefitGrant$inboundSchema = z.object({
    items: z.array(BenefitGrant$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceBenefitGrant$outboundSchema = z.object({
    items: z.array(BenefitGrant$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceBenefitGrant$;
(function (ListResourceBenefitGrant$) {
    /** @deprecated use `ListResourceBenefitGrant$inboundSchema` instead. */
    ListResourceBenefitGrant$.inboundSchema = ListResourceBenefitGrant$inboundSchema;
    /** @deprecated use `ListResourceBenefitGrant$outboundSchema` instead. */
    ListResourceBenefitGrant$.outboundSchema = ListResourceBenefitGrant$outboundSchema;
})(ListResourceBenefitGrant$ || (ListResourceBenefitGrant$ = {}));
export function listResourceBenefitGrantToJSON(listResourceBenefitGrant) {
    return JSON.stringify(ListResourceBenefitGrant$outboundSchema.parse(listResourceBenefitGrant));
}
export function listResourceBenefitGrantFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceBenefitGrant$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceBenefitGrant' from JSON`);
}
