// @polar-sh/sdk/dist/esm/models/components/listresourcecheckoutlink.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CheckoutLink$inboundSchema, CheckoutLink$outboundSchema, } from "./checkoutlink.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCheckoutLink$inboundSchema = z.object({
    items: z.array(CheckoutLink$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCheckoutLink$outboundSchema = z.object({
    items: z.array(CheckoutLink$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCheckoutLink$;
(function (ListResourceCheckoutLink$) {
    /** @deprecated use `ListResourceCheckoutLink$inboundSchema` instead. */
    ListResourceCheckoutLink$.inboundSchema = ListResourceCheckoutLink$inboundSchema;
    /** @deprecated use `ListResourceCheckoutLink$outboundSchema` instead. */
    ListResourceCheckoutLink$.outboundSchema = ListResourceCheckoutLink$outboundSchema;
})(ListResourceCheckoutLink$ || (ListResourceCheckoutLink$ = {}));
export function listResourceCheckoutLinkToJSON(listResourceCheckoutLink) {
    return JSON.stringify(ListResourceCheckoutLink$outboundSchema.parse(listResourceCheckoutLink));
}
export function listResourceCheckoutLinkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCheckoutLink$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCheckoutLink' from JSON`);
}
