// @polar-sh/sdk/dist/esm/models/components/listresourcebenefit.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Benefit$inboundSchema, Benefit$outboundSchema, } from "./benefit.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceBenefit$inboundSchema = z.object({
    items: z.array(Benefit$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceBenefit$outboundSchema = z.object({
    items: z.array(Benefit$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceBenefit$;
(function (ListResourceBenefit$) {
    /** @deprecated use `ListResourceBenefit$inboundSchema` instead. */
    ListResourceBenefit$.inboundSchema = ListResourceBenefit$inboundSchema;
    /** @deprecated use `ListResourceBenefit$outboundSchema` instead. */
    ListResourceBenefit$.outboundSchema = ListResourceBenefit$outboundSchema;
})(ListResourceBenefit$ || (ListResourceBenefit$ = {}));
export function listResourceBenefitToJSON(listResourceBenefit) {
    return JSON.stringify(ListResourceBenefit$outboundSchema.parse(listResourceBenefit));
}
export function listResourceBenefitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceBenefit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceBenefit' from JSON`);
}
