// @polar-sh/sdk/dist/esm/models/components/listresourcecustomerbenefitgrant.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerBenefitGrant$inboundSchema, CustomerBenefitGrant$outboundSchema, } from "./customerbenefitgrant.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomerBenefitGrant$inboundSchema = z.object({
    items: z.array(CustomerBenefitGrant$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomerBenefitGrant$outboundSchema = z.object({
    items: z.array(CustomerBenefitGrant$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomerBenefitGrant$;
(function (ListResourceCustomerBenefitGrant$) {
    /** @deprecated use `ListResourceCustomerBenefitGrant$inboundSchema` instead. */
    ListResourceCustomerBenefitGrant$.inboundSchema = ListResourceCustomerBenefitGrant$inboundSchema;
    /** @deprecated use `ListResourceCustomerBenefitGrant$outboundSchema` instead. */
    ListResourceCustomerBenefitGrant$.outboundSchema = ListResourceCustomerBenefitGrant$outboundSchema;
})(ListResourceCustomerBenefitGrant$ || (ListResourceCustomerBenefitGrant$ = {}));
export function listResourceCustomerBenefitGrantToJSON(listResourceCustomerBenefitGrant) {
    return JSON.stringify(ListResourceCustomerBenefitGrant$outboundSchema.parse(listResourceCustomerBenefitGrant));
}
export function listResourceCustomerBenefitGrantFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomerBenefitGrant$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerBenefitGrant' from JSON`);
}
