// @polar-sh/sdk/dist/esm/models/components/listresourceproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
import { Product$inboundSchema, Product$outboundSchema, } from "./product.js";
/** @internal */
export const ListResourceProduct$inboundSchema = z.object({
    items: z.array(Product$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceProduct$outboundSchema = z.object({
    items: z.array(Product$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceProduct$;
(function (ListResourceProduct$) {
    /** @deprecated use `ListResourceProduct$inboundSchema` instead. */
    ListResourceProduct$.inboundSchema = ListResourceProduct$inboundSchema;
    /** @deprecated use `ListResourceProduct$outboundSchema` instead. */
    ListResourceProduct$.outboundSchema = ListResourceProduct$outboundSchema;
})(ListResourceProduct$ || (ListResourceProduct$ = {}));
export function listResourceProductToJSON(listResourceProduct) {
    return JSON.stringify(ListResourceProduct$outboundSchema.parse(listResourceProduct));
}
export function listResourceProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceProduct' from JSON`);
}
