// @polar-sh/sdk/dist/esm/models/components/listresourcecustomersubscription.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSubscription$inboundSchema, CustomerSubscription$outboundSchema, } from "./customersubscription.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomerSubscription$inboundSchema = z.object({
    items: z.array(CustomerSubscription$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomerSubscription$outboundSchema = z.object({
    items: z.array(CustomerSubscription$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomerSubscription$;
(function (ListResourceCustomerSubscription$) {
    /** @deprecated use `ListResourceCustomerSubscription$inboundSchema` instead. */
    ListResourceCustomerSubscription$.inboundSchema = ListResourceCustomerSubscription$inboundSchema;
    /** @deprecated use `ListResourceCustomerSubscription$outboundSchema` instead. */
    ListResourceCustomerSubscription$.outboundSchema = ListResourceCustomerSubscription$outboundSchema;
})(ListResourceCustomerSubscription$ || (ListResourceCustomerSubscription$ = {}));
export function listResourceCustomerSubscriptionToJSON(listResourceCustomerSubscription) {
    return JSON.stringify(ListResourceCustomerSubscription$outboundSchema.parse(listResourceCustomerSubscription));
}
export function listResourceCustomerSubscriptionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomerSubscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerSubscription' from JSON`);
}
