// @polar-sh/sdk/dist/esm/models/components/listresourcecustomermeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerMeter$inboundSchema, CustomerMeter$outboundSchema, } from "./customermeter.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomerMeter$inboundSchema = z.object({
    items: z.array(CustomerMeter$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomerMeter$outboundSchema = z.object({
    items: z.array(CustomerMeter$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomerMeter$;
(function (ListResourceCustomerMeter$) {
    /** @deprecated use `ListResourceCustomerMeter$inboundSchema` instead. */
    ListResourceCustomerMeter$.inboundSchema = ListResourceCustomerMeter$inboundSchema;
    /** @deprecated use `ListResourceCustomerMeter$outboundSchema` instead. */
    ListResourceCustomerMeter$.outboundSchema = ListResourceCustomerMeter$outboundSchema;
})(ListResourceCustomerMeter$ || (ListResourceCustomerMeter$ = {}));
export function listResourceCustomerMeterToJSON(listResourceCustomerMeter) {
    return JSON.stringify(ListResourceCustomerMeter$outboundSchema.parse(listResourceCustomerMeter));
}
export function listResourceCustomerMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomerMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerMeter' from JSON`);
}
