// @polar-sh/sdk/dist/esm/models/components/listresourcecustomfield.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomField$inboundSchema, CustomField$outboundSchema, } from "./customfield.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomField$inboundSchema = z.object({
    items: z.array(CustomField$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomField$outboundSchema = z.object({
    items: z.array(CustomField$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomField$;
(function (ListResourceCustomField$) {
    /** @deprecated use `ListResourceCustomField$inboundSchema` instead. */
    ListResourceCustomField$.inboundSchema = ListResourceCustomField$inboundSchema;
    /** @deprecated use `ListResourceCustomField$outboundSchema` instead. */
    ListResourceCustomField$.outboundSchema = ListResourceCustomField$outboundSchema;
})(ListResourceCustomField$ || (ListResourceCustomField$ = {}));
export function listResourceCustomFieldToJSON(listResourceCustomField) {
    return JSON.stringify(ListResourceCustomField$outboundSchema.parse(listResourceCustomField));
}
export function listResourceCustomFieldFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomField$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomField' from JSON`);
}
