// @polar-sh/sdk/dist/esm/models/components/listresourcecustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomer$inboundSchema = z.object({
    items: z.array(Customer$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomer$outboundSchema = z.object({
    items: z.array(Customer$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomer$;
(function (ListResourceCustomer$) {
    /** @deprecated use `ListResourceCustomer$inboundSchema` instead. */
    ListResourceCustomer$.inboundSchema = ListResourceCustomer$inboundSchema;
    /** @deprecated use `ListResourceCustomer$outboundSchema` instead. */
    ListResourceCustomer$.outboundSchema = ListResourceCustomer$outboundSchema;
})(ListResourceCustomer$ || (ListResourceCustomer$ = {}));
export function listResourceCustomerToJSON(listResourceCustomer) {
    return JSON.stringify(ListResourceCustomer$outboundSchema.parse(listResourceCustomer));
}
export function listResourceCustomerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomer' from JSON`);
}
