// @polar-sh/sdk/dist/esm/models/components/metercreditedmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const MeterCreditedMetadata$inboundSchema = z.object({
    meter_id: z.string(),
    units: z.number().int(),
    rollover: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "meter_id": "meterId",
    });
});
/** @internal */
export const MeterCreditedMetadata$outboundSchema = z.object({
    meterId: z.string(),
    units: z.number().int(),
    rollover: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MeterCreditedMetadata$;
(function (MeterCreditedMetadata$) {
    /** @deprecated use `MeterCreditedMetadata$inboundSchema` instead. */
    MeterCreditedMetadata$.inboundSchema = MeterCreditedMetadata$inboundSchema;
    /** @deprecated use `MeterCreditedMetadata$outboundSchema` instead. */
    MeterCreditedMetadata$.outboundSchema = MeterCreditedMetadata$outboundSchema;
})(MeterCreditedMetadata$ || (MeterCreditedMetadata$ = {}));
export function meterCreditedMetadataToJSON(meterCreditedMetadata) {
    return JSON.stringify(MeterCreditedMetadata$outboundSchema.parse(meterCreditedMetadata));
}
export function meterCreditedMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MeterCreditedMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeterCreditedMetadata' from JSON`);
}
