// @polar-sh/sdk/dist/esm/models/components/meterresetevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { MeterResetMetadata$inboundSchema, MeterResetMetadata$outboundSchema, } from "./meterresetmetadata.js";
/** @internal */
export const MeterResetEvent$inboundSchema = z.object({
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("meter.reset"),
    metadata: MeterResetMetadata$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const MeterResetEvent$outboundSchema = z.object({
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("meter.reset"),
    metadata: MeterResetMetadata$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MeterResetEvent$;
(function (MeterResetEvent$) {
    /** @deprecated use `MeterResetEvent$inboundSchema` instead. */
    MeterResetEvent$.inboundSchema = MeterResetEvent$inboundSchema;
    /** @deprecated use `MeterResetEvent$outboundSchema` instead. */
    MeterResetEvent$.outboundSchema = MeterResetEvent$outboundSchema;
})(MeterResetEvent$ || (MeterResetEvent$ = {}));
export function meterResetEventToJSON(meterResetEvent) {
    return JSON.stringify(MeterResetEvent$outboundSchema.parse(meterResetEvent));
}
export function meterResetEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MeterResetEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeterResetEvent' from JSON`);
}
