// @polar-sh/sdk/dist/esm/models/components/ordersortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const OrderSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Status: "status",
    MinusStatus: "-status",
    InvoiceNumber: "invoice_number",
    MinusInvoiceNumber: "-invoice_number",
    Amount: "amount",
    MinusAmount: "-amount",
    NetAmount: "net_amount",
    MinusNetAmount: "-net_amount",
    Customer: "customer",
    MinusCustomer: "-customer",
    Product: "product",
    MinusProduct: "-product",
    Discount: "discount",
    MinusDiscount: "-discount",
    Subscription: "subscription",
    MinusSubscription: "-subscription",
};
/** @internal */
export const OrderSortProperty$inboundSchema = z.nativeEnum(OrderSortProperty);
/** @internal */
export const OrderSortProperty$outboundSchema = OrderSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrderSortProperty$;
(function (OrderSortProperty$) {
    /** @deprecated use `OrderSortProperty$inboundSchema` instead. */
    OrderSortProperty$.inboundSchema = OrderSortProperty$inboundSchema;
    /** @deprecated use `OrderSortProperty$outboundSchema` instead. */
    OrderSortProperty$.outboundSchema = OrderSortProperty$outboundSchema;
})(OrderSortProperty$ || (OrderSortProperty$ = {}));
