// @polar-sh/sdk/dist/esm/models/components/organizationsocialplatforms.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const OrganizationSocialPlatforms = {
    X: "x",
    Github: "github",
    Facebook: "facebook",
    Instagram: "instagram",
    Youtube: "youtube",
    Tiktok: "tiktok",
    Linkedin: "linkedin",
    Other: "other",
};
/** @internal */
export const OrganizationSocialPlatforms$inboundSchema = z.nativeEnum(OrganizationSocialPlatforms);
/** @internal */
export const OrganizationSocialPlatforms$outboundSchema = OrganizationSocialPlatforms$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrganizationSocialPlatforms$;
(function (OrganizationSocialPlatforms$) {
    /** @deprecated use `OrganizationSocialPlatforms$inboundSchema` instead. */
    OrganizationSocialPlatforms$.inboundSchema = OrganizationSocialPlatforms$inboundSchema;
    /** @deprecated use `OrganizationSocialPlatforms$outboundSchema` instead. */
    OrganizationSocialPlatforms$.outboundSchema = OrganizationSocialPlatforms$outboundSchema;
})(OrganizationSocialPlatforms$ || (OrganizationSocialPlatforms$ = {}));
