// @polar-sh/sdk/dist/esm/models/components/organizationavatarfileread.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrganizationAvatarFileRead$inboundSchema = z.object({
    id: z.string(),
    organization_id: z.string(),
    name: z.string(),
    path: z.string(),
    mime_type: z.string(),
    size: z.number().int(),
    storage_version: z.nullable(z.string()),
    checksum_etag: z.nullable(z.string()),
    checksum_sha256_base64: z.nullable(z.string()),
    checksum_sha256_hex: z.nullable(z.string()),
    last_modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    version: z.nullable(z.string()),
    service: z.literal("organization_avatar"),
    is_uploaded: z.boolean(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    size_readable: z.string(),
    public_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "mime_type": "mimeType",
        "storage_version": "storageVersion",
        "checksum_etag": "checksumEtag",
        "checksum_sha256_base64": "checksumSha256Base64",
        "checksum_sha256_hex": "checksumSha256Hex",
        "last_modified_at": "lastModifiedAt",
        "is_uploaded": "isUploaded",
        "created_at": "createdAt",
        "size_readable": "sizeReadable",
        "public_url": "publicUrl",
    });
});
/** @internal */
export const OrganizationAvatarFileRead$outboundSchema = z.object({
    id: z.string(),
    organizationId: z.string(),
    name: z.string(),
    path: z.string(),
    mimeType: z.string(),
    size: z.number().int(),
    storageVersion: z.nullable(z.string()),
    checksumEtag: z.nullable(z.string()),
    checksumSha256Base64: z.nullable(z.string()),
    checksumSha256Hex: z.nullable(z.string()),
    lastModifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    version: z.nullable(z.string()),
    service: z.literal("organization_avatar"),
    isUploaded: z.boolean(),
    createdAt: z.date().transform(v => v.toISOString()),
    sizeReadable: z.string(),
    publicUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        mimeType: "mime_type",
        storageVersion: "storage_version",
        checksumEtag: "checksum_etag",
        checksumSha256Base64: "checksum_sha256_base64",
        checksumSha256Hex: "checksum_sha256_hex",
        lastModifiedAt: "last_modified_at",
        isUploaded: "is_uploaded",
        createdAt: "created_at",
        sizeReadable: "size_readable",
        publicUrl: "public_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrganizationAvatarFileRead$;
(function (OrganizationAvatarFileRead$) {
    /** @deprecated use `OrganizationAvatarFileRead$inboundSchema` instead. */
    OrganizationAvatarFileRead$.inboundSchema = OrganizationAvatarFileRead$inboundSchema;
    /** @deprecated use `OrganizationAvatarFileRead$outboundSchema` instead. */
    OrganizationAvatarFileRead$.outboundSchema = OrganizationAvatarFileRead$outboundSchema;
})(OrganizationAvatarFileRead$ || (OrganizationAvatarFileRead$ = {}));
export function organizationAvatarFileReadToJSON(organizationAvatarFileRead) {
    return JSON.stringify(OrganizationAvatarFileRead$outboundSchema.parse(organizationAvatarFileRead));
}
export function organizationAvatarFileReadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationAvatarFileRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationAvatarFileRead' from JSON`);
}
