// @polar-sh/sdk/dist/esm/models/components/organizationupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { OrganizationDetails$inboundSchema, OrganizationDetails$outboundSchema, } from "./organizationdetails.js";
import { OrganizationFeatureSettings$inboundSchema, OrganizationFeatureSettings$outboundSchema, } from "./organizationfeaturesettings.js";
import { OrganizationNotificationSettings$inboundSchema, OrganizationNotificationSettings$outboundSchema, } from "./organizationnotificationsettings.js";
import { OrganizationSocialLink$inboundSchema, OrganizationSocialLink$outboundSchema, } from "./organizationsociallink.js";
import { OrganizationSubscriptionSettings$inboundSchema, OrganizationSubscriptionSettings$outboundSchema, } from "./organizationsubscriptionsettings.js";
/** @internal */
export const OrganizationUpdate$inboundSchema = z.object({
    name: z.nullable(z.string()).optional(),
    avatar_url: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    website: z.nullable(z.string()).optional(),
    socials: z.nullable(z.array(OrganizationSocialLink$inboundSchema)).optional(),
    details: z.nullable(OrganizationDetails$inboundSchema).optional(),
    feature_settings: z.nullable(OrganizationFeatureSettings$inboundSchema)
        .optional(),
    subscription_settings: z.nullable(OrganizationSubscriptionSettings$inboundSchema).optional(),
    notification_settings: z.nullable(OrganizationNotificationSettings$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "avatar_url": "avatarUrl",
        "feature_settings": "featureSettings",
        "subscription_settings": "subscriptionSettings",
        "notification_settings": "notificationSettings",
    });
});
/** @internal */
export const OrganizationUpdate$outboundSchema = z.object({
    name: z.nullable(z.string()).optional(),
    avatarUrl: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    website: z.nullable(z.string()).optional(),
    socials: z.nullable(z.array(OrganizationSocialLink$outboundSchema))
        .optional(),
    details: z.nullable(OrganizationDetails$outboundSchema).optional(),
    featureSettings: z.nullable(OrganizationFeatureSettings$outboundSchema)
        .optional(),
    subscriptionSettings: z.nullable(OrganizationSubscriptionSettings$outboundSchema).optional(),
    notificationSettings: z.nullable(OrganizationNotificationSettings$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        avatarUrl: "avatar_url",
        featureSettings: "feature_settings",
        subscriptionSettings: "subscription_settings",
        notificationSettings: "notification_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrganizationUpdate$;
(function (OrganizationUpdate$) {
    /** @deprecated use `OrganizationUpdate$inboundSchema` instead. */
    OrganizationUpdate$.inboundSchema = OrganizationUpdate$inboundSchema;
    /** @deprecated use `OrganizationUpdate$outboundSchema` instead. */
    OrganizationUpdate$.outboundSchema = OrganizationUpdate$outboundSchema;
})(OrganizationUpdate$ || (OrganizationUpdate$ = {}));
export function organizationUpdateToJSON(organizationUpdate) {
    return JSON.stringify(OrganizationUpdate$outboundSchema.parse(organizationUpdate));
}
export function organizationUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationUpdate' from JSON`);
}
