// @polar-sh/sdk/dist/esm/models/components/paymentsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const PaymentSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Status: "status",
    MinusStatus: "-status",
    Amount: "amount",
    MinusAmount: "-amount",
    Method: "method",
    MinusMethod: "-method",
};
/** @internal */
export const PaymentSortProperty$inboundSchema = z.nativeEnum(PaymentSortProperty);
/** @internal */
export const PaymentSortProperty$outboundSchema = PaymentSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PaymentSortProperty$;
(function (PaymentSortProperty$) {
    /** @deprecated use `PaymentSortProperty$inboundSchema` instead. */
    PaymentSortProperty$.inboundSchema = PaymentSortProperty$inboundSchema;
    /** @deprecated use `PaymentSortProperty$outboundSchema` instead. */
    PaymentSortProperty$.outboundSchema = PaymentSortProperty$outboundSchema;
})(PaymentSortProperty$ || (PaymentSortProperty$ = {}));
