// @polar-sh/sdk/dist/esm/models/components/productpricecustomcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ProductPriceCustomCreate$inboundSchema = z.object({
    amount_type: z.literal("custom"),
    price_currency: z.string().default("usd"),
    minimum_amount: z.nullable(z.number().int()).optional(),
    maximum_amount: z.nullable(z.number().int()).optional(),
    preset_amount: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return remap$(v, {
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
export const ProductPriceCustomCreate$outboundSchema = z.object({
    amountType: z.literal("custom"),
    priceCurrency: z.string().default("usd"),
    minimumAmount: z.nullable(z.number().int()).optional(),
    maximumAmount: z.nullable(z.number().int()).optional(),
    presetAmount: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceCustomCreate$;
(function (ProductPriceCustomCreate$) {
    /** @deprecated use `ProductPriceCustomCreate$inboundSchema` instead. */
    ProductPriceCustomCreate$.inboundSchema = ProductPriceCustomCreate$inboundSchema;
    /** @deprecated use `ProductPriceCustomCreate$outboundSchema` instead. */
    ProductPriceCustomCreate$.outboundSchema = ProductPriceCustomCreate$outboundSchema;
})(ProductPriceCustomCreate$ || (ProductPriceCustomCreate$ = {}));
export function productPriceCustomCreateToJSON(productPriceCustomCreate) {
    return JSON.stringify(ProductPriceCustomCreate$outboundSchema.parse(productPriceCustomCreate));
}
export function productPriceCustomCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceCustomCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceCustomCreate' from JSON`);
}
