// @polar-sh/sdk/dist/esm/models/components/productpricefixedcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ProductPriceFixedCreate$inboundSchema = z.object({
    amount_type: z.literal("fixed"),
    price_amount: z.number().int(),
    price_currency: z.string().default("usd"),
}).transform((v) => {
    return remap$(v, {
        "amount_type": "amountType",
        "price_amount": "priceAmount",
        "price_currency": "priceCurrency",
    });
});
/** @internal */
export const ProductPriceFixedCreate$outboundSchema = z.object({
    amountType: z.literal("fixed"),
    priceAmount: z.number().int(),
    priceCurrency: z.string().default("usd"),
}).transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceAmount: "price_amount",
        priceCurrency: "price_currency",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceFixedCreate$;
(function (ProductPriceFixedCreate$) {
    /** @deprecated use `ProductPriceFixedCreate$inboundSchema` instead. */
    ProductPriceFixedCreate$.inboundSchema = ProductPriceFixedCreate$inboundSchema;
    /** @deprecated use `ProductPriceFixedCreate$outboundSchema` instead. */
    ProductPriceFixedCreate$.outboundSchema = ProductPriceFixedCreate$outboundSchema;
})(ProductPriceFixedCreate$ || (ProductPriceFixedCreate$ = {}));
export function productPriceFixedCreateToJSON(productPriceFixedCreate) {
    return JSON.stringify(ProductPriceFixedCreate$outboundSchema.parse(productPriceFixedCreate));
}
export function productPriceFixedCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceFixedCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceFixedCreate' from JSON`);
}
