// @polar-sh/sdk/dist/esm/models/components/productpricefreecreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ProductPriceFreeCreate$inboundSchema = z.object({
    amount_type: z.literal("free"),
}).transform((v) => {
    return remap$(v, {
        "amount_type": "amountType",
    });
});
/** @internal */
export const ProductPriceFreeCreate$outboundSchema = z.object({
    amountType: z.literal("free"),
}).transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceFreeCreate$;
(function (ProductPriceFreeCreate$) {
    /** @deprecated use `ProductPriceFreeCreate$inboundSchema` instead. */
    ProductPriceFreeCreate$.inboundSchema = ProductPriceFreeCreate$inboundSchema;
    /** @deprecated use `ProductPriceFreeCreate$outboundSchema` instead. */
    ProductPriceFreeCreate$.outboundSchema = ProductPriceFreeCreate$outboundSchema;
})(ProductPriceFreeCreate$ || (ProductPriceFreeCreate$ = {}));
export function productPriceFreeCreateToJSON(productPriceFreeCreate) {
    return JSON.stringify(ProductPriceFreeCreate$outboundSchema.parse(productPriceFreeCreate));
}
export function productPriceFreeCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceFreeCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceFreeCreate' from JSON`);
}
