// @polar-sh/sdk/dist/esm/models/components/s3filecreatemultipart.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { S3FileCreatePart$inboundSchema, S3FileCreatePart$outboundSchema, } from "./s3filecreatepart.js";
/** @internal */
export const S3FileCreateMultipart$inboundSchema = z.object({
    parts: z.array(S3FileCreatePart$inboundSchema),
});
/** @internal */
export const S3FileCreateMultipart$outboundSchema = z.object({
    parts: z.array(S3FileCreatePart$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var S3FileCreateMultipart$;
(function (S3FileCreateMultipart$) {
    /** @deprecated use `S3FileCreateMultipart$inboundSchema` instead. */
    S3FileCreateMultipart$.inboundSchema = S3FileCreateMultipart$inboundSchema;
    /** @deprecated use `S3FileCreateMultipart$outboundSchema` instead. */
    S3FileCreateMultipart$.outboundSchema = S3FileCreateMultipart$outboundSchema;
})(S3FileCreateMultipart$ || (S3FileCreateMultipart$ = {}));
export function s3FileCreateMultipartToJSON(s3FileCreateMultipart) {
    return JSON.stringify(S3FileCreateMultipart$outboundSchema.parse(s3FileCreateMultipart));
}
export function s3FileCreateMultipartFromJSON(jsonString) {
    return safeParse(jsonString, (x) => S3FileCreateMultipart$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'S3FileCreateMultipart' from JSON`);
}
