// @polar-sh/sdk/dist/esm/models/components/subscriptionrevokedmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionRevokedMetadata$inboundSchema = z.object({
    subscription_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
export const SubscriptionRevokedMetadata$outboundSchema = z.object({
    subscriptionId: z.string(),
}).transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionRevokedMetadata$;
(function (SubscriptionRevokedMetadata$) {
    /** @deprecated use `SubscriptionRevokedMetadata$inboundSchema` instead. */
    SubscriptionRevokedMetadata$.inboundSchema = SubscriptionRevokedMetadata$inboundSchema;
    /** @deprecated use `SubscriptionRevokedMetadata$outboundSchema` instead. */
    SubscriptionRevokedMetadata$.outboundSchema = SubscriptionRevokedMetadata$outboundSchema;
})(SubscriptionRevokedMetadata$ || (SubscriptionRevokedMetadata$ = {}));
export function subscriptionRevokedMetadataToJSON(subscriptionRevokedMetadata) {
    return JSON.stringify(SubscriptionRevokedMetadata$outboundSchema.parse(subscriptionRevokedMetadata));
}
export function subscriptionRevokedMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionRevokedMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionRevokedMetadata' from JSON`);
}
