// @polar-sh/sdk/dist/esm/models/components/subscriptionproductupdatedevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { SubscriptionProductUpdatedMetadata$inboundSchema, SubscriptionProductUpdatedMetadata$outboundSchema, } from "./subscriptionproductupdatedmetadata.js";
/** @internal */
export const SubscriptionProductUpdatedEvent$inboundSchema = z.object({
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("subscription.product_updated"),
    metadata: SubscriptionProductUpdatedMetadata$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const SubscriptionProductUpdatedEvent$outboundSchema = z.object({
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("subscription.product_updated"),
    metadata: SubscriptionProductUpdatedMetadata$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionProductUpdatedEvent$;
(function (SubscriptionProductUpdatedEvent$) {
    /** @deprecated use `SubscriptionProductUpdatedEvent$inboundSchema` instead. */
    SubscriptionProductUpdatedEvent$.inboundSchema = SubscriptionProductUpdatedEvent$inboundSchema;
    /** @deprecated use `SubscriptionProductUpdatedEvent$outboundSchema` instead. */
    SubscriptionProductUpdatedEvent$.outboundSchema = SubscriptionProductUpdatedEvent$outboundSchema;
})(SubscriptionProductUpdatedEvent$ || (SubscriptionProductUpdatedEvent$ = {}));
export function subscriptionProductUpdatedEventToJSON(subscriptionProductUpdatedEvent) {
    return JSON.stringify(SubscriptionProductUpdatedEvent$outboundSchema.parse(subscriptionProductUpdatedEvent));
}
export function subscriptionProductUpdatedEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionProductUpdatedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionProductUpdatedEvent' from JSON`);
}
