// @polar-sh/sdk/dist/esm/models/components/subscriptionprorationbehavior.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const SubscriptionProrationBehavior = {
    Invoice: "invoice",
    Prorate: "prorate",
};
/** @internal */
export const SubscriptionProrationBehavior$inboundSchema = z.nativeEnum(SubscriptionProrationBehavior);
/** @internal */
export const SubscriptionProrationBehavior$outboundSchema = SubscriptionProrationBehavior$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionProrationBehavior$;
(function (SubscriptionProrationBehavior$) {
    /** @deprecated use `SubscriptionProrationBehavior$inboundSchema` instead. */
    SubscriptionProrationBehavior$.inboundSchema = SubscriptionProrationBehavior$inboundSchema;
    /** @deprecated use `SubscriptionProrationBehavior$outboundSchema` instead. */
    SubscriptionProrationBehavior$.outboundSchema = SubscriptionProrationBehavior$outboundSchema;
})(SubscriptionProrationBehavior$ || (SubscriptionProrationBehavior$ = {}));
