// @polar-sh/sdk/dist/esm/models/components/subscriptionrecurringinterval.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const SubscriptionRecurringInterval = {
    Day: "day",
    Week: "week",
    Month: "month",
    Year: "year",
};
/** @internal */
export const SubscriptionRecurringInterval$inboundSchema = z.nativeEnum(SubscriptionRecurringInterval);
/** @internal */
export const SubscriptionRecurringInterval$outboundSchema = SubscriptionRecurringInterval$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionRecurringInterval$;
(function (SubscriptionRecurringInterval$) {
    /** @deprecated use `SubscriptionRecurringInterval$inboundSchema` instead. */
    SubscriptionRecurringInterval$.inboundSchema = SubscriptionRecurringInterval$inboundSchema;
    /** @deprecated use `SubscriptionRecurringInterval$outboundSchema` instead. */
    SubscriptionRecurringInterval$.outboundSchema = SubscriptionRecurringInterval$outboundSchema;
})(SubscriptionRecurringInterval$ || (SubscriptionRecurringInterval$ = {}));
