// @polar-sh/sdk/dist/esm/models/components/subscriptionmeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Meter$inboundSchema, Meter$outboundSchema, } from "./meter.js";
/** @internal */
export const SubscriptionMeter$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    consumed_units: z.number(),
    credited_units: z.number().int(),
    amount: z.number().int(),
    meter_id: z.string(),
    meter: Meter$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
        "meter_id": "meterId",
    });
});
/** @internal */
export const SubscriptionMeter$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.number().int(),
    amount: z.number().int(),
    meterId: z.string(),
    meter: Meter$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionMeter$;
(function (SubscriptionMeter$) {
    /** @deprecated use `SubscriptionMeter$inboundSchema` instead. */
    SubscriptionMeter$.inboundSchema = SubscriptionMeter$inboundSchema;
    /** @deprecated use `SubscriptionMeter$outboundSchema` instead. */
    SubscriptionMeter$.outboundSchema = SubscriptionMeter$outboundSchema;
})(SubscriptionMeter$ || (SubscriptionMeter$ = {}));
export function subscriptionMeterToJSON(subscriptionMeter) {
    return JSON.stringify(SubscriptionMeter$outboundSchema.parse(subscriptionMeter));
}
export function subscriptionMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionMeter' from JSON`);
}
