// @polar-sh/sdk/dist/esm/models/components/subscriptioncycledmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionCycledMetadata$inboundSchema = z.object({
    subscription_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
export const SubscriptionCycledMetadata$outboundSchema = z.object({
    subscriptionId: z.string(),
}).transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionCycledMetadata$;
(function (SubscriptionCycledMetadata$) {
    /** @deprecated use `SubscriptionCycledMetadata$inboundSchema` instead. */
    SubscriptionCycledMetadata$.inboundSchema = SubscriptionCycledMetadata$inboundSchema;
    /** @deprecated use `SubscriptionCycledMetadata$outboundSchema` instead. */
    SubscriptionCycledMetadata$.outboundSchema = SubscriptionCycledMetadata$outboundSchema;
})(SubscriptionCycledMetadata$ || (SubscriptionCycledMetadata$ = {}));
export function subscriptionCycledMetadataToJSON(subscriptionCycledMetadata) {
    return JSON.stringify(SubscriptionCycledMetadata$outboundSchema.parse(subscriptionCycledMetadata));
}
export function subscriptionCycledMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCycledMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCycledMetadata' from JSON`);
}
