// @polar-sh/sdk/dist/esm/models/components/subscriptionupdateproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { SubscriptionProrationBehavior$inboundSchema, SubscriptionProrationBehavior$outboundSchema, } from "./subscriptionprorationbehavior.js";
/** @internal */
export const SubscriptionUpdateProduct$inboundSchema = z.object({
    product_id: z.string(),
    proration_behavior: z.nullable(SubscriptionProrationBehavior$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "product_id": "productId",
        "proration_behavior": "prorationBehavior",
    });
});
/** @internal */
export const SubscriptionUpdateProduct$outboundSchema = z.object({
    productId: z.string(),
    prorationBehavior: z.nullable(SubscriptionProrationBehavior$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        productId: "product_id",
        prorationBehavior: "proration_behavior",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionUpdateProduct$;
(function (SubscriptionUpdateProduct$) {
    /** @deprecated use `SubscriptionUpdateProduct$inboundSchema` instead. */
    SubscriptionUpdateProduct$.inboundSchema = SubscriptionUpdateProduct$inboundSchema;
    /** @deprecated use `SubscriptionUpdateProduct$outboundSchema` instead. */
    SubscriptionUpdateProduct$.outboundSchema = SubscriptionUpdateProduct$outboundSchema;
})(SubscriptionUpdateProduct$ || (SubscriptionUpdateProduct$ = {}));
export function subscriptionUpdateProductToJSON(subscriptionUpdateProduct) {
    return JSON.stringify(SubscriptionUpdateProduct$outboundSchema.parse(subscriptionUpdateProduct));
}
export function subscriptionUpdateProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionUpdateProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionUpdateProduct' from JSON`);
}
