// @polar-sh/sdk/dist/esm/models/components/subscriptionproductupdatedmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionProductUpdatedMetadata$inboundSchema = z.object({
    subscription_id: z.string(),
    old_product_id: z.string(),
    new_product_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
        "old_product_id": "oldProductId",
        "new_product_id": "newProductId",
    });
});
/** @internal */
export const SubscriptionProductUpdatedMetadata$outboundSchema = z.object({
    subscriptionId: z.string(),
    oldProductId: z.string(),
    newProductId: z.string(),
}).transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
        oldProductId: "old_product_id",
        newProductId: "new_product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionProductUpdatedMetadata$;
(function (SubscriptionProductUpdatedMetadata$) {
    /** @deprecated use `SubscriptionProductUpdatedMetadata$inboundSchema` instead. */
    SubscriptionProductUpdatedMetadata$.inboundSchema = SubscriptionProductUpdatedMetadata$inboundSchema;
    /** @deprecated use `SubscriptionProductUpdatedMetadata$outboundSchema` instead. */
    SubscriptionProductUpdatedMetadata$.outboundSchema = SubscriptionProductUpdatedMetadata$outboundSchema;
})(SubscriptionProductUpdatedMetadata$ || (SubscriptionProductUpdatedMetadata$ = {}));
export function subscriptionProductUpdatedMetadataToJSON(subscriptionProductUpdatedMetadata) {
    return JSON.stringify(SubscriptionProductUpdatedMetadata$outboundSchema.parse(subscriptionProductUpdatedMetadata));
}
export function subscriptionProductUpdatedMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionProductUpdatedMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionProductUpdatedMetadata' from JSON`);
}
