// @polar-sh/sdk/dist/esm/models/components/uniqueaggregation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const UniqueAggregation$inboundSchema = z.object({
    func: z.literal("unique").default("unique"),
    property: z.string(),
});
/** @internal */
export const UniqueAggregation$outboundSchema = z.object({
    func: z.literal("unique").default("unique"),
    property: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UniqueAggregation$;
(function (UniqueAggregation$) {
    /** @deprecated use `UniqueAggregation$inboundSchema` instead. */
    UniqueAggregation$.inboundSchema = UniqueAggregation$inboundSchema;
    /** @deprecated use `UniqueAggregation$outboundSchema` instead. */
    UniqueAggregation$.outboundSchema = UniqueAggregation$outboundSchema;
})(UniqueAggregation$ || (UniqueAggregation$ = {}));
export function uniqueAggregationToJSON(uniqueAggregation) {
    return JSON.stringify(UniqueAggregation$outboundSchema.parse(uniqueAggregation));
}
export function uniqueAggregationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UniqueAggregation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UniqueAggregation' from JSON`);
}
