// @polar-sh/sdk/dist/esm/models/components/webhookordercreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderCreatedPayload$inboundSchema = z.object({
    type: z.literal("order.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderCreatedPayload$outboundSchema = z.object({
    type: z.literal("order.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Order$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookOrderCreatedPayload$;
(function (WebhookOrderCreatedPayload$) {
    /** @deprecated use `WebhookOrderCreatedPayload$inboundSchema` instead. */
    WebhookOrderCreatedPayload$.inboundSchema = WebhookOrderCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrderCreatedPayload$outboundSchema` instead. */
    WebhookOrderCreatedPayload$.outboundSchema = WebhookOrderCreatedPayload$outboundSchema;
})(WebhookOrderCreatedPayload$ || (WebhookOrderCreatedPayload$ = {}));
export function webhookOrderCreatedPayloadToJSON(webhookOrderCreatedPayload) {
    return JSON.stringify(WebhookOrderCreatedPayload$outboundSchema.parse(webhookOrderCreatedPayload));
}
export function webhookOrderCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderCreatedPayload' from JSON`);
}
