// @polar-sh/sdk/dist/esm/models/components/webhookorderrefundedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderRefundedPayload$inboundSchema = z.object({
    type: z.literal("order.refunded"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderRefundedPayload$outboundSchema = z.object({
    type: z.literal("order.refunded"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Order$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookOrderRefundedPayload$;
(function (WebhookOrderRefundedPayload$) {
    /** @deprecated use `WebhookOrderRefundedPayload$inboundSchema` instead. */
    WebhookOrderRefundedPayload$.inboundSchema = WebhookOrderRefundedPayload$inboundSchema;
    /** @deprecated use `WebhookOrderRefundedPayload$outboundSchema` instead. */
    WebhookOrderRefundedPayload$.outboundSchema = WebhookOrderRefundedPayload$outboundSchema;
})(WebhookOrderRefundedPayload$ || (WebhookOrderRefundedPayload$ = {}));
export function webhookOrderRefundedPayloadToJSON(webhookOrderRefundedPayload) {
    return JSON.stringify(WebhookOrderRefundedPayload$outboundSchema.parse(webhookOrderRefundedPayload));
}
export function webhookOrderRefundedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderRefundedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderRefundedPayload' from JSON`);
}
