// @polar-sh/sdk/dist/esm/models/components/webhookorderupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderUpdatedPayload$inboundSchema = z.object({
    type: z.literal("order.updated"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderUpdatedPayload$outboundSchema = z.object({
    type: z.literal("order.updated"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Order$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookOrderUpdatedPayload$;
(function (WebhookOrderUpdatedPayload$) {
    /** @deprecated use `WebhookOrderUpdatedPayload$inboundSchema` instead. */
    WebhookOrderUpdatedPayload$.inboundSchema = WebhookOrderUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrderUpdatedPayload$outboundSchema` instead. */
    WebhookOrderUpdatedPayload$.outboundSchema = WebhookOrderUpdatedPayload$outboundSchema;
})(WebhookOrderUpdatedPayload$ || (WebhookOrderUpdatedPayload$ = {}));
export function webhookOrderUpdatedPayloadToJSON(webhookOrderUpdatedPayload) {
    return JSON.stringify(WebhookOrderUpdatedPayload$outboundSchema.parse(webhookOrderUpdatedPayload));
}
export function webhookOrderUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderUpdatedPayload' from JSON`);
}
