// @polar-sh/sdk/dist/esm/models/components/webhookbenefitgrantupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantWebhook$inboundSchema, BenefitGrantWebhook$outboundSchema, } from "./benefitgrantwebhook.js";
/** @internal */
export const WebhookBenefitGrantUpdatedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.updated"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: BenefitGrantWebhook$inboundSchema,
});
/** @internal */
export const WebhookBenefitGrantUpdatedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.updated"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookBenefitGrantUpdatedPayload$;
(function (WebhookBenefitGrantUpdatedPayload$) {
    /** @deprecated use `WebhookBenefitGrantUpdatedPayload$inboundSchema` instead. */
    WebhookBenefitGrantUpdatedPayload$.inboundSchema = WebhookBenefitGrantUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantUpdatedPayload$outboundSchema` instead. */
    WebhookBenefitGrantUpdatedPayload$.outboundSchema = WebhookBenefitGrantUpdatedPayload$outboundSchema;
})(WebhookBenefitGrantUpdatedPayload$ || (WebhookBenefitGrantUpdatedPayload$ = {}));
export function webhookBenefitGrantUpdatedPayloadToJSON(webhookBenefitGrantUpdatedPayload) {
    return JSON.stringify(WebhookBenefitGrantUpdatedPayload$outboundSchema.parse(webhookBenefitGrantUpdatedPayload));
}
export function webhookBenefitGrantUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitGrantUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantUpdatedPayload' from JSON`);
}
