// @polar-sh/sdk/dist/esm/models/components/webhookcheckoutcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Checkout$inboundSchema, Checkout$outboundSchema, } from "./checkout.js";
/** @internal */
export const WebhookCheckoutCreatedPayload$inboundSchema = z.object({
    type: z.literal("checkout.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Checkout$inboundSchema,
});
/** @internal */
export const WebhookCheckoutCreatedPayload$outboundSchema = z.object({
    type: z.literal("checkout.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Checkout$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookCheckoutCreatedPayload$;
(function (WebhookCheckoutCreatedPayload$) {
    /** @deprecated use `WebhookCheckoutCreatedPayload$inboundSchema` instead. */
    WebhookCheckoutCreatedPayload$.inboundSchema = WebhookCheckoutCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookCheckoutCreatedPayload$outboundSchema` instead. */
    WebhookCheckoutCreatedPayload$.outboundSchema = WebhookCheckoutCreatedPayload$outboundSchema;
})(WebhookCheckoutCreatedPayload$ || (WebhookCheckoutCreatedPayload$ = {}));
export function webhookCheckoutCreatedPayloadToJSON(webhookCheckoutCreatedPayload) {
    return JSON.stringify(WebhookCheckoutCreatedPayload$outboundSchema.parse(webhookCheckoutCreatedPayload));
}
export function webhookCheckoutCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCheckoutCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCheckoutCreatedPayload' from JSON`);
}
