// @polar-sh/sdk/dist/esm/models/components/webhookorderpaidpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderPaidPayload$inboundSchema = z.object({
    type: z.literal("order.paid"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderPaidPayload$outboundSchema = z.object({
    type: z.literal("order.paid"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Order$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookOrderPaidPayload$;
(function (WebhookOrderPaidPayload$) {
    /** @deprecated use `WebhookOrderPaidPayload$inboundSchema` instead. */
    WebhookOrderPaidPayload$.inboundSchema = WebhookOrderPaidPayload$inboundSchema;
    /** @deprecated use `WebhookOrderPaidPayload$outboundSchema` instead. */
    WebhookOrderPaidPayload$.outboundSchema = WebhookOrderPaidPayload$outboundSchema;
})(WebhookOrderPaidPayload$ || (WebhookOrderPaidPayload$ = {}));
export function webhookOrderPaidPayloadToJSON(webhookOrderPaidPayload) {
    return JSON.stringify(WebhookOrderPaidPayload$outboundSchema.parse(webhookOrderPaidPayload));
}
export function webhookOrderPaidPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderPaidPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderPaidPayload' from JSON`);
}
