// @polar-sh/sdk/dist/esm/models/components/webhookevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { WebhookEventType$inboundSchema, WebhookEventType$outboundSchema, } from "./webhookeventtype.js";
/** @internal */
export const WebhookEvent$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    last_http_code: z.nullable(z.number().int()).optional(),
    succeeded: z.nullable(z.boolean()).optional(),
    payload: z.nullable(z.string()),
    type: WebhookEventType$inboundSchema,
    is_archived: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "last_http_code": "lastHttpCode",
        "is_archived": "isArchived",
    });
});
/** @internal */
export const WebhookEvent$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    lastHttpCode: z.nullable(z.number().int()).optional(),
    succeeded: z.nullable(z.boolean()).optional(),
    payload: z.nullable(z.string()),
    type: WebhookEventType$outboundSchema,
    isArchived: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        lastHttpCode: "last_http_code",
        isArchived: "is_archived",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookEvent$;
(function (WebhookEvent$) {
    /** @deprecated use `WebhookEvent$inboundSchema` instead. */
    WebhookEvent$.inboundSchema = WebhookEvent$inboundSchema;
    /** @deprecated use `WebhookEvent$outboundSchema` instead. */
    WebhookEvent$.outboundSchema = WebhookEvent$outboundSchema;
})(WebhookEvent$ || (WebhookEvent$ = {}));
export function webhookEventToJSON(webhookEvent) {
    return JSON.stringify(WebhookEvent$outboundSchema.parse(webhookEvent));
}
export function webhookEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEvent' from JSON`);
}
