// @polar-sh/sdk/dist/esm/models/components/webhookcustomerdeletedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const WebhookCustomerDeletedPayload$inboundSchema = z.object({
    type: z.literal("customer.deleted"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Customer$inboundSchema,
});
/** @internal */
export const WebhookCustomerDeletedPayload$outboundSchema = z.object({
    type: z.literal("customer.deleted"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Customer$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookCustomerDeletedPayload$;
(function (WebhookCustomerDeletedPayload$) {
    /** @deprecated use `WebhookCustomerDeletedPayload$inboundSchema` instead. */
    WebhookCustomerDeletedPayload$.inboundSchema = WebhookCustomerDeletedPayload$inboundSchema;
    /** @deprecated use `WebhookCustomerDeletedPayload$outboundSchema` instead. */
    WebhookCustomerDeletedPayload$.outboundSchema = WebhookCustomerDeletedPayload$outboundSchema;
})(WebhookCustomerDeletedPayload$ || (WebhookCustomerDeletedPayload$ = {}));
export function webhookCustomerDeletedPayloadToJSON(webhookCustomerDeletedPayload) {
    return JSON.stringify(WebhookCustomerDeletedPayload$outboundSchema.parse(webhookCustomerDeletedPayload));
}
export function webhookCustomerDeletedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerDeletedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerDeletedPayload' from JSON`);
}
