// @polar-sh/sdk/dist/esm/models/components/webhookorganizationupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Organization$inboundSchema, Organization$outboundSchema, } from "./organization.js";
/** @internal */
export const WebhookOrganizationUpdatedPayload$inboundSchema = z.object({
    type: z.literal("organization.updated"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Organization$inboundSchema,
});
/** @internal */
export const WebhookOrganizationUpdatedPayload$outboundSchema = z.object({
    type: z.literal("organization.updated"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Organization$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookOrganizationUpdatedPayload$;
(function (WebhookOrganizationUpdatedPayload$) {
    /** @deprecated use `WebhookOrganizationUpdatedPayload$inboundSchema` instead. */
    WebhookOrganizationUpdatedPayload$.inboundSchema = WebhookOrganizationUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrganizationUpdatedPayload$outboundSchema` instead. */
    WebhookOrganizationUpdatedPayload$.outboundSchema = WebhookOrganizationUpdatedPayload$outboundSchema;
})(WebhookOrganizationUpdatedPayload$ || (WebhookOrganizationUpdatedPayload$ = {}));
export function webhookOrganizationUpdatedPayloadToJSON(webhookOrganizationUpdatedPayload) {
    return JSON.stringify(WebhookOrganizationUpdatedPayload$outboundSchema.parse(webhookOrganizationUpdatedPayload));
}
export function webhookOrganizationUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrganizationUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrganizationUpdatedPayload' from JSON`);
}
