// @polar-sh/sdk/dist/esm/models/components/webhookbenefitcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Benefit$inboundSchema, Benefit$outboundSchema, } from "./benefit.js";
/** @internal */
export const WebhookBenefitCreatedPayload$inboundSchema = z.object({
    type: z.literal("benefit.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Benefit$inboundSchema,
});
/** @internal */
export const WebhookBenefitCreatedPayload$outboundSchema = z.object({
    type: z.literal("benefit.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Benefit$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookBenefitCreatedPayload$;
(function (WebhookBenefitCreatedPayload$) {
    /** @deprecated use `WebhookBenefitCreatedPayload$inboundSchema` instead. */
    WebhookBenefitCreatedPayload$.inboundSchema = WebhookBenefitCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitCreatedPayload$outboundSchema` instead. */
    WebhookBenefitCreatedPayload$.outboundSchema = WebhookBenefitCreatedPayload$outboundSchema;
})(WebhookBenefitCreatedPayload$ || (WebhookBenefitCreatedPayload$ = {}));
export function webhookBenefitCreatedPayloadToJSON(webhookBenefitCreatedPayload) {
    return JSON.stringify(WebhookBenefitCreatedPayload$outboundSchema.parse(webhookBenefitCreatedPayload));
}
export function webhookBenefitCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitCreatedPayload' from JSON`);
}
