// @polar-sh/sdk/dist/esm/models/components/webhooksubscriptioncreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Subscription$inboundSchema, Subscription$outboundSchema, } from "./subscription.js";
/** @internal */
export const WebhookSubscriptionCreatedPayload$inboundSchema = z.object({
    type: z.literal("subscription.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Subscription$inboundSchema,
});
/** @internal */
export const WebhookSubscriptionCreatedPayload$outboundSchema = z.object({
    type: z.literal("subscription.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Subscription$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookSubscriptionCreatedPayload$;
(function (WebhookSubscriptionCreatedPayload$) {
    /** @deprecated use `WebhookSubscriptionCreatedPayload$inboundSchema` instead. */
    WebhookSubscriptionCreatedPayload$.inboundSchema = WebhookSubscriptionCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookSubscriptionCreatedPayload$outboundSchema` instead. */
    WebhookSubscriptionCreatedPayload$.outboundSchema = WebhookSubscriptionCreatedPayload$outboundSchema;
})(WebhookSubscriptionCreatedPayload$ || (WebhookSubscriptionCreatedPayload$ = {}));
export function webhookSubscriptionCreatedPayloadToJSON(webhookSubscriptionCreatedPayload) {
    return JSON.stringify(WebhookSubscriptionCreatedPayload$outboundSchema.parse(webhookSubscriptionCreatedPayload));
}
export function webhookSubscriptionCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookSubscriptionCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionCreatedPayload' from JSON`);
}
