// @polar-sh/sdk/dist/esm/models/components/webhookbenefitgrantrevokedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantWebhook$inboundSchema, BenefitGrantWebhook$outboundSchema, } from "./benefitgrantwebhook.js";
/** @internal */
export const WebhookBenefitGrantRevokedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.revoked"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: BenefitGrantWebhook$inboundSchema,
});
/** @internal */
export const WebhookBenefitGrantRevokedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.revoked"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookBenefitGrantRevokedPayload$;
(function (WebhookBenefitGrantRevokedPayload$) {
    /** @deprecated use `WebhookBenefitGrantRevokedPayload$inboundSchema` instead. */
    WebhookBenefitGrantRevokedPayload$.inboundSchema = WebhookBenefitGrantRevokedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantRevokedPayload$outboundSchema` instead. */
    WebhookBenefitGrantRevokedPayload$.outboundSchema = WebhookBenefitGrantRevokedPayload$outboundSchema;
})(WebhookBenefitGrantRevokedPayload$ || (WebhookBenefitGrantRevokedPayload$ = {}));
export function webhookBenefitGrantRevokedPayloadToJSON(webhookBenefitGrantRevokedPayload) {
    return JSON.stringify(WebhookBenefitGrantRevokedPayload$outboundSchema.parse(webhookBenefitGrantRevokedPayload));
}
export function webhookBenefitGrantRevokedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitGrantRevokedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantRevokedPayload' from JSON`);
}
