// @polar-sh/sdk/dist/esm/models/components/webhookcheckoutupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Checkout$inboundSchema, Checkout$outboundSchema, } from "./checkout.js";
/** @internal */
export const WebhookCheckoutUpdatedPayload$inboundSchema = z.object({
    type: z.literal("checkout.updated"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Checkout$inboundSchema,
});
/** @internal */
export const WebhookCheckoutUpdatedPayload$outboundSchema = z.object({
    type: z.literal("checkout.updated"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Checkout$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookCheckoutUpdatedPayload$;
(function (WebhookCheckoutUpdatedPayload$) {
    /** @deprecated use `WebhookCheckoutUpdatedPayload$inboundSchema` instead. */
    WebhookCheckoutUpdatedPayload$.inboundSchema = WebhookCheckoutUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookCheckoutUpdatedPayload$outboundSchema` instead. */
    WebhookCheckoutUpdatedPayload$.outboundSchema = WebhookCheckoutUpdatedPayload$outboundSchema;
})(WebhookCheckoutUpdatedPayload$ || (WebhookCheckoutUpdatedPayload$ = {}));
export function webhookCheckoutUpdatedPayloadToJSON(webhookCheckoutUpdatedPayload) {
    return JSON.stringify(WebhookCheckoutUpdatedPayload$outboundSchema.parse(webhookCheckoutUpdatedPayload));
}
export function webhookCheckoutUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCheckoutUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCheckoutUpdatedPayload' from JSON`);
}
