// @polar-sh/sdk/dist/commonjs/models/components/benefitgithubrepositorycreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryCreate$ = exports.BenefitGitHubRepositoryCreate$outboundSchema = exports.BenefitGitHubRepositoryCreate$inboundSchema = exports.BenefitGitHubRepositoryCreateMetadata$ = exports.BenefitGitHubRepositoryCreateMetadata$outboundSchema = exports.BenefitGitHubRepositoryCreateMetadata$inboundSchema = void 0;
exports.benefitGitHubRepositoryCreateMetadataToJSON = benefitGitHubRepositoryCreateMetadataToJSON;
exports.benefitGitHubRepositoryCreateMetadataFromJSON = benefitGitHubRepositoryCreateMetadataFromJSON;
exports.benefitGitHubRepositoryCreateToJSON = benefitGitHubRepositoryCreateToJSON;
exports.benefitGitHubRepositoryCreateFromJSON = benefitGitHubRepositoryCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitgithubrepositorycreateproperties_js_1 = require("./benefitgithubrepositorycreateproperties.js");
/** @internal */
exports.BenefitGitHubRepositoryCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitGitHubRepositoryCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreateMetadata$;
(function (BenefitGitHubRepositoryCreateMetadata$) {
    /** @deprecated use `BenefitGitHubRepositoryCreateMetadata$inboundSchema` instead. */
    BenefitGitHubRepositoryCreateMetadata$.inboundSchema = exports.BenefitGitHubRepositoryCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreateMetadata$outboundSchema` instead. */
    BenefitGitHubRepositoryCreateMetadata$.outboundSchema = exports.BenefitGitHubRepositoryCreateMetadata$outboundSchema;
})(BenefitGitHubRepositoryCreateMetadata$ || (exports.BenefitGitHubRepositoryCreateMetadata$ = BenefitGitHubRepositoryCreateMetadata$ = {}));
function benefitGitHubRepositoryCreateMetadataToJSON(benefitGitHubRepositoryCreateMetadata) {
    return JSON.stringify(exports.BenefitGitHubRepositoryCreateMetadata$outboundSchema.parse(benefitGitHubRepositoryCreateMetadata));
}
function benefitGitHubRepositoryCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositoryCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryCreateMetadata' from JSON`);
}
/** @internal */
exports.BenefitGitHubRepositoryCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("github_repository"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitGitHubRepositoryCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("github_repository"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreate$;
(function (BenefitGitHubRepositoryCreate$) {
    /** @deprecated use `BenefitGitHubRepositoryCreate$inboundSchema` instead. */
    BenefitGitHubRepositoryCreate$.inboundSchema = exports.BenefitGitHubRepositoryCreate$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreate$outboundSchema` instead. */
    BenefitGitHubRepositoryCreate$.outboundSchema = exports.BenefitGitHubRepositoryCreate$outboundSchema;
})(BenefitGitHubRepositoryCreate$ || (exports.BenefitGitHubRepositoryCreate$ = BenefitGitHubRepositoryCreate$ = {}));
function benefitGitHubRepositoryCreateToJSON(benefitGitHubRepositoryCreate) {
    return JSON.stringify(exports.BenefitGitHubRepositoryCreate$outboundSchema.parse(benefitGitHubRepositoryCreate));
}
function benefitGitHubRepositoryCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositoryCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryCreate' from JSON`);
}
