// @polar-sh/sdk/dist/commonjs/models/components/benefitgithubrepositorysubscriberproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositorySubscriberProperties$ = exports.BenefitGitHubRepositorySubscriberProperties$outboundSchema = exports.BenefitGitHubRepositorySubscriberProperties$inboundSchema = void 0;
exports.benefitGitHubRepositorySubscriberPropertiesToJSON = benefitGitHubRepositorySubscriberPropertiesToJSON;
exports.benefitGitHubRepositorySubscriberPropertiesFromJSON = benefitGitHubRepositorySubscriberPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitGitHubRepositorySubscriberProperties$inboundSchema = z.object({
    repository_owner: z.string(),
    repository_name: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
    });
});
/** @internal */
exports.BenefitGitHubRepositorySubscriberProperties$outboundSchema = z.object({
    repositoryOwner: z.string(),
    repositoryName: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositorySubscriberProperties$;
(function (BenefitGitHubRepositorySubscriberProperties$) {
    /** @deprecated use `BenefitGitHubRepositorySubscriberProperties$inboundSchema` instead. */
    BenefitGitHubRepositorySubscriberProperties$.inboundSchema = exports.BenefitGitHubRepositorySubscriberProperties$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositorySubscriberProperties$outboundSchema` instead. */
    BenefitGitHubRepositorySubscriberProperties$.outboundSchema = exports.BenefitGitHubRepositorySubscriberProperties$outboundSchema;
})(BenefitGitHubRepositorySubscriberProperties$ || (exports.BenefitGitHubRepositorySubscriberProperties$ = BenefitGitHubRepositorySubscriberProperties$ = {}));
function benefitGitHubRepositorySubscriberPropertiesToJSON(benefitGitHubRepositorySubscriberProperties) {
    return JSON.stringify(exports.BenefitGitHubRepositorySubscriberProperties$outboundSchema.parse(benefitGitHubRepositorySubscriberProperties));
}
function benefitGitHubRepositorySubscriberPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositorySubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositorySubscriberProperties' from JSON`);
}
