// @polar-sh/sdk/dist/commonjs/models/components/benefitgithubrepositoryupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryUpdate$ = exports.BenefitGitHubRepositoryUpdate$outboundSchema = exports.BenefitGitHubRepositoryUpdate$inboundSchema = exports.BenefitGitHubRepositoryUpdateMetadata$ = exports.BenefitGitHubRepositoryUpdateMetadata$outboundSchema = exports.BenefitGitHubRepositoryUpdateMetadata$inboundSchema = void 0;
exports.benefitGitHubRepositoryUpdateMetadataToJSON = benefitGitHubRepositoryUpdateMetadataToJSON;
exports.benefitGitHubRepositoryUpdateMetadataFromJSON = benefitGitHubRepositoryUpdateMetadataFromJSON;
exports.benefitGitHubRepositoryUpdateToJSON = benefitGitHubRepositoryUpdateToJSON;
exports.benefitGitHubRepositoryUpdateFromJSON = benefitGitHubRepositoryUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitgithubrepositorycreateproperties_js_1 = require("./benefitgithubrepositorycreateproperties.js");
/** @internal */
exports.BenefitGitHubRepositoryUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitGitHubRepositoryUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryUpdateMetadata$;
(function (BenefitGitHubRepositoryUpdateMetadata$) {
    /** @deprecated use `BenefitGitHubRepositoryUpdateMetadata$inboundSchema` instead. */
    BenefitGitHubRepositoryUpdateMetadata$.inboundSchema = exports.BenefitGitHubRepositoryUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryUpdateMetadata$outboundSchema` instead. */
    BenefitGitHubRepositoryUpdateMetadata$.outboundSchema = exports.BenefitGitHubRepositoryUpdateMetadata$outboundSchema;
})(BenefitGitHubRepositoryUpdateMetadata$ || (exports.BenefitGitHubRepositoryUpdateMetadata$ = BenefitGitHubRepositoryUpdateMetadata$ = {}));
function benefitGitHubRepositoryUpdateMetadataToJSON(benefitGitHubRepositoryUpdateMetadata) {
    return JSON.stringify(exports.BenefitGitHubRepositoryUpdateMetadata$outboundSchema.parse(benefitGitHubRepositoryUpdateMetadata));
}
function benefitGitHubRepositoryUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositoryUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryUpdateMetadata' from JSON`);
}
/** @internal */
exports.BenefitGitHubRepositoryUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("github_repository"),
    properties: z.nullable(benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitGitHubRepositoryUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("github_repository"),
    properties: z.nullable(benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryUpdate$;
(function (BenefitGitHubRepositoryUpdate$) {
    /** @deprecated use `BenefitGitHubRepositoryUpdate$inboundSchema` instead. */
    BenefitGitHubRepositoryUpdate$.inboundSchema = exports.BenefitGitHubRepositoryUpdate$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryUpdate$outboundSchema` instead. */
    BenefitGitHubRepositoryUpdate$.outboundSchema = exports.BenefitGitHubRepositoryUpdate$outboundSchema;
})(BenefitGitHubRepositoryUpdate$ || (exports.BenefitGitHubRepositoryUpdate$ = BenefitGitHubRepositoryUpdate$ = {}));
function benefitGitHubRepositoryUpdateToJSON(benefitGitHubRepositoryUpdate) {
    return JSON.stringify(exports.BenefitGitHubRepositoryUpdate$outboundSchema.parse(benefitGitHubRepositoryUpdate));
}
function benefitGitHubRepositoryUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositoryUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryUpdate' from JSON`);
}
