// @polar-sh/sdk/dist/commonjs/models/components/benefitgithubrepositorycreateproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryCreateProperties$ = exports.BenefitGitHubRepositoryCreateProperties$outboundSchema = exports.BenefitGitHubRepositoryCreateProperties$inboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$ = exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission = void 0;
exports.benefitGitHubRepositoryCreatePropertiesToJSON = benefitGitHubRepositoryCreatePropertiesToJSON;
exports.benefitGitHubRepositoryCreatePropertiesFromJSON = benefitGitHubRepositoryCreatePropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * The permission level to grant. Read more about roles and their permissions on [GitHub documentation](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role).
 */
exports.BenefitGitHubRepositoryCreatePropertiesPermission = {
    Pull: "pull",
    Triage: "triage",
    Push: "push",
    Maintain: "maintain",
    Admin: "admin",
};
/** @internal */
exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema = z
    .nativeEnum(exports.BenefitGitHubRepositoryCreatePropertiesPermission);
/** @internal */
exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreatePropertiesPermission$;
(function (BenefitGitHubRepositoryCreatePropertiesPermission$) {
    /** @deprecated use `BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema` instead. */
    BenefitGitHubRepositoryCreatePropertiesPermission$.inboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema` instead. */
    BenefitGitHubRepositoryCreatePropertiesPermission$.outboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema;
})(BenefitGitHubRepositoryCreatePropertiesPermission$ || (exports.BenefitGitHubRepositoryCreatePropertiesPermission$ = BenefitGitHubRepositoryCreatePropertiesPermission$ = {}));
/** @internal */
exports.BenefitGitHubRepositoryCreateProperties$inboundSchema = z.object({
    repository_owner: z.string(),
    repository_name: z.string(),
    permission: exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
    });
});
/** @internal */
exports.BenefitGitHubRepositoryCreateProperties$outboundSchema = z.object({
    repositoryOwner: z.string(),
    repositoryName: z.string(),
    permission: exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreateProperties$;
(function (BenefitGitHubRepositoryCreateProperties$) {
    /** @deprecated use `BenefitGitHubRepositoryCreateProperties$inboundSchema` instead. */
    BenefitGitHubRepositoryCreateProperties$.inboundSchema = exports.BenefitGitHubRepositoryCreateProperties$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreateProperties$outboundSchema` instead. */
    BenefitGitHubRepositoryCreateProperties$.outboundSchema = exports.BenefitGitHubRepositoryCreateProperties$outboundSchema;
})(BenefitGitHubRepositoryCreateProperties$ || (exports.BenefitGitHubRepositoryCreateProperties$ = BenefitGitHubRepositoryCreateProperties$ = {}));
function benefitGitHubRepositoryCreatePropertiesToJSON(benefitGitHubRepositoryCreateProperties) {
    return JSON.stringify(exports.BenefitGitHubRepositoryCreateProperties$outboundSchema.parse(benefitGitHubRepositoryCreateProperties));
}
function benefitGitHubRepositoryCreatePropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositoryCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryCreateProperties' from JSON`);
}
