// @polar-sh/sdk/dist/commonjs/models/components/benefitdownloadablessubscriberproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadablesSubscriberProperties$ = exports.BenefitDownloadablesSubscriberProperties$outboundSchema = exports.BenefitDownloadablesSubscriberProperties$inboundSchema = void 0;
exports.benefitDownloadablesSubscriberPropertiesToJSON = benefitDownloadablesSubscriberPropertiesToJSON;
exports.benefitDownloadablesSubscriberPropertiesFromJSON = benefitDownloadablesSubscriberPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitDownloadablesSubscriberProperties$inboundSchema = z.object({
    active_files: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "active_files": "activeFiles",
    });
});
/** @internal */
exports.BenefitDownloadablesSubscriberProperties$outboundSchema = z.object({
    activeFiles: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        activeFiles: "active_files",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesSubscriberProperties$;
(function (BenefitDownloadablesSubscriberProperties$) {
    /** @deprecated use `BenefitDownloadablesSubscriberProperties$inboundSchema` instead. */
    BenefitDownloadablesSubscriberProperties$.inboundSchema = exports.BenefitDownloadablesSubscriberProperties$inboundSchema;
    /** @deprecated use `BenefitDownloadablesSubscriberProperties$outboundSchema` instead. */
    BenefitDownloadablesSubscriberProperties$.outboundSchema = exports.BenefitDownloadablesSubscriberProperties$outboundSchema;
})(BenefitDownloadablesSubscriberProperties$ || (exports.BenefitDownloadablesSubscriberProperties$ = BenefitDownloadablesSubscriberProperties$ = {}));
function benefitDownloadablesSubscriberPropertiesToJSON(benefitDownloadablesSubscriberProperties) {
    return JSON.stringify(exports.BenefitDownloadablesSubscriberProperties$outboundSchema.parse(benefitDownloadablesSubscriberProperties));
}
function benefitDownloadablesSubscriberPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDownloadablesSubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesSubscriberProperties' from JSON`);
}
