// @polar-sh/sdk/dist/commonjs/models/components/benefitlicensekeys.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeys$ = exports.BenefitLicenseKeys$outboundSchema = exports.BenefitLicenseKeys$inboundSchema = exports.BenefitLicenseKeysMetadata$ = exports.BenefitLicenseKeysMetadata$outboundSchema = exports.BenefitLicenseKeysMetadata$inboundSchema = void 0;
exports.benefitLicenseKeysMetadataToJSON = benefitLicenseKeysMetadataToJSON;
exports.benefitLicenseKeysMetadataFromJSON = benefitLicenseKeysMetadataFromJSON;
exports.benefitLicenseKeysToJSON = benefitLicenseKeysToJSON;
exports.benefitLicenseKeysFromJSON = benefitLicenseKeysFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitlicensekeysproperties_js_1 = require("./benefitlicensekeysproperties.js");
/** @internal */
exports.BenefitLicenseKeysMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitLicenseKeysMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysMetadata$;
(function (BenefitLicenseKeysMetadata$) {
    /** @deprecated use `BenefitLicenseKeysMetadata$inboundSchema` instead. */
    BenefitLicenseKeysMetadata$.inboundSchema = exports.BenefitLicenseKeysMetadata$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysMetadata$outboundSchema` instead. */
    BenefitLicenseKeysMetadata$.outboundSchema = exports.BenefitLicenseKeysMetadata$outboundSchema;
})(BenefitLicenseKeysMetadata$ || (exports.BenefitLicenseKeysMetadata$ = BenefitLicenseKeysMetadata$ = {}));
function benefitLicenseKeysMetadataToJSON(benefitLicenseKeysMetadata) {
    return JSON.stringify(exports.BenefitLicenseKeysMetadata$outboundSchema.parse(benefitLicenseKeysMetadata));
}
function benefitLicenseKeysMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeysMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysMetadata' from JSON`);
}
/** @internal */
exports.BenefitLicenseKeys$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitlicensekeysproperties_js_1.BenefitLicenseKeysProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitLicenseKeys$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitlicensekeysproperties_js_1.BenefitLicenseKeysProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeys$;
(function (BenefitLicenseKeys$) {
    /** @deprecated use `BenefitLicenseKeys$inboundSchema` instead. */
    BenefitLicenseKeys$.inboundSchema = exports.BenefitLicenseKeys$inboundSchema;
    /** @deprecated use `BenefitLicenseKeys$outboundSchema` instead. */
    BenefitLicenseKeys$.outboundSchema = exports.BenefitLicenseKeys$outboundSchema;
})(BenefitLicenseKeys$ || (exports.BenefitLicenseKeys$ = BenefitLicenseKeys$ = {}));
function benefitLicenseKeysToJSON(benefitLicenseKeys) {
    return JSON.stringify(exports.BenefitLicenseKeys$outboundSchema.parse(benefitLicenseKeys));
}
function benefitLicenseKeysFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeys$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeys' from JSON`);
}
