// @polar-sh/sdk/dist/commonjs/models/components/benefitlicensekeyscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeysCreate$ = exports.BenefitLicenseKeysCreate$outboundSchema = exports.BenefitLicenseKeysCreate$inboundSchema = exports.BenefitLicenseKeysCreateMetadata$ = exports.BenefitLicenseKeysCreateMetadata$outboundSchema = exports.BenefitLicenseKeysCreateMetadata$inboundSchema = void 0;
exports.benefitLicenseKeysCreateMetadataToJSON = benefitLicenseKeysCreateMetadataToJSON;
exports.benefitLicenseKeysCreateMetadataFromJSON = benefitLicenseKeysCreateMetadataFromJSON;
exports.benefitLicenseKeysCreateToJSON = benefitLicenseKeysCreateToJSON;
exports.benefitLicenseKeysCreateFromJSON = benefitLicenseKeysCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitlicensekeyscreateproperties_js_1 = require("./benefitlicensekeyscreateproperties.js");
/** @internal */
exports.BenefitLicenseKeysCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitLicenseKeysCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysCreateMetadata$;
(function (BenefitLicenseKeysCreateMetadata$) {
    /** @deprecated use `BenefitLicenseKeysCreateMetadata$inboundSchema` instead. */
    BenefitLicenseKeysCreateMetadata$.inboundSchema = exports.BenefitLicenseKeysCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreateMetadata$outboundSchema` instead. */
    BenefitLicenseKeysCreateMetadata$.outboundSchema = exports.BenefitLicenseKeysCreateMetadata$outboundSchema;
})(BenefitLicenseKeysCreateMetadata$ || (exports.BenefitLicenseKeysCreateMetadata$ = BenefitLicenseKeysCreateMetadata$ = {}));
function benefitLicenseKeysCreateMetadataToJSON(benefitLicenseKeysCreateMetadata) {
    return JSON.stringify(exports.BenefitLicenseKeysCreateMetadata$outboundSchema.parse(benefitLicenseKeysCreateMetadata));
}
function benefitLicenseKeysCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeysCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysCreateMetadata' from JSON`);
}
/** @internal */
exports.BenefitLicenseKeysCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("license_keys"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitLicenseKeysCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("license_keys"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysCreate$;
(function (BenefitLicenseKeysCreate$) {
    /** @deprecated use `BenefitLicenseKeysCreate$inboundSchema` instead. */
    BenefitLicenseKeysCreate$.inboundSchema = exports.BenefitLicenseKeysCreate$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreate$outboundSchema` instead. */
    BenefitLicenseKeysCreate$.outboundSchema = exports.BenefitLicenseKeysCreate$outboundSchema;
})(BenefitLicenseKeysCreate$ || (exports.BenefitLicenseKeysCreate$ = BenefitLicenseKeysCreate$ = {}));
function benefitLicenseKeysCreateToJSON(benefitLicenseKeysCreate) {
    return JSON.stringify(exports.BenefitLicenseKeysCreate$outboundSchema.parse(benefitLicenseKeysCreate));
}
function benefitLicenseKeysCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeysCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysCreate' from JSON`);
}
