// @polar-sh/sdk/dist/commonjs/models/components/benefitdiscordcreateproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordCreateProperties$ = exports.BenefitDiscordCreateProperties$outboundSchema = exports.BenefitDiscordCreateProperties$inboundSchema = void 0;
exports.benefitDiscordCreatePropertiesToJSON = benefitDiscordCreatePropertiesToJSON;
exports.benefitDiscordCreatePropertiesFromJSON = benefitDiscordCreatePropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitDiscordCreateProperties$inboundSchema = z.object({
    guild_token: z.string(),
    role_id: z.string(),
    kick_member: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "guild_token": "guildToken",
        "role_id": "roleId",
        "kick_member": "kickMember",
    });
});
/** @internal */
exports.BenefitDiscordCreateProperties$outboundSchema = z.object({
    guildToken: z.string(),
    roleId: z.string(),
    kickMember: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        guildToken: "guild_token",
        roleId: "role_id",
        kickMember: "kick_member",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordCreateProperties$;
(function (BenefitDiscordCreateProperties$) {
    /** @deprecated use `BenefitDiscordCreateProperties$inboundSchema` instead. */
    BenefitDiscordCreateProperties$.inboundSchema = exports.BenefitDiscordCreateProperties$inboundSchema;
    /** @deprecated use `BenefitDiscordCreateProperties$outboundSchema` instead. */
    BenefitDiscordCreateProperties$.outboundSchema = exports.BenefitDiscordCreateProperties$outboundSchema;
})(BenefitDiscordCreateProperties$ || (exports.BenefitDiscordCreateProperties$ = BenefitDiscordCreateProperties$ = {}));
function benefitDiscordCreatePropertiesToJSON(benefitDiscordCreateProperties) {
    return JSON.stringify(exports.BenefitDiscordCreateProperties$outboundSchema.parse(benefitDiscordCreateProperties));
}
function benefitDiscordCreatePropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordCreateProperties' from JSON`);
}
