// @polar-sh/sdk/dist/commonjs/models/components/benefitdiscordsubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordSubscriber$ = exports.BenefitDiscordSubscriber$outboundSchema = exports.BenefitDiscordSubscriber$inboundSchema = exports.BenefitDiscordSubscriberMetadata$ = exports.BenefitDiscordSubscriberMetadata$outboundSchema = exports.BenefitDiscordSubscriberMetadata$inboundSchema = void 0;
exports.benefitDiscordSubscriberMetadataToJSON = benefitDiscordSubscriberMetadataToJSON;
exports.benefitDiscordSubscriberMetadataFromJSON = benefitDiscordSubscriberMetadataFromJSON;
exports.benefitDiscordSubscriberToJSON = benefitDiscordSubscriberToJSON;
exports.benefitDiscordSubscriberFromJSON = benefitDiscordSubscriberFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdiscordsubscriberproperties_js_1 = require("./benefitdiscordsubscriberproperties.js");
const organization_js_1 = require("./organization.js");
/** @internal */
exports.BenefitDiscordSubscriberMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitDiscordSubscriberMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordSubscriberMetadata$;
(function (BenefitDiscordSubscriberMetadata$) {
    /** @deprecated use `BenefitDiscordSubscriberMetadata$inboundSchema` instead. */
    BenefitDiscordSubscriberMetadata$.inboundSchema = exports.BenefitDiscordSubscriberMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordSubscriberMetadata$outboundSchema` instead. */
    BenefitDiscordSubscriberMetadata$.outboundSchema = exports.BenefitDiscordSubscriberMetadata$outboundSchema;
})(BenefitDiscordSubscriberMetadata$ || (exports.BenefitDiscordSubscriberMetadata$ = BenefitDiscordSubscriberMetadata$ = {}));
function benefitDiscordSubscriberMetadataToJSON(benefitDiscordSubscriberMetadata) {
    return JSON.stringify(exports.BenefitDiscordSubscriberMetadata$outboundSchema.parse(benefitDiscordSubscriberMetadata));
}
function benefitDiscordSubscriberMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordSubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordSubscriberMetadata' from JSON`);
}
/** @internal */
exports.BenefitDiscordSubscriber$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$inboundSchema,
    properties: benefitdiscordsubscriberproperties_js_1.BenefitDiscordSubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDiscordSubscriber$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$outboundSchema,
    properties: benefitdiscordsubscriberproperties_js_1.BenefitDiscordSubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordSubscriber$;
(function (BenefitDiscordSubscriber$) {
    /** @deprecated use `BenefitDiscordSubscriber$inboundSchema` instead. */
    BenefitDiscordSubscriber$.inboundSchema = exports.BenefitDiscordSubscriber$inboundSchema;
    /** @deprecated use `BenefitDiscordSubscriber$outboundSchema` instead. */
    BenefitDiscordSubscriber$.outboundSchema = exports.BenefitDiscordSubscriber$outboundSchema;
})(BenefitDiscordSubscriber$ || (exports.BenefitDiscordSubscriber$ = BenefitDiscordSubscriber$ = {}));
function benefitDiscordSubscriberToJSON(benefitDiscordSubscriber) {
    return JSON.stringify(exports.BenefitDiscordSubscriber$outboundSchema.parse(benefitDiscordSubscriber));
}
function benefitDiscordSubscriberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordSubscriber' from JSON`);
}
