// @polar-sh/sdk/dist/commonjs/models/components/benefitdownloadablessubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadablesSubscriber$ = exports.BenefitDownloadablesSubscriber$outboundSchema = exports.BenefitDownloadablesSubscriber$inboundSchema = exports.BenefitDownloadablesSubscriberMetadata$ = exports.BenefitDownloadablesSubscriberMetadata$outboundSchema = exports.BenefitDownloadablesSubscriberMetadata$inboundSchema = void 0;
exports.benefitDownloadablesSubscriberMetadataToJSON = benefitDownloadablesSubscriberMetadataToJSON;
exports.benefitDownloadablesSubscriberMetadataFromJSON = benefitDownloadablesSubscriberMetadataFromJSON;
exports.benefitDownloadablesSubscriberToJSON = benefitDownloadablesSubscriberToJSON;
exports.benefitDownloadablesSubscriberFromJSON = benefitDownloadablesSubscriberFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdownloadablessubscriberproperties_js_1 = require("./benefitdownloadablessubscriberproperties.js");
const organization_js_1 = require("./organization.js");
/** @internal */
exports.BenefitDownloadablesSubscriberMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitDownloadablesSubscriberMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesSubscriberMetadata$;
(function (BenefitDownloadablesSubscriberMetadata$) {
    /** @deprecated use `BenefitDownloadablesSubscriberMetadata$inboundSchema` instead. */
    BenefitDownloadablesSubscriberMetadata$.inboundSchema = exports.BenefitDownloadablesSubscriberMetadata$inboundSchema;
    /** @deprecated use `BenefitDownloadablesSubscriberMetadata$outboundSchema` instead. */
    BenefitDownloadablesSubscriberMetadata$.outboundSchema = exports.BenefitDownloadablesSubscriberMetadata$outboundSchema;
})(BenefitDownloadablesSubscriberMetadata$ || (exports.BenefitDownloadablesSubscriberMetadata$ = BenefitDownloadablesSubscriberMetadata$ = {}));
function benefitDownloadablesSubscriberMetadataToJSON(benefitDownloadablesSubscriberMetadata) {
    return JSON.stringify(exports.BenefitDownloadablesSubscriberMetadata$outboundSchema.parse(benefitDownloadablesSubscriberMetadata));
}
function benefitDownloadablesSubscriberMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDownloadablesSubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesSubscriberMetadata' from JSON`);
}
/** @internal */
exports.BenefitDownloadablesSubscriber$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$inboundSchema,
    properties: benefitdownloadablessubscriberproperties_js_1.BenefitDownloadablesSubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDownloadablesSubscriber$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$outboundSchema,
    properties: benefitdownloadablessubscriberproperties_js_1.BenefitDownloadablesSubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesSubscriber$;
(function (BenefitDownloadablesSubscriber$) {
    /** @deprecated use `BenefitDownloadablesSubscriber$inboundSchema` instead. */
    BenefitDownloadablesSubscriber$.inboundSchema = exports.BenefitDownloadablesSubscriber$inboundSchema;
    /** @deprecated use `BenefitDownloadablesSubscriber$outboundSchema` instead. */
    BenefitDownloadablesSubscriber$.outboundSchema = exports.BenefitDownloadablesSubscriber$outboundSchema;
})(BenefitDownloadablesSubscriber$ || (exports.BenefitDownloadablesSubscriber$ = BenefitDownloadablesSubscriber$ = {}));
function benefitDownloadablesSubscriberToJSON(benefitDownloadablesSubscriber) {
    return JSON.stringify(exports.BenefitDownloadablesSubscriber$outboundSchema.parse(benefitDownloadablesSubscriber));
}
function benefitDownloadablesSubscriberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDownloadablesSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesSubscriber' from JSON`);
}
