// @polar-sh/sdk/dist/commonjs/models/components/benefitdownloadables.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadables$ = exports.BenefitDownloadables$outboundSchema = exports.BenefitDownloadables$inboundSchema = exports.BenefitDownloadablesMetadata$ = exports.BenefitDownloadablesMetadata$outboundSchema = exports.BenefitDownloadablesMetadata$inboundSchema = void 0;
exports.benefitDownloadablesMetadataToJSON = benefitDownloadablesMetadataToJSON;
exports.benefitDownloadablesMetadataFromJSON = benefitDownloadablesMetadataFromJSON;
exports.benefitDownloadablesToJSON = benefitDownloadablesToJSON;
exports.benefitDownloadablesFromJSON = benefitDownloadablesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdownloadablesproperties_js_1 = require("./benefitdownloadablesproperties.js");
/** @internal */
exports.BenefitDownloadablesMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitDownloadablesMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesMetadata$;
(function (BenefitDownloadablesMetadata$) {
    /** @deprecated use `BenefitDownloadablesMetadata$inboundSchema` instead. */
    BenefitDownloadablesMetadata$.inboundSchema = exports.BenefitDownloadablesMetadata$inboundSchema;
    /** @deprecated use `BenefitDownloadablesMetadata$outboundSchema` instead. */
    BenefitDownloadablesMetadata$.outboundSchema = exports.BenefitDownloadablesMetadata$outboundSchema;
})(BenefitDownloadablesMetadata$ || (exports.BenefitDownloadablesMetadata$ = BenefitDownloadablesMetadata$ = {}));
function benefitDownloadablesMetadataToJSON(benefitDownloadablesMetadata) {
    return JSON.stringify(exports.BenefitDownloadablesMetadata$outboundSchema.parse(benefitDownloadablesMetadata));
}
function benefitDownloadablesMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDownloadablesMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesMetadata' from JSON`);
}
/** @internal */
exports.BenefitDownloadables$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitdownloadablesproperties_js_1.BenefitDownloadablesProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDownloadables$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitdownloadablesproperties_js_1.BenefitDownloadablesProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadables$;
(function (BenefitDownloadables$) {
    /** @deprecated use `BenefitDownloadables$inboundSchema` instead. */
    BenefitDownloadables$.inboundSchema = exports.BenefitDownloadables$inboundSchema;
    /** @deprecated use `BenefitDownloadables$outboundSchema` instead. */
    BenefitDownloadables$.outboundSchema = exports.BenefitDownloadables$outboundSchema;
})(BenefitDownloadables$ || (exports.BenefitDownloadables$ = BenefitDownloadables$ = {}));
function benefitDownloadablesToJSON(benefitDownloadables) {
    return JSON.stringify(exports.BenefitDownloadables$outboundSchema.parse(benefitDownloadables));
}
function benefitDownloadablesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDownloadables$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadables' from JSON`);
}
