// @polar-sh/sdk/dist/commonjs/models/components/benefitdiscordupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordUpdate$ = exports.BenefitDiscordUpdate$outboundSchema = exports.BenefitDiscordUpdate$inboundSchema = exports.BenefitDiscordUpdateMetadata$ = exports.BenefitDiscordUpdateMetadata$outboundSchema = exports.BenefitDiscordUpdateMetadata$inboundSchema = void 0;
exports.benefitDiscordUpdateMetadataToJSON = benefitDiscordUpdateMetadataToJSON;
exports.benefitDiscordUpdateMetadataFromJSON = benefitDiscordUpdateMetadataFromJSON;
exports.benefitDiscordUpdateToJSON = benefitDiscordUpdateToJSON;
exports.benefitDiscordUpdateFromJSON = benefitDiscordUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdiscordcreateproperties_js_1 = require("./benefitdiscordcreateproperties.js");
/** @internal */
exports.BenefitDiscordUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitDiscordUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordUpdateMetadata$;
(function (BenefitDiscordUpdateMetadata$) {
    /** @deprecated use `BenefitDiscordUpdateMetadata$inboundSchema` instead. */
    BenefitDiscordUpdateMetadata$.inboundSchema = exports.BenefitDiscordUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordUpdateMetadata$outboundSchema` instead. */
    BenefitDiscordUpdateMetadata$.outboundSchema = exports.BenefitDiscordUpdateMetadata$outboundSchema;
})(BenefitDiscordUpdateMetadata$ || (exports.BenefitDiscordUpdateMetadata$ = BenefitDiscordUpdateMetadata$ = {}));
function benefitDiscordUpdateMetadataToJSON(benefitDiscordUpdateMetadata) {
    return JSON.stringify(exports.BenefitDiscordUpdateMetadata$outboundSchema.parse(benefitDiscordUpdateMetadata));
}
function benefitDiscordUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordUpdateMetadata' from JSON`);
}
/** @internal */
exports.BenefitDiscordUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("discord"),
    properties: z.nullable(benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitDiscordUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("discord"),
    properties: z.nullable(benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordUpdate$;
(function (BenefitDiscordUpdate$) {
    /** @deprecated use `BenefitDiscordUpdate$inboundSchema` instead. */
    BenefitDiscordUpdate$.inboundSchema = exports.BenefitDiscordUpdate$inboundSchema;
    /** @deprecated use `BenefitDiscordUpdate$outboundSchema` instead. */
    BenefitDiscordUpdate$.outboundSchema = exports.BenefitDiscordUpdate$outboundSchema;
})(BenefitDiscordUpdate$ || (exports.BenefitDiscordUpdate$ = BenefitDiscordUpdate$ = {}));
function benefitDiscordUpdateToJSON(benefitDiscordUpdate) {
    return JSON.stringify(exports.BenefitDiscordUpdate$outboundSchema.parse(benefitDiscordUpdate));
}
function benefitDiscordUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordUpdate' from JSON`);
}
