// @polar-sh/sdk/dist/commonjs/models/components/benefitmetercreditsubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitMeterCreditSubscriber$ = exports.BenefitMeterCreditSubscriber$outboundSchema = exports.BenefitMeterCreditSubscriber$inboundSchema = exports.BenefitMeterCreditSubscriberMetadata$ = exports.BenefitMeterCreditSubscriberMetadata$outboundSchema = exports.BenefitMeterCreditSubscriberMetadata$inboundSchema = void 0;
exports.benefitMeterCreditSubscriberMetadataToJSON = benefitMeterCreditSubscriberMetadataToJSON;
exports.benefitMeterCreditSubscriberMetadataFromJSON = benefitMeterCreditSubscriberMetadataFromJSON;
exports.benefitMeterCreditSubscriberToJSON = benefitMeterCreditSubscriberToJSON;
exports.benefitMeterCreditSubscriberFromJSON = benefitMeterCreditSubscriberFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitmetercreditsubscriberproperties_js_1 = require("./benefitmetercreditsubscriberproperties.js");
const organization_js_1 = require("./organization.js");
/** @internal */
exports.BenefitMeterCreditSubscriberMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitMeterCreditSubscriberMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditSubscriberMetadata$;
(function (BenefitMeterCreditSubscriberMetadata$) {
    /** @deprecated use `BenefitMeterCreditSubscriberMetadata$inboundSchema` instead. */
    BenefitMeterCreditSubscriberMetadata$.inboundSchema = exports.BenefitMeterCreditSubscriberMetadata$inboundSchema;
    /** @deprecated use `BenefitMeterCreditSubscriberMetadata$outboundSchema` instead. */
    BenefitMeterCreditSubscriberMetadata$.outboundSchema = exports.BenefitMeterCreditSubscriberMetadata$outboundSchema;
})(BenefitMeterCreditSubscriberMetadata$ || (exports.BenefitMeterCreditSubscriberMetadata$ = BenefitMeterCreditSubscriberMetadata$ = {}));
function benefitMeterCreditSubscriberMetadataToJSON(benefitMeterCreditSubscriberMetadata) {
    return JSON.stringify(exports.BenefitMeterCreditSubscriberMetadata$outboundSchema.parse(benefitMeterCreditSubscriberMetadata));
}
function benefitMeterCreditSubscriberMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditSubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditSubscriberMetadata' from JSON`);
}
/** @internal */
exports.BenefitMeterCreditSubscriber$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("meter_credit"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$inboundSchema,
    properties: benefitmetercreditsubscriberproperties_js_1.BenefitMeterCreditSubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitMeterCreditSubscriber$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("meter_credit"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$outboundSchema,
    properties: benefitmetercreditsubscriberproperties_js_1.BenefitMeterCreditSubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditSubscriber$;
(function (BenefitMeterCreditSubscriber$) {
    /** @deprecated use `BenefitMeterCreditSubscriber$inboundSchema` instead. */
    BenefitMeterCreditSubscriber$.inboundSchema = exports.BenefitMeterCreditSubscriber$inboundSchema;
    /** @deprecated use `BenefitMeterCreditSubscriber$outboundSchema` instead. */
    BenefitMeterCreditSubscriber$.outboundSchema = exports.BenefitMeterCreditSubscriber$outboundSchema;
})(BenefitMeterCreditSubscriber$ || (exports.BenefitMeterCreditSubscriber$ = BenefitMeterCreditSubscriber$ = {}));
function benefitMeterCreditSubscriberToJSON(benefitMeterCreditSubscriber) {
    return JSON.stringify(exports.BenefitMeterCreditSubscriber$outboundSchema.parse(benefitMeterCreditSubscriber));
}
function benefitMeterCreditSubscriberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditSubscriber' from JSON`);
}
