// @polar-sh/sdk/dist/commonjs/models/components/checkoutsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutSortProperty$ = exports.CheckoutSortProperty$outboundSchema = exports.CheckoutSortProperty$inboundSchema = exports.CheckoutSortProperty = void 0;
const z = __importStar(require("zod"));
exports.CheckoutSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    ExpiresAt: "expires_at",
    MinusExpiresAt: "-expires_at",
    Status: "status",
    MinusStatus: "-status",
};
/** @internal */
exports.CheckoutSortProperty$inboundSchema = z.nativeEnum(exports.CheckoutSortProperty);
/** @internal */
exports.CheckoutSortProperty$outboundSchema = exports.CheckoutSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutSortProperty$;
(function (CheckoutSortProperty$) {
    /** @deprecated use `CheckoutSortProperty$inboundSchema` instead. */
    CheckoutSortProperty$.inboundSchema = exports.CheckoutSortProperty$inboundSchema;
    /** @deprecated use `CheckoutSortProperty$outboundSchema` instead. */
    CheckoutSortProperty$.outboundSchema = exports.CheckoutSortProperty$outboundSchema;
})(CheckoutSortProperty$ || (exports.CheckoutSortProperty$ = CheckoutSortProperty$ = {}));
