// @polar-sh/sdk/dist/commonjs/models/components/checkoutdiscountpercentagerepeatduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutDiscountPercentageRepeatDuration$ = exports.CheckoutDiscountPercentageRepeatDuration$outboundSchema = exports.CheckoutDiscountPercentageRepeatDuration$inboundSchema = void 0;
exports.checkoutDiscountPercentageRepeatDurationToJSON = checkoutDiscountPercentageRepeatDurationToJSON;
exports.checkoutDiscountPercentageRepeatDurationFromJSON = checkoutDiscountPercentageRepeatDurationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.CheckoutDiscountPercentageRepeatDuration$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    duration_in_months: z.number().int(),
    type: discounttype_js_1.DiscountType$inboundSchema,
    basis_points: z.number().int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "duration_in_months": "durationInMonths",
        "basis_points": "basisPoints",
    });
});
/** @internal */
exports.CheckoutDiscountPercentageRepeatDuration$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    durationInMonths: z.number().int(),
    type: discounttype_js_1.DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutDiscountPercentageRepeatDuration$;
(function (CheckoutDiscountPercentageRepeatDuration$) {
    /** @deprecated use `CheckoutDiscountPercentageRepeatDuration$inboundSchema` instead. */
    CheckoutDiscountPercentageRepeatDuration$.inboundSchema = exports.CheckoutDiscountPercentageRepeatDuration$inboundSchema;
    /** @deprecated use `CheckoutDiscountPercentageRepeatDuration$outboundSchema` instead. */
    CheckoutDiscountPercentageRepeatDuration$.outboundSchema = exports.CheckoutDiscountPercentageRepeatDuration$outboundSchema;
})(CheckoutDiscountPercentageRepeatDuration$ || (exports.CheckoutDiscountPercentageRepeatDuration$ = CheckoutDiscountPercentageRepeatDuration$ = {}));
function checkoutDiscountPercentageRepeatDurationToJSON(checkoutDiscountPercentageRepeatDuration) {
    return JSON.stringify(exports.CheckoutDiscountPercentageRepeatDuration$outboundSchema.parse(checkoutDiscountPercentageRepeatDuration));
}
function checkoutDiscountPercentageRepeatDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutDiscountPercentageRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountPercentageRepeatDuration' from JSON`);
}
