// @polar-sh/sdk/dist/commonjs/models/components/customerbenefitgrantlicensekeysupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerBenefitGrantLicenseKeysUpdate$ = exports.CustomerBenefitGrantLicenseKeysUpdate$outboundSchema = exports.CustomerBenefitGrantLicenseKeysUpdate$inboundSchema = void 0;
exports.customerBenefitGrantLicenseKeysUpdateToJSON = customerBenefitGrantLicenseKeysUpdateToJSON;
exports.customerBenefitGrantLicenseKeysUpdateFromJSON = customerBenefitGrantLicenseKeysUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerBenefitGrantLicenseKeysUpdate$inboundSchema = z.object({
    benefit_type: z.literal("license_keys"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "benefit_type": "benefitType",
    });
});
/** @internal */
exports.CustomerBenefitGrantLicenseKeysUpdate$outboundSchema = z.object({
    benefitType: z.literal("license_keys"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        benefitType: "benefit_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerBenefitGrantLicenseKeysUpdate$;
(function (CustomerBenefitGrantLicenseKeysUpdate$) {
    /** @deprecated use `CustomerBenefitGrantLicenseKeysUpdate$inboundSchema` instead. */
    CustomerBenefitGrantLicenseKeysUpdate$.inboundSchema = exports.CustomerBenefitGrantLicenseKeysUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantLicenseKeysUpdate$outboundSchema` instead. */
    CustomerBenefitGrantLicenseKeysUpdate$.outboundSchema = exports.CustomerBenefitGrantLicenseKeysUpdate$outboundSchema;
})(CustomerBenefitGrantLicenseKeysUpdate$ || (exports.CustomerBenefitGrantLicenseKeysUpdate$ = CustomerBenefitGrantLicenseKeysUpdate$ = {}));
function customerBenefitGrantLicenseKeysUpdateToJSON(customerBenefitGrantLicenseKeysUpdate) {
    return JSON.stringify(exports.CustomerBenefitGrantLicenseKeysUpdate$outboundSchema.parse(customerBenefitGrantLicenseKeysUpdate));
}
function customerBenefitGrantLicenseKeysUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerBenefitGrantLicenseKeysUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantLicenseKeysUpdate' from JSON`);
}
