// @polar-sh/sdk/dist/commonjs/models/components/customerpaymentmethodcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPaymentMethodCreate$ = exports.CustomerPaymentMethodCreate$outboundSchema = exports.CustomerPaymentMethodCreate$inboundSchema = void 0;
exports.customerPaymentMethodCreateToJSON = customerPaymentMethodCreateToJSON;
exports.customerPaymentMethodCreateFromJSON = customerPaymentMethodCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPaymentMethodCreate$inboundSchema = z.object({
    confirmation_token_id: z.string(),
    set_default: z.boolean(),
    return_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "confirmation_token_id": "confirmationTokenId",
        "set_default": "setDefault",
        "return_url": "returnUrl",
    });
});
/** @internal */
exports.CustomerPaymentMethodCreate$outboundSchema = z.object({
    confirmationTokenId: z.string(),
    setDefault: z.boolean(),
    returnUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        confirmationTokenId: "confirmation_token_id",
        setDefault: "set_default",
        returnUrl: "return_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPaymentMethodCreate$;
(function (CustomerPaymentMethodCreate$) {
    /** @deprecated use `CustomerPaymentMethodCreate$inboundSchema` instead. */
    CustomerPaymentMethodCreate$.inboundSchema = exports.CustomerPaymentMethodCreate$inboundSchema;
    /** @deprecated use `CustomerPaymentMethodCreate$outboundSchema` instead. */
    CustomerPaymentMethodCreate$.outboundSchema = exports.CustomerPaymentMethodCreate$outboundSchema;
})(CustomerPaymentMethodCreate$ || (exports.CustomerPaymentMethodCreate$ = CustomerPaymentMethodCreate$ = {}));
function customerPaymentMethodCreateToJSON(customerPaymentMethodCreate) {
    return JSON.stringify(exports.CustomerPaymentMethodCreate$outboundSchema.parse(customerPaymentMethodCreate));
}
function customerPaymentMethodCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPaymentMethodCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPaymentMethodCreate' from JSON`);
}
