// @polar-sh/sdk/dist/commonjs/models/components/customerbenefitgrantcustomupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerBenefitGrantCustomUpdate$ = exports.CustomerBenefitGrantCustomUpdate$outboundSchema = exports.CustomerBenefitGrantCustomUpdate$inboundSchema = void 0;
exports.customerBenefitGrantCustomUpdateToJSON = customerBenefitGrantCustomUpdateToJSON;
exports.customerBenefitGrantCustomUpdateFromJSON = customerBenefitGrantCustomUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerBenefitGrantCustomUpdate$inboundSchema = z.object({
    benefit_type: z.literal("custom"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "benefit_type": "benefitType",
    });
});
/** @internal */
exports.CustomerBenefitGrantCustomUpdate$outboundSchema = z.object({
    benefitType: z.literal("custom"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        benefitType: "benefit_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerBenefitGrantCustomUpdate$;
(function (CustomerBenefitGrantCustomUpdate$) {
    /** @deprecated use `CustomerBenefitGrantCustomUpdate$inboundSchema` instead. */
    CustomerBenefitGrantCustomUpdate$.inboundSchema = exports.CustomerBenefitGrantCustomUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantCustomUpdate$outboundSchema` instead. */
    CustomerBenefitGrantCustomUpdate$.outboundSchema = exports.CustomerBenefitGrantCustomUpdate$outboundSchema;
})(CustomerBenefitGrantCustomUpdate$ || (exports.CustomerBenefitGrantCustomUpdate$ = CustomerBenefitGrantCustomUpdate$ = {}));
function customerBenefitGrantCustomUpdateToJSON(customerBenefitGrantCustomUpdate) {
    return JSON.stringify(exports.CustomerBenefitGrantCustomUpdate$outboundSchema.parse(customerBenefitGrantCustomUpdate));
}
function customerBenefitGrantCustomUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerBenefitGrantCustomUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantCustomUpdate' from JSON`);
}
