// @polar-sh/sdk/dist/commonjs/models/components/customfieldtextproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldTextProperties$ = exports.CustomFieldTextProperties$outboundSchema = exports.CustomFieldTextProperties$inboundSchema = void 0;
exports.customFieldTextPropertiesToJSON = customFieldTextPropertiesToJSON;
exports.customFieldTextPropertiesFromJSON = customFieldTextPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomFieldTextProperties$inboundSchema = z.object({
    form_label: z.string().optional(),
    form_help_text: z.string().optional(),
    form_placeholder: z.string().optional(),
    textarea: z.boolean().optional(),
    min_length: z.number().int().optional(),
    max_length: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
        "min_length": "minLength",
        "max_length": "maxLength",
    });
});
/** @internal */
exports.CustomFieldTextProperties$outboundSchema = z.object({
    formLabel: z.string().optional(),
    formHelpText: z.string().optional(),
    formPlaceholder: z.string().optional(),
    textarea: z.boolean().optional(),
    minLength: z.number().int().optional(),
    maxLength: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
        minLength: "min_length",
        maxLength: "max_length",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldTextProperties$;
(function (CustomFieldTextProperties$) {
    /** @deprecated use `CustomFieldTextProperties$inboundSchema` instead. */
    CustomFieldTextProperties$.inboundSchema = exports.CustomFieldTextProperties$inboundSchema;
    /** @deprecated use `CustomFieldTextProperties$outboundSchema` instead. */
    CustomFieldTextProperties$.outboundSchema = exports.CustomFieldTextProperties$outboundSchema;
})(CustomFieldTextProperties$ || (exports.CustomFieldTextProperties$ = CustomFieldTextProperties$ = {}));
function customFieldTextPropertiesToJSON(customFieldTextProperties) {
    return JSON.stringify(exports.CustomFieldTextProperties$outboundSchema.parse(customFieldTextProperties));
}
function customFieldTextPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldTextProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldTextProperties' from JSON`);
}
