// @polar-sh/sdk/dist/commonjs/models/components/customfieldupdatetext.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldUpdateText$ = exports.CustomFieldUpdateText$outboundSchema = exports.CustomFieldUpdateText$inboundSchema = exports.CustomFieldUpdateTextMetadata$ = exports.CustomFieldUpdateTextMetadata$outboundSchema = exports.CustomFieldUpdateTextMetadata$inboundSchema = void 0;
exports.customFieldUpdateTextMetadataToJSON = customFieldUpdateTextMetadataToJSON;
exports.customFieldUpdateTextMetadataFromJSON = customFieldUpdateTextMetadataFromJSON;
exports.customFieldUpdateTextToJSON = customFieldUpdateTextToJSON;
exports.customFieldUpdateTextFromJSON = customFieldUpdateTextFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldtextproperties_js_1 = require("./customfieldtextproperties.js");
/** @internal */
exports.CustomFieldUpdateTextMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CustomFieldUpdateTextMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldUpdateTextMetadata$;
(function (CustomFieldUpdateTextMetadata$) {
    /** @deprecated use `CustomFieldUpdateTextMetadata$inboundSchema` instead. */
    CustomFieldUpdateTextMetadata$.inboundSchema = exports.CustomFieldUpdateTextMetadata$inboundSchema;
    /** @deprecated use `CustomFieldUpdateTextMetadata$outboundSchema` instead. */
    CustomFieldUpdateTextMetadata$.outboundSchema = exports.CustomFieldUpdateTextMetadata$outboundSchema;
})(CustomFieldUpdateTextMetadata$ || (exports.CustomFieldUpdateTextMetadata$ = CustomFieldUpdateTextMetadata$ = {}));
function customFieldUpdateTextMetadataToJSON(customFieldUpdateTextMetadata) {
    return JSON.stringify(exports.CustomFieldUpdateTextMetadata$outboundSchema.parse(customFieldUpdateTextMetadata));
}
function customFieldUpdateTextMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldUpdateTextMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdateTextMetadata' from JSON`);
}
/** @internal */
exports.CustomFieldUpdateText$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.nullable(z.string()).optional(),
    slug: z.nullable(z.string()).optional(),
    type: z.literal("text"),
    properties: z.nullable(customfieldtextproperties_js_1.CustomFieldTextProperties$inboundSchema).optional(),
});
/** @internal */
exports.CustomFieldUpdateText$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.nullable(z.string()).optional(),
    slug: z.nullable(z.string()).optional(),
    type: z.literal("text"),
    properties: z.nullable(customfieldtextproperties_js_1.CustomFieldTextProperties$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldUpdateText$;
(function (CustomFieldUpdateText$) {
    /** @deprecated use `CustomFieldUpdateText$inboundSchema` instead. */
    CustomFieldUpdateText$.inboundSchema = exports.CustomFieldUpdateText$inboundSchema;
    /** @deprecated use `CustomFieldUpdateText$outboundSchema` instead. */
    CustomFieldUpdateText$.outboundSchema = exports.CustomFieldUpdateText$outboundSchema;
})(CustomFieldUpdateText$ || (exports.CustomFieldUpdateText$ = CustomFieldUpdateText$ = {}));
function customFieldUpdateTextToJSON(customFieldUpdateText) {
    return JSON.stringify(exports.CustomFieldUpdateText$outboundSchema.parse(customFieldUpdateText));
}
function customFieldUpdateTextFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldUpdateText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdateText' from JSON`);
}
