// @polar-sh/sdk/dist/commonjs/models/components/customfieldtext.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldText$ = exports.CustomFieldText$outboundSchema = exports.CustomFieldText$inboundSchema = exports.CustomFieldTextMetadata$ = exports.CustomFieldTextMetadata$outboundSchema = exports.CustomFieldTextMetadata$inboundSchema = void 0;
exports.customFieldTextMetadataToJSON = customFieldTextMetadataToJSON;
exports.customFieldTextMetadataFromJSON = customFieldTextMetadataFromJSON;
exports.customFieldTextToJSON = customFieldTextToJSON;
exports.customFieldTextFromJSON = customFieldTextFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldtextproperties_js_1 = require("./customfieldtextproperties.js");
/** @internal */
exports.CustomFieldTextMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CustomFieldTextMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldTextMetadata$;
(function (CustomFieldTextMetadata$) {
    /** @deprecated use `CustomFieldTextMetadata$inboundSchema` instead. */
    CustomFieldTextMetadata$.inboundSchema = exports.CustomFieldTextMetadata$inboundSchema;
    /** @deprecated use `CustomFieldTextMetadata$outboundSchema` instead. */
    CustomFieldTextMetadata$.outboundSchema = exports.CustomFieldTextMetadata$outboundSchema;
})(CustomFieldTextMetadata$ || (exports.CustomFieldTextMetadata$ = CustomFieldTextMetadata$ = {}));
function customFieldTextMetadataToJSON(customFieldTextMetadata) {
    return JSON.stringify(exports.CustomFieldTextMetadata$outboundSchema.parse(customFieldTextMetadata));
}
function customFieldTextMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldTextMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldTextMetadata' from JSON`);
}
/** @internal */
exports.CustomFieldText$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    type: z.literal("text"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.string(),
    properties: customfieldtextproperties_js_1.CustomFieldTextProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.CustomFieldText$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    type: z.literal("text"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.string(),
    properties: customfieldtextproperties_js_1.CustomFieldTextProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldText$;
(function (CustomFieldText$) {
    /** @deprecated use `CustomFieldText$inboundSchema` instead. */
    CustomFieldText$.inboundSchema = exports.CustomFieldText$inboundSchema;
    /** @deprecated use `CustomFieldText$outboundSchema` instead. */
    CustomFieldText$.outboundSchema = exports.CustomFieldText$outboundSchema;
})(CustomFieldText$ || (exports.CustomFieldText$ = CustomFieldText$ = {}));
function customFieldTextToJSON(customFieldText) {
    return JSON.stringify(exports.CustomFieldText$outboundSchema.parse(customFieldText));
}
function customFieldTextFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldText' from JSON`);
}
