// @polar-sh/sdk/dist/commonjs/models/components/customerportalcustomer.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalCustomer$ = exports.CustomerPortalCustomer$outboundSchema = exports.CustomerPortalCustomer$inboundSchema = exports.CustomerPortalCustomerTaxId$ = exports.CustomerPortalCustomerTaxId$outboundSchema = exports.CustomerPortalCustomerTaxId$inboundSchema = void 0;
exports.customerPortalCustomerTaxIdToJSON = customerPortalCustomerTaxIdToJSON;
exports.customerPortalCustomerTaxIdFromJSON = customerPortalCustomerTaxIdFromJSON;
exports.customerPortalCustomerToJSON = customerPortalCustomerToJSON;
exports.customerPortalCustomerFromJSON = customerPortalCustomerFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const customerportaloauthaccount_js_1 = require("./customerportaloauthaccount.js");
const taxidformat_js_1 = require("./taxidformat.js");
/** @internal */
exports.CustomerPortalCustomerTaxId$inboundSchema = z.union([z.string(), taxidformat_js_1.TaxIDFormat$inboundSchema]);
/** @internal */
exports.CustomerPortalCustomerTaxId$outboundSchema = z.union([z.string(), taxidformat_js_1.TaxIDFormat$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomerTaxId$;
(function (CustomerPortalCustomerTaxId$) {
    /** @deprecated use `CustomerPortalCustomerTaxId$inboundSchema` instead. */
    CustomerPortalCustomerTaxId$.inboundSchema = exports.CustomerPortalCustomerTaxId$inboundSchema;
    /** @deprecated use `CustomerPortalCustomerTaxId$outboundSchema` instead. */
    CustomerPortalCustomerTaxId$.outboundSchema = exports.CustomerPortalCustomerTaxId$outboundSchema;
})(CustomerPortalCustomerTaxId$ || (exports.CustomerPortalCustomerTaxId$ = CustomerPortalCustomerTaxId$ = {}));
function customerPortalCustomerTaxIdToJSON(customerPortalCustomerTaxId) {
    return JSON.stringify(exports.CustomerPortalCustomerTaxId$outboundSchema.parse(customerPortalCustomerTaxId));
}
function customerPortalCustomerTaxIdFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomerTaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomerTaxId' from JSON`);
}
/** @internal */
exports.CustomerPortalCustomer$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    email: z.string(),
    email_verified: z.boolean(),
    name: z.nullable(z.string()),
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable(z.union([z.string(), taxidformat_js_1.TaxIDFormat$inboundSchema])))),
    oauth_accounts: z.record(customerportaloauthaccount_js_1.CustomerPortalOAuthAccount$inboundSchema),
    default_payment_method_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "email_verified": "emailVerified",
        "billing_name": "billingName",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "oauth_accounts": "oauthAccounts",
        "default_payment_method_id": "defaultPaymentMethodId",
    });
});
/** @internal */
exports.CustomerPortalCustomer$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    email: z.string(),
    emailVerified: z.boolean(),
    name: z.nullable(z.string()),
    billingName: z.nullable(z.string()),
    billingAddress: z.nullable(address_js_1.Address$outboundSchema),
    taxId: z.nullable(z.array(z.nullable(z.union([z.string(), taxidformat_js_1.TaxIDFormat$outboundSchema])))),
    oauthAccounts: z.record(customerportaloauthaccount_js_1.CustomerPortalOAuthAccount$outboundSchema),
    defaultPaymentMethodId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        emailVerified: "email_verified",
        billingName: "billing_name",
        billingAddress: "billing_address",
        taxId: "tax_id",
        oauthAccounts: "oauth_accounts",
        defaultPaymentMethodId: "default_payment_method_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomer$;
(function (CustomerPortalCustomer$) {
    /** @deprecated use `CustomerPortalCustomer$inboundSchema` instead. */
    CustomerPortalCustomer$.inboundSchema = exports.CustomerPortalCustomer$inboundSchema;
    /** @deprecated use `CustomerPortalCustomer$outboundSchema` instead. */
    CustomerPortalCustomer$.outboundSchema = exports.CustomerPortalCustomer$outboundSchema;
})(CustomerPortalCustomer$ || (exports.CustomerPortalCustomer$ = CustomerPortalCustomer$ = {}));
function customerPortalCustomerToJSON(customerPortalCustomer) {
    return JSON.stringify(exports.CustomerPortalCustomer$outboundSchema.parse(customerPortalCustomer));
}
function customerPortalCustomerFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomer' from JSON`);
}
