// @polar-sh/sdk/dist/commonjs/models/components/customfieldcreatecheckbox.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldCreateCheckbox$ = exports.CustomFieldCreateCheckbox$outboundSchema = exports.CustomFieldCreateCheckbox$inboundSchema = exports.CustomFieldCreateCheckboxMetadata$ = exports.CustomFieldCreateCheckboxMetadata$outboundSchema = exports.CustomFieldCreateCheckboxMetadata$inboundSchema = void 0;
exports.customFieldCreateCheckboxMetadataToJSON = customFieldCreateCheckboxMetadataToJSON;
exports.customFieldCreateCheckboxMetadataFromJSON = customFieldCreateCheckboxMetadataFromJSON;
exports.customFieldCreateCheckboxToJSON = customFieldCreateCheckboxToJSON;
exports.customFieldCreateCheckboxFromJSON = customFieldCreateCheckboxFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldcheckboxproperties_js_1 = require("./customfieldcheckboxproperties.js");
/** @internal */
exports.CustomFieldCreateCheckboxMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CustomFieldCreateCheckboxMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldCreateCheckboxMetadata$;
(function (CustomFieldCreateCheckboxMetadata$) {
    /** @deprecated use `CustomFieldCreateCheckboxMetadata$inboundSchema` instead. */
    CustomFieldCreateCheckboxMetadata$.inboundSchema = exports.CustomFieldCreateCheckboxMetadata$inboundSchema;
    /** @deprecated use `CustomFieldCreateCheckboxMetadata$outboundSchema` instead. */
    CustomFieldCreateCheckboxMetadata$.outboundSchema = exports.CustomFieldCreateCheckboxMetadata$outboundSchema;
})(CustomFieldCreateCheckboxMetadata$ || (exports.CustomFieldCreateCheckboxMetadata$ = CustomFieldCreateCheckboxMetadata$ = {}));
function customFieldCreateCheckboxMetadataToJSON(customFieldCreateCheckboxMetadata) {
    return JSON.stringify(exports.CustomFieldCreateCheckboxMetadata$outboundSchema.parse(customFieldCreateCheckboxMetadata));
}
function customFieldCreateCheckboxMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldCreateCheckboxMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCreateCheckboxMetadata' from JSON`);
}
/** @internal */
exports.CustomFieldCreateCheckbox$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: customfieldcheckboxproperties_js_1.CustomFieldCheckboxProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.CustomFieldCreateCheckbox$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: customfieldcheckboxproperties_js_1.CustomFieldCheckboxProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldCreateCheckbox$;
(function (CustomFieldCreateCheckbox$) {
    /** @deprecated use `CustomFieldCreateCheckbox$inboundSchema` instead. */
    CustomFieldCreateCheckbox$.inboundSchema = exports.CustomFieldCreateCheckbox$inboundSchema;
    /** @deprecated use `CustomFieldCreateCheckbox$outboundSchema` instead. */
    CustomFieldCreateCheckbox$.outboundSchema = exports.CustomFieldCreateCheckbox$outboundSchema;
})(CustomFieldCreateCheckbox$ || (exports.CustomFieldCreateCheckbox$ = CustomFieldCreateCheckbox$ = {}));
function customFieldCreateCheckboxToJSON(customFieldCreateCheckbox) {
    return JSON.stringify(exports.CustomFieldCreateCheckbox$outboundSchema.parse(customFieldCreateCheckbox));
}
function customFieldCreateCheckboxFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldCreateCheckbox$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCreateCheckbox' from JSON`);
}
