// @polar-sh/sdk/dist/commonjs/models/components/customercustomermetersortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerCustomerMeterSortProperty$ = exports.CustomerCustomerMeterSortProperty$outboundSchema = exports.CustomerCustomerMeterSortProperty$inboundSchema = exports.CustomerCustomerMeterSortProperty = void 0;
const z = __importStar(require("zod"));
exports.CustomerCustomerMeterSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    ModifiedAt: "modified_at",
    MinusModifiedAt: "-modified_at",
    MeterId: "meter_id",
    MinusMeterId: "-meter_id",
    MeterName: "meter_name",
    MinusMeterName: "-meter_name",
    ConsumedUnits: "consumed_units",
    MinusConsumedUnits: "-consumed_units",
    CreditedUnits: "credited_units",
    MinusCreditedUnits: "-credited_units",
    Balance: "balance",
    MinusBalance: "-balance",
};
/** @internal */
exports.CustomerCustomerMeterSortProperty$inboundSchema = z.nativeEnum(exports.CustomerCustomerMeterSortProperty);
/** @internal */
exports.CustomerCustomerMeterSortProperty$outboundSchema = exports.CustomerCustomerMeterSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerCustomerMeterSortProperty$;
(function (CustomerCustomerMeterSortProperty$) {
    /** @deprecated use `CustomerCustomerMeterSortProperty$inboundSchema` instead. */
    CustomerCustomerMeterSortProperty$.inboundSchema = exports.CustomerCustomerMeterSortProperty$inboundSchema;
    /** @deprecated use `CustomerCustomerMeterSortProperty$outboundSchema` instead. */
    CustomerCustomerMeterSortProperty$.outboundSchema = exports.CustomerCustomerMeterSortProperty$outboundSchema;
})(CustomerCustomerMeterSortProperty$ || (exports.CustomerCustomerMeterSortProperty$ = CustomerCustomerMeterSortProperty$ = {}));
