// @polar-sh/sdk/dist/commonjs/models/components/customerorderconfirmpayment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrderConfirmPayment$ = exports.CustomerOrderConfirmPayment$outboundSchema = exports.CustomerOrderConfirmPayment$inboundSchema = void 0;
exports.customerOrderConfirmPaymentToJSON = customerOrderConfirmPaymentToJSON;
exports.customerOrderConfirmPaymentFromJSON = customerOrderConfirmPaymentFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const paymentprocessor_js_1 = require("./paymentprocessor.js");
/** @internal */
exports.CustomerOrderConfirmPayment$inboundSchema = z.object({
    confirmation_token_id: z.nullable(z.string()).optional(),
    payment_method_id: z.nullable(z.string()).optional(),
    payment_processor: paymentprocessor_js_1.PaymentProcessor$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "confirmation_token_id": "confirmationTokenId",
        "payment_method_id": "paymentMethodId",
        "payment_processor": "paymentProcessor",
    });
});
/** @internal */
exports.CustomerOrderConfirmPayment$outboundSchema = z.object({
    confirmationTokenId: z.nullable(z.string()).optional(),
    paymentMethodId: z.nullable(z.string()).optional(),
    paymentProcessor: paymentprocessor_js_1.PaymentProcessor$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        confirmationTokenId: "confirmation_token_id",
        paymentMethodId: "payment_method_id",
        paymentProcessor: "payment_processor",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerOrderConfirmPayment$;
(function (CustomerOrderConfirmPayment$) {
    /** @deprecated use `CustomerOrderConfirmPayment$inboundSchema` instead. */
    CustomerOrderConfirmPayment$.inboundSchema = exports.CustomerOrderConfirmPayment$inboundSchema;
    /** @deprecated use `CustomerOrderConfirmPayment$outboundSchema` instead. */
    CustomerOrderConfirmPayment$.outboundSchema = exports.CustomerOrderConfirmPayment$outboundSchema;
})(CustomerOrderConfirmPayment$ || (exports.CustomerOrderConfirmPayment$ = CustomerOrderConfirmPayment$ = {}));
function customerOrderConfirmPaymentToJSON(customerOrderConfirmPayment) {
    return JSON.stringify(exports.CustomerOrderConfirmPayment$outboundSchema.parse(customerOrderConfirmPayment));
}
function customerOrderConfirmPaymentFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerOrderConfirmPayment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrderConfirmPayment' from JSON`);
}
