// @polar-sh/sdk/dist/commonjs/models/components/customerbenefitgrantdownloadables.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerBenefitGrantDownloadables$ = exports.CustomerBenefitGrantDownloadables$outboundSchema = exports.CustomerBenefitGrantDownloadables$inboundSchema = void 0;
exports.customerBenefitGrantDownloadablesToJSON = customerBenefitGrantDownloadablesToJSON;
exports.customerBenefitGrantDownloadablesFromJSON = customerBenefitGrantDownloadablesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdownloadablessubscriber_js_1 = require("./benefitdownloadablessubscriber.js");
const benefitgrantdownloadablesproperties_js_1 = require("./benefitgrantdownloadablesproperties.js");
const customerportalcustomer_js_1 = require("./customerportalcustomer.js");
/** @internal */
exports.CustomerBenefitGrantDownloadables$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    benefit_id: z.string(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    is_granted: z.boolean(),
    is_revoked: z.boolean(),
    customer: customerportalcustomer_js_1.CustomerPortalCustomer$inboundSchema,
    benefit: benefitdownloadablessubscriber_js_1.BenefitDownloadablesSubscriber$inboundSchema,
    properties: benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "revoked_at": "revokedAt",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "is_granted": "isGranted",
        "is_revoked": "isRevoked",
    });
});
/** @internal */
exports.CustomerBenefitGrantDownloadables$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    benefitId: z.string(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    isGranted: z.boolean(),
    isRevoked: z.boolean(),
    customer: customerportalcustomer_js_1.CustomerPortalCustomer$outboundSchema,
    benefit: benefitdownloadablessubscriber_js_1.BenefitDownloadablesSubscriber$outboundSchema,
    properties: benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        revokedAt: "revoked_at",
        customerId: "customer_id",
        benefitId: "benefit_id",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        isGranted: "is_granted",
        isRevoked: "is_revoked",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerBenefitGrantDownloadables$;
(function (CustomerBenefitGrantDownloadables$) {
    /** @deprecated use `CustomerBenefitGrantDownloadables$inboundSchema` instead. */
    CustomerBenefitGrantDownloadables$.inboundSchema = exports.CustomerBenefitGrantDownloadables$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantDownloadables$outboundSchema` instead. */
    CustomerBenefitGrantDownloadables$.outboundSchema = exports.CustomerBenefitGrantDownloadables$outboundSchema;
})(CustomerBenefitGrantDownloadables$ || (exports.CustomerBenefitGrantDownloadables$ = CustomerBenefitGrantDownloadables$ = {}));
function customerBenefitGrantDownloadablesToJSON(customerBenefitGrantDownloadables) {
    return JSON.stringify(exports.CustomerBenefitGrantDownloadables$outboundSchema.parse(customerBenefitGrantDownloadables));
}
function customerBenefitGrantDownloadablesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerBenefitGrantDownloadables$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantDownloadables' from JSON`);
}
