// @polar-sh/sdk/dist/commonjs/models/components/customfield.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomField$ = exports.CustomField$outboundSchema = exports.CustomField$inboundSchema = void 0;
exports.customFieldToJSON = customFieldToJSON;
exports.customFieldFromJSON = customFieldFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldcheckbox_js_1 = require("./customfieldcheckbox.js");
const customfielddate_js_1 = require("./customfielddate.js");
const customfieldnumber_js_1 = require("./customfieldnumber.js");
const customfieldselect_js_1 = require("./customfieldselect.js");
const customfieldtext_js_1 = require("./customfieldtext.js");
/** @internal */
exports.CustomField$inboundSchema = z.union([
    customfieldcheckbox_js_1.CustomFieldCheckbox$inboundSchema.and(z.object({ type: z.literal("checkbox") }).transform((v) => ({
        type: v.type,
    }))),
    customfielddate_js_1.CustomFieldDate$inboundSchema.and(z.object({ type: z.literal("date") }).transform((v) => ({ type: v.type }))),
    customfieldnumber_js_1.CustomFieldNumber$inboundSchema.and(z.object({ type: z.literal("number") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldselect_js_1.CustomFieldSelect$inboundSchema.and(z.object({ type: z.literal("select") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldtext_js_1.CustomFieldText$inboundSchema.and(z.object({ type: z.literal("text") }).transform((v) => ({ type: v.type }))),
]);
/** @internal */
exports.CustomField$outboundSchema = z.union([
    customfieldcheckbox_js_1.CustomFieldCheckbox$outboundSchema.and(z.object({ type: z.literal("checkbox") }).transform((v) => ({
        type: v.type,
    }))),
    customfielddate_js_1.CustomFieldDate$outboundSchema.and(z.object({ type: z.literal("date") }).transform((v) => ({ type: v.type }))),
    customfieldnumber_js_1.CustomFieldNumber$outboundSchema.and(z.object({ type: z.literal("number") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldselect_js_1.CustomFieldSelect$outboundSchema.and(z.object({ type: z.literal("select") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldtext_js_1.CustomFieldText$outboundSchema.and(z.object({ type: z.literal("text") }).transform((v) => ({ type: v.type }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomField$;
(function (CustomField$) {
    /** @deprecated use `CustomField$inboundSchema` instead. */
    CustomField$.inboundSchema = exports.CustomField$inboundSchema;
    /** @deprecated use `CustomField$outboundSchema` instead. */
    CustomField$.outboundSchema = exports.CustomField$outboundSchema;
})(CustomField$ || (exports.CustomField$ = CustomField$ = {}));
function customFieldToJSON(customField) {
    return JSON.stringify(exports.CustomField$outboundSchema.parse(customField));
}
function customFieldFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomField$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomField' from JSON`);
}
