// @polar-sh/sdk/dist/commonjs/models/components/customersessioncustomerexternalidcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSessionCustomerExternalIDCreate$ = exports.CustomerSessionCustomerExternalIDCreate$outboundSchema = exports.CustomerSessionCustomerExternalIDCreate$inboundSchema = void 0;
exports.customerSessionCustomerExternalIDCreateToJSON = customerSessionCustomerExternalIDCreateToJSON;
exports.customerSessionCustomerExternalIDCreateFromJSON = customerSessionCustomerExternalIDCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerSessionCustomerExternalIDCreate$inboundSchema = z.object({
    external_customer_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
exports.CustomerSessionCustomerExternalIDCreate$outboundSchema = z.object({
    externalCustomerId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerSessionCustomerExternalIDCreate$;
(function (CustomerSessionCustomerExternalIDCreate$) {
    /** @deprecated use `CustomerSessionCustomerExternalIDCreate$inboundSchema` instead. */
    CustomerSessionCustomerExternalIDCreate$.inboundSchema = exports.CustomerSessionCustomerExternalIDCreate$inboundSchema;
    /** @deprecated use `CustomerSessionCustomerExternalIDCreate$outboundSchema` instead. */
    CustomerSessionCustomerExternalIDCreate$.outboundSchema = exports.CustomerSessionCustomerExternalIDCreate$outboundSchema;
})(CustomerSessionCustomerExternalIDCreate$ || (exports.CustomerSessionCustomerExternalIDCreate$ = CustomerSessionCustomerExternalIDCreate$ = {}));
function customerSessionCustomerExternalIDCreateToJSON(customerSessionCustomerExternalIDCreate) {
    return JSON.stringify(exports.CustomerSessionCustomerExternalIDCreate$outboundSchema.parse(customerSessionCustomerExternalIDCreate));
}
function customerSessionCustomerExternalIDCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerSessionCustomerExternalIDCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSessionCustomerExternalIDCreate' from JSON`);
}
