// @polar-sh/sdk/dist/commonjs/models/components/customfieldcheckbox.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldCheckbox$ = exports.CustomFieldCheckbox$outboundSchema = exports.CustomFieldCheckbox$inboundSchema = exports.CustomFieldCheckboxMetadata$ = exports.CustomFieldCheckboxMetadata$outboundSchema = exports.CustomFieldCheckboxMetadata$inboundSchema = void 0;
exports.customFieldCheckboxMetadataToJSON = customFieldCheckboxMetadataToJSON;
exports.customFieldCheckboxMetadataFromJSON = customFieldCheckboxMetadataFromJSON;
exports.customFieldCheckboxToJSON = customFieldCheckboxToJSON;
exports.customFieldCheckboxFromJSON = customFieldCheckboxFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldcheckboxproperties_js_1 = require("./customfieldcheckboxproperties.js");
/** @internal */
exports.CustomFieldCheckboxMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CustomFieldCheckboxMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldCheckboxMetadata$;
(function (CustomFieldCheckboxMetadata$) {
    /** @deprecated use `CustomFieldCheckboxMetadata$inboundSchema` instead. */
    CustomFieldCheckboxMetadata$.inboundSchema = exports.CustomFieldCheckboxMetadata$inboundSchema;
    /** @deprecated use `CustomFieldCheckboxMetadata$outboundSchema` instead. */
    CustomFieldCheckboxMetadata$.outboundSchema = exports.CustomFieldCheckboxMetadata$outboundSchema;
})(CustomFieldCheckboxMetadata$ || (exports.CustomFieldCheckboxMetadata$ = CustomFieldCheckboxMetadata$ = {}));
function customFieldCheckboxMetadataToJSON(customFieldCheckboxMetadata) {
    return JSON.stringify(exports.CustomFieldCheckboxMetadata$outboundSchema.parse(customFieldCheckboxMetadata));
}
function customFieldCheckboxMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldCheckboxMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCheckboxMetadata' from JSON`);
}
/** @internal */
exports.CustomFieldCheckbox$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.string(),
    properties: customfieldcheckboxproperties_js_1.CustomFieldCheckboxProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.CustomFieldCheckbox$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.string(),
    properties: customfieldcheckboxproperties_js_1.CustomFieldCheckboxProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldCheckbox$;
(function (CustomFieldCheckbox$) {
    /** @deprecated use `CustomFieldCheckbox$inboundSchema` instead. */
    CustomFieldCheckbox$.inboundSchema = exports.CustomFieldCheckbox$inboundSchema;
    /** @deprecated use `CustomFieldCheckbox$outboundSchema` instead. */
    CustomFieldCheckbox$.outboundSchema = exports.CustomFieldCheckbox$outboundSchema;
})(CustomFieldCheckbox$ || (exports.CustomFieldCheckbox$ = CustomFieldCheckbox$ = {}));
function customFieldCheckboxToJSON(customFieldCheckbox) {
    return JSON.stringify(exports.CustomFieldCheckbox$outboundSchema.parse(customFieldCheckbox));
}
function customFieldCheckboxFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldCheckbox$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCheckbox' from JSON`);
}
