// @polar-sh/sdk/dist/commonjs/models/components/customerorder.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrder$ = exports.CustomerOrder$outboundSchema = exports.CustomerOrder$inboundSchema = void 0;
exports.customerOrderToJSON = customerOrderToJSON;
exports.customerOrderFromJSON = customerOrderFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const customerorderproduct_js_1 = require("./customerorderproduct.js");
const customerordersubscription_js_1 = require("./customerordersubscription.js");
const orderbillingreason_js_1 = require("./orderbillingreason.js");
const orderitemschema_js_1 = require("./orderitemschema.js");
const orderstatus_js_1 = require("./orderstatus.js");
/** @internal */
exports.CustomerOrder$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    status: orderstatus_js_1.OrderStatus$inboundSchema,
    paid: z.boolean(),
    subtotal_amount: z.number().int(),
    discount_amount: z.number().int(),
    net_amount: z.number().int(),
    tax_amount: z.number().int(),
    total_amount: z.number().int(),
    refunded_amount: z.number().int(),
    refunded_tax_amount: z.number().int(),
    currency: z.string(),
    billing_reason: orderbillingreason_js_1.OrderBillingReason$inboundSchema,
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
    is_invoice_generated: z.boolean(),
    customer_id: z.string(),
    product_id: z.string(),
    discount_id: z.nullable(z.string()),
    subscription_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    user_id: z.string(),
    product: customerorderproduct_js_1.CustomerOrderProduct$inboundSchema,
    subscription: z.nullable(customerordersubscription_js_1.CustomerOrderSubscription$inboundSchema),
    items: z.array(orderitemschema_js_1.OrderItemSchema$inboundSchema),
    next_payment_attempt_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subtotal_amount": "subtotalAmount",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "refunded_amount": "refundedAmount",
        "refunded_tax_amount": "refundedTaxAmount",
        "billing_reason": "billingReason",
        "billing_name": "billingName",
        "billing_address": "billingAddress",
        "is_invoice_generated": "isInvoiceGenerated",
        "customer_id": "customerId",
        "product_id": "productId",
        "discount_id": "discountId",
        "subscription_id": "subscriptionId",
        "checkout_id": "checkoutId",
        "user_id": "userId",
        "next_payment_attempt_at": "nextPaymentAttemptAt",
    });
});
/** @internal */
exports.CustomerOrder$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    status: orderstatus_js_1.OrderStatus$outboundSchema,
    paid: z.boolean(),
    subtotalAmount: z.number().int(),
    discountAmount: z.number().int(),
    netAmount: z.number().int(),
    taxAmount: z.number().int(),
    totalAmount: z.number().int(),
    refundedAmount: z.number().int(),
    refundedTaxAmount: z.number().int(),
    currency: z.string(),
    billingReason: orderbillingreason_js_1.OrderBillingReason$outboundSchema,
    billingName: z.nullable(z.string()),
    billingAddress: z.nullable(address_js_1.Address$outboundSchema),
    isInvoiceGenerated: z.boolean(),
    customerId: z.string(),
    productId: z.string(),
    discountId: z.nullable(z.string()),
    subscriptionId: z.nullable(z.string()),
    checkoutId: z.nullable(z.string()),
    userId: z.string(),
    product: customerorderproduct_js_1.CustomerOrderProduct$outboundSchema,
    subscription: z.nullable(customerordersubscription_js_1.CustomerOrderSubscription$outboundSchema),
    items: z.array(orderitemschema_js_1.OrderItemSchema$outboundSchema),
    nextPaymentAttemptAt: z.nullable(z.date().transform(v => v.toISOString()))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        subtotalAmount: "subtotal_amount",
        discountAmount: "discount_amount",
        netAmount: "net_amount",
        taxAmount: "tax_amount",
        totalAmount: "total_amount",
        refundedAmount: "refunded_amount",
        refundedTaxAmount: "refunded_tax_amount",
        billingReason: "billing_reason",
        billingName: "billing_name",
        billingAddress: "billing_address",
        isInvoiceGenerated: "is_invoice_generated",
        customerId: "customer_id",
        productId: "product_id",
        discountId: "discount_id",
        subscriptionId: "subscription_id",
        checkoutId: "checkout_id",
        userId: "user_id",
        nextPaymentAttemptAt: "next_payment_attempt_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerOrder$;
(function (CustomerOrder$) {
    /** @deprecated use `CustomerOrder$inboundSchema` instead. */
    CustomerOrder$.inboundSchema = exports.CustomerOrder$inboundSchema;
    /** @deprecated use `CustomerOrder$outboundSchema` instead. */
    CustomerOrder$.outboundSchema = exports.CustomerOrder$outboundSchema;
})(CustomerOrder$ || (exports.CustomerOrder$ = CustomerOrder$ = {}));
function customerOrderToJSON(customerOrder) {
    return JSON.stringify(exports.CustomerOrder$outboundSchema.parse(customerOrder));
}
function customerOrderFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerOrder$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrder' from JSON`);
}
