// @polar-sh/sdk/dist/commonjs/models/components/customerportaloauthaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalOAuthAccount$ = exports.CustomerPortalOAuthAccount$outboundSchema = exports.CustomerPortalOAuthAccount$inboundSchema = void 0;
exports.customerPortalOAuthAccountToJSON = customerPortalOAuthAccountToJSON;
exports.customerPortalOAuthAccountFromJSON = customerPortalOAuthAccountFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalOAuthAccount$inboundSchema = z.object({
    account_id: z.string(),
    account_username: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "account_username": "accountUsername",
    });
});
/** @internal */
exports.CustomerPortalOAuthAccount$outboundSchema = z.object({
    accountId: z.string(),
    accountUsername: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        accountUsername: "account_username",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOAuthAccount$;
(function (CustomerPortalOAuthAccount$) {
    /** @deprecated use `CustomerPortalOAuthAccount$inboundSchema` instead. */
    CustomerPortalOAuthAccount$.inboundSchema = exports.CustomerPortalOAuthAccount$inboundSchema;
    /** @deprecated use `CustomerPortalOAuthAccount$outboundSchema` instead. */
    CustomerPortalOAuthAccount$.outboundSchema = exports.CustomerPortalOAuthAccount$outboundSchema;
})(CustomerPortalOAuthAccount$ || (exports.CustomerPortalOAuthAccount$ = CustomerPortalOAuthAccount$ = {}));
function customerPortalOAuthAccountToJSON(customerPortalOAuthAccount) {
    return JSON.stringify(exports.CustomerPortalOAuthAccount$outboundSchema.parse(customerPortalOAuthAccount));
}
function customerPortalOAuthAccountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOAuthAccount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOAuthAccount' from JSON`);
}
