// @polar-sh/sdk/dist/commonjs/models/components/discountpercentageonceforeverdurationcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountPercentageOnceForeverDurationCreate$ = exports.DiscountPercentageOnceForeverDurationCreate$outboundSchema = exports.DiscountPercentageOnceForeverDurationCreate$inboundSchema = exports.DiscountPercentageOnceForeverDurationCreateMetadata$ = exports.DiscountPercentageOnceForeverDurationCreateMetadata$outboundSchema = exports.DiscountPercentageOnceForeverDurationCreateMetadata$inboundSchema = void 0;
exports.discountPercentageOnceForeverDurationCreateMetadataToJSON = discountPercentageOnceForeverDurationCreateMetadataToJSON;
exports.discountPercentageOnceForeverDurationCreateMetadataFromJSON = discountPercentageOnceForeverDurationCreateMetadataFromJSON;
exports.discountPercentageOnceForeverDurationCreateToJSON = discountPercentageOnceForeverDurationCreateToJSON;
exports.discountPercentageOnceForeverDurationCreateFromJSON = discountPercentageOnceForeverDurationCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountPercentageOnceForeverDurationCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountPercentageOnceForeverDurationCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageOnceForeverDurationCreateMetadata$;
(function (DiscountPercentageOnceForeverDurationCreateMetadata$) {
    /** @deprecated use `DiscountPercentageOnceForeverDurationCreateMetadata$inboundSchema` instead. */
    DiscountPercentageOnceForeverDurationCreateMetadata$.inboundSchema = exports.DiscountPercentageOnceForeverDurationCreateMetadata$inboundSchema;
    /** @deprecated use `DiscountPercentageOnceForeverDurationCreateMetadata$outboundSchema` instead. */
    DiscountPercentageOnceForeverDurationCreateMetadata$.outboundSchema = exports.DiscountPercentageOnceForeverDurationCreateMetadata$outboundSchema;
})(DiscountPercentageOnceForeverDurationCreateMetadata$ || (exports.DiscountPercentageOnceForeverDurationCreateMetadata$ = DiscountPercentageOnceForeverDurationCreateMetadata$ = {}));
function discountPercentageOnceForeverDurationCreateMetadataToJSON(discountPercentageOnceForeverDurationCreateMetadata) {
    return JSON.stringify(exports.DiscountPercentageOnceForeverDurationCreateMetadata$outboundSchema.parse(discountPercentageOnceForeverDurationCreateMetadata));
}
function discountPercentageOnceForeverDurationCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageOnceForeverDurationCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDurationCreateMetadata' from JSON`);
}
/** @internal */
exports.DiscountPercentageOnceForeverDurationCreate$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    type: discounttype_js_1.DiscountType$inboundSchema,
    basis_points: z.number().int(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    code: z.nullable(z.string()).optional(),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    max_redemptions: z.nullable(z.number().int()).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "basis_points": "basisPoints",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountPercentageOnceForeverDurationCreate$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    type: discounttype_js_1.DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    code: z.nullable(z.string()).optional(),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    maxRedemptions: z.nullable(z.number().int()).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        basisPoints: "basis_points",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageOnceForeverDurationCreate$;
(function (DiscountPercentageOnceForeverDurationCreate$) {
    /** @deprecated use `DiscountPercentageOnceForeverDurationCreate$inboundSchema` instead. */
    DiscountPercentageOnceForeverDurationCreate$.inboundSchema = exports.DiscountPercentageOnceForeverDurationCreate$inboundSchema;
    /** @deprecated use `DiscountPercentageOnceForeverDurationCreate$outboundSchema` instead. */
    DiscountPercentageOnceForeverDurationCreate$.outboundSchema = exports.DiscountPercentageOnceForeverDurationCreate$outboundSchema;
})(DiscountPercentageOnceForeverDurationCreate$ || (exports.DiscountPercentageOnceForeverDurationCreate$ = DiscountPercentageOnceForeverDurationCreate$ = {}));
function discountPercentageOnceForeverDurationCreateToJSON(discountPercentageOnceForeverDurationCreate) {
    return JSON.stringify(exports.DiscountPercentageOnceForeverDurationCreate$outboundSchema.parse(discountPercentageOnceForeverDurationCreate));
}
function discountPercentageOnceForeverDurationCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageOnceForeverDurationCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDurationCreate' from JSON`);
}
