// @polar-sh/sdk/dist/commonjs/models/components/discountpercentagerepeatduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountPercentageRepeatDuration$ = exports.DiscountPercentageRepeatDuration$outboundSchema = exports.DiscountPercentageRepeatDuration$inboundSchema = exports.DiscountPercentageRepeatDurationMetadata$ = exports.DiscountPercentageRepeatDurationMetadata$outboundSchema = exports.DiscountPercentageRepeatDurationMetadata$inboundSchema = void 0;
exports.discountPercentageRepeatDurationMetadataToJSON = discountPercentageRepeatDurationMetadataToJSON;
exports.discountPercentageRepeatDurationMetadataFromJSON = discountPercentageRepeatDurationMetadataFromJSON;
exports.discountPercentageRepeatDurationToJSON = discountPercentageRepeatDurationToJSON;
exports.discountPercentageRepeatDurationFromJSON = discountPercentageRepeatDurationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discountproduct_js_1 = require("./discountproduct.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountPercentageRepeatDurationMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountPercentageRepeatDurationMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageRepeatDurationMetadata$;
(function (DiscountPercentageRepeatDurationMetadata$) {
    /** @deprecated use `DiscountPercentageRepeatDurationMetadata$inboundSchema` instead. */
    DiscountPercentageRepeatDurationMetadata$.inboundSchema = exports.DiscountPercentageRepeatDurationMetadata$inboundSchema;
    /** @deprecated use `DiscountPercentageRepeatDurationMetadata$outboundSchema` instead. */
    DiscountPercentageRepeatDurationMetadata$.outboundSchema = exports.DiscountPercentageRepeatDurationMetadata$outboundSchema;
})(DiscountPercentageRepeatDurationMetadata$ || (exports.DiscountPercentageRepeatDurationMetadata$ = DiscountPercentageRepeatDurationMetadata$ = {}));
function discountPercentageRepeatDurationMetadataToJSON(discountPercentageRepeatDurationMetadata) {
    return JSON.stringify(exports.DiscountPercentageRepeatDurationMetadata$outboundSchema.parse(discountPercentageRepeatDurationMetadata));
}
function discountPercentageRepeatDurationMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageRepeatDurationMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageRepeatDurationMetadata' from JSON`);
}
/** @internal */
exports.DiscountPercentageRepeatDuration$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    duration_in_months: z.number().int(),
    type: discounttype_js_1.DiscountType$inboundSchema,
    basis_points: z.number().int(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "duration_in_months": "durationInMonths",
        "basis_points": "basisPoints",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountPercentageRepeatDuration$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    durationInMonths: z.number().int(),
    type: discounttype_js_1.DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageRepeatDuration$;
(function (DiscountPercentageRepeatDuration$) {
    /** @deprecated use `DiscountPercentageRepeatDuration$inboundSchema` instead. */
    DiscountPercentageRepeatDuration$.inboundSchema = exports.DiscountPercentageRepeatDuration$inboundSchema;
    /** @deprecated use `DiscountPercentageRepeatDuration$outboundSchema` instead. */
    DiscountPercentageRepeatDuration$.outboundSchema = exports.DiscountPercentageRepeatDuration$outboundSchema;
})(DiscountPercentageRepeatDuration$ || (exports.DiscountPercentageRepeatDuration$ = DiscountPercentageRepeatDuration$ = {}));
function discountPercentageRepeatDurationToJSON(discountPercentageRepeatDuration) {
    return JSON.stringify(exports.DiscountPercentageRepeatDuration$outboundSchema.parse(discountPercentageRepeatDuration));
}
function discountPercentageRepeatDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageRepeatDuration' from JSON`);
}
