// @polar-sh/sdk/dist/commonjs/models/components/discountfixedonceforeverduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountFixedOnceForeverDuration$ = exports.DiscountFixedOnceForeverDuration$outboundSchema = exports.DiscountFixedOnceForeverDuration$inboundSchema = exports.DiscountFixedOnceForeverDurationMetadata$ = exports.DiscountFixedOnceForeverDurationMetadata$outboundSchema = exports.DiscountFixedOnceForeverDurationMetadata$inboundSchema = void 0;
exports.discountFixedOnceForeverDurationMetadataToJSON = discountFixedOnceForeverDurationMetadataToJSON;
exports.discountFixedOnceForeverDurationMetadataFromJSON = discountFixedOnceForeverDurationMetadataFromJSON;
exports.discountFixedOnceForeverDurationToJSON = discountFixedOnceForeverDurationToJSON;
exports.discountFixedOnceForeverDurationFromJSON = discountFixedOnceForeverDurationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discountproduct_js_1 = require("./discountproduct.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountFixedOnceForeverDurationMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountFixedOnceForeverDurationMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountFixedOnceForeverDurationMetadata$;
(function (DiscountFixedOnceForeverDurationMetadata$) {
    /** @deprecated use `DiscountFixedOnceForeverDurationMetadata$inboundSchema` instead. */
    DiscountFixedOnceForeverDurationMetadata$.inboundSchema = exports.DiscountFixedOnceForeverDurationMetadata$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDurationMetadata$outboundSchema` instead. */
    DiscountFixedOnceForeverDurationMetadata$.outboundSchema = exports.DiscountFixedOnceForeverDurationMetadata$outboundSchema;
})(DiscountFixedOnceForeverDurationMetadata$ || (exports.DiscountFixedOnceForeverDurationMetadata$ = DiscountFixedOnceForeverDurationMetadata$ = {}));
function discountFixedOnceForeverDurationMetadataToJSON(discountFixedOnceForeverDurationMetadata) {
    return JSON.stringify(exports.DiscountFixedOnceForeverDurationMetadata$outboundSchema.parse(discountFixedOnceForeverDurationMetadata));
}
function discountFixedOnceForeverDurationMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountFixedOnceForeverDurationMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDurationMetadata' from JSON`);
}
/** @internal */
exports.DiscountFixedOnceForeverDuration$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    type: discounttype_js_1.DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountFixedOnceForeverDuration$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    type: discounttype_js_1.DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountFixedOnceForeverDuration$;
(function (DiscountFixedOnceForeverDuration$) {
    /** @deprecated use `DiscountFixedOnceForeverDuration$inboundSchema` instead. */
    DiscountFixedOnceForeverDuration$.inboundSchema = exports.DiscountFixedOnceForeverDuration$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDuration$outboundSchema` instead. */
    DiscountFixedOnceForeverDuration$.outboundSchema = exports.DiscountFixedOnceForeverDuration$outboundSchema;
})(DiscountFixedOnceForeverDuration$ || (exports.DiscountFixedOnceForeverDuration$ = DiscountFixedOnceForeverDuration$ = {}));
function discountFixedOnceForeverDurationToJSON(discountFixedOnceForeverDuration) {
    return JSON.stringify(exports.DiscountFixedOnceForeverDuration$outboundSchema.parse(discountFixedOnceForeverDuration));
}
function discountFixedOnceForeverDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountFixedOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDuration' from JSON`);
}
