// @polar-sh/sdk/dist/commonjs/models/components/discountfixedrepeatdurationbase.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountFixedRepeatDurationBase$ = exports.DiscountFixedRepeatDurationBase$outboundSchema = exports.DiscountFixedRepeatDurationBase$inboundSchema = exports.DiscountFixedRepeatDurationBaseMetadata$ = exports.DiscountFixedRepeatDurationBaseMetadata$outboundSchema = exports.DiscountFixedRepeatDurationBaseMetadata$inboundSchema = void 0;
exports.discountFixedRepeatDurationBaseMetadataToJSON = discountFixedRepeatDurationBaseMetadataToJSON;
exports.discountFixedRepeatDurationBaseMetadataFromJSON = discountFixedRepeatDurationBaseMetadataFromJSON;
exports.discountFixedRepeatDurationBaseToJSON = discountFixedRepeatDurationBaseToJSON;
exports.discountFixedRepeatDurationBaseFromJSON = discountFixedRepeatDurationBaseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountFixedRepeatDurationBaseMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountFixedRepeatDurationBaseMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountFixedRepeatDurationBaseMetadata$;
(function (DiscountFixedRepeatDurationBaseMetadata$) {
    /** @deprecated use `DiscountFixedRepeatDurationBaseMetadata$inboundSchema` instead. */
    DiscountFixedRepeatDurationBaseMetadata$.inboundSchema = exports.DiscountFixedRepeatDurationBaseMetadata$inboundSchema;
    /** @deprecated use `DiscountFixedRepeatDurationBaseMetadata$outboundSchema` instead. */
    DiscountFixedRepeatDurationBaseMetadata$.outboundSchema = exports.DiscountFixedRepeatDurationBaseMetadata$outboundSchema;
})(DiscountFixedRepeatDurationBaseMetadata$ || (exports.DiscountFixedRepeatDurationBaseMetadata$ = DiscountFixedRepeatDurationBaseMetadata$ = {}));
function discountFixedRepeatDurationBaseMetadataToJSON(discountFixedRepeatDurationBaseMetadata) {
    return JSON.stringify(exports.DiscountFixedRepeatDurationBaseMetadata$outboundSchema.parse(discountFixedRepeatDurationBaseMetadata));
}
function discountFixedRepeatDurationBaseMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountFixedRepeatDurationBaseMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedRepeatDurationBaseMetadata' from JSON`);
}
/** @internal */
exports.DiscountFixedRepeatDurationBase$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    duration_in_months: z.number().int(),
    type: discounttype_js_1.DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "duration_in_months": "durationInMonths",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountFixedRepeatDurationBase$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    durationInMonths: z.number().int(),
    type: discounttype_js_1.DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        durationInMonths: "duration_in_months",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountFixedRepeatDurationBase$;
(function (DiscountFixedRepeatDurationBase$) {
    /** @deprecated use `DiscountFixedRepeatDurationBase$inboundSchema` instead. */
    DiscountFixedRepeatDurationBase$.inboundSchema = exports.DiscountFixedRepeatDurationBase$inboundSchema;
    /** @deprecated use `DiscountFixedRepeatDurationBase$outboundSchema` instead. */
    DiscountFixedRepeatDurationBase$.outboundSchema = exports.DiscountFixedRepeatDurationBase$outboundSchema;
})(DiscountFixedRepeatDurationBase$ || (exports.DiscountFixedRepeatDurationBase$ = DiscountFixedRepeatDurationBase$ = {}));
function discountFixedRepeatDurationBaseToJSON(discountFixedRepeatDurationBase) {
    return JSON.stringify(exports.DiscountFixedRepeatDurationBase$outboundSchema.parse(discountFixedRepeatDurationBase));
}
function discountFixedRepeatDurationBaseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountFixedRepeatDurationBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedRepeatDurationBase' from JSON`);
}
