// @polar-sh/sdk/dist/commonjs/models/components/downloadablefilecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DownloadableFileCreate$ = exports.DownloadableFileCreate$outboundSchema = exports.DownloadableFileCreate$inboundSchema = void 0;
exports.downloadableFileCreateToJSON = downloadableFileCreateToJSON;
exports.downloadableFileCreateFromJSON = downloadableFileCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const s3filecreatemultipart_js_1 = require("./s3filecreatemultipart.js");
/** @internal */
exports.DownloadableFileCreate$inboundSchema = z.object({
    organization_id: z.nullable(z.string()).optional(),
    name: z.string(),
    mime_type: z.string(),
    size: z.number().int(),
    checksum_sha256_base64: z.nullable(z.string()).optional(),
    upload: s3filecreatemultipart_js_1.S3FileCreateMultipart$inboundSchema,
    service: z.literal("downloadable"),
    version: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "mime_type": "mimeType",
        "checksum_sha256_base64": "checksumSha256Base64",
    });
});
/** @internal */
exports.DownloadableFileCreate$outboundSchema = z.object({
    organizationId: z.nullable(z.string()).optional(),
    name: z.string(),
    mimeType: z.string(),
    size: z.number().int(),
    checksumSha256Base64: z.nullable(z.string()).optional(),
    upload: s3filecreatemultipart_js_1.S3FileCreateMultipart$outboundSchema,
    service: z.literal("downloadable"),
    version: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        mimeType: "mime_type",
        checksumSha256Base64: "checksum_sha256_base64",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DownloadableFileCreate$;
(function (DownloadableFileCreate$) {
    /** @deprecated use `DownloadableFileCreate$inboundSchema` instead. */
    DownloadableFileCreate$.inboundSchema = exports.DownloadableFileCreate$inboundSchema;
    /** @deprecated use `DownloadableFileCreate$outboundSchema` instead. */
    DownloadableFileCreate$.outboundSchema = exports.DownloadableFileCreate$outboundSchema;
})(DownloadableFileCreate$ || (exports.DownloadableFileCreate$ = DownloadableFileCreate$ = {}));
function downloadableFileCreateToJSON(downloadableFileCreate) {
    return JSON.stringify(exports.DownloadableFileCreate$outboundSchema.parse(downloadableFileCreate));
}
function downloadableFileCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DownloadableFileCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DownloadableFileCreate' from JSON`);
}
