// @polar-sh/sdk/dist/commonjs/models/components/downloadableread.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DownloadableRead$ = exports.DownloadableRead$outboundSchema = exports.DownloadableRead$inboundSchema = void 0;
exports.downloadableReadToJSON = downloadableReadToJSON;
exports.downloadableReadFromJSON = downloadableReadFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const filedownload_js_1 = require("./filedownload.js");
/** @internal */
exports.DownloadableRead$inboundSchema = z.object({
    id: z.string(),
    benefit_id: z.string(),
    file: filedownload_js_1.FileDownload$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "benefit_id": "benefitId",
    });
});
/** @internal */
exports.DownloadableRead$outboundSchema = z.object({
    id: z.string(),
    benefitId: z.string(),
    file: filedownload_js_1.FileDownload$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        benefitId: "benefit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DownloadableRead$;
(function (DownloadableRead$) {
    /** @deprecated use `DownloadableRead$inboundSchema` instead. */
    DownloadableRead$.inboundSchema = exports.DownloadableRead$inboundSchema;
    /** @deprecated use `DownloadableRead$outboundSchema` instead. */
    DownloadableRead$.outboundSchema = exports.DownloadableRead$outboundSchema;
})(DownloadableRead$ || (exports.DownloadableRead$ = DownloadableRead$ = {}));
function downloadableReadToJSON(downloadableRead) {
    return JSON.stringify(exports.DownloadableRead$outboundSchema.parse(downloadableRead));
}
function downloadableReadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DownloadableRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DownloadableRead' from JSON`);
}
