// @polar-sh/sdk/dist/commonjs/models/components/eventsingest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsIngest$ = exports.EventsIngest$outboundSchema = exports.EventsIngest$inboundSchema = exports.Events$ = exports.Events$outboundSchema = exports.Events$inboundSchema = void 0;
exports.eventsToJSON = eventsToJSON;
exports.eventsFromJSON = eventsFromJSON;
exports.eventsIngestToJSON = eventsIngestToJSON;
exports.eventsIngestFromJSON = eventsIngestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const eventcreatecustomer_js_1 = require("./eventcreatecustomer.js");
const eventcreateexternalcustomer_js_1 = require("./eventcreateexternalcustomer.js");
/** @internal */
exports.Events$inboundSchema = z
    .union([
    eventcreatecustomer_js_1.EventCreateCustomer$inboundSchema,
    eventcreateexternalcustomer_js_1.EventCreateExternalCustomer$inboundSchema,
]);
/** @internal */
exports.Events$outboundSchema = z.union([
    eventcreatecustomer_js_1.EventCreateCustomer$outboundSchema,
    eventcreateexternalcustomer_js_1.EventCreateExternalCustomer$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Events$;
(function (Events$) {
    /** @deprecated use `Events$inboundSchema` instead. */
    Events$.inboundSchema = exports.Events$inboundSchema;
    /** @deprecated use `Events$outboundSchema` instead. */
    Events$.outboundSchema = exports.Events$outboundSchema;
})(Events$ || (exports.Events$ = Events$ = {}));
function eventsToJSON(events) {
    return JSON.stringify(exports.Events$outboundSchema.parse(events));
}
function eventsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Events$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Events' from JSON`);
}
/** @internal */
exports.EventsIngest$inboundSchema = z.object({
    events: z.array(z.union([
        eventcreatecustomer_js_1.EventCreateCustomer$inboundSchema,
        eventcreateexternalcustomer_js_1.EventCreateExternalCustomer$inboundSchema,
    ])),
});
/** @internal */
exports.EventsIngest$outboundSchema = z.object({
    events: z.array(z.union([
        eventcreatecustomer_js_1.EventCreateCustomer$outboundSchema,
        eventcreateexternalcustomer_js_1.EventCreateExternalCustomer$outboundSchema,
    ])),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsIngest$;
(function (EventsIngest$) {
    /** @deprecated use `EventsIngest$inboundSchema` instead. */
    EventsIngest$.inboundSchema = exports.EventsIngest$inboundSchema;
    /** @deprecated use `EventsIngest$outboundSchema` instead. */
    EventsIngest$.outboundSchema = exports.EventsIngest$outboundSchema;
})(EventsIngest$ || (exports.EventsIngest$ = EventsIngest$ = {}));
function eventsIngestToJSON(eventsIngest) {
    return JSON.stringify(exports.EventsIngest$outboundSchema.parse(eventsIngest));
}
function eventsIngestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsIngest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsIngest' from JSON`);
}
