// @polar-sh/sdk/dist/commonjs/models/components/filecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileCreate$ = exports.FileCreate$outboundSchema = exports.FileCreate$inboundSchema = void 0;
exports.fileCreateToJSON = fileCreateToJSON;
exports.fileCreateFromJSON = fileCreateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const downloadablefilecreate_js_1 = require("./downloadablefilecreate.js");
const organizationavatarfilecreate_js_1 = require("./organizationavatarfilecreate.js");
const productmediafilecreate_js_1 = require("./productmediafilecreate.js");
/** @internal */
exports.FileCreate$inboundSchema = z.union([
    downloadablefilecreate_js_1.DownloadableFileCreate$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
    organizationavatarfilecreate_js_1.OrganizationAvatarFileCreate$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    productmediafilecreate_js_1.ProductMediaFileCreate$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
]);
/** @internal */
exports.FileCreate$outboundSchema = z.union([
    downloadablefilecreate_js_1.DownloadableFileCreate$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
    organizationavatarfilecreate_js_1.OrganizationAvatarFileCreate$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    productmediafilecreate_js_1.ProductMediaFileCreate$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileCreate$;
(function (FileCreate$) {
    /** @deprecated use `FileCreate$inboundSchema` instead. */
    FileCreate$.inboundSchema = exports.FileCreate$inboundSchema;
    /** @deprecated use `FileCreate$outboundSchema` instead. */
    FileCreate$.outboundSchema = exports.FileCreate$outboundSchema;
})(FileCreate$ || (exports.FileCreate$ = FileCreate$ = {}));
function fileCreateToJSON(fileCreate) {
    return JSON.stringify(exports.FileCreate$outboundSchema.parse(fileCreate));
}
function fileCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FileCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileCreate' from JSON`);
}
