// @polar-sh/sdk/dist/commonjs/models/components/listresourcecustomfield.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceCustomField$ = exports.ListResourceCustomField$outboundSchema = exports.ListResourceCustomField$inboundSchema = void 0;
exports.listResourceCustomFieldToJSON = listResourceCustomFieldToJSON;
exports.listResourceCustomFieldFromJSON = listResourceCustomFieldFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customfield_js_1 = require("./customfield.js");
const pagination_js_1 = require("./pagination.js");
/** @internal */
exports.ListResourceCustomField$inboundSchema = z.object({
    items: z.array(customfield_js_1.CustomField$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceCustomField$outboundSchema = z.object({
    items: z.array(customfield_js_1.CustomField$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceCustomField$;
(function (ListResourceCustomField$) {
    /** @deprecated use `ListResourceCustomField$inboundSchema` instead. */
    ListResourceCustomField$.inboundSchema = exports.ListResourceCustomField$inboundSchema;
    /** @deprecated use `ListResourceCustomField$outboundSchema` instead. */
    ListResourceCustomField$.outboundSchema = exports.ListResourceCustomField$outboundSchema;
})(ListResourceCustomField$ || (exports.ListResourceCustomField$ = ListResourceCustomField$ = {}));
function listResourceCustomFieldToJSON(listResourceCustomField) {
    return JSON.stringify(exports.ListResourceCustomField$outboundSchema.parse(listResourceCustomField));
}
function listResourceCustomFieldFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListResourceCustomField$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomField' from JSON`);
}
