// @polar-sh/sdk/dist/commonjs/models/components/listresourcecustomersubscription.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceCustomerSubscription$ = exports.ListResourceCustomerSubscription$outboundSchema = exports.ListResourceCustomerSubscription$inboundSchema = void 0;
exports.listResourceCustomerSubscriptionToJSON = listResourceCustomerSubscriptionToJSON;
exports.listResourceCustomerSubscriptionFromJSON = listResourceCustomerSubscriptionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customersubscription_js_1 = require("./customersubscription.js");
const pagination_js_1 = require("./pagination.js");
/** @internal */
exports.ListResourceCustomerSubscription$inboundSchema = z.object({
    items: z.array(customersubscription_js_1.CustomerSubscription$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceCustomerSubscription$outboundSchema = z.object({
    items: z.array(customersubscription_js_1.CustomerSubscription$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceCustomerSubscription$;
(function (ListResourceCustomerSubscription$) {
    /** @deprecated use `ListResourceCustomerSubscription$inboundSchema` instead. */
    ListResourceCustomerSubscription$.inboundSchema = exports.ListResourceCustomerSubscription$inboundSchema;
    /** @deprecated use `ListResourceCustomerSubscription$outboundSchema` instead. */
    ListResourceCustomerSubscription$.outboundSchema = exports.ListResourceCustomerSubscription$outboundSchema;
})(ListResourceCustomerSubscription$ || (exports.ListResourceCustomerSubscription$ = ListResourceCustomerSubscription$ = {}));
function listResourceCustomerSubscriptionToJSON(listResourceCustomerSubscription) {
    return JSON.stringify(exports.ListResourceCustomerSubscription$outboundSchema.parse(listResourceCustomerSubscription));
}
function listResourceCustomerSubscriptionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListResourceCustomerSubscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerSubscription' from JSON`);
}
