// @polar-sh/sdk/dist/commonjs/models/components/listresourcediscount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceDiscount$ = exports.ListResourceDiscount$outboundSchema = exports.ListResourceDiscount$inboundSchema = void 0;
exports.listResourceDiscountToJSON = listResourceDiscountToJSON;
exports.listResourceDiscountFromJSON = listResourceDiscountFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const discount_js_1 = require("./discount.js");
const pagination_js_1 = require("./pagination.js");
/** @internal */
exports.ListResourceDiscount$inboundSchema = z.object({
    items: z.array(discount_js_1.Discount$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceDiscount$outboundSchema = z.object({
    items: z.array(discount_js_1.Discount$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceDiscount$;
(function (ListResourceDiscount$) {
    /** @deprecated use `ListResourceDiscount$inboundSchema` instead. */
    ListResourceDiscount$.inboundSchema = exports.ListResourceDiscount$inboundSchema;
    /** @deprecated use `ListResourceDiscount$outboundSchema` instead. */
    ListResourceDiscount$.outboundSchema = exports.ListResourceDiscount$outboundSchema;
})(ListResourceDiscount$ || (exports.ListResourceDiscount$ = ListResourceDiscount$ = {}));
function listResourceDiscountToJSON(listResourceDiscount) {
    return JSON.stringify(exports.ListResourceDiscount$outboundSchema.parse(listResourceDiscount));
}
function listResourceDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListResourceDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceDiscount' from JSON`);
}
