// @polar-sh/sdk/dist/commonjs/models/components/organization.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization$ = exports.Organization$outboundSchema = exports.Organization$inboundSchema = void 0;
exports.organizationToJSON = organizationToJSON;
exports.organizationFromJSON = organizationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const organizationfeaturesettings_js_1 = require("./organizationfeaturesettings.js");
const organizationnotificationsettings_js_1 = require("./organizationnotificationsettings.js");
const organizationsociallink_js_1 = require("./organizationsociallink.js");
const organizationsubscriptionsettings_js_1 = require("./organizationsubscriptionsettings.js");
const status_js_1 = require("./status.js");
/** @internal */
exports.Organization$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatar_url: z.nullable(z.string()),
    email: z.nullable(z.string()),
    website: z.nullable(z.string()),
    socials: z.array(organizationsociallink_js_1.OrganizationSocialLink$inboundSchema),
    status: status_js_1.Status$inboundSchema,
    details_submitted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    feature_settings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$inboundSchema),
    subscription_settings: organizationsubscriptionsettings_js_1.OrganizationSubscriptionSettings$inboundSchema,
    notification_settings: organizationnotificationsettings_js_1.OrganizationNotificationSettings$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "avatar_url": "avatarUrl",
        "details_submitted_at": "detailsSubmittedAt",
        "feature_settings": "featureSettings",
        "subscription_settings": "subscriptionSettings",
        "notification_settings": "notificationSettings",
    });
});
/** @internal */
exports.Organization$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatarUrl: z.nullable(z.string()),
    email: z.nullable(z.string()),
    website: z.nullable(z.string()),
    socials: z.array(organizationsociallink_js_1.OrganizationSocialLink$outboundSchema),
    status: status_js_1.Status$outboundSchema,
    detailsSubmittedAt: z.nullable(z.date().transform(v => v.toISOString())),
    featureSettings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$outboundSchema),
    subscriptionSettings: organizationsubscriptionsettings_js_1.OrganizationSubscriptionSettings$outboundSchema,
    notificationSettings: organizationnotificationsettings_js_1.OrganizationNotificationSettings$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        avatarUrl: "avatar_url",
        detailsSubmittedAt: "details_submitted_at",
        featureSettings: "feature_settings",
        subscriptionSettings: "subscription_settings",
        notificationSettings: "notification_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Organization$;
(function (Organization$) {
    /** @deprecated use `Organization$inboundSchema` instead. */
    Organization$.inboundSchema = exports.Organization$inboundSchema;
    /** @deprecated use `Organization$outboundSchema` instead. */
    Organization$.outboundSchema = exports.Organization$outboundSchema;
})(Organization$ || (exports.Organization$ = Organization$ = {}));
function organizationToJSON(organization) {
    return JSON.stringify(exports.Organization$outboundSchema.parse(organization));
}
function organizationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Organization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Organization' from JSON`);
}
