// @polar-sh/sdk/dist/commonjs/models/components/productpricecustomcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceCustomCreate$ = exports.ProductPriceCustomCreate$outboundSchema = exports.ProductPriceCustomCreate$inboundSchema = void 0;
exports.productPriceCustomCreateToJSON = productPriceCustomCreateToJSON;
exports.productPriceCustomCreateFromJSON = productPriceCustomCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ProductPriceCustomCreate$inboundSchema = z.object({
    amount_type: z.literal("custom"),
    price_currency: z.string().default("usd"),
    minimum_amount: z.nullable(z.number().int()).optional(),
    maximum_amount: z.nullable(z.number().int()).optional(),
    preset_amount: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
exports.ProductPriceCustomCreate$outboundSchema = z.object({
    amountType: z.literal("custom"),
    priceCurrency: z.string().default("usd"),
    minimumAmount: z.nullable(z.number().int()).optional(),
    maximumAmount: z.nullable(z.number().int()).optional(),
    presetAmount: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceCustomCreate$;
(function (ProductPriceCustomCreate$) {
    /** @deprecated use `ProductPriceCustomCreate$inboundSchema` instead. */
    ProductPriceCustomCreate$.inboundSchema = exports.ProductPriceCustomCreate$inboundSchema;
    /** @deprecated use `ProductPriceCustomCreate$outboundSchema` instead. */
    ProductPriceCustomCreate$.outboundSchema = exports.ProductPriceCustomCreate$outboundSchema;
})(ProductPriceCustomCreate$ || (exports.ProductPriceCustomCreate$ = ProductPriceCustomCreate$ = {}));
function productPriceCustomCreateToJSON(productPriceCustomCreate) {
    return JSON.stringify(exports.ProductPriceCustomCreate$outboundSchema.parse(productPriceCustomCreate));
}
function productPriceCustomCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceCustomCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceCustomCreate' from JSON`);
}
