// @polar-sh/sdk/dist/commonjs/models/components/productpricefixedcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceFixedCreate$ = exports.ProductPriceFixedCreate$outboundSchema = exports.ProductPriceFixedCreate$inboundSchema = void 0;
exports.productPriceFixedCreateToJSON = productPriceFixedCreateToJSON;
exports.productPriceFixedCreateFromJSON = productPriceFixedCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ProductPriceFixedCreate$inboundSchema = z.object({
    amount_type: z.literal("fixed"),
    price_amount: z.number().int(),
    price_currency: z.string().default("usd"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "price_amount": "priceAmount",
        "price_currency": "priceCurrency",
    });
});
/** @internal */
exports.ProductPriceFixedCreate$outboundSchema = z.object({
    amountType: z.literal("fixed"),
    priceAmount: z.number().int(),
    priceCurrency: z.string().default("usd"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        priceAmount: "price_amount",
        priceCurrency: "price_currency",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceFixedCreate$;
(function (ProductPriceFixedCreate$) {
    /** @deprecated use `ProductPriceFixedCreate$inboundSchema` instead. */
    ProductPriceFixedCreate$.inboundSchema = exports.ProductPriceFixedCreate$inboundSchema;
    /** @deprecated use `ProductPriceFixedCreate$outboundSchema` instead. */
    ProductPriceFixedCreate$.outboundSchema = exports.ProductPriceFixedCreate$outboundSchema;
})(ProductPriceFixedCreate$ || (exports.ProductPriceFixedCreate$ = ProductPriceFixedCreate$ = {}));
function productPriceFixedCreateToJSON(productPriceFixedCreate) {
    return JSON.stringify(exports.ProductPriceFixedCreate$outboundSchema.parse(productPriceFixedCreate));
}
function productPriceFixedCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceFixedCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceFixedCreate' from JSON`);
}
