// @polar-sh/sdk/dist/commonjs/models/components/refreshtokenrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefreshTokenRequest$ = exports.RefreshTokenRequest$outboundSchema = exports.RefreshTokenRequest$inboundSchema = void 0;
exports.refreshTokenRequestToJSON = refreshTokenRequestToJSON;
exports.refreshTokenRequestFromJSON = refreshTokenRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.RefreshTokenRequest$inboundSchema = z.object({
    grant_type: z.literal("refresh_token"),
    client_id: z.string(),
    client_secret: z.string(),
    refresh_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "grant_type": "grantType",
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "refresh_token": "refreshToken",
    });
});
/** @internal */
exports.RefreshTokenRequest$outboundSchema = z.object({
    grantType: z.literal("refresh_token"),
    clientId: z.string(),
    clientSecret: z.string(),
    refreshToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        refreshToken: "refresh_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RefreshTokenRequest$;
(function (RefreshTokenRequest$) {
    /** @deprecated use `RefreshTokenRequest$inboundSchema` instead. */
    RefreshTokenRequest$.inboundSchema = exports.RefreshTokenRequest$inboundSchema;
    /** @deprecated use `RefreshTokenRequest$outboundSchema` instead. */
    RefreshTokenRequest$.outboundSchema = exports.RefreshTokenRequest$outboundSchema;
})(RefreshTokenRequest$ || (exports.RefreshTokenRequest$ = RefreshTokenRequest$ = {}));
function refreshTokenRequestToJSON(refreshTokenRequest) {
    return JSON.stringify(exports.RefreshTokenRequest$outboundSchema.parse(refreshTokenRequest));
}
function refreshTokenRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RefreshTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefreshTokenRequest' from JSON`);
}
