// @polar-sh/sdk/dist/commonjs/models/components/s3downloadurl.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3DownloadURL$ = exports.S3DownloadURL$outboundSchema = exports.S3DownloadURL$inboundSchema = void 0;
exports.s3DownloadURLToJSON = s3DownloadURLToJSON;
exports.s3DownloadURLFromJSON = s3DownloadURLFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.S3DownloadURL$inboundSchema = z.object({
    url: z.string(),
    headers: z.record(z.string()).optional(),
    expires_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "expires_at": "expiresAt",
    });
});
/** @internal */
exports.S3DownloadURL$outboundSchema = z.object({
    url: z.string(),
    headers: z.record(z.string()).optional(),
    expiresAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        expiresAt: "expires_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var S3DownloadURL$;
(function (S3DownloadURL$) {
    /** @deprecated use `S3DownloadURL$inboundSchema` instead. */
    S3DownloadURL$.inboundSchema = exports.S3DownloadURL$inboundSchema;
    /** @deprecated use `S3DownloadURL$outboundSchema` instead. */
    S3DownloadURL$.outboundSchema = exports.S3DownloadURL$outboundSchema;
})(S3DownloadURL$ || (exports.S3DownloadURL$ = S3DownloadURL$ = {}));
function s3DownloadURLToJSON(s3DownloadURL) {
    return JSON.stringify(exports.S3DownloadURL$outboundSchema.parse(s3DownloadURL));
}
function s3DownloadURLFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.S3DownloadURL$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'S3DownloadURL' from JSON`);
}
