// @polar-sh/sdk/dist/commonjs/models/components/subscriptioncycledmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionCycledMetadata$ = exports.SubscriptionCycledMetadata$outboundSchema = exports.SubscriptionCycledMetadata$inboundSchema = void 0;
exports.subscriptionCycledMetadataToJSON = subscriptionCycledMetadataToJSON;
exports.subscriptionCycledMetadataFromJSON = subscriptionCycledMetadataFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.SubscriptionCycledMetadata$inboundSchema = z.object({
    subscription_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
exports.SubscriptionCycledMetadata$outboundSchema = z.object({
    subscriptionId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionCycledMetadata$;
(function (SubscriptionCycledMetadata$) {
    /** @deprecated use `SubscriptionCycledMetadata$inboundSchema` instead. */
    SubscriptionCycledMetadata$.inboundSchema = exports.SubscriptionCycledMetadata$inboundSchema;
    /** @deprecated use `SubscriptionCycledMetadata$outboundSchema` instead. */
    SubscriptionCycledMetadata$.outboundSchema = exports.SubscriptionCycledMetadata$outboundSchema;
})(SubscriptionCycledMetadata$ || (exports.SubscriptionCycledMetadata$ = SubscriptionCycledMetadata$ = {}));
function subscriptionCycledMetadataToJSON(subscriptionCycledMetadata) {
    return JSON.stringify(exports.SubscriptionCycledMetadata$outboundSchema.parse(subscriptionCycledMetadata));
}
function subscriptionCycledMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionCycledMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCycledMetadata' from JSON`);
}
