// @polar-sh/sdk/dist/commonjs/models/components/subscriptioncustomer.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionCustomer$ = exports.SubscriptionCustomer$outboundSchema = exports.SubscriptionCustomer$inboundSchema = exports.TaxId$ = exports.TaxId$outboundSchema = exports.TaxId$inboundSchema = exports.SubscriptionCustomerMetadata$ = exports.SubscriptionCustomerMetadata$outboundSchema = exports.SubscriptionCustomerMetadata$inboundSchema = void 0;
exports.subscriptionCustomerMetadataToJSON = subscriptionCustomerMetadataToJSON;
exports.subscriptionCustomerMetadataFromJSON = subscriptionCustomerMetadataFromJSON;
exports.taxIdToJSON = taxIdToJSON;
exports.taxIdFromJSON = taxIdFromJSON;
exports.subscriptionCustomerToJSON = subscriptionCustomerToJSON;
exports.subscriptionCustomerFromJSON = subscriptionCustomerFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const taxidformat_js_1 = require("./taxidformat.js");
/** @internal */
exports.SubscriptionCustomerMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.SubscriptionCustomerMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionCustomerMetadata$;
(function (SubscriptionCustomerMetadata$) {
    /** @deprecated use `SubscriptionCustomerMetadata$inboundSchema` instead. */
    SubscriptionCustomerMetadata$.inboundSchema = exports.SubscriptionCustomerMetadata$inboundSchema;
    /** @deprecated use `SubscriptionCustomerMetadata$outboundSchema` instead. */
    SubscriptionCustomerMetadata$.outboundSchema = exports.SubscriptionCustomerMetadata$outboundSchema;
})(SubscriptionCustomerMetadata$ || (exports.SubscriptionCustomerMetadata$ = SubscriptionCustomerMetadata$ = {}));
function subscriptionCustomerMetadataToJSON(subscriptionCustomerMetadata) {
    return JSON.stringify(exports.SubscriptionCustomerMetadata$outboundSchema.parse(subscriptionCustomerMetadata));
}
function subscriptionCustomerMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionCustomerMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomerMetadata' from JSON`);
}
/** @internal */
exports.TaxId$inboundSchema = z
    .union([z.string(), taxidformat_js_1.TaxIDFormat$inboundSchema]);
/** @internal */
exports.TaxId$outboundSchema = z.union([z.string(), taxidformat_js_1.TaxIDFormat$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxId$;
(function (TaxId$) {
    /** @deprecated use `TaxId$inboundSchema` instead. */
    TaxId$.inboundSchema = exports.TaxId$inboundSchema;
    /** @deprecated use `TaxId$outboundSchema` instead. */
    TaxId$.outboundSchema = exports.TaxId$outboundSchema;
})(TaxId$ || (exports.TaxId$ = TaxId$ = {}));
function taxIdToJSON(taxId) {
    return JSON.stringify(exports.TaxId$outboundSchema.parse(taxId));
}
function taxIdFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TaxId' from JSON`);
}
/** @internal */
exports.SubscriptionCustomer$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    external_id: z.nullable(z.string()),
    email: z.string(),
    email_verified: z.boolean(),
    name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable(z.union([z.string(), taxidformat_js_1.TaxIDFormat$inboundSchema])))),
    organization_id: z.string(),
    deleted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    avatar_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "external_id": "externalId",
        "email_verified": "emailVerified",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "organization_id": "organizationId",
        "deleted_at": "deletedAt",
        "avatar_url": "avatarUrl",
    });
});
/** @internal */
exports.SubscriptionCustomer$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    externalId: z.nullable(z.string()),
    email: z.string(),
    emailVerified: z.boolean(),
    name: z.nullable(z.string()),
    billingAddress: z.nullable(address_js_1.Address$outboundSchema),
    taxId: z.nullable(z.array(z.nullable(z.union([z.string(), taxidformat_js_1.TaxIDFormat$outboundSchema])))),
    organizationId: z.string(),
    deletedAt: z.nullable(z.date().transform(v => v.toISOString())),
    avatarUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        externalId: "external_id",
        emailVerified: "email_verified",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
        deletedAt: "deleted_at",
        avatarUrl: "avatar_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionCustomer$;
(function (SubscriptionCustomer$) {
    /** @deprecated use `SubscriptionCustomer$inboundSchema` instead. */
    SubscriptionCustomer$.inboundSchema = exports.SubscriptionCustomer$inboundSchema;
    /** @deprecated use `SubscriptionCustomer$outboundSchema` instead. */
    SubscriptionCustomer$.outboundSchema = exports.SubscriptionCustomer$outboundSchema;
})(SubscriptionCustomer$ || (exports.SubscriptionCustomer$ = SubscriptionCustomer$ = {}));
function subscriptionCustomerToJSON(subscriptionCustomer) {
    return JSON.stringify(exports.SubscriptionCustomer$outboundSchema.parse(subscriptionCustomer));
}
function subscriptionCustomerFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomer' from JSON`);
}
