// @polar-sh/sdk/dist/commonjs/models/components/subscriptionsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionSortProperty$ = exports.SubscriptionSortProperty$outboundSchema = exports.SubscriptionSortProperty$inboundSchema = exports.SubscriptionSortProperty = void 0;
const z = __importStar(require("zod"));
exports.SubscriptionSortProperty = {
    Customer: "customer",
    MinusCustomer: "-customer",
    Status: "status",
    MinusStatus: "-status",
    StartedAt: "started_at",
    MinusStartedAt: "-started_at",
    CurrentPeriodEnd: "current_period_end",
    MinusCurrentPeriodEnd: "-current_period_end",
    Amount: "amount",
    MinusAmount: "-amount",
    Product: "product",
    MinusProduct: "-product",
    Discount: "discount",
    MinusDiscount: "-discount",
};
/** @internal */
exports.SubscriptionSortProperty$inboundSchema = z.nativeEnum(exports.SubscriptionSortProperty);
/** @internal */
exports.SubscriptionSortProperty$outboundSchema = exports.SubscriptionSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionSortProperty$;
(function (SubscriptionSortProperty$) {
    /** @deprecated use `SubscriptionSortProperty$inboundSchema` instead. */
    SubscriptionSortProperty$.inboundSchema = exports.SubscriptionSortProperty$inboundSchema;
    /** @deprecated use `SubscriptionSortProperty$outboundSchema` instead. */
    SubscriptionSortProperty$.outboundSchema = exports.SubscriptionSortProperty$outboundSchema;
})(SubscriptionSortProperty$ || (exports.SubscriptionSortProperty$ = SubscriptionSortProperty$ = {}));
