// @polar-sh/sdk/dist/commonjs/models/components/userevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserEvent$ = exports.UserEvent$outboundSchema = exports.UserEvent$inboundSchema = exports.UserEventMetadata$ = exports.UserEventMetadata$outboundSchema = exports.UserEventMetadata$inboundSchema = void 0;
exports.userEventMetadataToJSON = userEventMetadataToJSON;
exports.userEventMetadataFromJSON = userEventMetadataFromJSON;
exports.userEventToJSON = userEventToJSON;
exports.userEventFromJSON = userEventFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customer_js_1 = require("./customer.js");
/** @internal */
exports.UserEventMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.UserEventMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserEventMetadata$;
(function (UserEventMetadata$) {
    /** @deprecated use `UserEventMetadata$inboundSchema` instead. */
    UserEventMetadata$.inboundSchema = exports.UserEventMetadata$inboundSchema;
    /** @deprecated use `UserEventMetadata$outboundSchema` instead. */
    UserEventMetadata$.outboundSchema = exports.UserEventMetadata$outboundSchema;
})(UserEventMetadata$ || (exports.UserEventMetadata$ = UserEventMetadata$ = {}));
function userEventMetadataToJSON(userEventMetadata) {
    return JSON.stringify(exports.UserEventMetadata$outboundSchema.parse(userEventMetadata));
}
function userEventMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UserEventMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserEventMetadata' from JSON`);
}
/** @internal */
exports.UserEvent$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(customer_js_1.Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    name: z.string(),
    source: z.literal("user"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
exports.UserEvent$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(customer_js_1.Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    name: z.string(),
    source: z.literal("user"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserEvent$;
(function (UserEvent$) {
    /** @deprecated use `UserEvent$inboundSchema` instead. */
    UserEvent$.inboundSchema = exports.UserEvent$inboundSchema;
    /** @deprecated use `UserEvent$outboundSchema` instead. */
    UserEvent$.outboundSchema = exports.UserEvent$outboundSchema;
})(UserEvent$ || (exports.UserEvent$ = UserEvent$ = {}));
function userEventToJSON(userEvent) {
    return JSON.stringify(exports.UserEvent$outboundSchema.parse(userEvent));
}
function userEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UserEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserEvent' from JSON`);
}
