// @polar-sh/sdk/dist/commonjs/models/components/validatedlicensekey.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidatedLicenseKey$ = exports.ValidatedLicenseKey$outboundSchema = exports.ValidatedLicenseKey$inboundSchema = void 0;
exports.validatedLicenseKeyToJSON = validatedLicenseKeyToJSON;
exports.validatedLicenseKeyFromJSON = validatedLicenseKeyFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const licensekeyactivationbase_js_1 = require("./licensekeyactivationbase.js");
const licensekeycustomer_js_1 = require("./licensekeycustomer.js");
const licensekeystatus_js_1 = require("./licensekeystatus.js");
/** @internal */
exports.ValidatedLicenseKey$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    organization_id: z.string(),
    customer_id: z.string(),
    customer: licensekeycustomer_js_1.LicenseKeyCustomer$inboundSchema,
    benefit_id: z.string(),
    key: z.string(),
    display_key: z.string(),
    status: licensekeystatus_js_1.LicenseKeyStatus$inboundSchema,
    limit_activations: z.nullable(z.number().int()),
    usage: z.number().int(),
    limit_usage: z.nullable(z.number().int()),
    validations: z.number().int(),
    last_validated_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    expires_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    activation: z.nullable(licensekeyactivationbase_js_1.LicenseKeyActivationBase$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "display_key": "displayKey",
        "limit_activations": "limitActivations",
        "limit_usage": "limitUsage",
        "last_validated_at": "lastValidatedAt",
        "expires_at": "expiresAt",
    });
});
/** @internal */
exports.ValidatedLicenseKey$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    organizationId: z.string(),
    customerId: z.string(),
    customer: licensekeycustomer_js_1.LicenseKeyCustomer$outboundSchema,
    benefitId: z.string(),
    key: z.string(),
    displayKey: z.string(),
    status: licensekeystatus_js_1.LicenseKeyStatus$outboundSchema,
    limitActivations: z.nullable(z.number().int()),
    usage: z.number().int(),
    limitUsage: z.nullable(z.number().int()),
    validations: z.number().int(),
    lastValidatedAt: z.nullable(z.date().transform(v => v.toISOString())),
    expiresAt: z.nullable(z.date().transform(v => v.toISOString())),
    activation: z.nullable(licensekeyactivationbase_js_1.LicenseKeyActivationBase$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
        customerId: "customer_id",
        benefitId: "benefit_id",
        displayKey: "display_key",
        limitActivations: "limit_activations",
        limitUsage: "limit_usage",
        lastValidatedAt: "last_validated_at",
        expiresAt: "expires_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ValidatedLicenseKey$;
(function (ValidatedLicenseKey$) {
    /** @deprecated use `ValidatedLicenseKey$inboundSchema` instead. */
    ValidatedLicenseKey$.inboundSchema = exports.ValidatedLicenseKey$inboundSchema;
    /** @deprecated use `ValidatedLicenseKey$outboundSchema` instead. */
    ValidatedLicenseKey$.outboundSchema = exports.ValidatedLicenseKey$outboundSchema;
})(ValidatedLicenseKey$ || (exports.ValidatedLicenseKey$ = ValidatedLicenseKey$ = {}));
function validatedLicenseKeyToJSON(validatedLicenseKey) {
    return JSON.stringify(exports.ValidatedLicenseKey$outboundSchema.parse(validatedLicenseKey));
}
function validatedLicenseKeyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ValidatedLicenseKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidatedLicenseKey' from JSON`);
}
