// @polar-sh/sdk/dist/commonjs/models/components/webhookorganizationupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookOrganizationUpdatedPayload$ = exports.WebhookOrganizationUpdatedPayload$outboundSchema = exports.WebhookOrganizationUpdatedPayload$inboundSchema = void 0;
exports.webhookOrganizationUpdatedPayloadToJSON = webhookOrganizationUpdatedPayloadToJSON;
exports.webhookOrganizationUpdatedPayloadFromJSON = webhookOrganizationUpdatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const organization_js_1 = require("./organization.js");
/** @internal */
exports.WebhookOrganizationUpdatedPayload$inboundSchema = z.object({
    type: z.literal("organization.updated"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: organization_js_1.Organization$inboundSchema,
});
/** @internal */
exports.WebhookOrganizationUpdatedPayload$outboundSchema = z.object({
    type: z.literal("organization.updated"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: organization_js_1.Organization$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookOrganizationUpdatedPayload$;
(function (WebhookOrganizationUpdatedPayload$) {
    /** @deprecated use `WebhookOrganizationUpdatedPayload$inboundSchema` instead. */
    WebhookOrganizationUpdatedPayload$.inboundSchema = exports.WebhookOrganizationUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrganizationUpdatedPayload$outboundSchema` instead. */
    WebhookOrganizationUpdatedPayload$.outboundSchema = exports.WebhookOrganizationUpdatedPayload$outboundSchema;
})(WebhookOrganizationUpdatedPayload$ || (exports.WebhookOrganizationUpdatedPayload$ = WebhookOrganizationUpdatedPayload$ = {}));
function webhookOrganizationUpdatedPayloadToJSON(webhookOrganizationUpdatedPayload) {
    return JSON.stringify(exports.WebhookOrganizationUpdatedPayload$outboundSchema.parse(webhookOrganizationUpdatedPayload));
}
function webhookOrganizationUpdatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookOrganizationUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrganizationUpdatedPayload' from JSON`);
}
