// @polar-sh/sdk/dist/commonjs/models/components/webhookdelivery.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookDelivery$ = exports.WebhookDelivery$outboundSchema = exports.WebhookDelivery$inboundSchema = void 0;
exports.webhookDeliveryToJSON = webhookDeliveryToJSON;
exports.webhookDeliveryFromJSON = webhookDeliveryFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const webhookevent_js_1 = require("./webhookevent.js");
/** @internal */
exports.WebhookDelivery$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    succeeded: z.boolean(),
    http_code: z.nullable(z.number().int()),
    response: z.nullable(z.string()),
    webhook_event: webhookevent_js_1.WebhookEvent$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "http_code": "httpCode",
        "webhook_event": "webhookEvent",
    });
});
/** @internal */
exports.WebhookDelivery$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    succeeded: z.boolean(),
    httpCode: z.nullable(z.number().int()),
    response: z.nullable(z.string()),
    webhookEvent: webhookevent_js_1.WebhookEvent$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        httpCode: "http_code",
        webhookEvent: "webhook_event",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookDelivery$;
(function (WebhookDelivery$) {
    /** @deprecated use `WebhookDelivery$inboundSchema` instead. */
    WebhookDelivery$.inboundSchema = exports.WebhookDelivery$inboundSchema;
    /** @deprecated use `WebhookDelivery$outboundSchema` instead. */
    WebhookDelivery$.outboundSchema = exports.WebhookDelivery$outboundSchema;
})(WebhookDelivery$ || (exports.WebhookDelivery$ = WebhookDelivery$ = {}));
function webhookDeliveryToJSON(webhookDelivery) {
    return JSON.stringify(exports.WebhookDelivery$outboundSchema.parse(webhookDelivery));
}
function webhookDeliveryFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookDelivery$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookDelivery' from JSON`);
}
