// @polar-sh/sdk/dist/commonjs/models/components/webhookordercreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookOrderCreatedPayload$ = exports.WebhookOrderCreatedPayload$outboundSchema = exports.WebhookOrderCreatedPayload$inboundSchema = void 0;
exports.webhookOrderCreatedPayloadToJSON = webhookOrderCreatedPayloadToJSON;
exports.webhookOrderCreatedPayloadFromJSON = webhookOrderCreatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const order_js_1 = require("./order.js");
/** @internal */
exports.WebhookOrderCreatedPayload$inboundSchema = z.object({
    type: z.literal("order.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: order_js_1.Order$inboundSchema,
});
/** @internal */
exports.WebhookOrderCreatedPayload$outboundSchema = z.object({
    type: z.literal("order.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: order_js_1.Order$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookOrderCreatedPayload$;
(function (WebhookOrderCreatedPayload$) {
    /** @deprecated use `WebhookOrderCreatedPayload$inboundSchema` instead. */
    WebhookOrderCreatedPayload$.inboundSchema = exports.WebhookOrderCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrderCreatedPayload$outboundSchema` instead. */
    WebhookOrderCreatedPayload$.outboundSchema = exports.WebhookOrderCreatedPayload$outboundSchema;
})(WebhookOrderCreatedPayload$ || (exports.WebhookOrderCreatedPayload$ = WebhookOrderCreatedPayload$ = {}));
function webhookOrderCreatedPayloadToJSON(webhookOrderCreatedPayload) {
    return JSON.stringify(exports.WebhookOrderCreatedPayload$outboundSchema.parse(webhookOrderCreatedPayload));
}
function webhookOrderCreatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookOrderCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderCreatedPayload' from JSON`);
}
