// @polar-sh/sdk/dist/commonjs/models/components/webhookendpointupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEndpointUpdate$ = exports.WebhookEndpointUpdate$outboundSchema = exports.WebhookEndpointUpdate$inboundSchema = void 0;
exports.webhookEndpointUpdateToJSON = webhookEndpointUpdateToJSON;
exports.webhookEndpointUpdateFromJSON = webhookEndpointUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const webhookeventtype_js_1 = require("./webhookeventtype.js");
const webhookformat_js_1 = require("./webhookformat.js");
/** @internal */
exports.WebhookEndpointUpdate$inboundSchema = z.object({
    url: z.nullable(z.string()).optional(),
    secret: z.nullable(z.string()).optional(),
    format: z.nullable(webhookformat_js_1.WebhookFormat$inboundSchema).optional(),
    events: z.nullable(z.array(webhookeventtype_js_1.WebhookEventType$inboundSchema)).optional(),
});
/** @internal */
exports.WebhookEndpointUpdate$outboundSchema = z.object({
    url: z.nullable(z.string()).optional(),
    secret: z.nullable(z.string()).optional(),
    format: z.nullable(webhookformat_js_1.WebhookFormat$outboundSchema).optional(),
    events: z.nullable(z.array(webhookeventtype_js_1.WebhookEventType$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookEndpointUpdate$;
(function (WebhookEndpointUpdate$) {
    /** @deprecated use `WebhookEndpointUpdate$inboundSchema` instead. */
    WebhookEndpointUpdate$.inboundSchema = exports.WebhookEndpointUpdate$inboundSchema;
    /** @deprecated use `WebhookEndpointUpdate$outboundSchema` instead. */
    WebhookEndpointUpdate$.outboundSchema = exports.WebhookEndpointUpdate$outboundSchema;
})(WebhookEndpointUpdate$ || (exports.WebhookEndpointUpdate$ = WebhookEndpointUpdate$ = {}));
function webhookEndpointUpdateToJSON(webhookEndpointUpdate) {
    return JSON.stringify(exports.WebhookEndpointUpdate$outboundSchema.parse(webhookEndpointUpdate));
}
function webhookEndpointUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookEndpointUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEndpointUpdate' from JSON`);
}
