// @polar-sh/sdk/dist/commonjs/models/components/webhookbenefitgrantrevokedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookBenefitGrantRevokedPayload$ = exports.WebhookBenefitGrantRevokedPayload$outboundSchema = exports.WebhookBenefitGrantRevokedPayload$inboundSchema = void 0;
exports.webhookBenefitGrantRevokedPayloadToJSON = webhookBenefitGrantRevokedPayloadToJSON;
exports.webhookBenefitGrantRevokedPayloadFromJSON = webhookBenefitGrantRevokedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitgrantwebhook_js_1 = require("./benefitgrantwebhook.js");
/** @internal */
exports.WebhookBenefitGrantRevokedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.revoked"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$inboundSchema,
});
/** @internal */
exports.WebhookBenefitGrantRevokedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.revoked"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantRevokedPayload$;
(function (WebhookBenefitGrantRevokedPayload$) {
    /** @deprecated use `WebhookBenefitGrantRevokedPayload$inboundSchema` instead. */
    WebhookBenefitGrantRevokedPayload$.inboundSchema = exports.WebhookBenefitGrantRevokedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantRevokedPayload$outboundSchema` instead. */
    WebhookBenefitGrantRevokedPayload$.outboundSchema = exports.WebhookBenefitGrantRevokedPayload$outboundSchema;
})(WebhookBenefitGrantRevokedPayload$ || (exports.WebhookBenefitGrantRevokedPayload$ = WebhookBenefitGrantRevokedPayload$ = {}));
function webhookBenefitGrantRevokedPayloadToJSON(webhookBenefitGrantRevokedPayload) {
    return JSON.stringify(exports.WebhookBenefitGrantRevokedPayload$outboundSchema.parse(webhookBenefitGrantRevokedPayload));
}
function webhookBenefitGrantRevokedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookBenefitGrantRevokedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantRevokedPayload' from JSON`);
}
