// @polar-sh/sdk/dist/esm/models/errors/invoicealreadyexists.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class InvoiceAlreadyExists extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "InvoiceAlreadyExists";
    }
}
/** @internal */
export const InvoiceAlreadyExists$inboundSchema = z.object({
    error: z.literal("InvoiceAlreadyExists"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new InvoiceAlreadyExists(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const InvoiceAlreadyExists$outboundSchema = z.instanceof(InvoiceAlreadyExists)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("InvoiceAlreadyExists"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InvoiceAlreadyExists$;
(function (InvoiceAlreadyExists$) {
    /** @deprecated use `InvoiceAlreadyExists$inboundSchema` instead. */
    InvoiceAlreadyExists$.inboundSchema = InvoiceAlreadyExists$inboundSchema;
    /** @deprecated use `InvoiceAlreadyExists$outboundSchema` instead. */
    InvoiceAlreadyExists$.outboundSchema = InvoiceAlreadyExists$outboundSchema;
})(InvoiceAlreadyExists$ || (InvoiceAlreadyExists$ = {}));
