// @polar-sh/sdk/dist/esm/models/errors/notpaidorder.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class NotPaidOrder extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "NotPaidOrder";
    }
}
/** @internal */
export const NotPaidOrder$inboundSchema = z.object({
    error: z.literal("NotPaidOrder"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new NotPaidOrder(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const NotPaidOrder$outboundSchema = z.instanceof(NotPaidOrder)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("NotPaidOrder"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotPaidOrder$;
(function (NotPaidOrder$) {
    /** @deprecated use `NotPaidOrder$inboundSchema` instead. */
    NotPaidOrder$.inboundSchema = NotPaidOrder$inboundSchema;
    /** @deprecated use `NotPaidOrder$outboundSchema` instead. */
    NotPaidOrder$.outboundSchema = NotPaidOrder$outboundSchema;
})(NotPaidOrder$ || (NotPaidOrder$ = {}));
