// @polar-sh/sdk/dist/esm/models/errors/notpermitted.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class NotPermitted extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "NotPermitted";
    }
}
/** @internal */
export const NotPermitted$inboundSchema = z.object({
    error: z.literal("NotPermitted"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new NotPermitted(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const NotPermitted$outboundSchema = z.instanceof(NotPermitted)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("NotPermitted"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotPermitted$;
(function (NotPermitted$) {
    /** @deprecated use `NotPermitted$inboundSchema` instead. */
    NotPermitted$.inboundSchema = NotPermitted$inboundSchema;
    /** @deprecated use `NotPermitted$outboundSchema` instead. */
    NotPermitted$.outboundSchema = NotPermitted$outboundSchema;
})(NotPermitted$ || (NotPermitted$ = {}));
