// @polar-sh/sdk/dist/esm/models/errors/paymentmethodinusebyactivesubscription.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class PaymentMethodInUseByActiveSubscription extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "PaymentMethodInUseByActiveSubscription";
    }
}
/** @internal */
export const PaymentMethodInUseByActiveSubscription$inboundSchema = z.object({
    error: z.literal("PaymentMethodInUseByActiveSubscription"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new PaymentMethodInUseByActiveSubscription(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const PaymentMethodInUseByActiveSubscription$outboundSchema = z.instanceof(PaymentMethodInUseByActiveSubscription)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("PaymentMethodInUseByActiveSubscription"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PaymentMethodInUseByActiveSubscription$;
(function (PaymentMethodInUseByActiveSubscription$) {
    /** @deprecated use `PaymentMethodInUseByActiveSubscription$inboundSchema` instead. */
    PaymentMethodInUseByActiveSubscription$.inboundSchema = PaymentMethodInUseByActiveSubscription$inboundSchema;
    /** @deprecated use `PaymentMethodInUseByActiveSubscription$outboundSchema` instead. */
    PaymentMethodInUseByActiveSubscription$.outboundSchema = PaymentMethodInUseByActiveSubscription$outboundSchema;
})(PaymentMethodInUseByActiveSubscription$ || (PaymentMethodInUseByActiveSubscription$ = {}));
