// @polar-sh/sdk/dist/esm/models/errors/paymentalreadyinprogress.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class PaymentAlreadyInProgress extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "PaymentAlreadyInProgress";
    }
}
/** @internal */
export const PaymentAlreadyInProgress$inboundSchema = z.object({
    error: z.literal("PaymentAlreadyInProgress"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new PaymentAlreadyInProgress(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const PaymentAlreadyInProgress$outboundSchema = z.instanceof(PaymentAlreadyInProgress)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("PaymentAlreadyInProgress"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PaymentAlreadyInProgress$;
(function (PaymentAlreadyInProgress$) {
    /** @deprecated use `PaymentAlreadyInProgress$inboundSchema` instead. */
    PaymentAlreadyInProgress$.inboundSchema = PaymentAlreadyInProgress$inboundSchema;
    /** @deprecated use `PaymentAlreadyInProgress$outboundSchema` instead. */
    PaymentAlreadyInProgress$.outboundSchema = PaymentAlreadyInProgress$outboundSchema;
})(PaymentAlreadyInProgress$ || (PaymentAlreadyInProgress$ = {}));
