// @polar-sh/sdk/dist/esm/models/errors/paymentnotready.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class PaymentNotReady extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "PaymentNotReady";
    }
}
/** @internal */
export const PaymentNotReady$inboundSchema = z.object({
    error: z.literal("PaymentNotReady"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new PaymentNotReady(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const PaymentNotReady$outboundSchema = z.instanceof(PaymentNotReady)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("PaymentNotReady"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PaymentNotReady$;
(function (PaymentNotReady$) {
    /** @deprecated use `PaymentNotReady$inboundSchema` instead. */
    PaymentNotReady$.inboundSchema = PaymentNotReady$inboundSchema;
    /** @deprecated use `PaymentNotReady$outboundSchema` instead. */
    PaymentNotReady$.outboundSchema = PaymentNotReady$outboundSchema;
})(PaymentNotReady$ || (PaymentNotReady$ = {}));
