// @polar-sh/sdk/dist/esm/models/errors/subscriptionlocked.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class SubscriptionLocked extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "SubscriptionLocked";
    }
}
/** @internal */
export const SubscriptionLocked$inboundSchema = z.object({
    error: z.literal("SubscriptionLocked"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new SubscriptionLocked(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const SubscriptionLocked$outboundSchema = z.instanceof(SubscriptionLocked)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("SubscriptionLocked"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionLocked$;
(function (SubscriptionLocked$) {
    /** @deprecated use `SubscriptionLocked$inboundSchema` instead. */
    SubscriptionLocked$.inboundSchema = SubscriptionLocked$inboundSchema;
    /** @deprecated use `SubscriptionLocked$outboundSchema` instead. */
    SubscriptionLocked$.outboundSchema = SubscriptionLocked$outboundSchema;
})(SubscriptionLocked$ || (SubscriptionLocked$ = {}));
