// @polar-sh/sdk/dist/esm/funcs/customerPortalCustomerMetersList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { dlv } from "../lib/dlv.js";
import { encodeFormQuery } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { resolveSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import { HTTPValidationError$inboundSchema, } from "../models/errors/httpvalidationerror.js";
import { CustomerPortalCustomerMetersListRequest$outboundSchema, CustomerPortalCustomerMetersListResponse$inboundSchema, } from "../models/operations/customerportalcustomermeterslist.js";
import { APIPromise } from "../types/async.js";
import { createPageIterator, haltIterator, } from "../types/operations.js";
/**
 * List Meters
 *
 * @remarks
 * List meters of the authenticated customer.
 *
 * **Scopes**: `customer_portal:read` `customer_portal:write`
 */
export function customerPortalCustomerMetersList(client, security, request, options) {
    return new APIPromise($do(client, security, request, options));
}
async function $do(client, security, request, options) {
    const parsed = safeParse(request, (value) => CustomerPortalCustomerMetersListRequest$outboundSchema.parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [haltIterator(parsed), { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const path = pathToFunc("/v1/customer-portal/meters/")();
    const query = encodeFormQuery({
        "limit": payload.limit,
        "meter_id": payload.meter_id,
        "page": payload.page,
        "query": payload.query,
        "sorting": payload.sorting,
    });
    const headers = new Headers(compactMap({
        Accept: "application/json",
    }));
    const requestSecurity = resolveSecurity([
        {
            fieldName: "Authorization",
            type: "http:bearer",
            value: security?.customerSession,
        },
    ]);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "customer_portal:customer_meters:list",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: security,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        query: query,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [haltIterator(requestRes), { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["422", "4XX", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [haltIterator(doResult), { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result, raw] = await M.match(M.json(200, CustomerPortalCustomerMetersListResponse$inboundSchema, {
        key: "Result",
    }), M.jsonErr(422, HTTPValidationError$inboundSchema), M.fail("4XX"), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [haltIterator(result), {
                status: "complete",
                request: req,
                response,
            }];
    }
    const nextFunc = (responseData) => {
        const page = request?.page ?? 1;
        const nextPage = page + 1;
        const numPages = dlv(responseData, "pagination.max_page");
        if (typeof numPages !== "number" || numPages <= page) {
            return { next: () => null };
        }
        if (!responseData) {
            return { next: () => null };
        }
        const results = dlv(responseData, "items");
        if (!Array.isArray(results) || !results.length) {
            return { next: () => null };
        }
        const limit = request?.limit ?? 10;
        if (results.length < limit) {
            return { next: () => null };
        }
        const nextVal = () => customerPortalCustomerMetersList(client, security, {
            ...request,
            page: nextPage,
        }, options);
        return { next: nextVal, "~next": { page: nextPage } };
    };
    const page = { ...result, ...nextFunc(raw) };
    return [{ ...page, ...createPageIterator(page, (v) => !v.ok) }, {
            status: "complete",
            request: req,
            response,
        }];
}
