// @polar-sh/sdk/dist/esm/models/errors/alreadyactivesubscriptionerror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class AlreadyActiveSubscriptionError extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "AlreadyActiveSubscriptionError";
    }
}
/** @internal */
export const AlreadyActiveSubscriptionError$inboundSchema = z.object({
    error: z.literal("AlreadyActiveSubscriptionError"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new AlreadyActiveSubscriptionError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const AlreadyActiveSubscriptionError$outboundSchema = z.instanceof(AlreadyActiveSubscriptionError)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("AlreadyActiveSubscriptionError"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var AlreadyActiveSubscriptionError$;
(function (AlreadyActiveSubscriptionError$) {
    /** @deprecated use `AlreadyActiveSubscriptionError$inboundSchema` instead. */
    AlreadyActiveSubscriptionError$.inboundSchema = AlreadyActiveSubscriptionError$inboundSchema;
    /** @deprecated use `AlreadyActiveSubscriptionError$outboundSchema` instead. */
    AlreadyActiveSubscriptionError$.outboundSchema = AlreadyActiveSubscriptionError$outboundSchema;
})(AlreadyActiveSubscriptionError$ || (AlreadyActiveSubscriptionError$ = {}));
