// @polar-sh/sdk/dist/esm/models/errors/missinginvoicebillingdetails.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class MissingInvoiceBillingDetails extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "MissingInvoiceBillingDetails";
    }
}
/** @internal */
export const MissingInvoiceBillingDetails$inboundSchema = z.object({
    error: z.literal("MissingInvoiceBillingDetails"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new MissingInvoiceBillingDetails(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const MissingInvoiceBillingDetails$outboundSchema = z.instanceof(MissingInvoiceBillingDetails)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("MissingInvoiceBillingDetails"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MissingInvoiceBillingDetails$;
(function (MissingInvoiceBillingDetails$) {
    /** @deprecated use `MissingInvoiceBillingDetails$inboundSchema` instead. */
    MissingInvoiceBillingDetails$.inboundSchema = MissingInvoiceBillingDetails$inboundSchema;
    /** @deprecated use `MissingInvoiceBillingDetails$outboundSchema` instead. */
    MissingInvoiceBillingDetails$.outboundSchema = MissingInvoiceBillingDetails$outboundSchema;
})(MissingInvoiceBillingDetails$ || (MissingInvoiceBillingDetails$ = {}));
