// @polar-sh/sdk/dist/esm/models/errors/notopencheckout.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class NotOpenCheckout extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "NotOpenCheckout";
    }
}
/** @internal */
export const NotOpenCheckout$inboundSchema = z.object({
    error: z.literal("NotOpenCheckout"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new NotOpenCheckout(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const NotOpenCheckout$outboundSchema = z.instanceof(NotOpenCheckout)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("NotOpenCheckout"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotOpenCheckout$;
(function (NotOpenCheckout$) {
    /** @deprecated use `NotOpenCheckout$inboundSchema` instead. */
    NotOpenCheckout$.inboundSchema = NotOpenCheckout$inboundSchema;
    /** @deprecated use `NotOpenCheckout$outboundSchema` instead. */
    NotOpenCheckout$.outboundSchema = NotOpenCheckout$outboundSchema;
})(NotOpenCheckout$ || (NotOpenCheckout$ = {}));
