// @polar-sh/sdk/dist/esm/models/errors/paymenterror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class PaymentError extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "PaymentError";
    }
}
/** @internal */
export const PaymentError$inboundSchema = z.object({
    error: z.literal("PaymentError"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new PaymentError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const PaymentError$outboundSchema = z.instanceof(PaymentError)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("PaymentError"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PaymentError$;
(function (PaymentError$) {
    /** @deprecated use `PaymentError$inboundSchema` instead. */
    PaymentError$.inboundSchema = PaymentError$inboundSchema;
    /** @deprecated use `PaymentError$outboundSchema` instead. */
    PaymentError$.outboundSchema = PaymentError$outboundSchema;
})(PaymentError$ || (PaymentError$ = {}));
