// @polar-sh/sdk/dist/esm/models/errors/refundamounttoohigh.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class RefundAmountTooHigh extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "RefundAmountTooHigh";
    }
}
/** @internal */
export const RefundAmountTooHigh$inboundSchema = z.object({
    error: z.literal("RefundAmountTooHigh"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new RefundAmountTooHigh(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const RefundAmountTooHigh$outboundSchema = z.instanceof(RefundAmountTooHigh)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("RefundAmountTooHigh"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RefundAmountTooHigh$;
(function (RefundAmountTooHigh$) {
    /** @deprecated use `RefundAmountTooHigh$inboundSchema` instead. */
    RefundAmountTooHigh$.inboundSchema = RefundAmountTooHigh$inboundSchema;
    /** @deprecated use `RefundAmountTooHigh$outboundSchema` instead. */
    RefundAmountTooHigh$.outboundSchema = RefundAmountTooHigh$outboundSchema;
})(RefundAmountTooHigh$ || (RefundAmountTooHigh$ = {}));
