// @polar-sh/sdk/dist/esm/models/errors/resourcenotfound.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PolarError } from "./polarerror.js";
export class ResourceNotFound extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "ResourceNotFound";
    }
}
/** @internal */
export const ResourceNotFound$inboundSchema = z.object({
    error: z.literal("ResourceNotFound"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new ResourceNotFound(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const ResourceNotFound$outboundSchema = z.instanceof(ResourceNotFound)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("ResourceNotFound"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ResourceNotFound$;
(function (ResourceNotFound$) {
    /** @deprecated use `ResourceNotFound$inboundSchema` instead. */
    ResourceNotFound$.inboundSchema = ResourceNotFound$inboundSchema;
    /** @deprecated use `ResourceNotFound$outboundSchema` instead. */
    ResourceNotFound$.outboundSchema = ResourceNotFound$outboundSchema;
})(ResourceNotFound$ || (ResourceNotFound$ = {}));
