// @polar-sh/sdk/dist/commonjs/sdk/files.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Files = void 0;
const filesCreate_js_1 = require("../funcs/filesCreate.js");
const filesDelete_js_1 = require("../funcs/filesDelete.js");
const filesList_js_1 = require("../funcs/filesList.js");
const filesUpdate_js_1 = require("../funcs/filesUpdate.js");
const filesUploaded_js_1 = require("../funcs/filesUploaded.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const operations_js_1 = require("../types/operations.js");
class Files extends sdks_js_1.ClientSDK {
    /**
     * List Files
     *
     * @remarks
     * List files.
     *
     * **Scopes**: `files:read` `files:write`
     */
    async list(request, options) {
        return (0, operations_js_1.unwrapResultIterator)((0, filesList_js_1.filesList)(this, request, options));
    }
    /**
     * Create File
     *
     * @remarks
     * Create a file.
     *
     * **Scopes**: `files:write`
     */
    async create(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, filesCreate_js_1.filesCreate)(this, request, options));
    }
    /**
     * Complete File Upload
     *
     * @remarks
     * Complete a file upload.
     *
     * **Scopes**: `files:write`
     */
    async uploaded(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, filesUploaded_js_1.filesUploaded)(this, request, options));
    }
    /**
     * Update File
     *
     * @remarks
     * Update a file.
     *
     * **Scopes**: `files:write`
     */
    async update(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, filesUpdate_js_1.filesUpdate)(this, request, options));
    }
    /**
     * Delete File
     *
     * @remarks
     * Delete a file.
     *
     * **Scopes**: `files:write`
     */
    async delete(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, filesDelete_js_1.filesDelete)(this, request, options));
    }
}
exports.Files = Files;
