// @wasp.sh/lib-vite-ssr/dist/index.mjs
import assert from "node:assert/strict";
import * as path from "node:path";
import * as posixPath from "node:path/posix";
import * as fs from "node:fs/promises";
import { isRunnableDevEnvironment } from "vite";

//#region src/plugins/common/constants.ts
const PACKAGE_NAME = "@wasp.sh/lib-vite-ssr";
const ENVIRONMENT_NAMES = {
	SSR: "ssr",
	CLIENT: "client"
};

//#endregion
//#region src/plugins/build.ts
const ssrBuild = (routes, { ssrEntrySrc, clientEntrySrc }) => {
	let prerenderApp = null;
	return {
		name: `${PACKAGE_NAME}:build`,
		apply: "build",
		sharedDuringBuild: true,
		config() {
			return {
				environments: {
					[ENVIRONMENT_NAMES.SSR]: { build: {
						ssr: true,
						rollupOptions: { input: ssrEntrySrc }
					} },
					[ENVIRONMENT_NAMES.CLIENT]: { build: { rollupOptions: { input: routes.getAllIds() } } }
				},
				builder: { async buildApp(builder) {
					const { [ENVIRONMENT_NAMES.SSR]: ssrEnv, [ENVIRONMENT_NAMES.CLIENT]: clientEnv } = builder.environments;
					const ssrOutput = await builder.build(ssrEnv);
					assert(!Array.isArray(ssrOutput), "Expected ssr build output to be a single chunk");
					assert("output" in ssrOutput, "Watch mode is not supported for ssr production builds");
					const entryChunk = ssrOutput.output[0];
					assert(entryChunk, "Expected ssr build output to have at least one chunk");
					assert(entryChunk.exports.includes("default"), "Expected ssr build output chunk to export `default`");
					prerenderApp = (await import(path.resolve(ssrEnv.config.build.outDir, entryChunk.fileName))).default;
					await builder.build(clientEnv);
				} }
			};
		},
		resolveId: {
			filter: { id: /\.html$/ },
			handler(id, _, { ssr }) {
				if (ssr) return;
				assert(prerenderApp, "Expected ssr build to have completed before resolving ids in client build");
				if (routes.byId.has(id)) return id;
			}
		},
		load: {
			filter: { id: /\.html$/ },
			async handler(id, { ssr = false } = {}) {
				if (ssr) return;
				assert(prerenderApp, "Expected ssr build to have completed before resolving ids in client build");
				const route = routes.byId.get(id);
				assert(route, `Expected id ${id} to exist in ssrRoutes`);
				const html = await prerenderApp(route.path, { clientEntrySrc });
				if (!html) return;
				return html;
			}
		}
	};
};

//#endregion
//#region src/plugins/util/path.ts
function removeLeadingSlash(path) {
	return path.replace(/^\//, "");
}
async function pathExists(...params) {
	try {
		await fs.access(...params);
		return true;
	} catch {
		return false;
	}
}

//#endregion
//#region src/plugins/common/routes.ts
/**
* This class holds the collection of routes to be SSR'd, and precomputes lookup
* maps for them to quickly find a route while in the middle of a request or a
* build.
*/
var SsrRoutes = class {
	byId;
	byPath;
	spaFallbackFile;
	constructor(paths, spaFallbackFile) {
		this.spaFallbackFile = {
			path: spaFallbackFile,
			id: removeLeadingSlash(spaFallbackFile)
		};
		const routes = [...paths.map((path) => ({
			path,
			id: posixPath.join(removeLeadingSlash(path), "index.html")
		})), this.spaFallbackFile];
		this.byId = new Map(routes.map((route) => [route.id, route]));
		this.byPath = new Map(routes.map((route) => [route.path, route]));
	}
	getAllIds() {
		return Array.from(this.byId.keys());
	}
};

//#endregion
//#region src/plugins/util/html.ts
const appendToHead = (html, content) => html.replace(/<\/\s*head\s*>/i, (headClosingTag) => `${content}${headClosingTag}`);

//#endregion
//#region src/plugins/dev/css.ts
const REGISTER_CSS_EVENT_NAME = `${PACKAGE_NAME}:register-css`;
const appendCssRegistrationCode = (id, code) => code + `;

    if (import.meta.hot) {
      const register = () => import.meta.hot.send(${JSON.stringify(REGISTER_CSS_EVENT_NAME)}, { id: ${JSON.stringify(id)} });
      register();
      import.meta.hot.accept();
      import.meta.hot.prune(register);
    }
  `;
const withRegisteredCss = async (hot, makeHtml) => {
	const cssFiles = /* @__PURE__ */ new Set();
	const listener = ({ id }) => cssFiles.add(id);
	hot.on(REGISTER_CSS_EVENT_NAME, listener);
	try {
		const html = await makeHtml();
		if (!html) return html;
		return appendToHead(html, Array.from(cssFiles).map((id) => `<link rel="stylesheet" href="${id}">\n`).join(""));
	} finally {
		hot.off(REGISTER_CSS_EVENT_NAME, listener);
	}
};

//#endregion
//#region src/plugins/dev/index.ts
const ssrDev = (routes, { ssrEntrySrc, clientEntrySrc }) => {
	return {
		name: `${PACKAGE_NAME}:dev`,
		apply: "serve",
		transform: {
			filter: { id: /\.css$/ },
			order: "post",
			handler(code, id, opts) {
				if (opts?.ssr) return appendCssRegistrationCode(id, code);
			}
		},
		configureServer(server) {
			return () => {
				const ssrEnv = server.environments[ENVIRONMENT_NAMES.SSR];
				assert(isRunnableDevEnvironment(ssrEnv), "Expected ssr to be a runnable dev environment");
				server.middlewares.use(async ({ url, originalUrl }, res, next) => {
					if (!url || url !== "/index.html" || !originalUrl) return next();
					const route = routes.byPath.has(originalUrl) ? originalUrl : routes.spaFallbackFile.path;
					ssrEnv.runner.clearCache();
					const html = await withRegisteredCss(ssrEnv.hot, async () => {
						const { default: prerenderApp } = await ssrEnv.runner.import(ssrEntrySrc);
						return await prerenderApp(route, { clientEntrySrc });
					});
					if (!html) return next();
					const newHtml = await server.transformIndexHtml(originalUrl, html);
					res.statusCode = 200;
					res.setHeader("Content-Type", "text/html");
					res.end(newHtml);
				});
			};
		}
	};
};

//#endregion
//#region src/plugins/preview.ts
const ssrPreview = (routes) => {
	return {
		name: `${PACKAGE_NAME}:preview`,
		configurePreviewServer(server) {
			return () => {
				const clientOutDir = server.config.build.outDir;
				server.middlewares.use(async ({ url, originalUrl }, res, next) => {
					if (!url || url !== "/index.html" || !originalUrl) return next();
					const route = routes.byPath.get(originalUrl) ?? routes.spaFallbackFile;
					const htmlPath = path.resolve(clientOutDir, route.id);
					if (!await pathExists(htmlPath)) return next();
					const html = await fs.readFile(htmlPath, "utf-8");
					res.statusCode = 200;
					res.setHeader("Content-Type", "text/html");
					res.end(html);
				});
			};
		}
	};
};

//#endregion
//#region src/plugins/set-environments.ts
const ssrSetEnvironments = () => ({
	name: `${PACKAGE_NAME}:set-environments`,
	config: () => ({ environments: {
		[ENVIRONMENT_NAMES.SSR]: {},
		[ENVIRONMENT_NAMES.CLIENT]: {}
	} })
});

//#endregion
//#region src/plugins/index.ts
function ssr(options) {
	const routes = new SsrRoutes(options.ssrPaths, options.spaFallbackFile);
	return [
		ssrSetEnvironments(),
		ssrDev(routes, options),
		ssrBuild(routes, options),
		ssrPreview(routes)
	];
}

//#endregion
export { ssr as default };
