// @polar-sh/sdk/dist/esm/models/components/benefitdiscordproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitDiscordProperties$inboundSchema = z.object({
    guild_id: z.string(),
    role_id: z.string(),
    kick_member: z.boolean(),
    guild_token: z.string(),
}).transform((v) => {
    return remap$(v, {
        "guild_id": "guildId",
        "role_id": "roleId",
        "kick_member": "kickMember",
        "guild_token": "guildToken",
    });
});
/** @internal */
export const BenefitDiscordProperties$outboundSchema = z.object({
    guildId: z.string(),
    roleId: z.string(),
    kickMember: z.boolean(),
    guildToken: z.string(),
}).transform((v) => {
    return remap$(v, {
        guildId: "guild_id",
        roleId: "role_id",
        kickMember: "kick_member",
        guildToken: "guild_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordProperties$;
(function (BenefitDiscordProperties$) {
    /** @deprecated use `BenefitDiscordProperties$inboundSchema` instead. */
    BenefitDiscordProperties$.inboundSchema = BenefitDiscordProperties$inboundSchema;
    /** @deprecated use `BenefitDiscordProperties$outboundSchema` instead. */
    BenefitDiscordProperties$.outboundSchema = BenefitDiscordProperties$outboundSchema;
})(BenefitDiscordProperties$ || (BenefitDiscordProperties$ = {}));
export function benefitDiscordPropertiesToJSON(benefitDiscordProperties) {
    return JSON.stringify(BenefitDiscordProperties$outboundSchema.parse(benefitDiscordProperties));
}
export function benefitDiscordPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordProperties' from JSON`);
}
