// @polar-sh/sdk/dist/esm/models/components/benefitcustomcreateproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitCustomCreateProperties$inboundSchema = z.object({
    note: z.nullable(z.string()).optional(),
});
/** @internal */
export const BenefitCustomCreateProperties$outboundSchema = z.object({
    note: z.nullable(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitCustomCreateProperties$;
(function (BenefitCustomCreateProperties$) {
    /** @deprecated use `BenefitCustomCreateProperties$inboundSchema` instead. */
    BenefitCustomCreateProperties$.inboundSchema = BenefitCustomCreateProperties$inboundSchema;
    /** @deprecated use `BenefitCustomCreateProperties$outboundSchema` instead. */
    BenefitCustomCreateProperties$.outboundSchema = BenefitCustomCreateProperties$outboundSchema;
})(BenefitCustomCreateProperties$ || (BenefitCustomCreateProperties$ = {}));
export function benefitCustomCreatePropertiesToJSON(benefitCustomCreateProperties) {
    return JSON.stringify(BenefitCustomCreateProperties$outboundSchema.parse(benefitCustomCreateProperties));
}
export function benefitCustomCreatePropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomCreateProperties' from JSON`);
}
