// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablesupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDownloadablesCreateProperties$inboundSchema, BenefitDownloadablesCreateProperties$outboundSchema, } from "./benefitdownloadablescreateproperties.js";
/** @internal */
export const BenefitDownloadablesUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitDownloadablesUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDownloadablesUpdateMetadata$;
(function (BenefitDownloadablesUpdateMetadata$) {
    /** @deprecated use `BenefitDownloadablesUpdateMetadata$inboundSchema` instead. */
    BenefitDownloadablesUpdateMetadata$.inboundSchema = BenefitDownloadablesUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitDownloadablesUpdateMetadata$outboundSchema` instead. */
    BenefitDownloadablesUpdateMetadata$.outboundSchema = BenefitDownloadablesUpdateMetadata$outboundSchema;
})(BenefitDownloadablesUpdateMetadata$ || (BenefitDownloadablesUpdateMetadata$ = {}));
export function benefitDownloadablesUpdateMetadataToJSON(benefitDownloadablesUpdateMetadata) {
    return JSON.stringify(BenefitDownloadablesUpdateMetadata$outboundSchema.parse(benefitDownloadablesUpdateMetadata));
}
export function benefitDownloadablesUpdateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesUpdateMetadata' from JSON`);
}
/** @internal */
export const BenefitDownloadablesUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("downloadables"),
    properties: z.nullable(BenefitDownloadablesCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
export const BenefitDownloadablesUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("downloadables"),
    properties: z.nullable(BenefitDownloadablesCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDownloadablesUpdate$;
(function (BenefitDownloadablesUpdate$) {
    /** @deprecated use `BenefitDownloadablesUpdate$inboundSchema` instead. */
    BenefitDownloadablesUpdate$.inboundSchema = BenefitDownloadablesUpdate$inboundSchema;
    /** @deprecated use `BenefitDownloadablesUpdate$outboundSchema` instead. */
    BenefitDownloadablesUpdate$.outboundSchema = BenefitDownloadablesUpdate$outboundSchema;
})(BenefitDownloadablesUpdate$ || (BenefitDownloadablesUpdate$ = {}));
export function benefitDownloadablesUpdateToJSON(benefitDownloadablesUpdate) {
    return JSON.stringify(BenefitDownloadablesUpdate$outboundSchema.parse(benefitDownloadablesUpdate));
}
export function benefitDownloadablesUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesUpdate' from JSON`);
}
