// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitLicenseKeysProperties$inboundSchema, BenefitLicenseKeysProperties$outboundSchema, } from "./benefitlicensekeysproperties.js";
/** @internal */
export const BenefitLicenseKeysMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitLicenseKeysMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeysMetadata$;
(function (BenefitLicenseKeysMetadata$) {
    /** @deprecated use `BenefitLicenseKeysMetadata$inboundSchema` instead. */
    BenefitLicenseKeysMetadata$.inboundSchema = BenefitLicenseKeysMetadata$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysMetadata$outboundSchema` instead. */
    BenefitLicenseKeysMetadata$.outboundSchema = BenefitLicenseKeysMetadata$outboundSchema;
})(BenefitLicenseKeysMetadata$ || (BenefitLicenseKeysMetadata$ = {}));
export function benefitLicenseKeysMetadataToJSON(benefitLicenseKeysMetadata) {
    return JSON.stringify(BenefitLicenseKeysMetadata$outboundSchema.parse(benefitLicenseKeysMetadata));
}
export function benefitLicenseKeysMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysMetadata' from JSON`);
}
/** @internal */
export const BenefitLicenseKeys$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: BenefitLicenseKeysProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitLicenseKeys$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: BenefitLicenseKeysProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeys$;
(function (BenefitLicenseKeys$) {
    /** @deprecated use `BenefitLicenseKeys$inboundSchema` instead. */
    BenefitLicenseKeys$.inboundSchema = BenefitLicenseKeys$inboundSchema;
    /** @deprecated use `BenefitLicenseKeys$outboundSchema` instead. */
    BenefitLicenseKeys$.outboundSchema = BenefitLicenseKeys$outboundSchema;
})(BenefitLicenseKeys$ || (BenefitLicenseKeys$ = {}));
export function benefitLicenseKeysToJSON(benefitLicenseKeys) {
    return JSON.stringify(BenefitLicenseKeys$outboundSchema.parse(benefitLicenseKeys));
}
export function benefitLicenseKeysFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeys$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeys' from JSON`);
}
