// @polar-sh/sdk/dist/esm/models/components/benefitcustomsubscriberproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitCustomSubscriberProperties$inboundSchema = z.object({
    note: z.nullable(z.string()),
});
/** @internal */
export const BenefitCustomSubscriberProperties$outboundSchema = z.object({
    note: z.nullable(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitCustomSubscriberProperties$;
(function (BenefitCustomSubscriberProperties$) {
    /** @deprecated use `BenefitCustomSubscriberProperties$inboundSchema` instead. */
    BenefitCustomSubscriberProperties$.inboundSchema = BenefitCustomSubscriberProperties$inboundSchema;
    /** @deprecated use `BenefitCustomSubscriberProperties$outboundSchema` instead. */
    BenefitCustomSubscriberProperties$.outboundSchema = BenefitCustomSubscriberProperties$outboundSchema;
})(BenefitCustomSubscriberProperties$ || (BenefitCustomSubscriberProperties$ = {}));
export function benefitCustomSubscriberPropertiesToJSON(benefitCustomSubscriberProperties) {
    return JSON.stringify(BenefitCustomSubscriberProperties$outboundSchema.parse(benefitCustomSubscriberProperties));
}
export function benefitCustomSubscriberPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomSubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomSubscriberProperties' from JSON`);
}
