// @polar-sh/sdk/dist/esm/models/components/benefitgrantlicensekeysproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitGrantLicenseKeysProperties$inboundSchema = z.object({
    license_key_id: z.string().optional(),
    display_key: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "license_key_id": "licenseKeyId",
        "display_key": "displayKey",
    });
});
/** @internal */
export const BenefitGrantLicenseKeysProperties$outboundSchema = z.object({
    licenseKeyId: z.string().optional(),
    displayKey: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        licenseKeyId: "license_key_id",
        displayKey: "display_key",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantLicenseKeysProperties$;
(function (BenefitGrantLicenseKeysProperties$) {
    /** @deprecated use `BenefitGrantLicenseKeysProperties$inboundSchema` instead. */
    BenefitGrantLicenseKeysProperties$.inboundSchema = BenefitGrantLicenseKeysProperties$inboundSchema;
    /** @deprecated use `BenefitGrantLicenseKeysProperties$outboundSchema` instead. */
    BenefitGrantLicenseKeysProperties$.outboundSchema = BenefitGrantLicenseKeysProperties$outboundSchema;
})(BenefitGrantLicenseKeysProperties$ || (BenefitGrantLicenseKeysProperties$ = {}));
export function benefitGrantLicenseKeysPropertiesToJSON(benefitGrantLicenseKeysProperties) {
    return JSON.stringify(BenefitGrantLicenseKeysProperties$outboundSchema.parse(benefitGrantLicenseKeysProperties));
}
export function benefitGrantLicenseKeysPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantLicenseKeysProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantLicenseKeysProperties' from JSON`);
}
