// @polar-sh/sdk/dist/esm/models/components/benefitsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const BenefitSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Description: "description",
    MinusDescription: "-description",
    Type: "type",
    MinusType: "-type",
};
/** @internal */
export const BenefitSortProperty$inboundSchema = z.nativeEnum(BenefitSortProperty);
/** @internal */
export const BenefitSortProperty$outboundSchema = BenefitSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitSortProperty$;
(function (BenefitSortProperty$) {
    /** @deprecated use `BenefitSortProperty$inboundSchema` instead. */
    BenefitSortProperty$.inboundSchema = BenefitSortProperty$inboundSchema;
    /** @deprecated use `BenefitSortProperty$outboundSchema` instead. */
    BenefitSortProperty$.outboundSchema = BenefitSortProperty$outboundSchema;
})(BenefitSortProperty$ || (BenefitSortProperty$ = {}));
