// @polar-sh/sdk/dist/esm/models/components/benefitdiscordupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDiscordCreateProperties$inboundSchema, BenefitDiscordCreateProperties$outboundSchema, } from "./benefitdiscordcreateproperties.js";
/** @internal */
export const BenefitDiscordUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitDiscordUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordUpdateMetadata$;
(function (BenefitDiscordUpdateMetadata$) {
    /** @deprecated use `BenefitDiscordUpdateMetadata$inboundSchema` instead. */
    BenefitDiscordUpdateMetadata$.inboundSchema = BenefitDiscordUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordUpdateMetadata$outboundSchema` instead. */
    BenefitDiscordUpdateMetadata$.outboundSchema = BenefitDiscordUpdateMetadata$outboundSchema;
})(BenefitDiscordUpdateMetadata$ || (BenefitDiscordUpdateMetadata$ = {}));
export function benefitDiscordUpdateMetadataToJSON(benefitDiscordUpdateMetadata) {
    return JSON.stringify(BenefitDiscordUpdateMetadata$outboundSchema.parse(benefitDiscordUpdateMetadata));
}
export function benefitDiscordUpdateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordUpdateMetadata' from JSON`);
}
/** @internal */
export const BenefitDiscordUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("discord"),
    properties: z.nullable(BenefitDiscordCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
export const BenefitDiscordUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("discord"),
    properties: z.nullable(BenefitDiscordCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordUpdate$;
(function (BenefitDiscordUpdate$) {
    /** @deprecated use `BenefitDiscordUpdate$inboundSchema` instead. */
    BenefitDiscordUpdate$.inboundSchema = BenefitDiscordUpdate$inboundSchema;
    /** @deprecated use `BenefitDiscordUpdate$outboundSchema` instead. */
    BenefitDiscordUpdate$.outboundSchema = BenefitDiscordUpdate$outboundSchema;
})(BenefitDiscordUpdate$ || (BenefitDiscordUpdate$ = {}));
export function benefitDiscordUpdateToJSON(benefitDiscordUpdate) {
    return JSON.stringify(BenefitDiscordUpdate$outboundSchema.parse(benefitDiscordUpdate));
}
export function benefitDiscordUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordUpdate' from JSON`);
}
