// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyactivationcreateproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitLicenseKeyActivationCreateProperties$inboundSchema = z.object({
    limit: z.number().int(),
    enable_customer_admin: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "enable_customer_admin": "enableCustomerAdmin",
    });
});
/** @internal */
export const BenefitLicenseKeyActivationCreateProperties$outboundSchema = z.object({
    limit: z.number().int(),
    enableCustomerAdmin: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        enableCustomerAdmin: "enable_customer_admin",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeyActivationCreateProperties$;
(function (BenefitLicenseKeyActivationCreateProperties$) {
    /** @deprecated use `BenefitLicenseKeyActivationCreateProperties$inboundSchema` instead. */
    BenefitLicenseKeyActivationCreateProperties$.inboundSchema = BenefitLicenseKeyActivationCreateProperties$inboundSchema;
    /** @deprecated use `BenefitLicenseKeyActivationCreateProperties$outboundSchema` instead. */
    BenefitLicenseKeyActivationCreateProperties$.outboundSchema = BenefitLicenseKeyActivationCreateProperties$outboundSchema;
})(BenefitLicenseKeyActivationCreateProperties$ || (BenefitLicenseKeyActivationCreateProperties$ = {}));
export function benefitLicenseKeyActivationCreatePropertiesToJSON(benefitLicenseKeyActivationCreateProperties) {
    return JSON.stringify(BenefitLicenseKeyActivationCreateProperties$outboundSchema.parse(benefitLicenseKeyActivationCreateProperties));
}
export function benefitLicenseKeyActivationCreatePropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeyActivationCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeyActivationCreateProperties' from JSON`);
}
