// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitMeterCreditProperties$inboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meter_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "meter_id": "meterId",
    });
});
/** @internal */
export const BenefitMeterCreditProperties$outboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meterId: z.string(),
}).transform((v) => {
    return remap$(v, {
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitMeterCreditProperties$;
(function (BenefitMeterCreditProperties$) {
    /** @deprecated use `BenefitMeterCreditProperties$inboundSchema` instead. */
    BenefitMeterCreditProperties$.inboundSchema = BenefitMeterCreditProperties$inboundSchema;
    /** @deprecated use `BenefitMeterCreditProperties$outboundSchema` instead. */
    BenefitMeterCreditProperties$.outboundSchema = BenefitMeterCreditProperties$outboundSchema;
})(BenefitMeterCreditProperties$ || (BenefitMeterCreditProperties$ = {}));
export function benefitMeterCreditPropertiesToJSON(benefitMeterCreditProperties) {
    return JSON.stringify(BenefitMeterCreditProperties$outboundSchema.parse(benefitMeterCreditProperties));
}
export function benefitMeterCreditPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitMeterCreditProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditProperties' from JSON`);
}
