// @polar-sh/sdk/dist/esm/models/components/benefitcustomupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitCustomProperties$inboundSchema, BenefitCustomProperties$outboundSchema, } from "./benefitcustomproperties.js";
/** @internal */
export const BenefitCustomUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitCustomUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitCustomUpdateMetadata$;
(function (BenefitCustomUpdateMetadata$) {
    /** @deprecated use `BenefitCustomUpdateMetadata$inboundSchema` instead. */
    BenefitCustomUpdateMetadata$.inboundSchema = BenefitCustomUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitCustomUpdateMetadata$outboundSchema` instead. */
    BenefitCustomUpdateMetadata$.outboundSchema = BenefitCustomUpdateMetadata$outboundSchema;
})(BenefitCustomUpdateMetadata$ || (BenefitCustomUpdateMetadata$ = {}));
export function benefitCustomUpdateMetadataToJSON(benefitCustomUpdateMetadata) {
    return JSON.stringify(BenefitCustomUpdateMetadata$outboundSchema.parse(benefitCustomUpdateMetadata));
}
export function benefitCustomUpdateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomUpdateMetadata' from JSON`);
}
/** @internal */
export const BenefitCustomUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("custom"),
    properties: z.nullable(BenefitCustomProperties$inboundSchema).optional(),
});
/** @internal */
export const BenefitCustomUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("custom"),
    properties: z.nullable(BenefitCustomProperties$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitCustomUpdate$;
(function (BenefitCustomUpdate$) {
    /** @deprecated use `BenefitCustomUpdate$inboundSchema` instead. */
    BenefitCustomUpdate$.inboundSchema = BenefitCustomUpdate$inboundSchema;
    /** @deprecated use `BenefitCustomUpdate$outboundSchema` instead. */
    BenefitCustomUpdate$.outboundSchema = BenefitCustomUpdate$outboundSchema;
})(BenefitCustomUpdate$ || (BenefitCustomUpdate$ = {}));
export function benefitCustomUpdateToJSON(benefitCustomUpdate) {
    return JSON.stringify(BenefitCustomUpdate$outboundSchema.parse(benefitCustomUpdate));
}
export function benefitCustomUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomUpdate' from JSON`);
}
