// @polar-sh/sdk/dist/esm/models/components/benefitcustomcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitCustomCreateProperties$inboundSchema, BenefitCustomCreateProperties$outboundSchema, } from "./benefitcustomcreateproperties.js";
/** @internal */
export const BenefitCustomCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitCustomCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitCustomCreateMetadata$;
(function (BenefitCustomCreateMetadata$) {
    /** @deprecated use `BenefitCustomCreateMetadata$inboundSchema` instead. */
    BenefitCustomCreateMetadata$.inboundSchema = BenefitCustomCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitCustomCreateMetadata$outboundSchema` instead. */
    BenefitCustomCreateMetadata$.outboundSchema = BenefitCustomCreateMetadata$outboundSchema;
})(BenefitCustomCreateMetadata$ || (BenefitCustomCreateMetadata$ = {}));
export function benefitCustomCreateMetadataToJSON(benefitCustomCreateMetadata) {
    return JSON.stringify(BenefitCustomCreateMetadata$outboundSchema.parse(benefitCustomCreateMetadata));
}
export function benefitCustomCreateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomCreateMetadata' from JSON`);
}
/** @internal */
export const BenefitCustomCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("custom"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: BenefitCustomCreateProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const BenefitCustomCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("custom"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: BenefitCustomCreateProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitCustomCreate$;
(function (BenefitCustomCreate$) {
    /** @deprecated use `BenefitCustomCreate$inboundSchema` instead. */
    BenefitCustomCreate$.inboundSchema = BenefitCustomCreate$inboundSchema;
    /** @deprecated use `BenefitCustomCreate$outboundSchema` instead. */
    BenefitCustomCreate$.outboundSchema = BenefitCustomCreate$outboundSchema;
})(BenefitCustomCreate$ || (BenefitCustomCreate$ = {}));
export function benefitCustomCreateToJSON(benefitCustomCreate) {
    return JSON.stringify(BenefitCustomCreate$outboundSchema.parse(benefitCustomCreate));
}
export function benefitCustomCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomCreate' from JSON`);
}
