// @polar-sh/sdk/dist/esm/models/components/benefitgrantcustomwebhook.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitCustom$inboundSchema, BenefitCustom$outboundSchema, } from "./benefitcustom.js";
import { BenefitGrantCustomProperties$inboundSchema, BenefitGrantCustomProperties$outboundSchema, } from "./benefitgrantcustomproperties.js";
import { BenefitGrantError$inboundSchema, BenefitGrantError$outboundSchema, } from "./benefitgranterror.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const BenefitGrantCustomWebhook$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    is_granted: z.boolean(),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    is_revoked: z.boolean(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    customer_id: z.string(),
    benefit_id: z.string(),
    error: z.nullable(BenefitGrantError$inboundSchema).optional(),
    customer: Customer$inboundSchema,
    benefit: BenefitCustom$inboundSchema,
    properties: BenefitGrantCustomProperties$inboundSchema,
    previous_properties: z.nullable(BenefitGrantCustomProperties$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "is_granted": "isGranted",
        "revoked_at": "revokedAt",
        "is_revoked": "isRevoked",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "previous_properties": "previousProperties",
    });
});
/** @internal */
export const BenefitGrantCustomWebhook$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    isGranted: z.boolean(),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    isRevoked: z.boolean(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    customerId: z.string(),
    benefitId: z.string(),
    error: z.nullable(BenefitGrantError$outboundSchema).optional(),
    customer: Customer$outboundSchema,
    benefit: BenefitCustom$outboundSchema,
    properties: BenefitGrantCustomProperties$outboundSchema,
    previousProperties: z.nullable(BenefitGrantCustomProperties$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        isGranted: "is_granted",
        revokedAt: "revoked_at",
        isRevoked: "is_revoked",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        customerId: "customer_id",
        benefitId: "benefit_id",
        previousProperties: "previous_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitGrantCustomWebhook$;
(function (BenefitGrantCustomWebhook$) {
    /** @deprecated use `BenefitGrantCustomWebhook$inboundSchema` instead. */
    BenefitGrantCustomWebhook$.inboundSchema = BenefitGrantCustomWebhook$inboundSchema;
    /** @deprecated use `BenefitGrantCustomWebhook$outboundSchema` instead. */
    BenefitGrantCustomWebhook$.outboundSchema = BenefitGrantCustomWebhook$outboundSchema;
})(BenefitGrantCustomWebhook$ || (BenefitGrantCustomWebhook$ = {}));
export function benefitGrantCustomWebhookToJSON(benefitGrantCustomWebhook) {
    return JSON.stringify(BenefitGrantCustomWebhook$outboundSchema.parse(benefitGrantCustomWebhook));
}
export function benefitGrantCustomWebhookFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantCustomWebhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantCustomWebhook' from JSON`);
}
