// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablessubscriberproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitDownloadablesSubscriberProperties$inboundSchema = z.object({
    active_files: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        "active_files": "activeFiles",
    });
});
/** @internal */
export const BenefitDownloadablesSubscriberProperties$outboundSchema = z.object({
    activeFiles: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        activeFiles: "active_files",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDownloadablesSubscriberProperties$;
(function (BenefitDownloadablesSubscriberProperties$) {
    /** @deprecated use `BenefitDownloadablesSubscriberProperties$inboundSchema` instead. */
    BenefitDownloadablesSubscriberProperties$.inboundSchema = BenefitDownloadablesSubscriberProperties$inboundSchema;
    /** @deprecated use `BenefitDownloadablesSubscriberProperties$outboundSchema` instead. */
    BenefitDownloadablesSubscriberProperties$.outboundSchema = BenefitDownloadablesSubscriberProperties$outboundSchema;
})(BenefitDownloadablesSubscriberProperties$ || (BenefitDownloadablesSubscriberProperties$ = {}));
export function benefitDownloadablesSubscriberPropertiesToJSON(benefitDownloadablesSubscriberProperties) {
    return JSON.stringify(BenefitDownloadablesSubscriberProperties$outboundSchema.parse(benefitDownloadablesSubscriberProperties));
}
export function benefitDownloadablesSubscriberPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesSubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesSubscriberProperties' from JSON`);
}
