// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeysproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitLicenseKeyActivationProperties$inboundSchema, BenefitLicenseKeyActivationProperties$outboundSchema, } from "./benefitlicensekeyactivationproperties.js";
import { BenefitLicenseKeyExpirationProperties$inboundSchema, BenefitLicenseKeyExpirationProperties$outboundSchema, } from "./benefitlicensekeyexpirationproperties.js";
/** @internal */
export const BenefitLicenseKeysProperties$inboundSchema = z.object({
    prefix: z.nullable(z.string()),
    expires: z.nullable(BenefitLicenseKeyExpirationProperties$inboundSchema),
    activations: z.nullable(BenefitLicenseKeyActivationProperties$inboundSchema),
    limit_usage: z.nullable(z.number().int()),
}).transform((v) => {
    return remap$(v, {
        "limit_usage": "limitUsage",
    });
});
/** @internal */
export const BenefitLicenseKeysProperties$outboundSchema = z.object({
    prefix: z.nullable(z.string()),
    expires: z.nullable(BenefitLicenseKeyExpirationProperties$outboundSchema),
    activations: z.nullable(BenefitLicenseKeyActivationProperties$outboundSchema),
    limitUsage: z.nullable(z.number().int()),
}).transform((v) => {
    return remap$(v, {
        limitUsage: "limit_usage",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitLicenseKeysProperties$;
(function (BenefitLicenseKeysProperties$) {
    /** @deprecated use `BenefitLicenseKeysProperties$inboundSchema` instead. */
    BenefitLicenseKeysProperties$.inboundSchema = BenefitLicenseKeysProperties$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysProperties$outboundSchema` instead. */
    BenefitLicenseKeysProperties$.outboundSchema = BenefitLicenseKeysProperties$outboundSchema;
})(BenefitLicenseKeysProperties$ || (BenefitLicenseKeysProperties$ = {}));
export function benefitLicenseKeysPropertiesToJSON(benefitLicenseKeysProperties) {
    return JSON.stringify(BenefitLicenseKeysProperties$outboundSchema.parse(benefitLicenseKeysProperties));
}
export function benefitLicenseKeysPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysProperties' from JSON`);
}
