// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditsubscriberproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitMeterCreditSubscriberProperties$inboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meter_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "meter_id": "meterId",
    });
});
/** @internal */
export const BenefitMeterCreditSubscriberProperties$outboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meterId: z.string(),
}).transform((v) => {
    return remap$(v, {
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitMeterCreditSubscriberProperties$;
(function (BenefitMeterCreditSubscriberProperties$) {
    /** @deprecated use `BenefitMeterCreditSubscriberProperties$inboundSchema` instead. */
    BenefitMeterCreditSubscriberProperties$.inboundSchema = BenefitMeterCreditSubscriberProperties$inboundSchema;
    /** @deprecated use `BenefitMeterCreditSubscriberProperties$outboundSchema` instead. */
    BenefitMeterCreditSubscriberProperties$.outboundSchema = BenefitMeterCreditSubscriberProperties$outboundSchema;
})(BenefitMeterCreditSubscriberProperties$ || (BenefitMeterCreditSubscriberProperties$ = {}));
export function benefitMeterCreditSubscriberPropertiesToJSON(benefitMeterCreditSubscriberProperties) {
    return JSON.stringify(BenefitMeterCreditSubscriberProperties$outboundSchema.parse(benefitMeterCreditSubscriberProperties));
}
export function benefitMeterCreditSubscriberPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitMeterCreditSubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditSubscriberProperties' from JSON`);
}
