// @polar-sh/sdk/dist/esm/models/components/benefitdiscordcreateproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitDiscordCreateProperties$inboundSchema = z.object({
    guild_token: z.string(),
    role_id: z.string(),
    kick_member: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "guild_token": "guildToken",
        "role_id": "roleId",
        "kick_member": "kickMember",
    });
});
/** @internal */
export const BenefitDiscordCreateProperties$outboundSchema = z.object({
    guildToken: z.string(),
    roleId: z.string(),
    kickMember: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        guildToken: "guild_token",
        roleId: "role_id",
        kickMember: "kick_member",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitDiscordCreateProperties$;
(function (BenefitDiscordCreateProperties$) {
    /** @deprecated use `BenefitDiscordCreateProperties$inboundSchema` instead. */
    BenefitDiscordCreateProperties$.inboundSchema = BenefitDiscordCreateProperties$inboundSchema;
    /** @deprecated use `BenefitDiscordCreateProperties$outboundSchema` instead. */
    BenefitDiscordCreateProperties$.outboundSchema = BenefitDiscordCreateProperties$outboundSchema;
})(BenefitDiscordCreateProperties$ || (BenefitDiscordCreateProperties$ = {}));
export function benefitDiscordCreatePropertiesToJSON(benefitDiscordCreateProperties) {
    return JSON.stringify(BenefitDiscordCreateProperties$outboundSchema.parse(benefitDiscordCreateProperties));
}
export function benefitDiscordCreatePropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDiscordCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordCreateProperties' from JSON`);
}
