// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditcreateproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitMeterCreditCreateProperties$inboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meter_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "meter_id": "meterId",
    });
});
/** @internal */
export const BenefitMeterCreditCreateProperties$outboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meterId: z.string(),
}).transform((v) => {
    return remap$(v, {
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BenefitMeterCreditCreateProperties$;
(function (BenefitMeterCreditCreateProperties$) {
    /** @deprecated use `BenefitMeterCreditCreateProperties$inboundSchema` instead. */
    BenefitMeterCreditCreateProperties$.inboundSchema = BenefitMeterCreditCreateProperties$inboundSchema;
    /** @deprecated use `BenefitMeterCreditCreateProperties$outboundSchema` instead. */
    BenefitMeterCreditCreateProperties$.outboundSchema = BenefitMeterCreditCreateProperties$outboundSchema;
})(BenefitMeterCreditCreateProperties$ || (BenefitMeterCreditCreateProperties$ = {}));
export function benefitMeterCreditCreatePropertiesToJSON(benefitMeterCreditCreateProperties) {
    return JSON.stringify(BenefitMeterCreditCreateProperties$outboundSchema.parse(benefitMeterCreditCreateProperties));
}
export function benefitMeterCreditCreatePropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitMeterCreditCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditCreateProperties' from JSON`);
}
