// @polar-sh/sdk/dist/esm/models/components/cardpayment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CardPaymentMetadata$inboundSchema, CardPaymentMetadata$outboundSchema, } from "./cardpaymentmetadata.js";
import { PaymentProcessor$inboundSchema, PaymentProcessor$outboundSchema, } from "./paymentprocessor.js";
import { PaymentStatus$inboundSchema, PaymentStatus$outboundSchema, } from "./paymentstatus.js";
/** @internal */
export const CardPayment$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    processor: PaymentProcessor$inboundSchema,
    status: PaymentStatus$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    method: z.literal("card"),
    decline_reason: z.nullable(z.string()),
    decline_message: z.nullable(z.string()),
    organization_id: z.string(),
    checkout_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    processor_metadata: z.record(z.any()).optional(),
    method_metadata: CardPaymentMetadata$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "decline_reason": "declineReason",
        "decline_message": "declineMessage",
        "organization_id": "organizationId",
        "checkout_id": "checkoutId",
        "order_id": "orderId",
        "processor_metadata": "processorMetadata",
        "method_metadata": "methodMetadata",
    });
});
/** @internal */
export const CardPayment$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    processor: PaymentProcessor$outboundSchema,
    status: PaymentStatus$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    method: z.literal("card"),
    declineReason: z.nullable(z.string()),
    declineMessage: z.nullable(z.string()),
    organizationId: z.string(),
    checkoutId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    processorMetadata: z.record(z.any()).optional(),
    methodMetadata: CardPaymentMetadata$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        declineReason: "decline_reason",
        declineMessage: "decline_message",
        organizationId: "organization_id",
        checkoutId: "checkout_id",
        orderId: "order_id",
        processorMetadata: "processor_metadata",
        methodMetadata: "method_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CardPayment$;
(function (CardPayment$) {
    /** @deprecated use `CardPayment$inboundSchema` instead. */
    CardPayment$.inboundSchema = CardPayment$inboundSchema;
    /** @deprecated use `CardPayment$outboundSchema` instead. */
    CardPayment$.outboundSchema = CardPayment$outboundSchema;
})(CardPayment$ || (CardPayment$ = {}));
export function cardPaymentToJSON(cardPayment) {
    return JSON.stringify(CardPayment$outboundSchema.parse(cardPayment));
}
export function cardPaymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CardPayment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CardPayment' from JSON`);
}
