// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountpercentageonceforeverduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountPercentageOnceForeverDuration$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    basis_points: z.number().int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "basis_points": "basisPoints",
    });
});
/** @internal */
export const CheckoutDiscountPercentageOnceForeverDuration$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    type: DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        basisPoints: "basis_points",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutDiscountPercentageOnceForeverDuration$;
(function (CheckoutDiscountPercentageOnceForeverDuration$) {
    /** @deprecated use `CheckoutDiscountPercentageOnceForeverDuration$inboundSchema` instead. */
    CheckoutDiscountPercentageOnceForeverDuration$.inboundSchema = CheckoutDiscountPercentageOnceForeverDuration$inboundSchema;
    /** @deprecated use `CheckoutDiscountPercentageOnceForeverDuration$outboundSchema` instead. */
    CheckoutDiscountPercentageOnceForeverDuration$.outboundSchema = CheckoutDiscountPercentageOnceForeverDuration$outboundSchema;
})(CheckoutDiscountPercentageOnceForeverDuration$ || (CheckoutDiscountPercentageOnceForeverDuration$ = {}));
export function checkoutDiscountPercentageOnceForeverDurationToJSON(checkoutDiscountPercentageOnceForeverDuration) {
    return JSON.stringify(CheckoutDiscountPercentageOnceForeverDuration$outboundSchema.parse(checkoutDiscountPercentageOnceForeverDuration));
}
export function checkoutDiscountPercentageOnceForeverDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountPercentageOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountPercentageOnceForeverDuration' from JSON`);
}
