// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountfixedrepeatduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountFixedRepeatDuration$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    duration_in_months: z.number().int(),
    type: DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "duration_in_months": "durationInMonths",
    });
});
/** @internal */
export const CheckoutDiscountFixedRepeatDuration$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.number().int(),
    type: DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutDiscountFixedRepeatDuration$;
(function (CheckoutDiscountFixedRepeatDuration$) {
    /** @deprecated use `CheckoutDiscountFixedRepeatDuration$inboundSchema` instead. */
    CheckoutDiscountFixedRepeatDuration$.inboundSchema = CheckoutDiscountFixedRepeatDuration$inboundSchema;
    /** @deprecated use `CheckoutDiscountFixedRepeatDuration$outboundSchema` instead. */
    CheckoutDiscountFixedRepeatDuration$.outboundSchema = CheckoutDiscountFixedRepeatDuration$outboundSchema;
})(CheckoutDiscountFixedRepeatDuration$ || (CheckoutDiscountFixedRepeatDuration$ = {}));
export function checkoutDiscountFixedRepeatDurationToJSON(checkoutDiscountFixedRepeatDuration) {
    return JSON.stringify(CheckoutDiscountFixedRepeatDuration$outboundSchema.parse(checkoutDiscountFixedRepeatDuration));
}
export function checkoutDiscountFixedRepeatDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountFixedRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountFixedRepeatDuration' from JSON`);
}
