// @polar-sh/sdk/dist/esm/models/components/checkoutlinkcreateproducts.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CheckoutLinkCreateProductsMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const CheckoutLinkCreateProductsMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkCreateProductsMetadata$;
(function (CheckoutLinkCreateProductsMetadata$) {
    /** @deprecated use `CheckoutLinkCreateProductsMetadata$inboundSchema` instead. */
    CheckoutLinkCreateProductsMetadata$.inboundSchema = CheckoutLinkCreateProductsMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductsMetadata$outboundSchema` instead. */
    CheckoutLinkCreateProductsMetadata$.outboundSchema = CheckoutLinkCreateProductsMetadata$outboundSchema;
})(CheckoutLinkCreateProductsMetadata$ || (CheckoutLinkCreateProductsMetadata$ = {}));
export function checkoutLinkCreateProductsMetadataToJSON(checkoutLinkCreateProductsMetadata) {
    return JSON.stringify(CheckoutLinkCreateProductsMetadata$outboundSchema.parse(checkoutLinkCreateProductsMetadata));
}
export function checkoutLinkCreateProductsMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkCreateProductsMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductsMetadata' from JSON`);
}
/** @internal */
export const CheckoutLinkCreateProducts$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    payment_processor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    require_billing_address: z.boolean().default(false),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    products: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "success_url": "successUrl",
    });
});
/** @internal */
export const CheckoutLinkCreateProducts$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    paymentProcessor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    requireBillingAddress: z.boolean().default(false),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    products: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinkCreateProducts$;
(function (CheckoutLinkCreateProducts$) {
    /** @deprecated use `CheckoutLinkCreateProducts$inboundSchema` instead. */
    CheckoutLinkCreateProducts$.inboundSchema = CheckoutLinkCreateProducts$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProducts$outboundSchema` instead. */
    CheckoutLinkCreateProducts$.outboundSchema = CheckoutLinkCreateProducts$outboundSchema;
})(CheckoutLinkCreateProducts$ || (CheckoutLinkCreateProducts$ = {}));
export function checkoutLinkCreateProductsToJSON(checkoutLinkCreateProducts) {
    return JSON.stringify(CheckoutLinkCreateProducts$outboundSchema.parse(checkoutLinkCreateProducts));
}
export function checkoutLinkCreateProductsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkCreateProducts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProducts' from JSON`);
}
