// @polar-sh/sdk/dist/esm/models/components/customersubscriptionsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const CustomerSubscriptionSortProperty = {
    StartedAt: "started_at",
    MinusStartedAt: "-started_at",
    Amount: "amount",
    MinusAmount: "-amount",
    Status: "status",
    MinusStatus: "-status",
    Organization: "organization",
    MinusOrganization: "-organization",
    Product: "product",
    MinusProduct: "-product",
};
/** @internal */
export const CustomerSubscriptionSortProperty$inboundSchema = z.nativeEnum(CustomerSubscriptionSortProperty);
/** @internal */
export const CustomerSubscriptionSortProperty$outboundSchema = CustomerSubscriptionSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSubscriptionSortProperty$;
(function (CustomerSubscriptionSortProperty$) {
    /** @deprecated use `CustomerSubscriptionSortProperty$inboundSchema` instead. */
    CustomerSubscriptionSortProperty$.inboundSchema = CustomerSubscriptionSortProperty$inboundSchema;
    /** @deprecated use `CustomerSubscriptionSortProperty$outboundSchema` instead. */
    CustomerSubscriptionSortProperty$.outboundSchema = CustomerSubscriptionSortProperty$outboundSchema;
})(CustomerSubscriptionSortProperty$ || (CustomerSubscriptionSortProperty$ = {}));
