// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantcustom.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitCustomSubscriber$inboundSchema, BenefitCustomSubscriber$outboundSchema, } from "./benefitcustomsubscriber.js";
import { BenefitGrantCustomProperties$inboundSchema, BenefitGrantCustomProperties$outboundSchema, } from "./benefitgrantcustomproperties.js";
import { CustomerPortalCustomer$inboundSchema, CustomerPortalCustomer$outboundSchema, } from "./customerportalcustomer.js";
/** @internal */
export const CustomerBenefitGrantCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    benefit_id: z.string(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    is_granted: z.boolean(),
    is_revoked: z.boolean(),
    customer: CustomerPortalCustomer$inboundSchema,
    benefit: BenefitCustomSubscriber$inboundSchema,
    properties: BenefitGrantCustomProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "revoked_at": "revokedAt",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "is_granted": "isGranted",
        "is_revoked": "isRevoked",
    });
});
/** @internal */
export const CustomerBenefitGrantCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    benefitId: z.string(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    isGranted: z.boolean(),
    isRevoked: z.boolean(),
    customer: CustomerPortalCustomer$outboundSchema,
    benefit: BenefitCustomSubscriber$outboundSchema,
    properties: BenefitGrantCustomProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        revokedAt: "revoked_at",
        customerId: "customer_id",
        benefitId: "benefit_id",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        isGranted: "is_granted",
        isRevoked: "is_revoked",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantCustom$;
(function (CustomerBenefitGrantCustom$) {
    /** @deprecated use `CustomerBenefitGrantCustom$inboundSchema` instead. */
    CustomerBenefitGrantCustom$.inboundSchema = CustomerBenefitGrantCustom$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantCustom$outboundSchema` instead. */
    CustomerBenefitGrantCustom$.outboundSchema = CustomerBenefitGrantCustom$outboundSchema;
})(CustomerBenefitGrantCustom$ || (CustomerBenefitGrantCustom$ = {}));
export function customerBenefitGrantCustomToJSON(customerBenefitGrantCustom) {
    return JSON.stringify(CustomerBenefitGrantCustom$outboundSchema.parse(customerBenefitGrantCustom));
}
export function customerBenefitGrantCustomFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantCustom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantCustom' from JSON`);
}
