// @polar-sh/sdk/dist/esm/models/components/customfieldupdatecheckbox.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomFieldCheckboxProperties$inboundSchema, CustomFieldCheckboxProperties$outboundSchema, } from "./customfieldcheckboxproperties.js";
/** @internal */
export const CustomFieldUpdateCheckboxMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const CustomFieldUpdateCheckboxMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldUpdateCheckboxMetadata$;
(function (CustomFieldUpdateCheckboxMetadata$) {
    /** @deprecated use `CustomFieldUpdateCheckboxMetadata$inboundSchema` instead. */
    CustomFieldUpdateCheckboxMetadata$.inboundSchema = CustomFieldUpdateCheckboxMetadata$inboundSchema;
    /** @deprecated use `CustomFieldUpdateCheckboxMetadata$outboundSchema` instead. */
    CustomFieldUpdateCheckboxMetadata$.outboundSchema = CustomFieldUpdateCheckboxMetadata$outboundSchema;
})(CustomFieldUpdateCheckboxMetadata$ || (CustomFieldUpdateCheckboxMetadata$ = {}));
export function customFieldUpdateCheckboxMetadataToJSON(customFieldUpdateCheckboxMetadata) {
    return JSON.stringify(CustomFieldUpdateCheckboxMetadata$outboundSchema.parse(customFieldUpdateCheckboxMetadata));
}
export function customFieldUpdateCheckboxMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldUpdateCheckboxMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdateCheckboxMetadata' from JSON`);
}
/** @internal */
export const CustomFieldUpdateCheckbox$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.nullable(z.string()).optional(),
    slug: z.nullable(z.string()).optional(),
    type: z.literal("checkbox"),
    properties: z.nullable(CustomFieldCheckboxProperties$inboundSchema)
        .optional(),
});
/** @internal */
export const CustomFieldUpdateCheckbox$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.nullable(z.string()).optional(),
    slug: z.nullable(z.string()).optional(),
    type: z.literal("checkbox"),
    properties: z.nullable(CustomFieldCheckboxProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldUpdateCheckbox$;
(function (CustomFieldUpdateCheckbox$) {
    /** @deprecated use `CustomFieldUpdateCheckbox$inboundSchema` instead. */
    CustomFieldUpdateCheckbox$.inboundSchema = CustomFieldUpdateCheckbox$inboundSchema;
    /** @deprecated use `CustomFieldUpdateCheckbox$outboundSchema` instead. */
    CustomFieldUpdateCheckbox$.outboundSchema = CustomFieldUpdateCheckbox$outboundSchema;
})(CustomFieldUpdateCheckbox$ || (CustomFieldUpdateCheckbox$ = {}));
export function customFieldUpdateCheckboxToJSON(customFieldUpdateCheckbox) {
    return JSON.stringify(CustomFieldUpdateCheckbox$outboundSchema.parse(customFieldUpdateCheckbox));
}
export function customFieldUpdateCheckboxFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldUpdateCheckbox$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdateCheckbox' from JSON`);
}
