// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantlicensekeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantLicenseKeysProperties$inboundSchema, BenefitGrantLicenseKeysProperties$outboundSchema, } from "./benefitgrantlicensekeysproperties.js";
import { BenefitLicenseKeysSubscriber$inboundSchema, BenefitLicenseKeysSubscriber$outboundSchema, } from "./benefitlicensekeyssubscriber.js";
import { CustomerPortalCustomer$inboundSchema, CustomerPortalCustomer$outboundSchema, } from "./customerportalcustomer.js";
/** @internal */
export const CustomerBenefitGrantLicenseKeys$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    benefit_id: z.string(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    is_granted: z.boolean(),
    is_revoked: z.boolean(),
    customer: CustomerPortalCustomer$inboundSchema,
    benefit: BenefitLicenseKeysSubscriber$inboundSchema,
    properties: BenefitGrantLicenseKeysProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "revoked_at": "revokedAt",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "is_granted": "isGranted",
        "is_revoked": "isRevoked",
    });
});
/** @internal */
export const CustomerBenefitGrantLicenseKeys$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    benefitId: z.string(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    isGranted: z.boolean(),
    isRevoked: z.boolean(),
    customer: CustomerPortalCustomer$outboundSchema,
    benefit: BenefitLicenseKeysSubscriber$outboundSchema,
    properties: BenefitGrantLicenseKeysProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        revokedAt: "revoked_at",
        customerId: "customer_id",
        benefitId: "benefit_id",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        isGranted: "is_granted",
        isRevoked: "is_revoked",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantLicenseKeys$;
(function (CustomerBenefitGrantLicenseKeys$) {
    /** @deprecated use `CustomerBenefitGrantLicenseKeys$inboundSchema` instead. */
    CustomerBenefitGrantLicenseKeys$.inboundSchema = CustomerBenefitGrantLicenseKeys$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantLicenseKeys$outboundSchema` instead. */
    CustomerBenefitGrantLicenseKeys$.outboundSchema = CustomerBenefitGrantLicenseKeys$outboundSchema;
})(CustomerBenefitGrantLicenseKeys$ || (CustomerBenefitGrantLicenseKeys$ = {}));
export function customerBenefitGrantLicenseKeysToJSON(customerBenefitGrantLicenseKeys) {
    return JSON.stringify(CustomerBenefitGrantLicenseKeys$outboundSchema.parse(customerBenefitGrantLicenseKeys));
}
export function customerBenefitGrantLicenseKeysFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantLicenseKeys$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantLicenseKeys' from JSON`);
}
