// @polar-sh/sdk/dist/esm/models/components/customfieldcheckboxproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomFieldCheckboxProperties$inboundSchema = z.object({
    form_label: z.string().optional(),
    form_help_text: z.string().optional(),
    form_placeholder: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
});
/** @internal */
export const CustomFieldCheckboxProperties$outboundSchema = z.object({
    formLabel: z.string().optional(),
    formHelpText: z.string().optional(),
    formPlaceholder: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldCheckboxProperties$;
(function (CustomFieldCheckboxProperties$) {
    /** @deprecated use `CustomFieldCheckboxProperties$inboundSchema` instead. */
    CustomFieldCheckboxProperties$.inboundSchema = CustomFieldCheckboxProperties$inboundSchema;
    /** @deprecated use `CustomFieldCheckboxProperties$outboundSchema` instead. */
    CustomFieldCheckboxProperties$.outboundSchema = CustomFieldCheckboxProperties$outboundSchema;
})(CustomFieldCheckboxProperties$ || (CustomFieldCheckboxProperties$ = {}));
export function customFieldCheckboxPropertiesToJSON(customFieldCheckboxProperties) {
    return JSON.stringify(CustomFieldCheckboxProperties$outboundSchema.parse(customFieldCheckboxProperties));
}
export function customFieldCheckboxPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldCheckboxProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCheckboxProperties' from JSON`);
}
