// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantmetercreditupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerBenefitGrantMeterCreditUpdate$inboundSchema = z.object({
    benefit_type: z.literal("meter_credit"),
}).transform((v) => {
    return remap$(v, {
        "benefit_type": "benefitType",
    });
});
/** @internal */
export const CustomerBenefitGrantMeterCreditUpdate$outboundSchema = z.object({
    benefitType: z.literal("meter_credit"),
}).transform((v) => {
    return remap$(v, {
        benefitType: "benefit_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantMeterCreditUpdate$;
(function (CustomerBenefitGrantMeterCreditUpdate$) {
    /** @deprecated use `CustomerBenefitGrantMeterCreditUpdate$inboundSchema` instead. */
    CustomerBenefitGrantMeterCreditUpdate$.inboundSchema = CustomerBenefitGrantMeterCreditUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantMeterCreditUpdate$outboundSchema` instead. */
    CustomerBenefitGrantMeterCreditUpdate$.outboundSchema = CustomerBenefitGrantMeterCreditUpdate$outboundSchema;
})(CustomerBenefitGrantMeterCreditUpdate$ || (CustomerBenefitGrantMeterCreditUpdate$ = {}));
export function customerBenefitGrantMeterCreditUpdateToJSON(customerBenefitGrantMeterCreditUpdate) {
    return JSON.stringify(CustomerBenefitGrantMeterCreditUpdate$outboundSchema.parse(customerBenefitGrantMeterCreditUpdate));
}
export function customerBenefitGrantMeterCreditUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantMeterCreditUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantMeterCreditUpdate' from JSON`);
}
