// @polar-sh/sdk/dist/esm/models/components/customersessioncustomerexternalidcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSessionCustomerExternalIDCreate$inboundSchema = z.object({
    external_customer_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const CustomerSessionCustomerExternalIDCreate$outboundSchema = z.object({
    externalCustomerId: z.string(),
}).transform((v) => {
    return remap$(v, {
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSessionCustomerExternalIDCreate$;
(function (CustomerSessionCustomerExternalIDCreate$) {
    /** @deprecated use `CustomerSessionCustomerExternalIDCreate$inboundSchema` instead. */
    CustomerSessionCustomerExternalIDCreate$.inboundSchema = CustomerSessionCustomerExternalIDCreate$inboundSchema;
    /** @deprecated use `CustomerSessionCustomerExternalIDCreate$outboundSchema` instead. */
    CustomerSessionCustomerExternalIDCreate$.outboundSchema = CustomerSessionCustomerExternalIDCreate$outboundSchema;
})(CustomerSessionCustomerExternalIDCreate$ || (CustomerSessionCustomerExternalIDCreate$ = {}));
export function customerSessionCustomerExternalIDCreateToJSON(customerSessionCustomerExternalIDCreate) {
    return JSON.stringify(CustomerSessionCustomerExternalIDCreate$outboundSchema.parse(customerSessionCustomerExternalIDCreate));
}
export function customerSessionCustomerExternalIDCreateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSessionCustomerExternalIDCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSessionCustomerExternalIDCreate' from JSON`);
}
