// @polar-sh/sdk/dist/esm/models/components/customerstatesubscriptionmeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerStateSubscriptionMeter$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    consumed_units: z.number(),
    credited_units: z.number().int(),
    amount: z.number().int(),
    meter_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
        "meter_id": "meterId",
    });
});
/** @internal */
export const CustomerStateSubscriptionMeter$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.number().int(),
    amount: z.number().int(),
    meterId: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerStateSubscriptionMeter$;
(function (CustomerStateSubscriptionMeter$) {
    /** @deprecated use `CustomerStateSubscriptionMeter$inboundSchema` instead. */
    CustomerStateSubscriptionMeter$.inboundSchema = CustomerStateSubscriptionMeter$inboundSchema;
    /** @deprecated use `CustomerStateSubscriptionMeter$outboundSchema` instead. */
    CustomerStateSubscriptionMeter$.outboundSchema = CustomerStateSubscriptionMeter$outboundSchema;
})(CustomerStateSubscriptionMeter$ || (CustomerStateSubscriptionMeter$ = {}));
export function customerStateSubscriptionMeterToJSON(customerStateSubscriptionMeter) {
    return JSON.stringify(CustomerStateSubscriptionMeter$outboundSchema.parse(customerStateSubscriptionMeter));
}
export function customerStateSubscriptionMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateSubscriptionMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateSubscriptionMeter' from JSON`);
}
