// @polar-sh/sdk/dist/esm/models/components/customerorderpaymentstatus.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerOrderPaymentStatus$inboundSchema = z.object({
    status: z.string(),
    error: z.nullable(z.string()).optional(),
});
/** @internal */
export const CustomerOrderPaymentStatus$outboundSchema = z.object({
    status: z.string(),
    error: z.nullable(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerOrderPaymentStatus$;
(function (CustomerOrderPaymentStatus$) {
    /** @deprecated use `CustomerOrderPaymentStatus$inboundSchema` instead. */
    CustomerOrderPaymentStatus$.inboundSchema = CustomerOrderPaymentStatus$inboundSchema;
    /** @deprecated use `CustomerOrderPaymentStatus$outboundSchema` instead. */
    CustomerOrderPaymentStatus$.outboundSchema = CustomerOrderPaymentStatus$outboundSchema;
})(CustomerOrderPaymentStatus$ || (CustomerOrderPaymentStatus$ = {}));
export function customerOrderPaymentStatusToJSON(customerOrderPaymentStatus) {
    return JSON.stringify(CustomerOrderPaymentStatus$outboundSchema.parse(customerOrderPaymentStatus));
}
export function customerOrderPaymentStatusFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerOrderPaymentStatus$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrderPaymentStatus' from JSON`);
}
