// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantgithubrepositorypropertiesupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$inboundSchema = z.object({
    account_id: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "account_id": "accountId",
    });
});
/** @internal */
export const CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$outboundSchema = z.object({
    accountId: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        accountId: "account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$;
(function (CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$) {
    /** @deprecated use `CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$inboundSchema` instead. */
    CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$.inboundSchema = CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$outboundSchema` instead. */
    CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$.outboundSchema = CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$outboundSchema;
})(CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$ || (CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$ = {}));
export function customerBenefitGrantGitHubRepositoryPropertiesUpdateToJSON(customerBenefitGrantGitHubRepositoryPropertiesUpdate) {
    return JSON.stringify(CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$outboundSchema.parse(customerBenefitGrantGitHubRepositoryPropertiesUpdate));
}
export function customerBenefitGrantGitHubRepositoryPropertiesUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantGitHubRepositoryPropertiesUpdate' from JSON`);
}
