// @polar-sh/sdk/dist/esm/models/components/customersubscriptionmeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSubscriptionMeterMeter$inboundSchema, CustomerSubscriptionMeterMeter$outboundSchema, } from "./customersubscriptionmetermeter.js";
/** @internal */
export const CustomerSubscriptionMeter$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    consumed_units: z.number(),
    credited_units: z.number().int(),
    amount: z.number().int(),
    meter_id: z.string(),
    meter: CustomerSubscriptionMeterMeter$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
        "meter_id": "meterId",
    });
});
/** @internal */
export const CustomerSubscriptionMeter$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.number().int(),
    amount: z.number().int(),
    meterId: z.string(),
    meter: CustomerSubscriptionMeterMeter$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSubscriptionMeter$;
(function (CustomerSubscriptionMeter$) {
    /** @deprecated use `CustomerSubscriptionMeter$inboundSchema` instead. */
    CustomerSubscriptionMeter$.inboundSchema = CustomerSubscriptionMeter$inboundSchema;
    /** @deprecated use `CustomerSubscriptionMeter$outboundSchema` instead. */
    CustomerSubscriptionMeter$.outboundSchema = CustomerSubscriptionMeter$outboundSchema;
})(CustomerSubscriptionMeter$ || (CustomerSubscriptionMeter$ = {}));
export function customerSubscriptionMeterToJSON(customerSubscriptionMeter) {
    return JSON.stringify(CustomerSubscriptionMeter$outboundSchema.parse(customerSubscriptionMeter));
}
export function customerSubscriptionMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionMeter' from JSON`);
}
