// @polar-sh/sdk/dist/esm/models/components/customerbenefitgrantcustomupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerBenefitGrantCustomUpdate$inboundSchema = z.object({
    benefit_type: z.literal("custom"),
}).transform((v) => {
    return remap$(v, {
        "benefit_type": "benefitType",
    });
});
/** @internal */
export const CustomerBenefitGrantCustomUpdate$outboundSchema = z.object({
    benefitType: z.literal("custom"),
}).transform((v) => {
    return remap$(v, {
        benefitType: "benefit_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerBenefitGrantCustomUpdate$;
(function (CustomerBenefitGrantCustomUpdate$) {
    /** @deprecated use `CustomerBenefitGrantCustomUpdate$inboundSchema` instead. */
    CustomerBenefitGrantCustomUpdate$.inboundSchema = CustomerBenefitGrantCustomUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantCustomUpdate$outboundSchema` instead. */
    CustomerBenefitGrantCustomUpdate$.outboundSchema = CustomerBenefitGrantCustomUpdate$outboundSchema;
})(CustomerBenefitGrantCustomUpdate$ || (CustomerBenefitGrantCustomUpdate$ = {}));
export function customerBenefitGrantCustomUpdateToJSON(customerBenefitGrantCustomUpdate) {
    return JSON.stringify(CustomerBenefitGrantCustomUpdate$outboundSchema.parse(customerBenefitGrantCustomUpdate));
}
export function customerBenefitGrantCustomUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerBenefitGrantCustomUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantCustomUpdate' from JSON`);
}
