// @polar-sh/sdk/dist/esm/models/components/customfieldcreatecheckbox.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomFieldCheckboxProperties$inboundSchema, CustomFieldCheckboxProperties$outboundSchema, } from "./customfieldcheckboxproperties.js";
/** @internal */
export const CustomFieldCreateCheckboxMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const CustomFieldCreateCheckboxMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldCreateCheckboxMetadata$;
(function (CustomFieldCreateCheckboxMetadata$) {
    /** @deprecated use `CustomFieldCreateCheckboxMetadata$inboundSchema` instead. */
    CustomFieldCreateCheckboxMetadata$.inboundSchema = CustomFieldCreateCheckboxMetadata$inboundSchema;
    /** @deprecated use `CustomFieldCreateCheckboxMetadata$outboundSchema` instead. */
    CustomFieldCreateCheckboxMetadata$.outboundSchema = CustomFieldCreateCheckboxMetadata$outboundSchema;
})(CustomFieldCreateCheckboxMetadata$ || (CustomFieldCreateCheckboxMetadata$ = {}));
export function customFieldCreateCheckboxMetadataToJSON(customFieldCreateCheckboxMetadata) {
    return JSON.stringify(CustomFieldCreateCheckboxMetadata$outboundSchema.parse(customFieldCreateCheckboxMetadata));
}
export function customFieldCreateCheckboxMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldCreateCheckboxMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCreateCheckboxMetadata' from JSON`);
}
/** @internal */
export const CustomFieldCreateCheckbox$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: CustomFieldCheckboxProperties$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const CustomFieldCreateCheckbox$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: CustomFieldCheckboxProperties$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldCreateCheckbox$;
(function (CustomFieldCreateCheckbox$) {
    /** @deprecated use `CustomFieldCreateCheckbox$inboundSchema` instead. */
    CustomFieldCreateCheckbox$.inboundSchema = CustomFieldCreateCheckbox$inboundSchema;
    /** @deprecated use `CustomFieldCreateCheckbox$outboundSchema` instead. */
    CustomFieldCreateCheckbox$.outboundSchema = CustomFieldCreateCheckbox$outboundSchema;
})(CustomFieldCreateCheckbox$ || (CustomFieldCreateCheckbox$ = {}));
export function customFieldCreateCheckboxToJSON(customFieldCreateCheckbox) {
    return JSON.stringify(CustomFieldCreateCheckbox$outboundSchema.parse(customFieldCreateCheckbox));
}
export function customFieldCreateCheckboxFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldCreateCheckbox$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCreateCheckbox' from JSON`);
}
