// @polar-sh/sdk/dist/esm/models/components/customercustomermetermeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerCustomerMeterMeter$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
    });
});
/** @internal */
export const CustomerCustomerMeterMeter$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerCustomerMeterMeter$;
(function (CustomerCustomerMeterMeter$) {
    /** @deprecated use `CustomerCustomerMeterMeter$inboundSchema` instead. */
    CustomerCustomerMeterMeter$.inboundSchema = CustomerCustomerMeterMeter$inboundSchema;
    /** @deprecated use `CustomerCustomerMeterMeter$outboundSchema` instead. */
    CustomerCustomerMeterMeter$.outboundSchema = CustomerCustomerMeterMeter$outboundSchema;
})(CustomerCustomerMeterMeter$ || (CustomerCustomerMeterMeter$ = {}));
export function customerCustomerMeterMeterToJSON(customerCustomerMeterMeter) {
    return JSON.stringify(CustomerCustomerMeterMeter$outboundSchema.parse(customerCustomerMeterMeter));
}
export function customerCustomerMeterMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerCustomerMeterMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerCustomerMeterMeter' from JSON`);
}
