// @polar-sh/sdk/dist/esm/models/components/customfieldnumberproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomFieldNumberProperties$inboundSchema = z.object({
    form_label: z.string().optional(),
    form_help_text: z.string().optional(),
    form_placeholder: z.string().optional(),
    ge: z.number().int().optional(),
    le: z.number().int().optional(),
}).transform((v) => {
    return remap$(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
});
/** @internal */
export const CustomFieldNumberProperties$outboundSchema = z.object({
    formLabel: z.string().optional(),
    formHelpText: z.string().optional(),
    formPlaceholder: z.string().optional(),
    ge: z.number().int().optional(),
    le: z.number().int().optional(),
}).transform((v) => {
    return remap$(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldNumberProperties$;
(function (CustomFieldNumberProperties$) {
    /** @deprecated use `CustomFieldNumberProperties$inboundSchema` instead. */
    CustomFieldNumberProperties$.inboundSchema = CustomFieldNumberProperties$inboundSchema;
    /** @deprecated use `CustomFieldNumberProperties$outboundSchema` instead. */
    CustomFieldNumberProperties$.outboundSchema = CustomFieldNumberProperties$outboundSchema;
})(CustomFieldNumberProperties$ || (CustomFieldNumberProperties$ = {}));
export function customFieldNumberPropertiesToJSON(customFieldNumberProperties) {
    return JSON.stringify(CustomFieldNumberProperties$outboundSchema.parse(customFieldNumberProperties));
}
export function customFieldNumberPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldNumberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldNumberProperties' from JSON`);
}
