// @polar-sh/sdk/dist/esm/models/components/discountfixedonceforeverdurationbase.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const DiscountFixedOnceForeverDurationBaseMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const DiscountFixedOnceForeverDurationBaseMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountFixedOnceForeverDurationBaseMetadata$;
(function (DiscountFixedOnceForeverDurationBaseMetadata$) {
    /** @deprecated use `DiscountFixedOnceForeverDurationBaseMetadata$inboundSchema` instead. */
    DiscountFixedOnceForeverDurationBaseMetadata$.inboundSchema = DiscountFixedOnceForeverDurationBaseMetadata$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDurationBaseMetadata$outboundSchema` instead. */
    DiscountFixedOnceForeverDurationBaseMetadata$.outboundSchema = DiscountFixedOnceForeverDurationBaseMetadata$outboundSchema;
})(DiscountFixedOnceForeverDurationBaseMetadata$ || (DiscountFixedOnceForeverDurationBaseMetadata$ = {}));
export function discountFixedOnceForeverDurationBaseMetadataToJSON(discountFixedOnceForeverDurationBaseMetadata) {
    return JSON.stringify(DiscountFixedOnceForeverDurationBaseMetadata$outboundSchema.parse(discountFixedOnceForeverDurationBaseMetadata));
}
export function discountFixedOnceForeverDurationBaseMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountFixedOnceForeverDurationBaseMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDurationBaseMetadata' from JSON`);
}
/** @internal */
export const DiscountFixedOnceForeverDurationBase$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const DiscountFixedOnceForeverDurationBase$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    type: DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountFixedOnceForeverDurationBase$;
(function (DiscountFixedOnceForeverDurationBase$) {
    /** @deprecated use `DiscountFixedOnceForeverDurationBase$inboundSchema` instead. */
    DiscountFixedOnceForeverDurationBase$.inboundSchema = DiscountFixedOnceForeverDurationBase$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDurationBase$outboundSchema` instead. */
    DiscountFixedOnceForeverDurationBase$.outboundSchema = DiscountFixedOnceForeverDurationBase$outboundSchema;
})(DiscountFixedOnceForeverDurationBase$ || (DiscountFixedOnceForeverDurationBase$ = {}));
export function discountFixedOnceForeverDurationBaseToJSON(discountFixedOnceForeverDurationBase) {
    return JSON.stringify(DiscountFixedOnceForeverDurationBase$outboundSchema.parse(discountFixedOnceForeverDurationBase));
}
export function discountFixedOnceForeverDurationBaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountFixedOnceForeverDurationBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDurationBase' from JSON`);
}
