// @polar-sh/sdk/dist/esm/models/components/discountsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const DiscountSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Name: "name",
    MinusName: "-name",
    Code: "code",
    MinusCode: "-code",
    RedemptionsCount: "redemptions_count",
    MinusRedemptionsCount: "-redemptions_count",
};
/** @internal */
export const DiscountSortProperty$inboundSchema = z.nativeEnum(DiscountSortProperty);
/** @internal */
export const DiscountSortProperty$outboundSchema = DiscountSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountSortProperty$;
(function (DiscountSortProperty$) {
    /** @deprecated use `DiscountSortProperty$inboundSchema` instead. */
    DiscountSortProperty$.inboundSchema = DiscountSortProperty$inboundSchema;
    /** @deprecated use `DiscountSortProperty$outboundSchema` instead. */
    DiscountSortProperty$.outboundSchema = DiscountSortProperty$outboundSchema;
})(DiscountSortProperty$ || (DiscountSortProperty$ = {}));
