// @polar-sh/sdk/dist/esm/models/components/discountfixedonceforeverduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountProduct$inboundSchema, DiscountProduct$outboundSchema, } from "./discountproduct.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const DiscountFixedOnceForeverDurationMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
export const DiscountFixedOnceForeverDurationMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountFixedOnceForeverDurationMetadata$;
(function (DiscountFixedOnceForeverDurationMetadata$) {
    /** @deprecated use `DiscountFixedOnceForeverDurationMetadata$inboundSchema` instead. */
    DiscountFixedOnceForeverDurationMetadata$.inboundSchema = DiscountFixedOnceForeverDurationMetadata$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDurationMetadata$outboundSchema` instead. */
    DiscountFixedOnceForeverDurationMetadata$.outboundSchema = DiscountFixedOnceForeverDurationMetadata$outboundSchema;
})(DiscountFixedOnceForeverDurationMetadata$ || (DiscountFixedOnceForeverDurationMetadata$ = {}));
export function discountFixedOnceForeverDurationMetadataToJSON(discountFixedOnceForeverDurationMetadata) {
    return JSON.stringify(DiscountFixedOnceForeverDurationMetadata$outboundSchema.parse(discountFixedOnceForeverDurationMetadata));
}
export function discountFixedOnceForeverDurationMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountFixedOnceForeverDurationMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDurationMetadata' from JSON`);
}
/** @internal */
export const DiscountFixedOnceForeverDuration$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
    products: z.array(DiscountProduct$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
export const DiscountFixedOnceForeverDuration$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    type: DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
    products: z.array(DiscountProduct$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountFixedOnceForeverDuration$;
(function (DiscountFixedOnceForeverDuration$) {
    /** @deprecated use `DiscountFixedOnceForeverDuration$inboundSchema` instead. */
    DiscountFixedOnceForeverDuration$.inboundSchema = DiscountFixedOnceForeverDuration$inboundSchema;
    /** @deprecated use `DiscountFixedOnceForeverDuration$outboundSchema` instead. */
    DiscountFixedOnceForeverDuration$.outboundSchema = DiscountFixedOnceForeverDuration$outboundSchema;
})(DiscountFixedOnceForeverDuration$ || (DiscountFixedOnceForeverDuration$ = {}));
export function discountFixedOnceForeverDurationToJSON(discountFixedOnceForeverDuration) {
    return JSON.stringify(DiscountFixedOnceForeverDuration$outboundSchema.parse(discountFixedOnceForeverDuration));
}
export function discountFixedOnceForeverDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountFixedOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedOnceForeverDuration' from JSON`);
}
