// @polar-sh/sdk/dist/esm/models/components/legacyrecurringproductpricefixed.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const LegacyRecurringProductPriceFixed$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("fixed"),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: z.literal("recurring"),
    recurring_interval: SubscriptionRecurringInterval$inboundSchema,
    price_currency: z.string(),
    price_amount: z.number().int(),
    legacy: z.literal(true),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "price_currency": "priceCurrency",
        "price_amount": "priceAmount",
    });
});
/** @internal */
export const LegacyRecurringProductPriceFixed$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("fixed"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: z.literal("recurring"),
    recurringInterval: SubscriptionRecurringInterval$outboundSchema,
    priceCurrency: z.string(),
    priceAmount: z.number().int(),
    legacy: z.literal(true),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
        priceCurrency: "price_currency",
        priceAmount: "price_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var LegacyRecurringProductPriceFixed$;
(function (LegacyRecurringProductPriceFixed$) {
    /** @deprecated use `LegacyRecurringProductPriceFixed$inboundSchema` instead. */
    LegacyRecurringProductPriceFixed$.inboundSchema = LegacyRecurringProductPriceFixed$inboundSchema;
    /** @deprecated use `LegacyRecurringProductPriceFixed$outboundSchema` instead. */
    LegacyRecurringProductPriceFixed$.outboundSchema = LegacyRecurringProductPriceFixed$outboundSchema;
})(LegacyRecurringProductPriceFixed$ || (LegacyRecurringProductPriceFixed$ = {}));
export function legacyRecurringProductPriceFixedToJSON(legacyRecurringProductPriceFixed) {
    return JSON.stringify(LegacyRecurringProductPriceFixed$outboundSchema.parse(legacyRecurringProductPriceFixed));
}
export function legacyRecurringProductPriceFixedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => LegacyRecurringProductPriceFixed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LegacyRecurringProductPriceFixed' from JSON`);
}
