// @polar-sh/sdk/dist/esm/models/components/listresourceorganization.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Organization$inboundSchema, Organization$outboundSchema, } from "./organization.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceOrganization$inboundSchema = z.object({
    items: z.array(Organization$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceOrganization$outboundSchema = z.object({
    items: z.array(Organization$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceOrganization$;
(function (ListResourceOrganization$) {
    /** @deprecated use `ListResourceOrganization$inboundSchema` instead. */
    ListResourceOrganization$.inboundSchema = ListResourceOrganization$inboundSchema;
    /** @deprecated use `ListResourceOrganization$outboundSchema` instead. */
    ListResourceOrganization$.outboundSchema = ListResourceOrganization$outboundSchema;
})(ListResourceOrganization$ || (ListResourceOrganization$ = {}));
export function listResourceOrganizationToJSON(listResourceOrganization) {
    return JSON.stringify(ListResourceOrganization$outboundSchema.parse(listResourceOrganization));
}
export function listResourceOrganizationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceOrganization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceOrganization' from JSON`);
}
