// @polar-sh/sdk/dist/esm/models/components/listresourcediscount.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Discount$inboundSchema, Discount$outboundSchema, } from "./discount.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceDiscount$inboundSchema = z.object({
    items: z.array(Discount$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceDiscount$outboundSchema = z.object({
    items: z.array(Discount$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceDiscount$;
(function (ListResourceDiscount$) {
    /** @deprecated use `ListResourceDiscount$inboundSchema` instead. */
    ListResourceDiscount$.inboundSchema = ListResourceDiscount$inboundSchema;
    /** @deprecated use `ListResourceDiscount$outboundSchema` instead. */
    ListResourceDiscount$.outboundSchema = ListResourceDiscount$outboundSchema;
})(ListResourceDiscount$ || (ListResourceDiscount$ = {}));
export function listResourceDiscountToJSON(listResourceDiscount) {
    return JSON.stringify(ListResourceDiscount$outboundSchema.parse(listResourceDiscount));
}
export function listResourceDiscountFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceDiscount' from JSON`);
}
