// @polar-sh/sdk/dist/esm/models/components/listresourcecustomerorder.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerOrder$inboundSchema, CustomerOrder$outboundSchema, } from "./customerorder.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomerOrder$inboundSchema = z.object({
    items: z.array(CustomerOrder$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomerOrder$outboundSchema = z.object({
    items: z.array(CustomerOrder$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomerOrder$;
(function (ListResourceCustomerOrder$) {
    /** @deprecated use `ListResourceCustomerOrder$inboundSchema` instead. */
    ListResourceCustomerOrder$.inboundSchema = ListResourceCustomerOrder$inboundSchema;
    /** @deprecated use `ListResourceCustomerOrder$outboundSchema` instead. */
    ListResourceCustomerOrder$.outboundSchema = ListResourceCustomerOrder$outboundSchema;
})(ListResourceCustomerOrder$ || (ListResourceCustomerOrder$ = {}));
export function listResourceCustomerOrderToJSON(listResourceCustomerOrder) {
    return JSON.stringify(ListResourceCustomerOrder$outboundSchema.parse(listResourceCustomerOrder));
}
export function listResourceCustomerOrderFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomerOrder$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerOrder' from JSON`);
}
