// @polar-sh/sdk/dist/esm/models/components/listresourcecustomercustomermeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerCustomerMeter$inboundSchema, CustomerCustomerMeter$outboundSchema, } from "./customercustomermeter.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomerCustomerMeter$inboundSchema = z.object({
    items: z.array(CustomerCustomerMeter$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomerCustomerMeter$outboundSchema = z.object({
    items: z.array(CustomerCustomerMeter$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomerCustomerMeter$;
(function (ListResourceCustomerCustomerMeter$) {
    /** @deprecated use `ListResourceCustomerCustomerMeter$inboundSchema` instead. */
    ListResourceCustomerCustomerMeter$.inboundSchema = ListResourceCustomerCustomerMeter$inboundSchema;
    /** @deprecated use `ListResourceCustomerCustomerMeter$outboundSchema` instead. */
    ListResourceCustomerCustomerMeter$.outboundSchema = ListResourceCustomerCustomerMeter$outboundSchema;
})(ListResourceCustomerCustomerMeter$ || (ListResourceCustomerCustomerMeter$ = {}));
export function listResourceCustomerCustomerMeterToJSON(listResourceCustomerCustomerMeter) {
    return JSON.stringify(ListResourceCustomerCustomerMeter$outboundSchema.parse(listResourceCustomerCustomerMeter));
}
export function listResourceCustomerCustomerMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomerCustomerMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerCustomerMeter' from JSON`);
}
