// @polar-sh/sdk/dist/esm/models/components/listresourcecheckout.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Checkout$inboundSchema, Checkout$outboundSchema, } from "./checkout.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCheckout$inboundSchema = z.object({
    items: z.array(Checkout$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCheckout$outboundSchema = z.object({
    items: z.array(Checkout$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCheckout$;
(function (ListResourceCheckout$) {
    /** @deprecated use `ListResourceCheckout$inboundSchema` instead. */
    ListResourceCheckout$.inboundSchema = ListResourceCheckout$inboundSchema;
    /** @deprecated use `ListResourceCheckout$outboundSchema` instead. */
    ListResourceCheckout$.outboundSchema = ListResourceCheckout$outboundSchema;
})(ListResourceCheckout$ || (ListResourceCheckout$ = {}));
export function listResourceCheckoutToJSON(listResourceCheckout) {
    return JSON.stringify(ListResourceCheckout$outboundSchema.parse(listResourceCheckout));
}
export function listResourceCheckoutFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCheckout$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCheckout' from JSON`);
}
