// @polar-sh/sdk/dist/esm/models/components/listresourcewebhookendpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
import { WebhookEndpoint$inboundSchema, WebhookEndpoint$outboundSchema, } from "./webhookendpoint.js";
/** @internal */
export const ListResourceWebhookEndpoint$inboundSchema = z.object({
    items: z.array(WebhookEndpoint$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceWebhookEndpoint$outboundSchema = z.object({
    items: z.array(WebhookEndpoint$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceWebhookEndpoint$;
(function (ListResourceWebhookEndpoint$) {
    /** @deprecated use `ListResourceWebhookEndpoint$inboundSchema` instead. */
    ListResourceWebhookEndpoint$.inboundSchema = ListResourceWebhookEndpoint$inboundSchema;
    /** @deprecated use `ListResourceWebhookEndpoint$outboundSchema` instead. */
    ListResourceWebhookEndpoint$.outboundSchema = ListResourceWebhookEndpoint$outboundSchema;
})(ListResourceWebhookEndpoint$ || (ListResourceWebhookEndpoint$ = {}));
export function listResourceWebhookEndpointToJSON(listResourceWebhookEndpoint) {
    return JSON.stringify(ListResourceWebhookEndpoint$outboundSchema.parse(listResourceWebhookEndpoint));
}
export function listResourceWebhookEndpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceWebhookEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceWebhookEndpoint' from JSON`);
}
