// @polar-sh/sdk/dist/esm/models/components/listresourcecustomerpaymentmethod.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { CustomerPaymentMethod$inboundSchema, CustomerPaymentMethod$outboundSchema, } from "./customerpaymentmethod.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/** @internal */
export const ListResourceCustomerPaymentMethod$inboundSchema = z.object({
    items: z.array(CustomerPaymentMethod$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceCustomerPaymentMethod$outboundSchema = z.object({
    items: z.array(CustomerPaymentMethod$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceCustomerPaymentMethod$;
(function (ListResourceCustomerPaymentMethod$) {
    /** @deprecated use `ListResourceCustomerPaymentMethod$inboundSchema` instead. */
    ListResourceCustomerPaymentMethod$.inboundSchema = ListResourceCustomerPaymentMethod$inboundSchema;
    /** @deprecated use `ListResourceCustomerPaymentMethod$outboundSchema` instead. */
    ListResourceCustomerPaymentMethod$.outboundSchema = ListResourceCustomerPaymentMethod$outboundSchema;
})(ListResourceCustomerPaymentMethod$ || (ListResourceCustomerPaymentMethod$ = {}));
export function listResourceCustomerPaymentMethodToJSON(listResourceCustomerPaymentMethod) {
    return JSON.stringify(ListResourceCustomerPaymentMethod$outboundSchema.parse(listResourceCustomerPaymentMethod));
}
export function listResourceCustomerPaymentMethodFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceCustomerPaymentMethod$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceCustomerPaymentMethod' from JSON`);
}
