// @polar-sh/sdk/dist/esm/models/components/listresourcewebhookdelivery.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
import { WebhookDelivery$inboundSchema, WebhookDelivery$outboundSchema, } from "./webhookdelivery.js";
/** @internal */
export const ListResourceWebhookDelivery$inboundSchema = z.object({
    items: z.array(WebhookDelivery$inboundSchema),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListResourceWebhookDelivery$outboundSchema = z.object({
    items: z.array(WebhookDelivery$outboundSchema),
    pagination: Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListResourceWebhookDelivery$;
(function (ListResourceWebhookDelivery$) {
    /** @deprecated use `ListResourceWebhookDelivery$inboundSchema` instead. */
    ListResourceWebhookDelivery$.inboundSchema = ListResourceWebhookDelivery$inboundSchema;
    /** @deprecated use `ListResourceWebhookDelivery$outboundSchema` instead. */
    ListResourceWebhookDelivery$.outboundSchema = ListResourceWebhookDelivery$outboundSchema;
})(ListResourceWebhookDelivery$ || (ListResourceWebhookDelivery$ = {}));
export function listResourceWebhookDeliveryToJSON(listResourceWebhookDelivery) {
    return JSON.stringify(ListResourceWebhookDelivery$outboundSchema.parse(listResourceWebhookDelivery));
}
export function listResourceWebhookDeliveryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListResourceWebhookDelivery$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceWebhookDelivery' from JSON`);
}
