// @polar-sh/sdk/dist/esm/models/components/metercreditevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { MeterCreditedMetadata$inboundSchema, MeterCreditedMetadata$outboundSchema, } from "./metercreditedmetadata.js";
/** @internal */
export const MeterCreditEvent$inboundSchema = z.object({
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("meter.credited"),
    metadata: MeterCreditedMetadata$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const MeterCreditEvent$outboundSchema = z.object({
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("meter.credited"),
    metadata: MeterCreditedMetadata$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MeterCreditEvent$;
(function (MeterCreditEvent$) {
    /** @deprecated use `MeterCreditEvent$inboundSchema` instead. */
    MeterCreditEvent$.inboundSchema = MeterCreditEvent$inboundSchema;
    /** @deprecated use `MeterCreditEvent$outboundSchema` instead. */
    MeterCreditEvent$.outboundSchema = MeterCreditEvent$outboundSchema;
})(MeterCreditEvent$ || (MeterCreditEvent$ = {}));
export function meterCreditEventToJSON(meterCreditEvent) {
    return JSON.stringify(MeterCreditEvent$outboundSchema.parse(meterCreditEvent));
}
export function meterCreditEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MeterCreditEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeterCreditEvent' from JSON`);
}
