// @polar-sh/sdk/dist/esm/models/components/orderupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
/** @internal */
export const OrderUpdate$inboundSchema = z.object({
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "billing_name": "billingName",
        "billing_address": "billingAddress",
    });
});
/** @internal */
export const OrderUpdate$outboundSchema = z.object({
    billingName: z.nullable(z.string()),
    billingAddress: z.nullable(Address$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        billingName: "billing_name",
        billingAddress: "billing_address",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrderUpdate$;
(function (OrderUpdate$) {
    /** @deprecated use `OrderUpdate$inboundSchema` instead. */
    OrderUpdate$.inboundSchema = OrderUpdate$inboundSchema;
    /** @deprecated use `OrderUpdate$outboundSchema` instead. */
    OrderUpdate$.outboundSchema = OrderUpdate$outboundSchema;
})(OrderUpdate$ || (OrderUpdate$ = {}));
export function orderUpdateToJSON(orderUpdate) {
    return JSON.stringify(OrderUpdate$outboundSchema.parse(orderUpdate));
}
export function orderUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrderUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderUpdate' from JSON`);
}
