// @polar-sh/sdk/dist/esm/models/components/organizationsociallink.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { OrganizationSocialPlatforms$inboundSchema, OrganizationSocialPlatforms$outboundSchema, } from "./organizationsocialplatforms.js";
/** @internal */
export const OrganizationSocialLink$inboundSchema = z.object({
    platform: OrganizationSocialPlatforms$inboundSchema,
    url: z.string(),
});
/** @internal */
export const OrganizationSocialLink$outboundSchema = z.object({
    platform: OrganizationSocialPlatforms$outboundSchema,
    url: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrganizationSocialLink$;
(function (OrganizationSocialLink$) {
    /** @deprecated use `OrganizationSocialLink$inboundSchema` instead. */
    OrganizationSocialLink$.inboundSchema = OrganizationSocialLink$inboundSchema;
    /** @deprecated use `OrganizationSocialLink$outboundSchema` instead. */
    OrganizationSocialLink$.outboundSchema = OrganizationSocialLink$outboundSchema;
})(OrganizationSocialLink$ || (OrganizationSocialLink$ = {}));
export function organizationSocialLinkToJSON(organizationSocialLink) {
    return JSON.stringify(OrganizationSocialLink$outboundSchema.parse(organizationSocialLink));
}
export function organizationSocialLinkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationSocialLink$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationSocialLink' from JSON`);
}
