// @polar-sh/sdk/dist/esm/models/components/organizationsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const OrganizationSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Slug: "slug",
    MinusSlug: "-slug",
    Name: "name",
    MinusName: "-name",
};
/** @internal */
export const OrganizationSortProperty$inboundSchema = z.nativeEnum(OrganizationSortProperty);
/** @internal */
export const OrganizationSortProperty$outboundSchema = OrganizationSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrganizationSortProperty$;
(function (OrganizationSortProperty$) {
    /** @deprecated use `OrganizationSortProperty$inboundSchema` instead. */
    OrganizationSortProperty$.inboundSchema = OrganizationSortProperty$inboundSchema;
    /** @deprecated use `OrganizationSortProperty$outboundSchema` instead. */
    OrganizationSortProperty$.outboundSchema = OrganizationSortProperty$outboundSchema;
})(OrganizationSortProperty$ || (OrganizationSortProperty$ = {}));
