// @polar-sh/sdk/dist/esm/models/components/productpricecustom.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceType$inboundSchema, ProductPriceType$outboundSchema, } from "./productpricetype.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const ProductPriceCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("custom"),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: ProductPriceType$inboundSchema,
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    price_currency: z.string(),
    minimum_amount: z.nullable(z.number().int()),
    maximum_amount: z.nullable(z.number().int()),
    preset_amount: z.nullable(z.number().int()),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
export const ProductPriceCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("custom"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: ProductPriceType$outboundSchema,
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
    priceCurrency: z.string(),
    minimumAmount: z.nullable(z.number().int()),
    maximumAmount: z.nullable(z.number().int()),
    presetAmount: z.nullable(z.number().int()),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceCustom$;
(function (ProductPriceCustom$) {
    /** @deprecated use `ProductPriceCustom$inboundSchema` instead. */
    ProductPriceCustom$.inboundSchema = ProductPriceCustom$inboundSchema;
    /** @deprecated use `ProductPriceCustom$outboundSchema` instead. */
    ProductPriceCustom$.outboundSchema = ProductPriceCustom$outboundSchema;
})(ProductPriceCustom$ || (ProductPriceCustom$ = {}));
export function productPriceCustomToJSON(productPriceCustom) {
    return JSON.stringify(ProductPriceCustom$outboundSchema.parse(productPriceCustom));
}
export function productPriceCustomFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceCustom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceCustom' from JSON`);
}
