// @polar-sh/sdk/dist/esm/models/components/productpricefree.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceType$inboundSchema, ProductPriceType$outboundSchema, } from "./productpricetype.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
/** @internal */
export const ProductPriceFree$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("free"),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: ProductPriceType$inboundSchema,
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
export const ProductPriceFree$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("free"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: ProductPriceType$outboundSchema,
    recurringInterval: z.nullable(SubscriptionRecurringInterval$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductPriceFree$;
(function (ProductPriceFree$) {
    /** @deprecated use `ProductPriceFree$inboundSchema` instead. */
    ProductPriceFree$.inboundSchema = ProductPriceFree$inboundSchema;
    /** @deprecated use `ProductPriceFree$outboundSchema` instead. */
    ProductPriceFree$.outboundSchema = ProductPriceFree$outboundSchema;
})(ProductPriceFree$ || (ProductPriceFree$ = {}));
export function productPriceFreeToJSON(productPriceFree) {
    return JSON.stringify(ProductPriceFree$outboundSchema.parse(productPriceFree));
}
export function productPriceFreeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceFree$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceFree' from JSON`);
}
