// @polar-sh/sdk/dist/esm/models/components/productsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const ProductSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Name: "name",
    MinusName: "-name",
    PriceAmountType: "price_amount_type",
    MinusPriceAmountType: "-price_amount_type",
    PriceAmount: "price_amount",
    MinusPriceAmount: "-price_amount",
};
/** @internal */
export const ProductSortProperty$inboundSchema = z.nativeEnum(ProductSortProperty);
/** @internal */
export const ProductSortProperty$outboundSchema = ProductSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductSortProperty$;
(function (ProductSortProperty$) {
    /** @deprecated use `ProductSortProperty$inboundSchema` instead. */
    ProductSortProperty$.inboundSchema = ProductSortProperty$inboundSchema;
    /** @deprecated use `ProductSortProperty$outboundSchema` instead. */
    ProductSortProperty$.outboundSchema = ProductSortProperty$outboundSchema;
})(ProductSortProperty$ || (ProductSortProperty$ = {}));
