// @polar-sh/sdk/dist/esm/models/components/propertyaggregation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
export const Func = {
    Sum: "sum",
    Max: "max",
    Min: "min",
    Avg: "avg",
};
/** @internal */
export const Func$inboundSchema = z.nativeEnum(Func);
/** @internal */
export const Func$outboundSchema = Func$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Func$;
(function (Func$) {
    /** @deprecated use `Func$inboundSchema` instead. */
    Func$.inboundSchema = Func$inboundSchema;
    /** @deprecated use `Func$outboundSchema` instead. */
    Func$.outboundSchema = Func$outboundSchema;
})(Func$ || (Func$ = {}));
/** @internal */
export const PropertyAggregation$inboundSchema = z.object({
    func: Func$inboundSchema,
    property: z.string(),
});
/** @internal */
export const PropertyAggregation$outboundSchema = z.object({
    func: Func$outboundSchema,
    property: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PropertyAggregation$;
(function (PropertyAggregation$) {
    /** @deprecated use `PropertyAggregation$inboundSchema` instead. */
    PropertyAggregation$.inboundSchema = PropertyAggregation$inboundSchema;
    /** @deprecated use `PropertyAggregation$outboundSchema` instead. */
    PropertyAggregation$.outboundSchema = PropertyAggregation$outboundSchema;
})(PropertyAggregation$ || (PropertyAggregation$ = {}));
export function propertyAggregationToJSON(propertyAggregation) {
    return JSON.stringify(PropertyAggregation$outboundSchema.parse(propertyAggregation));
}
export function propertyAggregationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PropertyAggregation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PropertyAggregation' from JSON`);
}
