// @polar-sh/sdk/dist/esm/models/components/refreshtokenrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const RefreshTokenRequest$inboundSchema = z.object({
    grant_type: z.literal("refresh_token"),
    client_id: z.string(),
    client_secret: z.string(),
    refresh_token: z.string(),
}).transform((v) => {
    return remap$(v, {
        "grant_type": "grantType",
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "refresh_token": "refreshToken",
    });
});
/** @internal */
export const RefreshTokenRequest$outboundSchema = z.object({
    grantType: z.literal("refresh_token"),
    clientId: z.string(),
    clientSecret: z.string(),
    refreshToken: z.string(),
}).transform((v) => {
    return remap$(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        refreshToken: "refresh_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RefreshTokenRequest$;
(function (RefreshTokenRequest$) {
    /** @deprecated use `RefreshTokenRequest$inboundSchema` instead. */
    RefreshTokenRequest$.inboundSchema = RefreshTokenRequest$inboundSchema;
    /** @deprecated use `RefreshTokenRequest$outboundSchema` instead. */
    RefreshTokenRequest$.outboundSchema = RefreshTokenRequest$outboundSchema;
})(RefreshTokenRequest$ || (RefreshTokenRequest$ = {}));
export function refreshTokenRequestToJSON(refreshTokenRequest) {
    return JSON.stringify(RefreshTokenRequest$outboundSchema.parse(refreshTokenRequest));
}
export function refreshTokenRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RefreshTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefreshTokenRequest' from JSON`);
}
