// @polar-sh/sdk/dist/esm/models/components/refundreason.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const RefundReason = {
    Duplicate: "duplicate",
    Fraudulent: "fraudulent",
    CustomerRequest: "customer_request",
    ServiceDisruption: "service_disruption",
    SatisfactionGuarantee: "satisfaction_guarantee",
    Other: "other",
};
/** @internal */
export const RefundReason$inboundSchema = z.nativeEnum(RefundReason);
/** @internal */
export const RefundReason$outboundSchema = RefundReason$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RefundReason$;
(function (RefundReason$) {
    /** @deprecated use `RefundReason$inboundSchema` instead. */
    RefundReason$.inboundSchema = RefundReason$inboundSchema;
    /** @deprecated use `RefundReason$outboundSchema` instead. */
    RefundReason$.outboundSchema = RefundReason$outboundSchema;
})(RefundReason$ || (RefundReason$ = {}));
