// @polar-sh/sdk/dist/esm/models/components/refundsortproperty.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const RefundSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Amount: "amount",
    MinusAmount: "-amount",
};
/** @internal */
export const RefundSortProperty$inboundSchema = z.nativeEnum(RefundSortProperty);
/** @internal */
export const RefundSortProperty$outboundSchema = RefundSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RefundSortProperty$;
(function (RefundSortProperty$) {
    /** @deprecated use `RefundSortProperty$inboundSchema` instead. */
    RefundSortProperty$.inboundSchema = RefundSortProperty$inboundSchema;
    /** @deprecated use `RefundSortProperty$outboundSchema` instead. */
    RefundSortProperty$.outboundSchema = RefundSortProperty$outboundSchema;
})(RefundSortProperty$ || (RefundSortProperty$ = {}));
