// @polar-sh/sdk/dist/esm/models/components/subscriptioncancel.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerCancellationReason$inboundSchema, CustomerCancellationReason$outboundSchema, } from "./customercancellationreason.js";
/** @internal */
export const SubscriptionCancel$inboundSchema = z.object({
    customer_cancellation_reason: z.nullable(CustomerCancellationReason$inboundSchema).optional(),
    customer_cancellation_comment: z.nullable(z.string()).optional(),
    cancel_at_period_end: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
        "cancel_at_period_end": "cancelAtPeriodEnd",
    });
});
/** @internal */
export const SubscriptionCancel$outboundSchema = z.object({
    customerCancellationReason: z.nullable(CustomerCancellationReason$outboundSchema).optional(),
    customerCancellationComment: z.nullable(z.string()).optional(),
    cancelAtPeriodEnd: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        customerCancellationReason: "customer_cancellation_reason",
        customerCancellationComment: "customer_cancellation_comment",
        cancelAtPeriodEnd: "cancel_at_period_end",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionCancel$;
(function (SubscriptionCancel$) {
    /** @deprecated use `SubscriptionCancel$inboundSchema` instead. */
    SubscriptionCancel$.inboundSchema = SubscriptionCancel$inboundSchema;
    /** @deprecated use `SubscriptionCancel$outboundSchema` instead. */
    SubscriptionCancel$.outboundSchema = SubscriptionCancel$outboundSchema;
})(SubscriptionCancel$ || (SubscriptionCancel$ = {}));
export function subscriptionCancelToJSON(subscriptionCancel) {
    return JSON.stringify(SubscriptionCancel$outboundSchema.parse(subscriptionCancel));
}
export function subscriptionCancelFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCancel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCancel' from JSON`);
}
