// @polar-sh/sdk/dist/esm/models/components/subscriptionrevokedevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { SubscriptionRevokedMetadata$inboundSchema, SubscriptionRevokedMetadata$outboundSchema, } from "./subscriptionrevokedmetadata.js";
/** @internal */
export const SubscriptionRevokedEvent$inboundSchema = z.object({
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("subscription.revoked"),
    metadata: SubscriptionRevokedMetadata$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const SubscriptionRevokedEvent$outboundSchema = z.object({
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("subscription.revoked"),
    metadata: SubscriptionRevokedMetadata$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionRevokedEvent$;
(function (SubscriptionRevokedEvent$) {
    /** @deprecated use `SubscriptionRevokedEvent$inboundSchema` instead. */
    SubscriptionRevokedEvent$.inboundSchema = SubscriptionRevokedEvent$inboundSchema;
    /** @deprecated use `SubscriptionRevokedEvent$outboundSchema` instead. */
    SubscriptionRevokedEvent$.outboundSchema = SubscriptionRevokedEvent$outboundSchema;
})(SubscriptionRevokedEvent$ || (SubscriptionRevokedEvent$ = {}));
export function subscriptionRevokedEventToJSON(subscriptionRevokedEvent) {
    return JSON.stringify(SubscriptionRevokedEvent$outboundSchema.parse(subscriptionRevokedEvent));
}
export function subscriptionRevokedEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionRevokedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionRevokedEvent' from JSON`);
}
