// @polar-sh/sdk/dist/esm/models/components/subscriptionstatus.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const SubscriptionStatus = {
    Incomplete: "incomplete",
    IncompleteExpired: "incomplete_expired",
    Trialing: "trialing",
    Active: "active",
    PastDue: "past_due",
    Canceled: "canceled",
    Unpaid: "unpaid",
};
/** @internal */
export const SubscriptionStatus$inboundSchema = z.nativeEnum(SubscriptionStatus);
/** @internal */
export const SubscriptionStatus$outboundSchema = SubscriptionStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionStatus$;
(function (SubscriptionStatus$) {
    /** @deprecated use `SubscriptionStatus$inboundSchema` instead. */
    SubscriptionStatus$.inboundSchema = SubscriptionStatus$inboundSchema;
    /** @deprecated use `SubscriptionStatus$outboundSchema` instead. */
    SubscriptionStatus$.outboundSchema = SubscriptionStatus$outboundSchema;
})(SubscriptionStatus$ || (SubscriptionStatus$ = {}));
