// @polar-sh/sdk/dist/esm/models/components/validatedlicensekey.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { LicenseKeyActivationBase$inboundSchema, LicenseKeyActivationBase$outboundSchema, } from "./licensekeyactivationbase.js";
import { LicenseKeyCustomer$inboundSchema, LicenseKeyCustomer$outboundSchema, } from "./licensekeycustomer.js";
import { LicenseKeyStatus$inboundSchema, LicenseKeyStatus$outboundSchema, } from "./licensekeystatus.js";
/** @internal */
export const ValidatedLicenseKey$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    organization_id: z.string(),
    customer_id: z.string(),
    customer: LicenseKeyCustomer$inboundSchema,
    benefit_id: z.string(),
    key: z.string(),
    display_key: z.string(),
    status: LicenseKeyStatus$inboundSchema,
    limit_activations: z.nullable(z.number().int()),
    usage: z.number().int(),
    limit_usage: z.nullable(z.number().int()),
    validations: z.number().int(),
    last_validated_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    expires_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    activation: z.nullable(LicenseKeyActivationBase$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "display_key": "displayKey",
        "limit_activations": "limitActivations",
        "limit_usage": "limitUsage",
        "last_validated_at": "lastValidatedAt",
        "expires_at": "expiresAt",
    });
});
/** @internal */
export const ValidatedLicenseKey$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    organizationId: z.string(),
    customerId: z.string(),
    customer: LicenseKeyCustomer$outboundSchema,
    benefitId: z.string(),
    key: z.string(),
    displayKey: z.string(),
    status: LicenseKeyStatus$outboundSchema,
    limitActivations: z.nullable(z.number().int()),
    usage: z.number().int(),
    limitUsage: z.nullable(z.number().int()),
    validations: z.number().int(),
    lastValidatedAt: z.nullable(z.date().transform(v => v.toISOString())),
    expiresAt: z.nullable(z.date().transform(v => v.toISOString())),
    activation: z.nullable(LicenseKeyActivationBase$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
        customerId: "customer_id",
        benefitId: "benefit_id",
        displayKey: "display_key",
        limitActivations: "limit_activations",
        limitUsage: "limit_usage",
        lastValidatedAt: "last_validated_at",
        expiresAt: "expires_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ValidatedLicenseKey$;
(function (ValidatedLicenseKey$) {
    /** @deprecated use `ValidatedLicenseKey$inboundSchema` instead. */
    ValidatedLicenseKey$.inboundSchema = ValidatedLicenseKey$inboundSchema;
    /** @deprecated use `ValidatedLicenseKey$outboundSchema` instead. */
    ValidatedLicenseKey$.outboundSchema = ValidatedLicenseKey$outboundSchema;
})(ValidatedLicenseKey$ || (ValidatedLicenseKey$ = {}));
export function validatedLicenseKeyToJSON(validatedLicenseKey) {
    return JSON.stringify(ValidatedLicenseKey$outboundSchema.parse(validatedLicenseKey));
}
export function validatedLicenseKeyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ValidatedLicenseKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidatedLicenseKey' from JSON`);
}
