// @polar-sh/sdk/dist/esm/models/components/webhookcustomercreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const WebhookCustomerCreatedPayload$inboundSchema = z.object({
    type: z.literal("customer.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Customer$inboundSchema,
});
/** @internal */
export const WebhookCustomerCreatedPayload$outboundSchema = z.object({
    type: z.literal("customer.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Customer$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookCustomerCreatedPayload$;
(function (WebhookCustomerCreatedPayload$) {
    /** @deprecated use `WebhookCustomerCreatedPayload$inboundSchema` instead. */
    WebhookCustomerCreatedPayload$.inboundSchema = WebhookCustomerCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookCustomerCreatedPayload$outboundSchema` instead. */
    WebhookCustomerCreatedPayload$.outboundSchema = WebhookCustomerCreatedPayload$outboundSchema;
})(WebhookCustomerCreatedPayload$ || (WebhookCustomerCreatedPayload$ = {}));
export function webhookCustomerCreatedPayloadToJSON(webhookCustomerCreatedPayload) {
    return JSON.stringify(WebhookCustomerCreatedPayload$outboundSchema.parse(webhookCustomerCreatedPayload));
}
export function webhookCustomerCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerCreatedPayload' from JSON`);
}
