// @polar-sh/sdk/dist/esm/models/components/webhookrefundcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Refund$inboundSchema, Refund$outboundSchema, } from "./refund.js";
/** @internal */
export const WebhookRefundCreatedPayload$inboundSchema = z.object({
    type: z.literal("refund.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Refund$inboundSchema,
});
/** @internal */
export const WebhookRefundCreatedPayload$outboundSchema = z.object({
    type: z.literal("refund.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Refund$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookRefundCreatedPayload$;
(function (WebhookRefundCreatedPayload$) {
    /** @deprecated use `WebhookRefundCreatedPayload$inboundSchema` instead. */
    WebhookRefundCreatedPayload$.inboundSchema = WebhookRefundCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookRefundCreatedPayload$outboundSchema` instead. */
    WebhookRefundCreatedPayload$.outboundSchema = WebhookRefundCreatedPayload$outboundSchema;
})(WebhookRefundCreatedPayload$ || (WebhookRefundCreatedPayload$ = {}));
export function webhookRefundCreatedPayloadToJSON(webhookRefundCreatedPayload) {
    return JSON.stringify(WebhookRefundCreatedPayload$outboundSchema.parse(webhookRefundCreatedPayload));
}
export function webhookRefundCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookRefundCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookRefundCreatedPayload' from JSON`);
}
