// @polar-sh/sdk/dist/esm/models/components/webhookbenefitgrantcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantWebhook$inboundSchema, BenefitGrantWebhook$outboundSchema, } from "./benefitgrantwebhook.js";
/** @internal */
export const WebhookBenefitGrantCreatedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: BenefitGrantWebhook$inboundSchema,
});
/** @internal */
export const WebhookBenefitGrantCreatedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookBenefitGrantCreatedPayload$;
(function (WebhookBenefitGrantCreatedPayload$) {
    /** @deprecated use `WebhookBenefitGrantCreatedPayload$inboundSchema` instead. */
    WebhookBenefitGrantCreatedPayload$.inboundSchema = WebhookBenefitGrantCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantCreatedPayload$outboundSchema` instead. */
    WebhookBenefitGrantCreatedPayload$.outboundSchema = WebhookBenefitGrantCreatedPayload$outboundSchema;
})(WebhookBenefitGrantCreatedPayload$ || (WebhookBenefitGrantCreatedPayload$ = {}));
export function webhookBenefitGrantCreatedPayloadToJSON(webhookBenefitGrantCreatedPayload) {
    return JSON.stringify(WebhookBenefitGrantCreatedPayload$outboundSchema.parse(webhookBenefitGrantCreatedPayload));
}
export function webhookBenefitGrantCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitGrantCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantCreatedPayload' from JSON`);
}
