// @polar-sh/sdk/dist/esm/models/components/webhookproductcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Product$inboundSchema, Product$outboundSchema, } from "./product.js";
/** @internal */
export const WebhookProductCreatedPayload$inboundSchema = z.object({
    type: z.literal("product.created"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Product$inboundSchema,
});
/** @internal */
export const WebhookProductCreatedPayload$outboundSchema = z.object({
    type: z.literal("product.created"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Product$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookProductCreatedPayload$;
(function (WebhookProductCreatedPayload$) {
    /** @deprecated use `WebhookProductCreatedPayload$inboundSchema` instead. */
    WebhookProductCreatedPayload$.inboundSchema = WebhookProductCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookProductCreatedPayload$outboundSchema` instead. */
    WebhookProductCreatedPayload$.outboundSchema = WebhookProductCreatedPayload$outboundSchema;
})(WebhookProductCreatedPayload$ || (WebhookProductCreatedPayload$ = {}));
export function webhookProductCreatedPayloadToJSON(webhookProductCreatedPayload) {
    return JSON.stringify(WebhookProductCreatedPayload$outboundSchema.parse(webhookProductCreatedPayload));
}
export function webhookProductCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookProductCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookProductCreatedPayload' from JSON`);
}
