// @polar-sh/sdk/dist/esm/models/components/webhookeventtype.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
export const WebhookEventType = {
    CheckoutCreated: "checkout.created",
    CheckoutUpdated: "checkout.updated",
    CustomerCreated: "customer.created",
    CustomerUpdated: "customer.updated",
    CustomerDeleted: "customer.deleted",
    CustomerStateChanged: "customer.state_changed",
    OrderCreated: "order.created",
    OrderUpdated: "order.updated",
    OrderPaid: "order.paid",
    OrderRefunded: "order.refunded",
    SubscriptionCreated: "subscription.created",
    SubscriptionUpdated: "subscription.updated",
    SubscriptionActive: "subscription.active",
    SubscriptionCanceled: "subscription.canceled",
    SubscriptionUncanceled: "subscription.uncanceled",
    SubscriptionRevoked: "subscription.revoked",
    RefundCreated: "refund.created",
    RefundUpdated: "refund.updated",
    ProductCreated: "product.created",
    ProductUpdated: "product.updated",
    BenefitCreated: "benefit.created",
    BenefitUpdated: "benefit.updated",
    BenefitGrantCreated: "benefit_grant.created",
    BenefitGrantCycled: "benefit_grant.cycled",
    BenefitGrantUpdated: "benefit_grant.updated",
    BenefitGrantRevoked: "benefit_grant.revoked",
    OrganizationUpdated: "organization.updated",
};
/** @internal */
export const WebhookEventType$inboundSchema = z.nativeEnum(WebhookEventType);
/** @internal */
export const WebhookEventType$outboundSchema = WebhookEventType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookEventType$;
(function (WebhookEventType$) {
    /** @deprecated use `WebhookEventType$inboundSchema` instead. */
    WebhookEventType$.inboundSchema = WebhookEventType$inboundSchema;
    /** @deprecated use `WebhookEventType$outboundSchema` instead. */
    WebhookEventType$.outboundSchema = WebhookEventType$outboundSchema;
})(WebhookEventType$ || (WebhookEventType$ = {}));
