// @polar-sh/sdk/dist/esm/models/components/webhookbenefitgrantcycledpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantWebhook$inboundSchema, BenefitGrantWebhook$outboundSchema, } from "./benefitgrantwebhook.js";
/** @internal */
export const WebhookBenefitGrantCycledPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.cycled"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: BenefitGrantWebhook$inboundSchema,
});
/** @internal */
export const WebhookBenefitGrantCycledPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.cycled"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookBenefitGrantCycledPayload$;
(function (WebhookBenefitGrantCycledPayload$) {
    /** @deprecated use `WebhookBenefitGrantCycledPayload$inboundSchema` instead. */
    WebhookBenefitGrantCycledPayload$.inboundSchema = WebhookBenefitGrantCycledPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantCycledPayload$outboundSchema` instead. */
    WebhookBenefitGrantCycledPayload$.outboundSchema = WebhookBenefitGrantCycledPayload$outboundSchema;
})(WebhookBenefitGrantCycledPayload$ || (WebhookBenefitGrantCycledPayload$ = {}));
export function webhookBenefitGrantCycledPayloadToJSON(webhookBenefitGrantCycledPayload) {
    return JSON.stringify(WebhookBenefitGrantCycledPayload$outboundSchema.parse(webhookBenefitGrantCycledPayload));
}
export function webhookBenefitGrantCycledPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitGrantCycledPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantCycledPayload' from JSON`);
}
