// @polar-sh/sdk/dist/esm/models/components/webhookproductupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Product$inboundSchema, Product$outboundSchema, } from "./product.js";
/** @internal */
export const WebhookProductUpdatedPayload$inboundSchema = z.object({
    type: z.literal("product.updated"),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    data: Product$inboundSchema,
});
/** @internal */
export const WebhookProductUpdatedPayload$outboundSchema = z.object({
    type: z.literal("product.updated"),
    timestamp: z.date().transform(v => v.toISOString()),
    data: Product$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhookProductUpdatedPayload$;
(function (WebhookProductUpdatedPayload$) {
    /** @deprecated use `WebhookProductUpdatedPayload$inboundSchema` instead. */
    WebhookProductUpdatedPayload$.inboundSchema = WebhookProductUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookProductUpdatedPayload$outboundSchema` instead. */
    WebhookProductUpdatedPayload$.outboundSchema = WebhookProductUpdatedPayload$outboundSchema;
})(WebhookProductUpdatedPayload$ || (WebhookProductUpdatedPayload$ = {}));
export function webhookProductUpdatedPayloadToJSON(webhookProductUpdatedPayload) {
    return JSON.stringify(WebhookProductUpdatedPayload$outboundSchema.parse(webhookProductUpdatedPayload));
}
export function webhookProductUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookProductUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookProductUpdatedPayload' from JSON`);
}
