// @polar-sh/sdk/dist/esm/sdk/licensekeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { licenseKeysActivate } from "../funcs/licenseKeysActivate.js";
import { licenseKeysDeactivate } from "../funcs/licenseKeysDeactivate.js";
import { licenseKeysGet } from "../funcs/licenseKeysGet.js";
import { licenseKeysGetActivation } from "../funcs/licenseKeysGetActivation.js";
import { licenseKeysList } from "../funcs/licenseKeysList.js";
import { licenseKeysUpdate } from "../funcs/licenseKeysUpdate.js";
import { licenseKeysValidate } from "../funcs/licenseKeysValidate.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class LicenseKeys extends ClientSDK {
    /**
     * List License Keys
     *
     * @remarks
     * Get license keys connected to the given organization & filters.
     *
     * **Scopes**: `license_keys:read` `license_keys:write`
     */
    async list(request, options) {
        return unwrapResultIterator(licenseKeysList(this, request, options));
    }
    /**
     * Get License Key
     *
     * @remarks
     * Get a license key.
     *
     * **Scopes**: `license_keys:read` `license_keys:write`
     */
    async get(request, options) {
        return unwrapAsync(licenseKeysGet(this, request, options));
    }
    /**
     * Update License Key
     *
     * @remarks
     * Update a license key.
     *
     * **Scopes**: `license_keys:write`
     */
    async update(request, options) {
        return unwrapAsync(licenseKeysUpdate(this, request, options));
    }
    /**
     * Get Activation
     *
     * @remarks
     * Get a license key activation.
     *
     * **Scopes**: `license_keys:read` `license_keys:write`
     */
    async getActivation(request, options) {
        return unwrapAsync(licenseKeysGetActivation(this, request, options));
    }
    /**
     * Validate License Key
     *
     * @remarks
     * Validate a license key.
     *
     * **Scopes**: `license_keys:write`
     */
    async validate(request, options) {
        return unwrapAsync(licenseKeysValidate(this, request, options));
    }
    /**
     * Activate License Key
     *
     * @remarks
     * Activate a license key instance.
     *
     * **Scopes**: `license_keys:write`
     */
    async activate(request, options) {
        return unwrapAsync(licenseKeysActivate(this, request, options));
    }
    /**
     * Deactivate License Key
     *
     * @remarks
     * Deactivate a license key instance.
     *
     * **Scopes**: `license_keys:write`
     */
    async deactivate(request, options) {
        return unwrapAsync(licenseKeysDeactivate(this, request, options));
    }
}
