// @polar-sh/sdk/dist/commonjs/mcp-server/server.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMCPServer = createMCPServer;
const mcp_js_1 = require("@modelcontextprotocol/sdk/server/mcp.js");
const core_js_1 = require("../core.js");
const prompts_js_1 = require("./prompts.js");
const resources_js_1 = require("./resources.js");
const tools_js_1 = require("./tools.js");
const customerMetersGet_js_1 = require("./tools/customerMetersGet.js");
const customerMetersList_js_1 = require("./tools/customerMetersList.js");
const customersCreate_js_1 = require("./tools/customersCreate.js");
const customersDelete_js_1 = require("./tools/customersDelete.js");
const customersDeleteExternal_js_1 = require("./tools/customersDeleteExternal.js");
const customersGet_js_1 = require("./tools/customersGet.js");
const customersGetExternal_js_1 = require("./tools/customersGetExternal.js");
const customersGetState_js_1 = require("./tools/customersGetState.js");
const customersGetStateExternal_js_1 = require("./tools/customersGetStateExternal.js");
const customersList_js_1 = require("./tools/customersList.js");
const customersUpdate_js_1 = require("./tools/customersUpdate.js");
const customersUpdateExternal_js_1 = require("./tools/customersUpdateExternal.js");
const metricsGet_js_1 = require("./tools/metricsGet.js");
const metricsLimits_js_1 = require("./tools/metricsLimits.js");
const ordersGenerateInvoice_js_1 = require("./tools/ordersGenerateInvoice.js");
const ordersGet_js_1 = require("./tools/ordersGet.js");
const ordersInvoice_js_1 = require("./tools/ordersInvoice.js");
const ordersList_js_1 = require("./tools/ordersList.js");
const ordersUpdate_js_1 = require("./tools/ordersUpdate.js");
const paymentsGet_js_1 = require("./tools/paymentsGet.js");
const paymentsList_js_1 = require("./tools/paymentsList.js");
const productsCreate_js_1 = require("./tools/productsCreate.js");
const productsGet_js_1 = require("./tools/productsGet.js");
const productsList_js_1 = require("./tools/productsList.js");
const productsUpdate_js_1 = require("./tools/productsUpdate.js");
const productsUpdateBenefits_js_1 = require("./tools/productsUpdateBenefits.js");
const subscriptionsExport_js_1 = require("./tools/subscriptionsExport.js");
const subscriptionsGet_js_1 = require("./tools/subscriptionsGet.js");
const subscriptionsList_js_1 = require("./tools/subscriptionsList.js");
const subscriptionsRevoke_js_1 = require("./tools/subscriptionsRevoke.js");
const subscriptionsUpdate_js_1 = require("./tools/subscriptionsUpdate.js");
function createMCPServer(deps) {
    const server = new mcp_js_1.McpServer({
        name: "Polar",
        version: "0.34.17",
    });
    const client = new core_js_1.PolarCore({
        accessToken: deps.accessToken,
        serverURL: deps.serverURL,
        server: deps.server,
    });
    const scopes = new Set(deps.scopes);
    const allowedTools = deps.allowedTools && new Set(deps.allowedTools);
    const tool = (0, tools_js_1.createRegisterTool)(deps.logger, server, client, scopes, allowedTools);
    const resource = (0, resources_js_1.createRegisterResource)(deps.logger, server, client, scopes);
    const resourceTemplate = (0, resources_js_1.createRegisterResourceTemplate)(deps.logger, server, client, scopes);
    const prompt = (0, prompts_js_1.createRegisterPrompt)(deps.logger, server, client, scopes);
    const register = { tool, resource, resourceTemplate, prompt };
    void register; // suppress unused warnings
    tool(subscriptionsList_js_1.tool$subscriptionsList);
    tool(subscriptionsExport_js_1.tool$subscriptionsExport);
    tool(subscriptionsGet_js_1.tool$subscriptionsGet);
    tool(subscriptionsUpdate_js_1.tool$subscriptionsUpdate);
    tool(subscriptionsRevoke_js_1.tool$subscriptionsRevoke);
    tool(productsList_js_1.tool$productsList);
    tool(productsCreate_js_1.tool$productsCreate);
    tool(productsGet_js_1.tool$productsGet);
    tool(productsUpdate_js_1.tool$productsUpdate);
    tool(productsUpdateBenefits_js_1.tool$productsUpdateBenefits);
    tool(ordersList_js_1.tool$ordersList);
    tool(ordersGet_js_1.tool$ordersGet);
    tool(ordersUpdate_js_1.tool$ordersUpdate);
    tool(ordersGenerateInvoice_js_1.tool$ordersGenerateInvoice);
    tool(ordersInvoice_js_1.tool$ordersInvoice);
    tool(metricsGet_js_1.tool$metricsGet);
    tool(metricsLimits_js_1.tool$metricsLimits);
    tool(customersList_js_1.tool$customersList);
    tool(customersCreate_js_1.tool$customersCreate);
    tool(customersGet_js_1.tool$customersGet);
    tool(customersUpdate_js_1.tool$customersUpdate);
    tool(customersDelete_js_1.tool$customersDelete);
    tool(customersGetExternal_js_1.tool$customersGetExternal);
    tool(customersUpdateExternal_js_1.tool$customersUpdateExternal);
    tool(customersDeleteExternal_js_1.tool$customersDeleteExternal);
    tool(customersGetState_js_1.tool$customersGetState);
    tool(customersGetStateExternal_js_1.tool$customersGetStateExternal);
    tool(customerMetersList_js_1.tool$customerMetersList);
    tool(customerMetersGet_js_1.tool$customerMetersGet);
    tool(paymentsList_js_1.tool$paymentsList);
    tool(paymentsGet_js_1.tool$paymentsGet);
    return server;
}
