// @polar-sh/sdk/dist/commonjs/models/operations/customerportalsubscriptionsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalSubscriptionsUpdateRequest$ = exports.CustomerPortalSubscriptionsUpdateRequest$outboundSchema = exports.CustomerPortalSubscriptionsUpdateRequest$inboundSchema = exports.CustomerPortalSubscriptionsUpdateSecurity$ = exports.CustomerPortalSubscriptionsUpdateSecurity$outboundSchema = exports.CustomerPortalSubscriptionsUpdateSecurity$inboundSchema = void 0;
exports.customerPortalSubscriptionsUpdateSecurityToJSON = customerPortalSubscriptionsUpdateSecurityToJSON;
exports.customerPortalSubscriptionsUpdateSecurityFromJSON = customerPortalSubscriptionsUpdateSecurityFromJSON;
exports.customerPortalSubscriptionsUpdateRequestToJSON = customerPortalSubscriptionsUpdateRequestToJSON;
exports.customerPortalSubscriptionsUpdateRequestFromJSON = customerPortalSubscriptionsUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customersubscriptionupdate_js_1 = require("../components/customersubscriptionupdate.js");
/** @internal */
exports.CustomerPortalSubscriptionsUpdateSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalSubscriptionsUpdateSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSubscriptionsUpdateSecurity$;
(function (CustomerPortalSubscriptionsUpdateSecurity$) {
    /** @deprecated use `CustomerPortalSubscriptionsUpdateSecurity$inboundSchema` instead. */
    CustomerPortalSubscriptionsUpdateSecurity$.inboundSchema = exports.CustomerPortalSubscriptionsUpdateSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSubscriptionsUpdateSecurity$outboundSchema` instead. */
    CustomerPortalSubscriptionsUpdateSecurity$.outboundSchema = exports.CustomerPortalSubscriptionsUpdateSecurity$outboundSchema;
})(CustomerPortalSubscriptionsUpdateSecurity$ || (exports.CustomerPortalSubscriptionsUpdateSecurity$ = CustomerPortalSubscriptionsUpdateSecurity$ = {}));
function customerPortalSubscriptionsUpdateSecurityToJSON(customerPortalSubscriptionsUpdateSecurity) {
    return JSON.stringify(exports.CustomerPortalSubscriptionsUpdateSecurity$outboundSchema.parse(customerPortalSubscriptionsUpdateSecurity));
}
function customerPortalSubscriptionsUpdateSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSubscriptionsUpdateSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsUpdateSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalSubscriptionsUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    CustomerSubscriptionUpdate: customersubscriptionupdate_js_1.CustomerSubscriptionUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CustomerSubscriptionUpdate": "customerSubscriptionUpdate",
    });
});
/** @internal */
exports.CustomerPortalSubscriptionsUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    customerSubscriptionUpdate: customersubscriptionupdate_js_1.CustomerSubscriptionUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSubscriptionUpdate: "CustomerSubscriptionUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSubscriptionsUpdateRequest$;
(function (CustomerPortalSubscriptionsUpdateRequest$) {
    /** @deprecated use `CustomerPortalSubscriptionsUpdateRequest$inboundSchema` instead. */
    CustomerPortalSubscriptionsUpdateRequest$.inboundSchema = exports.CustomerPortalSubscriptionsUpdateRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSubscriptionsUpdateRequest$outboundSchema` instead. */
    CustomerPortalSubscriptionsUpdateRequest$.outboundSchema = exports.CustomerPortalSubscriptionsUpdateRequest$outboundSchema;
})(CustomerPortalSubscriptionsUpdateRequest$ || (exports.CustomerPortalSubscriptionsUpdateRequest$ = CustomerPortalSubscriptionsUpdateRequest$ = {}));
function customerPortalSubscriptionsUpdateRequestToJSON(customerPortalSubscriptionsUpdateRequest) {
    return JSON.stringify(exports.CustomerPortalSubscriptionsUpdateRequest$outboundSchema.parse(customerPortalSubscriptionsUpdateRequest));
}
function customerPortalSubscriptionsUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSubscriptionsUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsUpdateRequest' from JSON`);
}
