// @polar-sh/sdk/dist/commonjs/models/operations/customersgetexternal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomersGetExternalRequest$ = exports.CustomersGetExternalRequest$outboundSchema = exports.CustomersGetExternalRequest$inboundSchema = void 0;
exports.customersGetExternalRequestToJSON = customersGetExternalRequestToJSON;
exports.customersGetExternalRequestFromJSON = customersGetExternalRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomersGetExternalRequest$inboundSchema = z.object({
    external_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "external_id": "externalId",
    });
});
/** @internal */
exports.CustomersGetExternalRequest$outboundSchema = z.object({
    externalId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        externalId: "external_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomersGetExternalRequest$;
(function (CustomersGetExternalRequest$) {
    /** @deprecated use `CustomersGetExternalRequest$inboundSchema` instead. */
    CustomersGetExternalRequest$.inboundSchema = exports.CustomersGetExternalRequest$inboundSchema;
    /** @deprecated use `CustomersGetExternalRequest$outboundSchema` instead. */
    CustomersGetExternalRequest$.outboundSchema = exports.CustomersGetExternalRequest$outboundSchema;
})(CustomersGetExternalRequest$ || (exports.CustomersGetExternalRequest$ = CustomersGetExternalRequest$ = {}));
function customersGetExternalRequestToJSON(customersGetExternalRequest) {
    return JSON.stringify(exports.CustomersGetExternalRequest$outboundSchema.parse(customersGetExternalRequest));
}
function customersGetExternalRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomersGetExternalRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersGetExternalRequest' from JSON`);
}
