// @polar-sh/sdk/dist/commonjs/models/operations/customerportalordersconfirmretrypayment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalOrdersConfirmRetryPaymentRequest$ = exports.CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentRequest$inboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$ = exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$inboundSchema = void 0;
exports.customerPortalOrdersConfirmRetryPaymentSecurityToJSON = customerPortalOrdersConfirmRetryPaymentSecurityToJSON;
exports.customerPortalOrdersConfirmRetryPaymentSecurityFromJSON = customerPortalOrdersConfirmRetryPaymentSecurityFromJSON;
exports.customerPortalOrdersConfirmRetryPaymentRequestToJSON = customerPortalOrdersConfirmRetryPaymentRequestToJSON;
exports.customerPortalOrdersConfirmRetryPaymentRequestFromJSON = customerPortalOrdersConfirmRetryPaymentRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerorderconfirmpayment_js_1 = require("../components/customerorderconfirmpayment.js");
/** @internal */
exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersConfirmRetryPaymentSecurity$;
(function (CustomerPortalOrdersConfirmRetryPaymentSecurity$) {
    /** @deprecated use `CustomerPortalOrdersConfirmRetryPaymentSecurity$inboundSchema` instead. */
    CustomerPortalOrdersConfirmRetryPaymentSecurity$.inboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema` instead. */
    CustomerPortalOrdersConfirmRetryPaymentSecurity$.outboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema;
})(CustomerPortalOrdersConfirmRetryPaymentSecurity$ || (exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$ = CustomerPortalOrdersConfirmRetryPaymentSecurity$ = {}));
function customerPortalOrdersConfirmRetryPaymentSecurityToJSON(customerPortalOrdersConfirmRetryPaymentSecurity) {
    return JSON.stringify(exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema.parse(customerPortalOrdersConfirmRetryPaymentSecurity));
}
function customerPortalOrdersConfirmRetryPaymentSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersConfirmRetryPaymentSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersConfirmRetryPaymentSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersConfirmRetryPaymentRequest$inboundSchema = z.object({
    id: z.string(),
    CustomerOrderConfirmPayment: customerorderconfirmpayment_js_1.CustomerOrderConfirmPayment$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CustomerOrderConfirmPayment": "customerOrderConfirmPayment",
    });
});
/** @internal */
exports.CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema = z.object({
    id: z.string(),
    customerOrderConfirmPayment: customerorderconfirmpayment_js_1.CustomerOrderConfirmPayment$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerOrderConfirmPayment: "CustomerOrderConfirmPayment",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersConfirmRetryPaymentRequest$;
(function (CustomerPortalOrdersConfirmRetryPaymentRequest$) {
    /** @deprecated use `CustomerPortalOrdersConfirmRetryPaymentRequest$inboundSchema` instead. */
    CustomerPortalOrdersConfirmRetryPaymentRequest$.inboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema` instead. */
    CustomerPortalOrdersConfirmRetryPaymentRequest$.outboundSchema = exports.CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema;
})(CustomerPortalOrdersConfirmRetryPaymentRequest$ || (exports.CustomerPortalOrdersConfirmRetryPaymentRequest$ = CustomerPortalOrdersConfirmRetryPaymentRequest$ = {}));
function customerPortalOrdersConfirmRetryPaymentRequestToJSON(customerPortalOrdersConfirmRetryPaymentRequest) {
    return JSON.stringify(exports.CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema.parse(customerPortalOrdersConfirmRetryPaymentRequest));
}
function customerPortalOrdersConfirmRetryPaymentRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersConfirmRetryPaymentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersConfirmRetryPaymentRequest' from JSON`);
}
