// @polar-sh/sdk/dist/commonjs/models/operations/customerportallicensekeysget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalLicenseKeysGetRequest$ = exports.CustomerPortalLicenseKeysGetRequest$outboundSchema = exports.CustomerPortalLicenseKeysGetRequest$inboundSchema = exports.CustomerPortalLicenseKeysGetSecurity$ = exports.CustomerPortalLicenseKeysGetSecurity$outboundSchema = exports.CustomerPortalLicenseKeysGetSecurity$inboundSchema = void 0;
exports.customerPortalLicenseKeysGetSecurityToJSON = customerPortalLicenseKeysGetSecurityToJSON;
exports.customerPortalLicenseKeysGetSecurityFromJSON = customerPortalLicenseKeysGetSecurityFromJSON;
exports.customerPortalLicenseKeysGetRequestToJSON = customerPortalLicenseKeysGetRequestToJSON;
exports.customerPortalLicenseKeysGetRequestFromJSON = customerPortalLicenseKeysGetRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalLicenseKeysGetSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalLicenseKeysGetSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalLicenseKeysGetSecurity$;
(function (CustomerPortalLicenseKeysGetSecurity$) {
    /** @deprecated use `CustomerPortalLicenseKeysGetSecurity$inboundSchema` instead. */
    CustomerPortalLicenseKeysGetSecurity$.inboundSchema = exports.CustomerPortalLicenseKeysGetSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalLicenseKeysGetSecurity$outboundSchema` instead. */
    CustomerPortalLicenseKeysGetSecurity$.outboundSchema = exports.CustomerPortalLicenseKeysGetSecurity$outboundSchema;
})(CustomerPortalLicenseKeysGetSecurity$ || (exports.CustomerPortalLicenseKeysGetSecurity$ = CustomerPortalLicenseKeysGetSecurity$ = {}));
function customerPortalLicenseKeysGetSecurityToJSON(customerPortalLicenseKeysGetSecurity) {
    return JSON.stringify(exports.CustomerPortalLicenseKeysGetSecurity$outboundSchema.parse(customerPortalLicenseKeysGetSecurity));
}
function customerPortalLicenseKeysGetSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalLicenseKeysGetSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalLicenseKeysGetSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalLicenseKeysGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalLicenseKeysGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalLicenseKeysGetRequest$;
(function (CustomerPortalLicenseKeysGetRequest$) {
    /** @deprecated use `CustomerPortalLicenseKeysGetRequest$inboundSchema` instead. */
    CustomerPortalLicenseKeysGetRequest$.inboundSchema = exports.CustomerPortalLicenseKeysGetRequest$inboundSchema;
    /** @deprecated use `CustomerPortalLicenseKeysGetRequest$outboundSchema` instead. */
    CustomerPortalLicenseKeysGetRequest$.outboundSchema = exports.CustomerPortalLicenseKeysGetRequest$outboundSchema;
})(CustomerPortalLicenseKeysGetRequest$ || (exports.CustomerPortalLicenseKeysGetRequest$ = CustomerPortalLicenseKeysGetRequest$ = {}));
function customerPortalLicenseKeysGetRequestToJSON(customerPortalLicenseKeysGetRequest) {
    return JSON.stringify(exports.CustomerPortalLicenseKeysGetRequest$outboundSchema.parse(customerPortalLicenseKeysGetRequest));
}
function customerPortalLicenseKeysGetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalLicenseKeysGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalLicenseKeysGetRequest' from JSON`);
}
