// @polar-sh/sdk/dist/commonjs/models/operations/customerportalsubscriptionscancel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalSubscriptionsCancelRequest$ = exports.CustomerPortalSubscriptionsCancelRequest$outboundSchema = exports.CustomerPortalSubscriptionsCancelRequest$inboundSchema = exports.CustomerPortalSubscriptionsCancelSecurity$ = exports.CustomerPortalSubscriptionsCancelSecurity$outboundSchema = exports.CustomerPortalSubscriptionsCancelSecurity$inboundSchema = void 0;
exports.customerPortalSubscriptionsCancelSecurityToJSON = customerPortalSubscriptionsCancelSecurityToJSON;
exports.customerPortalSubscriptionsCancelSecurityFromJSON = customerPortalSubscriptionsCancelSecurityFromJSON;
exports.customerPortalSubscriptionsCancelRequestToJSON = customerPortalSubscriptionsCancelRequestToJSON;
exports.customerPortalSubscriptionsCancelRequestFromJSON = customerPortalSubscriptionsCancelRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalSubscriptionsCancelSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalSubscriptionsCancelSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSubscriptionsCancelSecurity$;
(function (CustomerPortalSubscriptionsCancelSecurity$) {
    /** @deprecated use `CustomerPortalSubscriptionsCancelSecurity$inboundSchema` instead. */
    CustomerPortalSubscriptionsCancelSecurity$.inboundSchema = exports.CustomerPortalSubscriptionsCancelSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSubscriptionsCancelSecurity$outboundSchema` instead. */
    CustomerPortalSubscriptionsCancelSecurity$.outboundSchema = exports.CustomerPortalSubscriptionsCancelSecurity$outboundSchema;
})(CustomerPortalSubscriptionsCancelSecurity$ || (exports.CustomerPortalSubscriptionsCancelSecurity$ = CustomerPortalSubscriptionsCancelSecurity$ = {}));
function customerPortalSubscriptionsCancelSecurityToJSON(customerPortalSubscriptionsCancelSecurity) {
    return JSON.stringify(exports.CustomerPortalSubscriptionsCancelSecurity$outboundSchema.parse(customerPortalSubscriptionsCancelSecurity));
}
function customerPortalSubscriptionsCancelSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSubscriptionsCancelSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsCancelSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalSubscriptionsCancelRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalSubscriptionsCancelRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalSubscriptionsCancelRequest$;
(function (CustomerPortalSubscriptionsCancelRequest$) {
    /** @deprecated use `CustomerPortalSubscriptionsCancelRequest$inboundSchema` instead. */
    CustomerPortalSubscriptionsCancelRequest$.inboundSchema = exports.CustomerPortalSubscriptionsCancelRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSubscriptionsCancelRequest$outboundSchema` instead. */
    CustomerPortalSubscriptionsCancelRequest$.outboundSchema = exports.CustomerPortalSubscriptionsCancelRequest$outboundSchema;
})(CustomerPortalSubscriptionsCancelRequest$ || (exports.CustomerPortalSubscriptionsCancelRequest$ = CustomerPortalSubscriptionsCancelRequest$ = {}));
function customerPortalSubscriptionsCancelRequestToJSON(customerPortalSubscriptionsCancelRequest) {
    return JSON.stringify(exports.CustomerPortalSubscriptionsCancelRequest$outboundSchema.parse(customerPortalSubscriptionsCancelRequest));
}
function customerPortalSubscriptionsCancelRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalSubscriptionsCancelRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsCancelRequest' from JSON`);
}
