// @polar-sh/sdk/dist/commonjs/models/operations/eventslistnames.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsListNamesResponse$ = exports.EventsListNamesResponse$outboundSchema = exports.EventsListNamesResponse$inboundSchema = exports.EventsListNamesRequest$ = exports.EventsListNamesRequest$outboundSchema = exports.EventsListNamesRequest$inboundSchema = exports.QueryParamSourceFilter$ = exports.QueryParamSourceFilter$outboundSchema = exports.QueryParamSourceFilter$inboundSchema = exports.EventsListNamesQueryParamExternalCustomerIDFilter$ = exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema = exports.EventsListNamesQueryParamExternalCustomerIDFilter$inboundSchema = exports.EventsListNamesQueryParamCustomerIDFilter$ = exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema = exports.EventsListNamesQueryParamCustomerIDFilter$inboundSchema = exports.EventsListNamesQueryParamOrganizationIDFilter$ = exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema = exports.EventsListNamesQueryParamOrganizationIDFilter$inboundSchema = void 0;
exports.eventsListNamesQueryParamOrganizationIDFilterToJSON = eventsListNamesQueryParamOrganizationIDFilterToJSON;
exports.eventsListNamesQueryParamOrganizationIDFilterFromJSON = eventsListNamesQueryParamOrganizationIDFilterFromJSON;
exports.eventsListNamesQueryParamCustomerIDFilterToJSON = eventsListNamesQueryParamCustomerIDFilterToJSON;
exports.eventsListNamesQueryParamCustomerIDFilterFromJSON = eventsListNamesQueryParamCustomerIDFilterFromJSON;
exports.eventsListNamesQueryParamExternalCustomerIDFilterToJSON = eventsListNamesQueryParamExternalCustomerIDFilterToJSON;
exports.eventsListNamesQueryParamExternalCustomerIDFilterFromJSON = eventsListNamesQueryParamExternalCustomerIDFilterFromJSON;
exports.queryParamSourceFilterToJSON = queryParamSourceFilterToJSON;
exports.queryParamSourceFilterFromJSON = queryParamSourceFilterFromJSON;
exports.eventsListNamesRequestToJSON = eventsListNamesRequestToJSON;
exports.eventsListNamesRequestFromJSON = eventsListNamesRequestFromJSON;
exports.eventsListNamesResponseToJSON = eventsListNamesResponseToJSON;
exports.eventsListNamesResponseFromJSON = eventsListNamesResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const eventnamessortproperty_js_1 = require("../components/eventnamessortproperty.js");
const eventsource_js_1 = require("../components/eventsource.js");
const listresourceeventname_js_1 = require("../components/listresourceeventname.js");
/** @internal */
exports.EventsListNamesQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsListNamesQueryParamOrganizationIDFilter$;
(function (EventsListNamesQueryParamOrganizationIDFilter$) {
    /** @deprecated use `EventsListNamesQueryParamOrganizationIDFilter$inboundSchema` instead. */
    EventsListNamesQueryParamOrganizationIDFilter$.inboundSchema = exports.EventsListNamesQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `EventsListNamesQueryParamOrganizationIDFilter$outboundSchema` instead. */
    EventsListNamesQueryParamOrganizationIDFilter$.outboundSchema = exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema;
})(EventsListNamesQueryParamOrganizationIDFilter$ || (exports.EventsListNamesQueryParamOrganizationIDFilter$ = EventsListNamesQueryParamOrganizationIDFilter$ = {}));
function eventsListNamesQueryParamOrganizationIDFilterToJSON(eventsListNamesQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema.parse(eventsListNamesQueryParamOrganizationIDFilter));
}
function eventsListNamesQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsListNamesQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListNamesQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.EventsListNamesQueryParamCustomerIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsListNamesQueryParamCustomerIDFilter$;
(function (EventsListNamesQueryParamCustomerIDFilter$) {
    /** @deprecated use `EventsListNamesQueryParamCustomerIDFilter$inboundSchema` instead. */
    EventsListNamesQueryParamCustomerIDFilter$.inboundSchema = exports.EventsListNamesQueryParamCustomerIDFilter$inboundSchema;
    /** @deprecated use `EventsListNamesQueryParamCustomerIDFilter$outboundSchema` instead. */
    EventsListNamesQueryParamCustomerIDFilter$.outboundSchema = exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema;
})(EventsListNamesQueryParamCustomerIDFilter$ || (exports.EventsListNamesQueryParamCustomerIDFilter$ = EventsListNamesQueryParamCustomerIDFilter$ = {}));
function eventsListNamesQueryParamCustomerIDFilterToJSON(eventsListNamesQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema.parse(eventsListNamesQueryParamCustomerIDFilter));
}
function eventsListNamesQueryParamCustomerIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsListNamesQueryParamCustomerIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListNamesQueryParamCustomerIDFilter' from JSON`);
}
/** @internal */
exports.EventsListNamesQueryParamExternalCustomerIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsListNamesQueryParamExternalCustomerIDFilter$;
(function (EventsListNamesQueryParamExternalCustomerIDFilter$) {
    /** @deprecated use `EventsListNamesQueryParamExternalCustomerIDFilter$inboundSchema` instead. */
    EventsListNamesQueryParamExternalCustomerIDFilter$.inboundSchema = exports.EventsListNamesQueryParamExternalCustomerIDFilter$inboundSchema;
    /** @deprecated use `EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema` instead. */
    EventsListNamesQueryParamExternalCustomerIDFilter$.outboundSchema = exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema;
})(EventsListNamesQueryParamExternalCustomerIDFilter$ || (exports.EventsListNamesQueryParamExternalCustomerIDFilter$ = EventsListNamesQueryParamExternalCustomerIDFilter$ = {}));
function eventsListNamesQueryParamExternalCustomerIDFilterToJSON(eventsListNamesQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema.parse(eventsListNamesQueryParamExternalCustomerIDFilter));
}
function eventsListNamesQueryParamExternalCustomerIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsListNamesQueryParamExternalCustomerIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListNamesQueryParamExternalCustomerIDFilter' from JSON`);
}
/** @internal */
exports.QueryParamSourceFilter$inboundSchema = z.union([eventsource_js_1.EventSource$inboundSchema, z.array(eventsource_js_1.EventSource$inboundSchema)]);
/** @internal */
exports.QueryParamSourceFilter$outboundSchema = z.union([eventsource_js_1.EventSource$outboundSchema, z.array(eventsource_js_1.EventSource$outboundSchema)]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamSourceFilter$;
(function (QueryParamSourceFilter$) {
    /** @deprecated use `QueryParamSourceFilter$inboundSchema` instead. */
    QueryParamSourceFilter$.inboundSchema = exports.QueryParamSourceFilter$inboundSchema;
    /** @deprecated use `QueryParamSourceFilter$outboundSchema` instead. */
    QueryParamSourceFilter$.outboundSchema = exports.QueryParamSourceFilter$outboundSchema;
})(QueryParamSourceFilter$ || (exports.QueryParamSourceFilter$ = QueryParamSourceFilter$ = {}));
function queryParamSourceFilterToJSON(queryParamSourceFilter) {
    return JSON.stringify(exports.QueryParamSourceFilter$outboundSchema.parse(queryParamSourceFilter));
}
function queryParamSourceFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QueryParamSourceFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryParamSourceFilter' from JSON`);
}
/** @internal */
exports.EventsListNamesRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    customer_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    external_customer_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    source: z.nullable(z.union([eventsource_js_1.EventSource$inboundSchema, z.array(eventsource_js_1.EventSource$inboundSchema)])).optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(eventnamessortproperty_js_1.EventNamesSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
exports.EventsListNamesRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    customerId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    externalCustomerId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    source: z.nullable(z.union([eventsource_js_1.EventSource$outboundSchema, z.array(eventsource_js_1.EventSource$outboundSchema)])).optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(eventnamessortproperty_js_1.EventNamesSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsListNamesRequest$;
(function (EventsListNamesRequest$) {
    /** @deprecated use `EventsListNamesRequest$inboundSchema` instead. */
    EventsListNamesRequest$.inboundSchema = exports.EventsListNamesRequest$inboundSchema;
    /** @deprecated use `EventsListNamesRequest$outboundSchema` instead. */
    EventsListNamesRequest$.outboundSchema = exports.EventsListNamesRequest$outboundSchema;
})(EventsListNamesRequest$ || (exports.EventsListNamesRequest$ = EventsListNamesRequest$ = {}));
function eventsListNamesRequestToJSON(eventsListNamesRequest) {
    return JSON.stringify(exports.EventsListNamesRequest$outboundSchema.parse(eventsListNamesRequest));
}
function eventsListNamesRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsListNamesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListNamesRequest' from JSON`);
}
/** @internal */
exports.EventsListNamesResponse$inboundSchema = z.object({
    Result: listresourceeventname_js_1.ListResourceEventName$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.EventsListNamesResponse$outboundSchema = z.object({
    result: listresourceeventname_js_1.ListResourceEventName$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsListNamesResponse$;
(function (EventsListNamesResponse$) {
    /** @deprecated use `EventsListNamesResponse$inboundSchema` instead. */
    EventsListNamesResponse$.inboundSchema = exports.EventsListNamesResponse$inboundSchema;
    /** @deprecated use `EventsListNamesResponse$outboundSchema` instead. */
    EventsListNamesResponse$.outboundSchema = exports.EventsListNamesResponse$outboundSchema;
})(EventsListNamesResponse$ || (exports.EventsListNamesResponse$ = EventsListNamesResponse$ = {}));
function eventsListNamesResponseToJSON(eventsListNamesResponse) {
    return JSON.stringify(exports.EventsListNamesResponse$outboundSchema.parse(eventsListNamesResponse));
}
function eventsListNamesResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsListNamesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListNamesResponse' from JSON`);
}
