// @polar-sh/sdk/dist/commonjs/models/operations/ordersupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrdersUpdateRequest$ = exports.OrdersUpdateRequest$outboundSchema = exports.OrdersUpdateRequest$inboundSchema = void 0;
exports.ordersUpdateRequestToJSON = ordersUpdateRequestToJSON;
exports.ordersUpdateRequestFromJSON = ordersUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const orderupdate_js_1 = require("../components/orderupdate.js");
/** @internal */
exports.OrdersUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    OrderUpdate: orderupdate_js_1.OrderUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "OrderUpdate": "orderUpdate",
    });
});
/** @internal */
exports.OrdersUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    orderUpdate: orderupdate_js_1.OrderUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        orderUpdate: "OrderUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersUpdateRequest$;
(function (OrdersUpdateRequest$) {
    /** @deprecated use `OrdersUpdateRequest$inboundSchema` instead. */
    OrdersUpdateRequest$.inboundSchema = exports.OrdersUpdateRequest$inboundSchema;
    /** @deprecated use `OrdersUpdateRequest$outboundSchema` instead. */
    OrdersUpdateRequest$.outboundSchema = exports.OrdersUpdateRequest$outboundSchema;
})(OrdersUpdateRequest$ || (exports.OrdersUpdateRequest$ = OrdersUpdateRequest$ = {}));
function ordersUpdateRequestToJSON(ordersUpdateRequest) {
    return JSON.stringify(exports.OrdersUpdateRequest$outboundSchema.parse(ordersUpdateRequest));
}
function ordersUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrdersUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersUpdateRequest' from JSON`);
}
