// @polar-sh/sdk/dist/commonjs/models/operations/organizationslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationsListResponse$ = exports.OrganizationsListResponse$outboundSchema = exports.OrganizationsListResponse$inboundSchema = exports.OrganizationsListRequest$ = exports.OrganizationsListRequest$outboundSchema = exports.OrganizationsListRequest$inboundSchema = void 0;
exports.organizationsListRequestToJSON = organizationsListRequestToJSON;
exports.organizationsListRequestFromJSON = organizationsListRequestFromJSON;
exports.organizationsListResponseToJSON = organizationsListResponseToJSON;
exports.organizationsListResponseFromJSON = organizationsListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourceorganization_js_1 = require("../components/listresourceorganization.js");
const organizationsortproperty_js_1 = require("../components/organizationsortproperty.js");
/** @internal */
exports.OrganizationsListRequest$inboundSchema = z.object({
    slug: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(organizationsortproperty_js_1.OrganizationSortProperty$inboundSchema))
        .optional(),
});
/** @internal */
exports.OrganizationsListRequest$outboundSchema = z.object({
    slug: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(organizationsortproperty_js_1.OrganizationSortProperty$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationsListRequest$;
(function (OrganizationsListRequest$) {
    /** @deprecated use `OrganizationsListRequest$inboundSchema` instead. */
    OrganizationsListRequest$.inboundSchema = exports.OrganizationsListRequest$inboundSchema;
    /** @deprecated use `OrganizationsListRequest$outboundSchema` instead. */
    OrganizationsListRequest$.outboundSchema = exports.OrganizationsListRequest$outboundSchema;
})(OrganizationsListRequest$ || (exports.OrganizationsListRequest$ = OrganizationsListRequest$ = {}));
function organizationsListRequestToJSON(organizationsListRequest) {
    return JSON.stringify(exports.OrganizationsListRequest$outboundSchema.parse(organizationsListRequest));
}
function organizationsListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationsListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationsListRequest' from JSON`);
}
/** @internal */
exports.OrganizationsListResponse$inboundSchema = z.object({
    Result: listresourceorganization_js_1.ListResourceOrganization$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.OrganizationsListResponse$outboundSchema = z.object({
    result: listresourceorganization_js_1.ListResourceOrganization$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationsListResponse$;
(function (OrganizationsListResponse$) {
    /** @deprecated use `OrganizationsListResponse$inboundSchema` instead. */
    OrganizationsListResponse$.inboundSchema = exports.OrganizationsListResponse$inboundSchema;
    /** @deprecated use `OrganizationsListResponse$outboundSchema` instead. */
    OrganizationsListResponse$.outboundSchema = exports.OrganizationsListResponse$outboundSchema;
})(OrganizationsListResponse$ || (exports.OrganizationsListResponse$ = OrganizationsListResponse$ = {}));
function organizationsListResponseToJSON(organizationsListResponse) {
    return JSON.stringify(exports.OrganizationsListResponse$outboundSchema.parse(organizationsListResponse));
}
function organizationsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationsListResponse' from JSON`);
}
