// @polar-sh/sdk/dist/commonjs/models/operations/subscriptionsexport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionsExportRequest$ = exports.SubscriptionsExportRequest$outboundSchema = exports.SubscriptionsExportRequest$inboundSchema = exports.OrganizationId$ = exports.OrganizationId$outboundSchema = exports.OrganizationId$inboundSchema = void 0;
exports.organizationIdToJSON = organizationIdToJSON;
exports.organizationIdFromJSON = organizationIdFromJSON;
exports.subscriptionsExportRequestToJSON = subscriptionsExportRequestToJSON;
exports.subscriptionsExportRequestFromJSON = subscriptionsExportRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.OrganizationId$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrganizationId$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationId$;
(function (OrganizationId$) {
    /** @deprecated use `OrganizationId$inboundSchema` instead. */
    OrganizationId$.inboundSchema = exports.OrganizationId$inboundSchema;
    /** @deprecated use `OrganizationId$outboundSchema` instead. */
    OrganizationId$.outboundSchema = exports.OrganizationId$outboundSchema;
})(OrganizationId$ || (exports.OrganizationId$ = OrganizationId$ = {}));
function organizationIdToJSON(organizationId) {
    return JSON.stringify(exports.OrganizationId$outboundSchema.parse(organizationId));
}
function organizationIdFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationId' from JSON`);
}
/** @internal */
exports.SubscriptionsExportRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.SubscriptionsExportRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionsExportRequest$;
(function (SubscriptionsExportRequest$) {
    /** @deprecated use `SubscriptionsExportRequest$inboundSchema` instead. */
    SubscriptionsExportRequest$.inboundSchema = exports.SubscriptionsExportRequest$inboundSchema;
    /** @deprecated use `SubscriptionsExportRequest$outboundSchema` instead. */
    SubscriptionsExportRequest$.outboundSchema = exports.SubscriptionsExportRequest$outboundSchema;
})(SubscriptionsExportRequest$ || (exports.SubscriptionsExportRequest$ = SubscriptionsExportRequest$ = {}));
function subscriptionsExportRequestToJSON(subscriptionsExportRequest) {
    return JSON.stringify(exports.SubscriptionsExportRequest$outboundSchema.parse(subscriptionsExportRequest));
}
function subscriptionsExportRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionsExportRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionsExportRequest' from JSON`);
}
