// @polar-sh/sdk/dist/commonjs/models/operations/webhookslistwebhookendpoints.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksListWebhookEndpointsResponse$ = exports.WebhooksListWebhookEndpointsResponse$outboundSchema = exports.WebhooksListWebhookEndpointsResponse$inboundSchema = exports.WebhooksListWebhookEndpointsRequest$ = exports.WebhooksListWebhookEndpointsRequest$outboundSchema = exports.WebhooksListWebhookEndpointsRequest$inboundSchema = exports.QueryParamOrganizationId$ = exports.QueryParamOrganizationId$outboundSchema = exports.QueryParamOrganizationId$inboundSchema = void 0;
exports.queryParamOrganizationIdToJSON = queryParamOrganizationIdToJSON;
exports.queryParamOrganizationIdFromJSON = queryParamOrganizationIdFromJSON;
exports.webhooksListWebhookEndpointsRequestToJSON = webhooksListWebhookEndpointsRequestToJSON;
exports.webhooksListWebhookEndpointsRequestFromJSON = webhooksListWebhookEndpointsRequestFromJSON;
exports.webhooksListWebhookEndpointsResponseToJSON = webhooksListWebhookEndpointsResponseToJSON;
exports.webhooksListWebhookEndpointsResponseFromJSON = webhooksListWebhookEndpointsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourcewebhookendpoint_js_1 = require("../components/listresourcewebhookendpoint.js");
/** @internal */
exports.QueryParamOrganizationId$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.QueryParamOrganizationId$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamOrganizationId$;
(function (QueryParamOrganizationId$) {
    /** @deprecated use `QueryParamOrganizationId$inboundSchema` instead. */
    QueryParamOrganizationId$.inboundSchema = exports.QueryParamOrganizationId$inboundSchema;
    /** @deprecated use `QueryParamOrganizationId$outboundSchema` instead. */
    QueryParamOrganizationId$.outboundSchema = exports.QueryParamOrganizationId$outboundSchema;
})(QueryParamOrganizationId$ || (exports.QueryParamOrganizationId$ = QueryParamOrganizationId$ = {}));
function queryParamOrganizationIdToJSON(queryParamOrganizationId) {
    return JSON.stringify(exports.QueryParamOrganizationId$outboundSchema.parse(queryParamOrganizationId));
}
function queryParamOrganizationIdFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QueryParamOrganizationId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryParamOrganizationId' from JSON`);
}
/** @internal */
exports.WebhooksListWebhookEndpointsRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.WebhooksListWebhookEndpointsRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhooksListWebhookEndpointsRequest$;
(function (WebhooksListWebhookEndpointsRequest$) {
    /** @deprecated use `WebhooksListWebhookEndpointsRequest$inboundSchema` instead. */
    WebhooksListWebhookEndpointsRequest$.inboundSchema = exports.WebhooksListWebhookEndpointsRequest$inboundSchema;
    /** @deprecated use `WebhooksListWebhookEndpointsRequest$outboundSchema` instead. */
    WebhooksListWebhookEndpointsRequest$.outboundSchema = exports.WebhooksListWebhookEndpointsRequest$outboundSchema;
})(WebhooksListWebhookEndpointsRequest$ || (exports.WebhooksListWebhookEndpointsRequest$ = WebhooksListWebhookEndpointsRequest$ = {}));
function webhooksListWebhookEndpointsRequestToJSON(webhooksListWebhookEndpointsRequest) {
    return JSON.stringify(exports.WebhooksListWebhookEndpointsRequest$outboundSchema.parse(webhooksListWebhookEndpointsRequest));
}
function webhooksListWebhookEndpointsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhooksListWebhookEndpointsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookEndpointsRequest' from JSON`);
}
/** @internal */
exports.WebhooksListWebhookEndpointsResponse$inboundSchema = z.object({
    Result: listresourcewebhookendpoint_js_1.ListResourceWebhookEndpoint$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.WebhooksListWebhookEndpointsResponse$outboundSchema = z.object({
    result: listresourcewebhookendpoint_js_1.ListResourceWebhookEndpoint$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhooksListWebhookEndpointsResponse$;
(function (WebhooksListWebhookEndpointsResponse$) {
    /** @deprecated use `WebhooksListWebhookEndpointsResponse$inboundSchema` instead. */
    WebhooksListWebhookEndpointsResponse$.inboundSchema = exports.WebhooksListWebhookEndpointsResponse$inboundSchema;
    /** @deprecated use `WebhooksListWebhookEndpointsResponse$outboundSchema` instead. */
    WebhooksListWebhookEndpointsResponse$.outboundSchema = exports.WebhooksListWebhookEndpointsResponse$outboundSchema;
})(WebhooksListWebhookEndpointsResponse$ || (exports.WebhooksListWebhookEndpointsResponse$ = WebhooksListWebhookEndpointsResponse$ = {}));
function webhooksListWebhookEndpointsResponseToJSON(webhooksListWebhookEndpointsResponse) {
    return JSON.stringify(exports.WebhooksListWebhookEndpointsResponse$outboundSchema.parse(webhooksListWebhookEndpointsResponse));
}
function webhooksListWebhookEndpointsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhooksListWebhookEndpointsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookEndpointsResponse' from JSON`);
}
