// @polar-sh/sdk/dist/commonjs/models/operations/webhookslistwebhookdeliveries.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksListWebhookDeliveriesResponse$ = exports.WebhooksListWebhookDeliveriesResponse$outboundSchema = exports.WebhooksListWebhookDeliveriesResponse$inboundSchema = exports.WebhooksListWebhookDeliveriesRequest$ = exports.WebhooksListWebhookDeliveriesRequest$outboundSchema = exports.WebhooksListWebhookDeliveriesRequest$inboundSchema = exports.EndpointId$ = exports.EndpointId$outboundSchema = exports.EndpointId$inboundSchema = void 0;
exports.endpointIdToJSON = endpointIdToJSON;
exports.endpointIdFromJSON = endpointIdFromJSON;
exports.webhooksListWebhookDeliveriesRequestToJSON = webhooksListWebhookDeliveriesRequestToJSON;
exports.webhooksListWebhookDeliveriesRequestFromJSON = webhooksListWebhookDeliveriesRequestFromJSON;
exports.webhooksListWebhookDeliveriesResponseToJSON = webhooksListWebhookDeliveriesResponseToJSON;
exports.webhooksListWebhookDeliveriesResponseFromJSON = webhooksListWebhookDeliveriesResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourcewebhookdelivery_js_1 = require("../components/listresourcewebhookdelivery.js");
/** @internal */
exports.EndpointId$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.EndpointId$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EndpointId$;
(function (EndpointId$) {
    /** @deprecated use `EndpointId$inboundSchema` instead. */
    EndpointId$.inboundSchema = exports.EndpointId$inboundSchema;
    /** @deprecated use `EndpointId$outboundSchema` instead. */
    EndpointId$.outboundSchema = exports.EndpointId$outboundSchema;
})(EndpointId$ || (exports.EndpointId$ = EndpointId$ = {}));
function endpointIdToJSON(endpointId) {
    return JSON.stringify(exports.EndpointId$outboundSchema.parse(endpointId));
}
function endpointIdFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EndpointId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EndpointId' from JSON`);
}
/** @internal */
exports.WebhooksListWebhookDeliveriesRequest$inboundSchema = z.object({
    endpoint_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "endpoint_id": "endpointId",
    });
});
/** @internal */
exports.WebhooksListWebhookDeliveriesRequest$outboundSchema = z.object({
    endpointId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        endpointId: "endpoint_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhooksListWebhookDeliveriesRequest$;
(function (WebhooksListWebhookDeliveriesRequest$) {
    /** @deprecated use `WebhooksListWebhookDeliveriesRequest$inboundSchema` instead. */
    WebhooksListWebhookDeliveriesRequest$.inboundSchema = exports.WebhooksListWebhookDeliveriesRequest$inboundSchema;
    /** @deprecated use `WebhooksListWebhookDeliveriesRequest$outboundSchema` instead. */
    WebhooksListWebhookDeliveriesRequest$.outboundSchema = exports.WebhooksListWebhookDeliveriesRequest$outboundSchema;
})(WebhooksListWebhookDeliveriesRequest$ || (exports.WebhooksListWebhookDeliveriesRequest$ = WebhooksListWebhookDeliveriesRequest$ = {}));
function webhooksListWebhookDeliveriesRequestToJSON(webhooksListWebhookDeliveriesRequest) {
    return JSON.stringify(exports.WebhooksListWebhookDeliveriesRequest$outboundSchema.parse(webhooksListWebhookDeliveriesRequest));
}
function webhooksListWebhookDeliveriesRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhooksListWebhookDeliveriesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookDeliveriesRequest' from JSON`);
}
/** @internal */
exports.WebhooksListWebhookDeliveriesResponse$inboundSchema = z.object({
    Result: listresourcewebhookdelivery_js_1.ListResourceWebhookDelivery$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.WebhooksListWebhookDeliveriesResponse$outboundSchema = z.object({
    result: listresourcewebhookdelivery_js_1.ListResourceWebhookDelivery$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhooksListWebhookDeliveriesResponse$;
(function (WebhooksListWebhookDeliveriesResponse$) {
    /** @deprecated use `WebhooksListWebhookDeliveriesResponse$inboundSchema` instead. */
    WebhooksListWebhookDeliveriesResponse$.inboundSchema = exports.WebhooksListWebhookDeliveriesResponse$inboundSchema;
    /** @deprecated use `WebhooksListWebhookDeliveriesResponse$outboundSchema` instead. */
    WebhooksListWebhookDeliveriesResponse$.outboundSchema = exports.WebhooksListWebhookDeliveriesResponse$outboundSchema;
})(WebhooksListWebhookDeliveriesResponse$ || (exports.WebhooksListWebhookDeliveriesResponse$ = WebhooksListWebhookDeliveriesResponse$ = {}));
function webhooksListWebhookDeliveriesResponseToJSON(webhooksListWebhookDeliveriesResponse) {
    return JSON.stringify(exports.WebhooksListWebhookDeliveriesResponse$outboundSchema.parse(webhooksListWebhookDeliveriesResponse));
}
function webhooksListWebhookDeliveriesResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhooksListWebhookDeliveriesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookDeliveriesResponse' from JSON`);
}
