// @polar-sh/sdk/dist/commonjs/models/components/benefitmetercreditproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitMeterCreditProperties$ = exports.BenefitMeterCreditProperties$outboundSchema = exports.BenefitMeterCreditProperties$inboundSchema = void 0;
exports.benefitMeterCreditPropertiesToJSON = benefitMeterCreditPropertiesToJSON;
exports.benefitMeterCreditPropertiesFromJSON = benefitMeterCreditPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitMeterCreditProperties$inboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meter_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "meter_id": "meterId",
    });
});
/** @internal */
exports.BenefitMeterCreditProperties$outboundSchema = z.object({
    units: z.number().int(),
    rollover: z.boolean(),
    meterId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        meterId: "meter_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditProperties$;
(function (BenefitMeterCreditProperties$) {
    /** @deprecated use `BenefitMeterCreditProperties$inboundSchema` instead. */
    BenefitMeterCreditProperties$.inboundSchema = exports.BenefitMeterCreditProperties$inboundSchema;
    /** @deprecated use `BenefitMeterCreditProperties$outboundSchema` instead. */
    BenefitMeterCreditProperties$.outboundSchema = exports.BenefitMeterCreditProperties$outboundSchema;
})(BenefitMeterCreditProperties$ || (exports.BenefitMeterCreditProperties$ = BenefitMeterCreditProperties$ = {}));
function benefitMeterCreditPropertiesToJSON(benefitMeterCreditProperties) {
    return JSON.stringify(exports.BenefitMeterCreditProperties$outboundSchema.parse(benefitMeterCreditProperties));
}
function benefitMeterCreditPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditProperties' from JSON`);
}
