// @polar-sh/sdk/dist/commonjs/models/components/benefitcustomupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCustomUpdate$ = exports.BenefitCustomUpdate$outboundSchema = exports.BenefitCustomUpdate$inboundSchema = exports.BenefitCustomUpdateMetadata$ = exports.BenefitCustomUpdateMetadata$outboundSchema = exports.BenefitCustomUpdateMetadata$inboundSchema = void 0;
exports.benefitCustomUpdateMetadataToJSON = benefitCustomUpdateMetadataToJSON;
exports.benefitCustomUpdateMetadataFromJSON = benefitCustomUpdateMetadataFromJSON;
exports.benefitCustomUpdateToJSON = benefitCustomUpdateToJSON;
exports.benefitCustomUpdateFromJSON = benefitCustomUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitcustomproperties_js_1 = require("./benefitcustomproperties.js");
/** @internal */
exports.BenefitCustomUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitCustomUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomUpdateMetadata$;
(function (BenefitCustomUpdateMetadata$) {
    /** @deprecated use `BenefitCustomUpdateMetadata$inboundSchema` instead. */
    BenefitCustomUpdateMetadata$.inboundSchema = exports.BenefitCustomUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitCustomUpdateMetadata$outboundSchema` instead. */
    BenefitCustomUpdateMetadata$.outboundSchema = exports.BenefitCustomUpdateMetadata$outboundSchema;
})(BenefitCustomUpdateMetadata$ || (exports.BenefitCustomUpdateMetadata$ = BenefitCustomUpdateMetadata$ = {}));
function benefitCustomUpdateMetadataToJSON(benefitCustomUpdateMetadata) {
    return JSON.stringify(exports.BenefitCustomUpdateMetadata$outboundSchema.parse(benefitCustomUpdateMetadata));
}
function benefitCustomUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitCustomUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomUpdateMetadata' from JSON`);
}
/** @internal */
exports.BenefitCustomUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("custom"),
    properties: z.nullable(benefitcustomproperties_js_1.BenefitCustomProperties$inboundSchema).optional(),
});
/** @internal */
exports.BenefitCustomUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("custom"),
    properties: z.nullable(benefitcustomproperties_js_1.BenefitCustomProperties$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomUpdate$;
(function (BenefitCustomUpdate$) {
    /** @deprecated use `BenefitCustomUpdate$inboundSchema` instead. */
    BenefitCustomUpdate$.inboundSchema = exports.BenefitCustomUpdate$inboundSchema;
    /** @deprecated use `BenefitCustomUpdate$outboundSchema` instead. */
    BenefitCustomUpdate$.outboundSchema = exports.BenefitCustomUpdate$outboundSchema;
})(BenefitCustomUpdate$ || (exports.BenefitCustomUpdate$ = BenefitCustomUpdate$ = {}));
function benefitCustomUpdateToJSON(benefitCustomUpdate) {
    return JSON.stringify(exports.BenefitCustomUpdate$outboundSchema.parse(benefitCustomUpdate));
}
function benefitCustomUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitCustomUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomUpdate' from JSON`);
}
