// @polar-sh/sdk/dist/commonjs/models/components/benefitgrantdiscordproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantDiscordProperties$ = exports.BenefitGrantDiscordProperties$outboundSchema = exports.BenefitGrantDiscordProperties$inboundSchema = void 0;
exports.benefitGrantDiscordPropertiesToJSON = benefitGrantDiscordPropertiesToJSON;
exports.benefitGrantDiscordPropertiesFromJSON = benefitGrantDiscordPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitGrantDiscordProperties$inboundSchema = z.object({
    account_id: z.nullable(z.string()).optional(),
    guild_id: z.string().optional(),
    role_id: z.string().optional(),
    granted_account_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "guild_id": "guildId",
        "role_id": "roleId",
        "granted_account_id": "grantedAccountId",
    });
});
/** @internal */
exports.BenefitGrantDiscordProperties$outboundSchema = z.object({
    accountId: z.nullable(z.string()).optional(),
    guildId: z.string().optional(),
    roleId: z.string().optional(),
    grantedAccountId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        guildId: "guild_id",
        roleId: "role_id",
        grantedAccountId: "granted_account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantDiscordProperties$;
(function (BenefitGrantDiscordProperties$) {
    /** @deprecated use `BenefitGrantDiscordProperties$inboundSchema` instead. */
    BenefitGrantDiscordProperties$.inboundSchema = exports.BenefitGrantDiscordProperties$inboundSchema;
    /** @deprecated use `BenefitGrantDiscordProperties$outboundSchema` instead. */
    BenefitGrantDiscordProperties$.outboundSchema = exports.BenefitGrantDiscordProperties$outboundSchema;
})(BenefitGrantDiscordProperties$ || (exports.BenefitGrantDiscordProperties$ = BenefitGrantDiscordProperties$ = {}));
function benefitGrantDiscordPropertiesToJSON(benefitGrantDiscordProperties) {
    return JSON.stringify(exports.BenefitGrantDiscordProperties$outboundSchema.parse(benefitGrantDiscordProperties));
}
function benefitGrantDiscordPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantDiscordProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantDiscordProperties' from JSON`);
}
