// @polar-sh/sdk/dist/commonjs/models/components/benefitlicensekeysupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeysUpdate$ = exports.BenefitLicenseKeysUpdate$outboundSchema = exports.BenefitLicenseKeysUpdate$inboundSchema = exports.BenefitLicenseKeysUpdateMetadata$ = exports.BenefitLicenseKeysUpdateMetadata$outboundSchema = exports.BenefitLicenseKeysUpdateMetadata$inboundSchema = void 0;
exports.benefitLicenseKeysUpdateMetadataToJSON = benefitLicenseKeysUpdateMetadataToJSON;
exports.benefitLicenseKeysUpdateMetadataFromJSON = benefitLicenseKeysUpdateMetadataFromJSON;
exports.benefitLicenseKeysUpdateToJSON = benefitLicenseKeysUpdateToJSON;
exports.benefitLicenseKeysUpdateFromJSON = benefitLicenseKeysUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitlicensekeyscreateproperties_js_1 = require("./benefitlicensekeyscreateproperties.js");
/** @internal */
exports.BenefitLicenseKeysUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitLicenseKeysUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysUpdateMetadata$;
(function (BenefitLicenseKeysUpdateMetadata$) {
    /** @deprecated use `BenefitLicenseKeysUpdateMetadata$inboundSchema` instead. */
    BenefitLicenseKeysUpdateMetadata$.inboundSchema = exports.BenefitLicenseKeysUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysUpdateMetadata$outboundSchema` instead. */
    BenefitLicenseKeysUpdateMetadata$.outboundSchema = exports.BenefitLicenseKeysUpdateMetadata$outboundSchema;
})(BenefitLicenseKeysUpdateMetadata$ || (exports.BenefitLicenseKeysUpdateMetadata$ = BenefitLicenseKeysUpdateMetadata$ = {}));
function benefitLicenseKeysUpdateMetadataToJSON(benefitLicenseKeysUpdateMetadata) {
    return JSON.stringify(exports.BenefitLicenseKeysUpdateMetadata$outboundSchema.parse(benefitLicenseKeysUpdateMetadata));
}
function benefitLicenseKeysUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeysUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysUpdateMetadata' from JSON`);
}
/** @internal */
exports.BenefitLicenseKeysUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("license_keys"),
    properties: z.nullable(benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitLicenseKeysUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("license_keys"),
    properties: z.nullable(benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysUpdate$;
(function (BenefitLicenseKeysUpdate$) {
    /** @deprecated use `BenefitLicenseKeysUpdate$inboundSchema` instead. */
    BenefitLicenseKeysUpdate$.inboundSchema = exports.BenefitLicenseKeysUpdate$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysUpdate$outboundSchema` instead. */
    BenefitLicenseKeysUpdate$.outboundSchema = exports.BenefitLicenseKeysUpdate$outboundSchema;
})(BenefitLicenseKeysUpdate$ || (exports.BenefitLicenseKeysUpdate$ = BenefitLicenseKeysUpdate$ = {}));
function benefitLicenseKeysUpdateToJSON(benefitLicenseKeysUpdate) {
    return JSON.stringify(exports.BenefitLicenseKeysUpdate$outboundSchema.parse(benefitLicenseKeysUpdate));
}
function benefitLicenseKeysUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeysUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysUpdate' from JSON`);
}
