// @polar-sh/sdk/dist/commonjs/models/components/benefitlicensekeyscreateproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeysCreateProperties$ = exports.BenefitLicenseKeysCreateProperties$outboundSchema = exports.BenefitLicenseKeysCreateProperties$inboundSchema = void 0;
exports.benefitLicenseKeysCreatePropertiesToJSON = benefitLicenseKeysCreatePropertiesToJSON;
exports.benefitLicenseKeysCreatePropertiesFromJSON = benefitLicenseKeysCreatePropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitlicensekeyactivationcreateproperties_js_1 = require("./benefitlicensekeyactivationcreateproperties.js");
const benefitlicensekeyexpirationproperties_js_1 = require("./benefitlicensekeyexpirationproperties.js");
/** @internal */
exports.BenefitLicenseKeysCreateProperties$inboundSchema = z.object({
    prefix: z.nullable(z.string()).optional(),
    expires: z.nullable(benefitlicensekeyexpirationproperties_js_1.BenefitLicenseKeyExpirationProperties$inboundSchema)
        .optional(),
    activations: z.nullable(benefitlicensekeyactivationcreateproperties_js_1.BenefitLicenseKeyActivationCreateProperties$inboundSchema).optional(),
    limit_usage: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "limit_usage": "limitUsage",
    });
});
/** @internal */
exports.BenefitLicenseKeysCreateProperties$outboundSchema = z.object({
    prefix: z.nullable(z.string()).optional(),
    expires: z.nullable(benefitlicensekeyexpirationproperties_js_1.BenefitLicenseKeyExpirationProperties$outboundSchema)
        .optional(),
    activations: z.nullable(benefitlicensekeyactivationcreateproperties_js_1.BenefitLicenseKeyActivationCreateProperties$outboundSchema).optional(),
    limitUsage: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        limitUsage: "limit_usage",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysCreateProperties$;
(function (BenefitLicenseKeysCreateProperties$) {
    /** @deprecated use `BenefitLicenseKeysCreateProperties$inboundSchema` instead. */
    BenefitLicenseKeysCreateProperties$.inboundSchema = exports.BenefitLicenseKeysCreateProperties$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreateProperties$outboundSchema` instead. */
    BenefitLicenseKeysCreateProperties$.outboundSchema = exports.BenefitLicenseKeysCreateProperties$outboundSchema;
})(BenefitLicenseKeysCreateProperties$ || (exports.BenefitLicenseKeysCreateProperties$ = BenefitLicenseKeysCreateProperties$ = {}));
function benefitLicenseKeysCreatePropertiesToJSON(benefitLicenseKeysCreateProperties) {
    return JSON.stringify(exports.BenefitLicenseKeysCreateProperties$outboundSchema.parse(benefitLicenseKeysCreateProperties));
}
function benefitLicenseKeysCreatePropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitLicenseKeysCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysCreateProperties' from JSON`);
}
