// @polar-sh/sdk/dist/commonjs/models/components/benefitdiscord.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscord$ = exports.BenefitDiscord$outboundSchema = exports.BenefitDiscord$inboundSchema = exports.BenefitDiscordMetadata$ = exports.BenefitDiscordMetadata$outboundSchema = exports.BenefitDiscordMetadata$inboundSchema = void 0;
exports.benefitDiscordMetadataToJSON = benefitDiscordMetadataToJSON;
exports.benefitDiscordMetadataFromJSON = benefitDiscordMetadataFromJSON;
exports.benefitDiscordToJSON = benefitDiscordToJSON;
exports.benefitDiscordFromJSON = benefitDiscordFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdiscordproperties_js_1 = require("./benefitdiscordproperties.js");
/** @internal */
exports.BenefitDiscordMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitDiscordMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordMetadata$;
(function (BenefitDiscordMetadata$) {
    /** @deprecated use `BenefitDiscordMetadata$inboundSchema` instead. */
    BenefitDiscordMetadata$.inboundSchema = exports.BenefitDiscordMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordMetadata$outboundSchema` instead. */
    BenefitDiscordMetadata$.outboundSchema = exports.BenefitDiscordMetadata$outboundSchema;
})(BenefitDiscordMetadata$ || (exports.BenefitDiscordMetadata$ = BenefitDiscordMetadata$ = {}));
function benefitDiscordMetadataToJSON(benefitDiscordMetadata) {
    return JSON.stringify(exports.BenefitDiscordMetadata$outboundSchema.parse(benefitDiscordMetadata));
}
function benefitDiscordMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordMetadata' from JSON`);
}
/** @internal */
exports.BenefitDiscord$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitdiscordproperties_js_1.BenefitDiscordProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDiscord$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitdiscordproperties_js_1.BenefitDiscordProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscord$;
(function (BenefitDiscord$) {
    /** @deprecated use `BenefitDiscord$inboundSchema` instead. */
    BenefitDiscord$.inboundSchema = exports.BenefitDiscord$inboundSchema;
    /** @deprecated use `BenefitDiscord$outboundSchema` instead. */
    BenefitDiscord$.outboundSchema = exports.BenefitDiscord$outboundSchema;
})(BenefitDiscord$ || (exports.BenefitDiscord$ = BenefitDiscord$ = {}));
function benefitDiscordToJSON(benefitDiscord) {
    return JSON.stringify(exports.BenefitDiscord$outboundSchema.parse(benefitDiscord));
}
function benefitDiscordFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscord$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscord' from JSON`);
}
