// @polar-sh/sdk/dist/commonjs/models/components/benefitgrantmetercreditproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantMeterCreditProperties$ = exports.BenefitGrantMeterCreditProperties$outboundSchema = exports.BenefitGrantMeterCreditProperties$inboundSchema = void 0;
exports.benefitGrantMeterCreditPropertiesToJSON = benefitGrantMeterCreditPropertiesToJSON;
exports.benefitGrantMeterCreditPropertiesFromJSON = benefitGrantMeterCreditPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitGrantMeterCreditProperties$inboundSchema = z.object({
    last_credited_meter_id: z.string().optional(),
    last_credited_units: z.number().int().optional(),
    last_credited_at: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "last_credited_meter_id": "lastCreditedMeterId",
        "last_credited_units": "lastCreditedUnits",
        "last_credited_at": "lastCreditedAt",
    });
});
/** @internal */
exports.BenefitGrantMeterCreditProperties$outboundSchema = z.object({
    lastCreditedMeterId: z.string().optional(),
    lastCreditedUnits: z.number().int().optional(),
    lastCreditedAt: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        lastCreditedMeterId: "last_credited_meter_id",
        lastCreditedUnits: "last_credited_units",
        lastCreditedAt: "last_credited_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantMeterCreditProperties$;
(function (BenefitGrantMeterCreditProperties$) {
    /** @deprecated use `BenefitGrantMeterCreditProperties$inboundSchema` instead. */
    BenefitGrantMeterCreditProperties$.inboundSchema = exports.BenefitGrantMeterCreditProperties$inboundSchema;
    /** @deprecated use `BenefitGrantMeterCreditProperties$outboundSchema` instead. */
    BenefitGrantMeterCreditProperties$.outboundSchema = exports.BenefitGrantMeterCreditProperties$outboundSchema;
})(BenefitGrantMeterCreditProperties$ || (exports.BenefitGrantMeterCreditProperties$ = BenefitGrantMeterCreditProperties$ = {}));
function benefitGrantMeterCreditPropertiesToJSON(benefitGrantMeterCreditProperties) {
    return JSON.stringify(exports.BenefitGrantMeterCreditProperties$outboundSchema.parse(benefitGrantMeterCreditProperties));
}
function benefitGrantMeterCreditPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantMeterCreditProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantMeterCreditProperties' from JSON`);
}
