// @polar-sh/sdk/dist/commonjs/models/components/benefitmetercreditcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitMeterCreditCreate$ = exports.BenefitMeterCreditCreate$outboundSchema = exports.BenefitMeterCreditCreate$inboundSchema = exports.BenefitMeterCreditCreateMetadata$ = exports.BenefitMeterCreditCreateMetadata$outboundSchema = exports.BenefitMeterCreditCreateMetadata$inboundSchema = void 0;
exports.benefitMeterCreditCreateMetadataToJSON = benefitMeterCreditCreateMetadataToJSON;
exports.benefitMeterCreditCreateMetadataFromJSON = benefitMeterCreditCreateMetadataFromJSON;
exports.benefitMeterCreditCreateToJSON = benefitMeterCreditCreateToJSON;
exports.benefitMeterCreditCreateFromJSON = benefitMeterCreditCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitmetercreditcreateproperties_js_1 = require("./benefitmetercreditcreateproperties.js");
/** @internal */
exports.BenefitMeterCreditCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitMeterCreditCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditCreateMetadata$;
(function (BenefitMeterCreditCreateMetadata$) {
    /** @deprecated use `BenefitMeterCreditCreateMetadata$inboundSchema` instead. */
    BenefitMeterCreditCreateMetadata$.inboundSchema = exports.BenefitMeterCreditCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitMeterCreditCreateMetadata$outboundSchema` instead. */
    BenefitMeterCreditCreateMetadata$.outboundSchema = exports.BenefitMeterCreditCreateMetadata$outboundSchema;
})(BenefitMeterCreditCreateMetadata$ || (exports.BenefitMeterCreditCreateMetadata$ = BenefitMeterCreditCreateMetadata$ = {}));
function benefitMeterCreditCreateMetadataToJSON(benefitMeterCreditCreateMetadata) {
    return JSON.stringify(exports.BenefitMeterCreditCreateMetadata$outboundSchema.parse(benefitMeterCreditCreateMetadata));
}
function benefitMeterCreditCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditCreateMetadata' from JSON`);
}
/** @internal */
exports.BenefitMeterCreditCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("meter_credit"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitmetercreditcreateproperties_js_1.BenefitMeterCreditCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitMeterCreditCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("meter_credit"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitmetercreditcreateproperties_js_1.BenefitMeterCreditCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditCreate$;
(function (BenefitMeterCreditCreate$) {
    /** @deprecated use `BenefitMeterCreditCreate$inboundSchema` instead. */
    BenefitMeterCreditCreate$.inboundSchema = exports.BenefitMeterCreditCreate$inboundSchema;
    /** @deprecated use `BenefitMeterCreditCreate$outboundSchema` instead. */
    BenefitMeterCreditCreate$.outboundSchema = exports.BenefitMeterCreditCreate$outboundSchema;
})(BenefitMeterCreditCreate$ || (exports.BenefitMeterCreditCreate$ = BenefitMeterCreditCreate$ = {}));
function benefitMeterCreditCreateToJSON(benefitMeterCreditCreate) {
    return JSON.stringify(exports.BenefitMeterCreditCreate$outboundSchema.parse(benefitMeterCreditCreate));
}
function benefitMeterCreditCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditCreate' from JSON`);
}
