// @polar-sh/sdk/dist/commonjs/models/components/benefitmetercreditupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitMeterCreditUpdate$ = exports.BenefitMeterCreditUpdate$outboundSchema = exports.BenefitMeterCreditUpdate$inboundSchema = exports.BenefitMeterCreditUpdateMetadata$ = exports.BenefitMeterCreditUpdateMetadata$outboundSchema = exports.BenefitMeterCreditUpdateMetadata$inboundSchema = void 0;
exports.benefitMeterCreditUpdateMetadataToJSON = benefitMeterCreditUpdateMetadataToJSON;
exports.benefitMeterCreditUpdateMetadataFromJSON = benefitMeterCreditUpdateMetadataFromJSON;
exports.benefitMeterCreditUpdateToJSON = benefitMeterCreditUpdateToJSON;
exports.benefitMeterCreditUpdateFromJSON = benefitMeterCreditUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitmetercreditcreateproperties_js_1 = require("./benefitmetercreditcreateproperties.js");
/** @internal */
exports.BenefitMeterCreditUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitMeterCreditUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditUpdateMetadata$;
(function (BenefitMeterCreditUpdateMetadata$) {
    /** @deprecated use `BenefitMeterCreditUpdateMetadata$inboundSchema` instead. */
    BenefitMeterCreditUpdateMetadata$.inboundSchema = exports.BenefitMeterCreditUpdateMetadata$inboundSchema;
    /** @deprecated use `BenefitMeterCreditUpdateMetadata$outboundSchema` instead. */
    BenefitMeterCreditUpdateMetadata$.outboundSchema = exports.BenefitMeterCreditUpdateMetadata$outboundSchema;
})(BenefitMeterCreditUpdateMetadata$ || (exports.BenefitMeterCreditUpdateMetadata$ = BenefitMeterCreditUpdateMetadata$ = {}));
function benefitMeterCreditUpdateMetadataToJSON(benefitMeterCreditUpdateMetadata) {
    return JSON.stringify(exports.BenefitMeterCreditUpdateMetadata$outboundSchema.parse(benefitMeterCreditUpdateMetadata));
}
function benefitMeterCreditUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditUpdateMetadata' from JSON`);
}
/** @internal */
exports.BenefitMeterCreditUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("meter_credit"),
    properties: z.nullable(benefitmetercreditcreateproperties_js_1.BenefitMeterCreditCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitMeterCreditUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    description: z.nullable(z.string()).optional(),
    type: z.literal("meter_credit"),
    properties: z.nullable(benefitmetercreditcreateproperties_js_1.BenefitMeterCreditCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitMeterCreditUpdate$;
(function (BenefitMeterCreditUpdate$) {
    /** @deprecated use `BenefitMeterCreditUpdate$inboundSchema` instead. */
    BenefitMeterCreditUpdate$.inboundSchema = exports.BenefitMeterCreditUpdate$inboundSchema;
    /** @deprecated use `BenefitMeterCreditUpdate$outboundSchema` instead. */
    BenefitMeterCreditUpdate$.outboundSchema = exports.BenefitMeterCreditUpdate$outboundSchema;
})(BenefitMeterCreditUpdate$ || (exports.BenefitMeterCreditUpdate$ = BenefitMeterCreditUpdate$ = {}));
function benefitMeterCreditUpdateToJSON(benefitMeterCreditUpdate) {
    return JSON.stringify(exports.BenefitMeterCreditUpdate$outboundSchema.parse(benefitMeterCreditUpdate));
}
function benefitMeterCreditUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitMeterCreditUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditUpdate' from JSON`);
}
