// @polar-sh/sdk/dist/commonjs/models/components/benefitdiscordcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordCreate$ = exports.BenefitDiscordCreate$outboundSchema = exports.BenefitDiscordCreate$inboundSchema = exports.BenefitDiscordCreateMetadata$ = exports.BenefitDiscordCreateMetadata$outboundSchema = exports.BenefitDiscordCreateMetadata$inboundSchema = void 0;
exports.benefitDiscordCreateMetadataToJSON = benefitDiscordCreateMetadataToJSON;
exports.benefitDiscordCreateMetadataFromJSON = benefitDiscordCreateMetadataFromJSON;
exports.benefitDiscordCreateToJSON = benefitDiscordCreateToJSON;
exports.benefitDiscordCreateFromJSON = benefitDiscordCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitdiscordcreateproperties_js_1 = require("./benefitdiscordcreateproperties.js");
/** @internal */
exports.BenefitDiscordCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitDiscordCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordCreateMetadata$;
(function (BenefitDiscordCreateMetadata$) {
    /** @deprecated use `BenefitDiscordCreateMetadata$inboundSchema` instead. */
    BenefitDiscordCreateMetadata$.inboundSchema = exports.BenefitDiscordCreateMetadata$inboundSchema;
    /** @deprecated use `BenefitDiscordCreateMetadata$outboundSchema` instead. */
    BenefitDiscordCreateMetadata$.outboundSchema = exports.BenefitDiscordCreateMetadata$outboundSchema;
})(BenefitDiscordCreateMetadata$ || (exports.BenefitDiscordCreateMetadata$ = BenefitDiscordCreateMetadata$ = {}));
function benefitDiscordCreateMetadataToJSON(benefitDiscordCreateMetadata) {
    return JSON.stringify(exports.BenefitDiscordCreateMetadata$outboundSchema.parse(benefitDiscordCreateMetadata));
}
function benefitDiscordCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordCreateMetadata' from JSON`);
}
/** @internal */
exports.BenefitDiscordCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("discord"),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDiscordCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    type: z.literal("discord"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordCreate$;
(function (BenefitDiscordCreate$) {
    /** @deprecated use `BenefitDiscordCreate$inboundSchema` instead. */
    BenefitDiscordCreate$.inboundSchema = exports.BenefitDiscordCreate$inboundSchema;
    /** @deprecated use `BenefitDiscordCreate$outboundSchema` instead. */
    BenefitDiscordCreate$.outboundSchema = exports.BenefitDiscordCreate$outboundSchema;
})(BenefitDiscordCreate$ || (exports.BenefitDiscordCreate$ = BenefitDiscordCreate$ = {}));
function benefitDiscordCreateToJSON(benefitDiscordCreate) {
    return JSON.stringify(exports.BenefitDiscordCreate$outboundSchema.parse(benefitDiscordCreate));
}
function benefitDiscordCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDiscordCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDiscordCreate' from JSON`);
}
