// @polar-sh/sdk/dist/commonjs/models/components/benefitcustom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCustom$ = exports.BenefitCustom$outboundSchema = exports.BenefitCustom$inboundSchema = exports.BenefitCustomMetadata$ = exports.BenefitCustomMetadata$outboundSchema = exports.BenefitCustomMetadata$inboundSchema = void 0;
exports.benefitCustomMetadataToJSON = benefitCustomMetadataToJSON;
exports.benefitCustomMetadataFromJSON = benefitCustomMetadataFromJSON;
exports.benefitCustomToJSON = benefitCustomToJSON;
exports.benefitCustomFromJSON = benefitCustomFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitcustomproperties_js_1 = require("./benefitcustomproperties.js");
/** @internal */
exports.BenefitCustomMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitCustomMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomMetadata$;
(function (BenefitCustomMetadata$) {
    /** @deprecated use `BenefitCustomMetadata$inboundSchema` instead. */
    BenefitCustomMetadata$.inboundSchema = exports.BenefitCustomMetadata$inboundSchema;
    /** @deprecated use `BenefitCustomMetadata$outboundSchema` instead. */
    BenefitCustomMetadata$.outboundSchema = exports.BenefitCustomMetadata$outboundSchema;
})(BenefitCustomMetadata$ || (exports.BenefitCustomMetadata$ = BenefitCustomMetadata$ = {}));
function benefitCustomMetadataToJSON(benefitCustomMetadata) {
    return JSON.stringify(exports.BenefitCustomMetadata$outboundSchema.parse(benefitCustomMetadata));
}
function benefitCustomMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitCustomMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomMetadata' from JSON`);
}
/** @internal */
exports.BenefitCustom$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("custom"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitcustomproperties_js_1.BenefitCustomProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitCustom$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("custom"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    properties: benefitcustomproperties_js_1.BenefitCustomProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustom$;
(function (BenefitCustom$) {
    /** @deprecated use `BenefitCustom$inboundSchema` instead. */
    BenefitCustom$.inboundSchema = exports.BenefitCustom$inboundSchema;
    /** @deprecated use `BenefitCustom$outboundSchema` instead. */
    BenefitCustom$.outboundSchema = exports.BenefitCustom$outboundSchema;
})(BenefitCustom$ || (exports.BenefitCustom$ = BenefitCustom$ = {}));
function benefitCustomToJSON(benefitCustom) {
    return JSON.stringify(exports.BenefitCustom$outboundSchema.parse(benefitCustom));
}
function benefitCustomFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitCustom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustom' from JSON`);
}
