// @polar-sh/sdk/dist/commonjs/models/components/benefitgithubrepositorysubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositorySubscriber$ = exports.BenefitGitHubRepositorySubscriber$outboundSchema = exports.BenefitGitHubRepositorySubscriber$inboundSchema = exports.BenefitGitHubRepositorySubscriberMetadata$ = exports.BenefitGitHubRepositorySubscriberMetadata$outboundSchema = exports.BenefitGitHubRepositorySubscriberMetadata$inboundSchema = void 0;
exports.benefitGitHubRepositorySubscriberMetadataToJSON = benefitGitHubRepositorySubscriberMetadataToJSON;
exports.benefitGitHubRepositorySubscriberMetadataFromJSON = benefitGitHubRepositorySubscriberMetadataFromJSON;
exports.benefitGitHubRepositorySubscriberToJSON = benefitGitHubRepositorySubscriberToJSON;
exports.benefitGitHubRepositorySubscriberFromJSON = benefitGitHubRepositorySubscriberFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitgithubrepositorysubscriberproperties_js_1 = require("./benefitgithubrepositorysubscriberproperties.js");
const organization_js_1 = require("./organization.js");
/** @internal */
exports.BenefitGitHubRepositorySubscriberMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.BenefitGitHubRepositorySubscriberMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositorySubscriberMetadata$;
(function (BenefitGitHubRepositorySubscriberMetadata$) {
    /** @deprecated use `BenefitGitHubRepositorySubscriberMetadata$inboundSchema` instead. */
    BenefitGitHubRepositorySubscriberMetadata$.inboundSchema = exports.BenefitGitHubRepositorySubscriberMetadata$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositorySubscriberMetadata$outboundSchema` instead. */
    BenefitGitHubRepositorySubscriberMetadata$.outboundSchema = exports.BenefitGitHubRepositorySubscriberMetadata$outboundSchema;
})(BenefitGitHubRepositorySubscriberMetadata$ || (exports.BenefitGitHubRepositorySubscriberMetadata$ = BenefitGitHubRepositorySubscriberMetadata$ = {}));
function benefitGitHubRepositorySubscriberMetadataToJSON(benefitGitHubRepositorySubscriberMetadata) {
    return JSON.stringify(exports.BenefitGitHubRepositorySubscriberMetadata$outboundSchema.parse(benefitGitHubRepositorySubscriberMetadata));
}
function benefitGitHubRepositorySubscriberMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositorySubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositorySubscriberMetadata' from JSON`);
}
/** @internal */
exports.BenefitGitHubRepositorySubscriber$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    type: z.literal("github_repository"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$inboundSchema,
    properties: benefitgithubrepositorysubscriberproperties_js_1.BenefitGitHubRepositorySubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitGitHubRepositorySubscriber$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    type: z.literal("github_repository"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    organization: organization_js_1.Organization$outboundSchema,
    properties: benefitgithubrepositorysubscriberproperties_js_1.BenefitGitHubRepositorySubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositorySubscriber$;
(function (BenefitGitHubRepositorySubscriber$) {
    /** @deprecated use `BenefitGitHubRepositorySubscriber$inboundSchema` instead. */
    BenefitGitHubRepositorySubscriber$.inboundSchema = exports.BenefitGitHubRepositorySubscriber$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositorySubscriber$outboundSchema` instead. */
    BenefitGitHubRepositorySubscriber$.outboundSchema = exports.BenefitGitHubRepositorySubscriber$outboundSchema;
})(BenefitGitHubRepositorySubscriber$ || (exports.BenefitGitHubRepositorySubscriber$ = BenefitGitHubRepositorySubscriber$ = {}));
function benefitGitHubRepositorySubscriberToJSON(benefitGitHubRepositorySubscriber) {
    return JSON.stringify(exports.BenefitGitHubRepositorySubscriber$outboundSchema.parse(benefitGitHubRepositorySubscriber));
}
function benefitGitHubRepositorySubscriberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGitHubRepositorySubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositorySubscriber' from JSON`);
}
