// @polar-sh/sdk/dist/commonjs/models/components/checkoutbillingaddressfields.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutBillingAddressFields$ = exports.CheckoutBillingAddressFields$outboundSchema = exports.CheckoutBillingAddressFields$inboundSchema = void 0;
exports.checkoutBillingAddressFieldsToJSON = checkoutBillingAddressFieldsToJSON;
exports.checkoutBillingAddressFieldsFromJSON = checkoutBillingAddressFieldsFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const billingaddressfieldmode_js_1 = require("./billingaddressfieldmode.js");
/** @internal */
exports.CheckoutBillingAddressFields$inboundSchema = z.object({
    country: billingaddressfieldmode_js_1.BillingAddressFieldMode$inboundSchema,
    state: billingaddressfieldmode_js_1.BillingAddressFieldMode$inboundSchema,
    city: billingaddressfieldmode_js_1.BillingAddressFieldMode$inboundSchema,
    postal_code: billingaddressfieldmode_js_1.BillingAddressFieldMode$inboundSchema,
    line1: billingaddressfieldmode_js_1.BillingAddressFieldMode$inboundSchema,
    line2: billingaddressfieldmode_js_1.BillingAddressFieldMode$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "postal_code": "postalCode",
    });
});
/** @internal */
exports.CheckoutBillingAddressFields$outboundSchema = z.object({
    country: billingaddressfieldmode_js_1.BillingAddressFieldMode$outboundSchema,
    state: billingaddressfieldmode_js_1.BillingAddressFieldMode$outboundSchema,
    city: billingaddressfieldmode_js_1.BillingAddressFieldMode$outboundSchema,
    postalCode: billingaddressfieldmode_js_1.BillingAddressFieldMode$outboundSchema,
    line1: billingaddressfieldmode_js_1.BillingAddressFieldMode$outboundSchema,
    line2: billingaddressfieldmode_js_1.BillingAddressFieldMode$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        postalCode: "postal_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutBillingAddressFields$;
(function (CheckoutBillingAddressFields$) {
    /** @deprecated use `CheckoutBillingAddressFields$inboundSchema` instead. */
    CheckoutBillingAddressFields$.inboundSchema = exports.CheckoutBillingAddressFields$inboundSchema;
    /** @deprecated use `CheckoutBillingAddressFields$outboundSchema` instead. */
    CheckoutBillingAddressFields$.outboundSchema = exports.CheckoutBillingAddressFields$outboundSchema;
})(CheckoutBillingAddressFields$ || (exports.CheckoutBillingAddressFields$ = CheckoutBillingAddressFields$ = {}));
function checkoutBillingAddressFieldsToJSON(checkoutBillingAddressFields) {
    return JSON.stringify(exports.CheckoutBillingAddressFields$outboundSchema.parse(checkoutBillingAddressFields));
}
function checkoutBillingAddressFieldsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutBillingAddressFields$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutBillingAddressFields' from JSON`);
}
