// @polar-sh/sdk/dist/commonjs/models/components/checkoutlinkcreateproductprice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkCreateProductPrice$ = exports.CheckoutLinkCreateProductPrice$outboundSchema = exports.CheckoutLinkCreateProductPrice$inboundSchema = exports.CheckoutLinkCreateProductPriceMetadata$ = exports.CheckoutLinkCreateProductPriceMetadata$outboundSchema = exports.CheckoutLinkCreateProductPriceMetadata$inboundSchema = void 0;
exports.checkoutLinkCreateProductPriceMetadataToJSON = checkoutLinkCreateProductPriceMetadataToJSON;
exports.checkoutLinkCreateProductPriceMetadataFromJSON = checkoutLinkCreateProductPriceMetadataFromJSON;
exports.checkoutLinkCreateProductPriceToJSON = checkoutLinkCreateProductPriceToJSON;
exports.checkoutLinkCreateProductPriceFromJSON = checkoutLinkCreateProductPriceFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CheckoutLinkCreateProductPriceMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CheckoutLinkCreateProductPriceMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkCreateProductPriceMetadata$;
(function (CheckoutLinkCreateProductPriceMetadata$) {
    /** @deprecated use `CheckoutLinkCreateProductPriceMetadata$inboundSchema` instead. */
    CheckoutLinkCreateProductPriceMetadata$.inboundSchema = exports.CheckoutLinkCreateProductPriceMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductPriceMetadata$outboundSchema` instead. */
    CheckoutLinkCreateProductPriceMetadata$.outboundSchema = exports.CheckoutLinkCreateProductPriceMetadata$outboundSchema;
})(CheckoutLinkCreateProductPriceMetadata$ || (exports.CheckoutLinkCreateProductPriceMetadata$ = CheckoutLinkCreateProductPriceMetadata$ = {}));
function checkoutLinkCreateProductPriceMetadataToJSON(checkoutLinkCreateProductPriceMetadata) {
    return JSON.stringify(exports.CheckoutLinkCreateProductPriceMetadata$outboundSchema.parse(checkoutLinkCreateProductPriceMetadata));
}
function checkoutLinkCreateProductPriceMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkCreateProductPriceMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductPriceMetadata' from JSON`);
}
/** @internal */
exports.CheckoutLinkCreateProductPrice$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    payment_processor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    require_billing_address: z.boolean().default(false),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    product_price_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "success_url": "successUrl",
        "product_price_id": "productPriceId",
    });
});
/** @internal */
exports.CheckoutLinkCreateProductPrice$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    paymentProcessor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    requireBillingAddress: z.boolean().default(false),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    productPriceId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
        productPriceId: "product_price_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkCreateProductPrice$;
(function (CheckoutLinkCreateProductPrice$) {
    /** @deprecated use `CheckoutLinkCreateProductPrice$inboundSchema` instead. */
    CheckoutLinkCreateProductPrice$.inboundSchema = exports.CheckoutLinkCreateProductPrice$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductPrice$outboundSchema` instead. */
    CheckoutLinkCreateProductPrice$.outboundSchema = exports.CheckoutLinkCreateProductPrice$outboundSchema;
})(CheckoutLinkCreateProductPrice$ || (exports.CheckoutLinkCreateProductPrice$ = CheckoutLinkCreateProductPrice$ = {}));
function checkoutLinkCreateProductPriceToJSON(checkoutLinkCreateProductPrice) {
    return JSON.stringify(exports.CheckoutLinkCreateProductPrice$outboundSchema.parse(checkoutLinkCreateProductPrice));
}
function checkoutLinkCreateProductPriceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkCreateProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductPrice' from JSON`);
}
