// @polar-sh/sdk/dist/commonjs/models/components/checkoutdiscountfixedonceforeverduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutDiscountFixedOnceForeverDuration$ = exports.CheckoutDiscountFixedOnceForeverDuration$outboundSchema = exports.CheckoutDiscountFixedOnceForeverDuration$inboundSchema = void 0;
exports.checkoutDiscountFixedOnceForeverDurationToJSON = checkoutDiscountFixedOnceForeverDurationToJSON;
exports.checkoutDiscountFixedOnceForeverDurationFromJSON = checkoutDiscountFixedOnceForeverDurationFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.CheckoutDiscountFixedOnceForeverDuration$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    type: discounttype_js_1.DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
});
/** @internal */
exports.CheckoutDiscountFixedOnceForeverDuration$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    type: discounttype_js_1.DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutDiscountFixedOnceForeverDuration$;
(function (CheckoutDiscountFixedOnceForeverDuration$) {
    /** @deprecated use `CheckoutDiscountFixedOnceForeverDuration$inboundSchema` instead. */
    CheckoutDiscountFixedOnceForeverDuration$.inboundSchema = exports.CheckoutDiscountFixedOnceForeverDuration$inboundSchema;
    /** @deprecated use `CheckoutDiscountFixedOnceForeverDuration$outboundSchema` instead. */
    CheckoutDiscountFixedOnceForeverDuration$.outboundSchema = exports.CheckoutDiscountFixedOnceForeverDuration$outboundSchema;
})(CheckoutDiscountFixedOnceForeverDuration$ || (exports.CheckoutDiscountFixedOnceForeverDuration$ = CheckoutDiscountFixedOnceForeverDuration$ = {}));
function checkoutDiscountFixedOnceForeverDurationToJSON(checkoutDiscountFixedOnceForeverDuration) {
    return JSON.stringify(exports.CheckoutDiscountFixedOnceForeverDuration$outboundSchema.parse(checkoutDiscountFixedOnceForeverDuration));
}
function checkoutDiscountFixedOnceForeverDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutDiscountFixedOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountFixedOnceForeverDuration' from JSON`);
}
