// @polar-sh/sdk/dist/commonjs/models/components/checkoutlinkcreateproducts.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkCreateProducts$ = exports.CheckoutLinkCreateProducts$outboundSchema = exports.CheckoutLinkCreateProducts$inboundSchema = exports.CheckoutLinkCreateProductsMetadata$ = exports.CheckoutLinkCreateProductsMetadata$outboundSchema = exports.CheckoutLinkCreateProductsMetadata$inboundSchema = void 0;
exports.checkoutLinkCreateProductsMetadataToJSON = checkoutLinkCreateProductsMetadataToJSON;
exports.checkoutLinkCreateProductsMetadataFromJSON = checkoutLinkCreateProductsMetadataFromJSON;
exports.checkoutLinkCreateProductsToJSON = checkoutLinkCreateProductsToJSON;
exports.checkoutLinkCreateProductsFromJSON = checkoutLinkCreateProductsFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CheckoutLinkCreateProductsMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CheckoutLinkCreateProductsMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkCreateProductsMetadata$;
(function (CheckoutLinkCreateProductsMetadata$) {
    /** @deprecated use `CheckoutLinkCreateProductsMetadata$inboundSchema` instead. */
    CheckoutLinkCreateProductsMetadata$.inboundSchema = exports.CheckoutLinkCreateProductsMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProductsMetadata$outboundSchema` instead. */
    CheckoutLinkCreateProductsMetadata$.outboundSchema = exports.CheckoutLinkCreateProductsMetadata$outboundSchema;
})(CheckoutLinkCreateProductsMetadata$ || (exports.CheckoutLinkCreateProductsMetadata$ = CheckoutLinkCreateProductsMetadata$ = {}));
function checkoutLinkCreateProductsMetadataToJSON(checkoutLinkCreateProductsMetadata) {
    return JSON.stringify(exports.CheckoutLinkCreateProductsMetadata$outboundSchema.parse(checkoutLinkCreateProductsMetadata));
}
function checkoutLinkCreateProductsMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkCreateProductsMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProductsMetadata' from JSON`);
}
/** @internal */
exports.CheckoutLinkCreateProducts$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    payment_processor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    require_billing_address: z.boolean().default(false),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    products: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "success_url": "successUrl",
    });
});
/** @internal */
exports.CheckoutLinkCreateProducts$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    paymentProcessor: z.literal("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    requireBillingAddress: z.boolean().default(false),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    products: z.array(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkCreateProducts$;
(function (CheckoutLinkCreateProducts$) {
    /** @deprecated use `CheckoutLinkCreateProducts$inboundSchema` instead. */
    CheckoutLinkCreateProducts$.inboundSchema = exports.CheckoutLinkCreateProducts$inboundSchema;
    /** @deprecated use `CheckoutLinkCreateProducts$outboundSchema` instead. */
    CheckoutLinkCreateProducts$.outboundSchema = exports.CheckoutLinkCreateProducts$outboundSchema;
})(CheckoutLinkCreateProducts$ || (exports.CheckoutLinkCreateProducts$ = CheckoutLinkCreateProducts$ = {}));
function checkoutLinkCreateProductsToJSON(checkoutLinkCreateProducts) {
    return JSON.stringify(exports.CheckoutLinkCreateProducts$outboundSchema.parse(checkoutLinkCreateProducts));
}
function checkoutLinkCreateProductsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkCreateProducts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkCreateProducts' from JSON`);
}
