// @polar-sh/sdk/dist/commonjs/models/components/checkoutlinkupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkUpdate$ = exports.CheckoutLinkUpdate$outboundSchema = exports.CheckoutLinkUpdate$inboundSchema = exports.CheckoutLinkUpdateMetadata$ = exports.CheckoutLinkUpdateMetadata$outboundSchema = exports.CheckoutLinkUpdateMetadata$inboundSchema = void 0;
exports.checkoutLinkUpdateMetadataToJSON = checkoutLinkUpdateMetadataToJSON;
exports.checkoutLinkUpdateMetadataFromJSON = checkoutLinkUpdateMetadataFromJSON;
exports.checkoutLinkUpdateToJSON = checkoutLinkUpdateToJSON;
exports.checkoutLinkUpdateFromJSON = checkoutLinkUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CheckoutLinkUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.CheckoutLinkUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkUpdateMetadata$;
(function (CheckoutLinkUpdateMetadata$) {
    /** @deprecated use `CheckoutLinkUpdateMetadata$inboundSchema` instead. */
    CheckoutLinkUpdateMetadata$.inboundSchema = exports.CheckoutLinkUpdateMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkUpdateMetadata$outboundSchema` instead. */
    CheckoutLinkUpdateMetadata$.outboundSchema = exports.CheckoutLinkUpdateMetadata$outboundSchema;
})(CheckoutLinkUpdateMetadata$ || (exports.CheckoutLinkUpdateMetadata$ = CheckoutLinkUpdateMetadata$ = {}));
function checkoutLinkUpdateMetadataToJSON(checkoutLinkUpdateMetadata) {
    return JSON.stringify(exports.CheckoutLinkUpdateMetadata$outboundSchema.parse(checkoutLinkUpdateMetadata));
}
function checkoutLinkUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkUpdateMetadata' from JSON`);
}
/** @internal */
exports.CheckoutLinkUpdate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.nullable(z.boolean()).optional(),
    require_billing_address: z.nullable(z.boolean()).optional(),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "success_url": "successUrl",
    });
});
/** @internal */
exports.CheckoutLinkUpdate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.nullable(z.boolean()).optional(),
    requireBillingAddress: z.nullable(z.boolean()).optional(),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkUpdate$;
(function (CheckoutLinkUpdate$) {
    /** @deprecated use `CheckoutLinkUpdate$inboundSchema` instead. */
    CheckoutLinkUpdate$.inboundSchema = exports.CheckoutLinkUpdate$inboundSchema;
    /** @deprecated use `CheckoutLinkUpdate$outboundSchema` instead. */
    CheckoutLinkUpdate$.outboundSchema = exports.CheckoutLinkUpdate$outboundSchema;
})(CheckoutLinkUpdate$ || (exports.CheckoutLinkUpdate$ = CheckoutLinkUpdate$ = {}));
function checkoutLinkUpdateToJSON(checkoutLinkUpdate) {
    return JSON.stringify(exports.CheckoutLinkUpdate$outboundSchema.parse(checkoutLinkUpdate));
}
function checkoutLinkUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkUpdate' from JSON`);
}
