// @polar-sh/sdk/dist/commonjs/models/components/customerbenefitgrantgithubrepositoryupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerBenefitGrantGitHubRepositoryUpdate$ = exports.CustomerBenefitGrantGitHubRepositoryUpdate$outboundSchema = exports.CustomerBenefitGrantGitHubRepositoryUpdate$inboundSchema = void 0;
exports.customerBenefitGrantGitHubRepositoryUpdateToJSON = customerBenefitGrantGitHubRepositoryUpdateToJSON;
exports.customerBenefitGrantGitHubRepositoryUpdateFromJSON = customerBenefitGrantGitHubRepositoryUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerbenefitgrantgithubrepositorypropertiesupdate_js_1 = require("./customerbenefitgrantgithubrepositorypropertiesupdate.js");
/** @internal */
exports.CustomerBenefitGrantGitHubRepositoryUpdate$inboundSchema = z.object({
    benefit_type: z.literal("github_repository"),
    properties: customerbenefitgrantgithubrepositorypropertiesupdate_js_1.CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "benefit_type": "benefitType",
    });
});
/** @internal */
exports.CustomerBenefitGrantGitHubRepositoryUpdate$outboundSchema = z.object({
    benefitType: z.literal("github_repository"),
    properties: customerbenefitgrantgithubrepositorypropertiesupdate_js_1.CustomerBenefitGrantGitHubRepositoryPropertiesUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        benefitType: "benefit_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerBenefitGrantGitHubRepositoryUpdate$;
(function (CustomerBenefitGrantGitHubRepositoryUpdate$) {
    /** @deprecated use `CustomerBenefitGrantGitHubRepositoryUpdate$inboundSchema` instead. */
    CustomerBenefitGrantGitHubRepositoryUpdate$.inboundSchema = exports.CustomerBenefitGrantGitHubRepositoryUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantGitHubRepositoryUpdate$outboundSchema` instead. */
    CustomerBenefitGrantGitHubRepositoryUpdate$.outboundSchema = exports.CustomerBenefitGrantGitHubRepositoryUpdate$outboundSchema;
})(CustomerBenefitGrantGitHubRepositoryUpdate$ || (exports.CustomerBenefitGrantGitHubRepositoryUpdate$ = CustomerBenefitGrantGitHubRepositoryUpdate$ = {}));
function customerBenefitGrantGitHubRepositoryUpdateToJSON(customerBenefitGrantGitHubRepositoryUpdate) {
    return JSON.stringify(exports.CustomerBenefitGrantGitHubRepositoryUpdate$outboundSchema.parse(customerBenefitGrantGitHubRepositoryUpdate));
}
function customerBenefitGrantGitHubRepositoryUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerBenefitGrantGitHubRepositoryUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantGitHubRepositoryUpdate' from JSON`);
}
