// @polar-sh/sdk/dist/commonjs/models/components/customerorderpaymentconfirmation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrderPaymentConfirmation$ = exports.CustomerOrderPaymentConfirmation$outboundSchema = exports.CustomerOrderPaymentConfirmation$inboundSchema = void 0;
exports.customerOrderPaymentConfirmationToJSON = customerOrderPaymentConfirmationToJSON;
exports.customerOrderPaymentConfirmationFromJSON = customerOrderPaymentConfirmationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerOrderPaymentConfirmation$inboundSchema = z.object({
    status: z.string(),
    client_secret: z.nullable(z.string()).optional(),
    error: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_secret": "clientSecret",
    });
});
/** @internal */
exports.CustomerOrderPaymentConfirmation$outboundSchema = z.object({
    status: z.string(),
    clientSecret: z.nullable(z.string()).optional(),
    error: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientSecret: "client_secret",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerOrderPaymentConfirmation$;
(function (CustomerOrderPaymentConfirmation$) {
    /** @deprecated use `CustomerOrderPaymentConfirmation$inboundSchema` instead. */
    CustomerOrderPaymentConfirmation$.inboundSchema = exports.CustomerOrderPaymentConfirmation$inboundSchema;
    /** @deprecated use `CustomerOrderPaymentConfirmation$outboundSchema` instead. */
    CustomerOrderPaymentConfirmation$.outboundSchema = exports.CustomerOrderPaymentConfirmation$outboundSchema;
})(CustomerOrderPaymentConfirmation$ || (exports.CustomerOrderPaymentConfirmation$ = CustomerOrderPaymentConfirmation$ = {}));
function customerOrderPaymentConfirmationToJSON(customerOrderPaymentConfirmation) {
    return JSON.stringify(exports.CustomerOrderPaymentConfirmation$outboundSchema.parse(customerOrderPaymentConfirmation));
}
function customerOrderPaymentConfirmationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerOrderPaymentConfirmation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrderPaymentConfirmation' from JSON`);
}
