// @polar-sh/sdk/dist/commonjs/models/components/customfieldupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldUpdate$ = exports.CustomFieldUpdate$outboundSchema = exports.CustomFieldUpdate$inboundSchema = void 0;
exports.customFieldUpdateToJSON = customFieldUpdateToJSON;
exports.customFieldUpdateFromJSON = customFieldUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldupdatecheckbox_js_1 = require("./customfieldupdatecheckbox.js");
const customfieldupdatedate_js_1 = require("./customfieldupdatedate.js");
const customfieldupdatenumber_js_1 = require("./customfieldupdatenumber.js");
const customfieldupdateselect_js_1 = require("./customfieldupdateselect.js");
const customfieldupdatetext_js_1 = require("./customfieldupdatetext.js");
/** @internal */
exports.CustomFieldUpdate$inboundSchema = z.union([
    customfieldupdatecheckbox_js_1.CustomFieldUpdateCheckbox$inboundSchema.and(z.object({ type: z.literal("checkbox") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldupdatedate_js_1.CustomFieldUpdateDate$inboundSchema.and(z.object({ type: z.literal("date") }).transform((v) => ({ type: v.type }))),
    customfieldupdatenumber_js_1.CustomFieldUpdateNumber$inboundSchema.and(z.object({ type: z.literal("number") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldupdateselect_js_1.CustomFieldUpdateSelect$inboundSchema.and(z.object({ type: z.literal("select") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldupdatetext_js_1.CustomFieldUpdateText$inboundSchema.and(z.object({ type: z.literal("text") }).transform((v) => ({ type: v.type }))),
]);
/** @internal */
exports.CustomFieldUpdate$outboundSchema = z.union([
    customfieldupdatecheckbox_js_1.CustomFieldUpdateCheckbox$outboundSchema.and(z.object({ type: z.literal("checkbox") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldupdatedate_js_1.CustomFieldUpdateDate$outboundSchema.and(z.object({ type: z.literal("date") }).transform((v) => ({ type: v.type }))),
    customfieldupdatenumber_js_1.CustomFieldUpdateNumber$outboundSchema.and(z.object({ type: z.literal("number") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldupdateselect_js_1.CustomFieldUpdateSelect$outboundSchema.and(z.object({ type: z.literal("select") }).transform((v) => ({
        type: v.type,
    }))),
    customfieldupdatetext_js_1.CustomFieldUpdateText$outboundSchema.and(z.object({ type: z.literal("text") }).transform((v) => ({ type: v.type }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldUpdate$;
(function (CustomFieldUpdate$) {
    /** @deprecated use `CustomFieldUpdate$inboundSchema` instead. */
    CustomFieldUpdate$.inboundSchema = exports.CustomFieldUpdate$inboundSchema;
    /** @deprecated use `CustomFieldUpdate$outboundSchema` instead. */
    CustomFieldUpdate$.outboundSchema = exports.CustomFieldUpdate$outboundSchema;
})(CustomFieldUpdate$ || (exports.CustomFieldUpdate$ = CustomFieldUpdate$ = {}));
function customFieldUpdateToJSON(customFieldUpdate) {
    return JSON.stringify(exports.CustomFieldUpdate$outboundSchema.parse(customFieldUpdate));
}
function customFieldUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdate' from JSON`);
}
