// @polar-sh/sdk/dist/commonjs/models/components/customerbenefitgrantupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerBenefitGrantUpdate$ = exports.CustomerBenefitGrantUpdate$outboundSchema = exports.CustomerBenefitGrantUpdate$inboundSchema = void 0;
exports.customerBenefitGrantUpdateToJSON = customerBenefitGrantUpdateToJSON;
exports.customerBenefitGrantUpdateFromJSON = customerBenefitGrantUpdateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customerbenefitgrantcustomupdate_js_1 = require("./customerbenefitgrantcustomupdate.js");
const customerbenefitgrantdiscordupdate_js_1 = require("./customerbenefitgrantdiscordupdate.js");
const customerbenefitgrantdownloadablesupdate_js_1 = require("./customerbenefitgrantdownloadablesupdate.js");
const customerbenefitgrantgithubrepositoryupdate_js_1 = require("./customerbenefitgrantgithubrepositoryupdate.js");
const customerbenefitgrantlicensekeysupdate_js_1 = require("./customerbenefitgrantlicensekeysupdate.js");
const customerbenefitgrantmetercreditupdate_js_1 = require("./customerbenefitgrantmetercreditupdate.js");
/** @internal */
exports.CustomerBenefitGrantUpdate$inboundSchema = z.union([
    customerbenefitgrantdiscordupdate_js_1.CustomerBenefitGrantDiscordUpdate$inboundSchema.and(z.object({ benefit_type: z.literal("discord") }).transform((v) => ({
        benefitType: v.benefit_type,
    }))),
    customerbenefitgrantgithubrepositoryupdate_js_1.CustomerBenefitGrantGitHubRepositoryUpdate$inboundSchema.and(z.object({ benefit_type: z.literal("github_repository") }).transform((v) => ({ benefitType: v.benefit_type }))),
    customerbenefitgrantcustomupdate_js_1.CustomerBenefitGrantCustomUpdate$inboundSchema.and(z.object({ benefit_type: z.literal("custom") }).transform((v) => ({
        benefitType: v.benefit_type,
    }))),
    customerbenefitgrantdownloadablesupdate_js_1.CustomerBenefitGrantDownloadablesUpdate$inboundSchema.and(z.object({ benefit_type: z.literal("downloadables") }).transform((v) => ({
        benefitType: v.benefit_type,
    }))),
    customerbenefitgrantlicensekeysupdate_js_1.CustomerBenefitGrantLicenseKeysUpdate$inboundSchema.and(z.object({ benefit_type: z.literal("license_keys") }).transform((v) => ({
        benefitType: v.benefit_type,
    }))),
    customerbenefitgrantmetercreditupdate_js_1.CustomerBenefitGrantMeterCreditUpdate$inboundSchema.and(z.object({ benefit_type: z.literal("meter_credit") }).transform((v) => ({
        benefitType: v.benefit_type,
    }))),
]);
/** @internal */
exports.CustomerBenefitGrantUpdate$outboundSchema = z.union([
    customerbenefitgrantdiscordupdate_js_1.CustomerBenefitGrantDiscordUpdate$outboundSchema.and(z.object({ benefitType: z.literal("discord") }).transform((v) => ({
        benefit_type: v.benefitType,
    }))),
    customerbenefitgrantgithubrepositoryupdate_js_1.CustomerBenefitGrantGitHubRepositoryUpdate$outboundSchema.and(z.object({ benefitType: z.literal("github_repository") }).transform((v) => ({ benefit_type: v.benefitType }))),
    customerbenefitgrantcustomupdate_js_1.CustomerBenefitGrantCustomUpdate$outboundSchema.and(z.object({ benefitType: z.literal("custom") }).transform((v) => ({
        benefit_type: v.benefitType,
    }))),
    customerbenefitgrantdownloadablesupdate_js_1.CustomerBenefitGrantDownloadablesUpdate$outboundSchema.and(z.object({ benefitType: z.literal("downloadables") }).transform((v) => ({
        benefit_type: v.benefitType,
    }))),
    customerbenefitgrantlicensekeysupdate_js_1.CustomerBenefitGrantLicenseKeysUpdate$outboundSchema.and(z.object({ benefitType: z.literal("license_keys") }).transform((v) => ({
        benefit_type: v.benefitType,
    }))),
    customerbenefitgrantmetercreditupdate_js_1.CustomerBenefitGrantMeterCreditUpdate$outboundSchema.and(z.object({ benefitType: z.literal("meter_credit") }).transform((v) => ({
        benefit_type: v.benefitType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerBenefitGrantUpdate$;
(function (CustomerBenefitGrantUpdate$) {
    /** @deprecated use `CustomerBenefitGrantUpdate$inboundSchema` instead. */
    CustomerBenefitGrantUpdate$.inboundSchema = exports.CustomerBenefitGrantUpdate$inboundSchema;
    /** @deprecated use `CustomerBenefitGrantUpdate$outboundSchema` instead. */
    CustomerBenefitGrantUpdate$.outboundSchema = exports.CustomerBenefitGrantUpdate$outboundSchema;
})(CustomerBenefitGrantUpdate$ || (exports.CustomerBenefitGrantUpdate$ = CustomerBenefitGrantUpdate$ = {}));
function customerBenefitGrantUpdateToJSON(customerBenefitGrantUpdate) {
    return JSON.stringify(exports.CustomerBenefitGrantUpdate$outboundSchema.parse(customerBenefitGrantUpdate));
}
function customerBenefitGrantUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerBenefitGrantUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerBenefitGrantUpdate' from JSON`);
}
