// @polar-sh/sdk/dist/commonjs/models/components/customersubscriptioncancel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSubscriptionCancel$ = exports.CustomerSubscriptionCancel$outboundSchema = exports.CustomerSubscriptionCancel$inboundSchema = void 0;
exports.customerSubscriptionCancelToJSON = customerSubscriptionCancelToJSON;
exports.customerSubscriptionCancelFromJSON = customerSubscriptionCancelFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customercancellationreason_js_1 = require("./customercancellationreason.js");
/** @internal */
exports.CustomerSubscriptionCancel$inboundSchema = z.object({
    cancel_at_period_end: z.nullable(z.boolean()).optional(),
    cancellation_reason: z.nullable(customercancellationreason_js_1.CustomerCancellationReason$inboundSchema)
        .optional(),
    cancellation_comment: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "cancel_at_period_end": "cancelAtPeriodEnd",
        "cancellation_reason": "cancellationReason",
        "cancellation_comment": "cancellationComment",
    });
});
/** @internal */
exports.CustomerSubscriptionCancel$outboundSchema = z.object({
    cancelAtPeriodEnd: z.nullable(z.boolean()).optional(),
    cancellationReason: z.nullable(customercancellationreason_js_1.CustomerCancellationReason$outboundSchema)
        .optional(),
    cancellationComment: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        cancelAtPeriodEnd: "cancel_at_period_end",
        cancellationReason: "cancellation_reason",
        cancellationComment: "cancellation_comment",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerSubscriptionCancel$;
(function (CustomerSubscriptionCancel$) {
    /** @deprecated use `CustomerSubscriptionCancel$inboundSchema` instead. */
    CustomerSubscriptionCancel$.inboundSchema = exports.CustomerSubscriptionCancel$inboundSchema;
    /** @deprecated use `CustomerSubscriptionCancel$outboundSchema` instead. */
    CustomerSubscriptionCancel$.outboundSchema = exports.CustomerSubscriptionCancel$outboundSchema;
})(CustomerSubscriptionCancel$ || (exports.CustomerSubscriptionCancel$ = CustomerSubscriptionCancel$ = {}));
function customerSubscriptionCancelToJSON(customerSubscriptionCancel) {
    return JSON.stringify(exports.CustomerSubscriptionCancel$outboundSchema.parse(customerSubscriptionCancel));
}
function customerSubscriptionCancelFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerSubscriptionCancel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionCancel' from JSON`);
}
