// @polar-sh/sdk/dist/commonjs/models/components/customfieldcheckboxproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldCheckboxProperties$ = exports.CustomFieldCheckboxProperties$outboundSchema = exports.CustomFieldCheckboxProperties$inboundSchema = void 0;
exports.customFieldCheckboxPropertiesToJSON = customFieldCheckboxPropertiesToJSON;
exports.customFieldCheckboxPropertiesFromJSON = customFieldCheckboxPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomFieldCheckboxProperties$inboundSchema = z.object({
    form_label: z.string().optional(),
    form_help_text: z.string().optional(),
    form_placeholder: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
});
/** @internal */
exports.CustomFieldCheckboxProperties$outboundSchema = z.object({
    formLabel: z.string().optional(),
    formHelpText: z.string().optional(),
    formPlaceholder: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldCheckboxProperties$;
(function (CustomFieldCheckboxProperties$) {
    /** @deprecated use `CustomFieldCheckboxProperties$inboundSchema` instead. */
    CustomFieldCheckboxProperties$.inboundSchema = exports.CustomFieldCheckboxProperties$inboundSchema;
    /** @deprecated use `CustomFieldCheckboxProperties$outboundSchema` instead. */
    CustomFieldCheckboxProperties$.outboundSchema = exports.CustomFieldCheckboxProperties$outboundSchema;
})(CustomFieldCheckboxProperties$ || (exports.CustomFieldCheckboxProperties$ = CustomFieldCheckboxProperties$ = {}));
function customFieldCheckboxPropertiesToJSON(customFieldCheckboxProperties) {
    return JSON.stringify(exports.CustomFieldCheckboxProperties$outboundSchema.parse(customFieldCheckboxProperties));
}
function customFieldCheckboxPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldCheckboxProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCheckboxProperties' from JSON`);
}
