// @polar-sh/sdk/dist/commonjs/models/components/customersubscriptionsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSubscriptionSortProperty$ = exports.CustomerSubscriptionSortProperty$outboundSchema = exports.CustomerSubscriptionSortProperty$inboundSchema = exports.CustomerSubscriptionSortProperty = void 0;
const z = __importStar(require("zod"));
exports.CustomerSubscriptionSortProperty = {
    StartedAt: "started_at",
    MinusStartedAt: "-started_at",
    Amount: "amount",
    MinusAmount: "-amount",
    Status: "status",
    MinusStatus: "-status",
    Organization: "organization",
    MinusOrganization: "-organization",
    Product: "product",
    MinusProduct: "-product",
};
/** @internal */
exports.CustomerSubscriptionSortProperty$inboundSchema = z.nativeEnum(exports.CustomerSubscriptionSortProperty);
/** @internal */
exports.CustomerSubscriptionSortProperty$outboundSchema = exports.CustomerSubscriptionSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerSubscriptionSortProperty$;
(function (CustomerSubscriptionSortProperty$) {
    /** @deprecated use `CustomerSubscriptionSortProperty$inboundSchema` instead. */
    CustomerSubscriptionSortProperty$.inboundSchema = exports.CustomerSubscriptionSortProperty$inboundSchema;
    /** @deprecated use `CustomerSubscriptionSortProperty$outboundSchema` instead. */
    CustomerSubscriptionSortProperty$.outboundSchema = exports.CustomerSubscriptionSortProperty$outboundSchema;
})(CustomerSubscriptionSortProperty$ || (exports.CustomerSubscriptionSortProperty$ = CustomerSubscriptionSortProperty$ = {}));
