// @polar-sh/sdk/dist/commonjs/models/components/customermeter.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerMeter$ = exports.CustomerMeter$outboundSchema = exports.CustomerMeter$inboundSchema = void 0;
exports.customerMeterToJSON = customerMeterToJSON;
exports.customerMeterFromJSON = customerMeterFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customer_js_1 = require("./customer.js");
const meter_js_1 = require("./meter.js");
/** @internal */
exports.CustomerMeter$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    meter_id: z.string(),
    consumed_units: z.number(),
    credited_units: z.number().int(),
    balance: z.number(),
    customer: customer_js_1.Customer$inboundSchema,
    meter: meter_js_1.Meter$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "customer_id": "customerId",
        "meter_id": "meterId",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
    });
});
/** @internal */
exports.CustomerMeter$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    meterId: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.number().int(),
    balance: z.number(),
    customer: customer_js_1.Customer$outboundSchema,
    meter: meter_js_1.Meter$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customerId: "customer_id",
        meterId: "meter_id",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerMeter$;
(function (CustomerMeter$) {
    /** @deprecated use `CustomerMeter$inboundSchema` instead. */
    CustomerMeter$.inboundSchema = exports.CustomerMeter$inboundSchema;
    /** @deprecated use `CustomerMeter$outboundSchema` instead. */
    CustomerMeter$.outboundSchema = exports.CustomerMeter$outboundSchema;
})(CustomerMeter$ || (exports.CustomerMeter$ = CustomerMeter$ = {}));
function customerMeterToJSON(customerMeter) {
    return JSON.stringify(exports.CustomerMeter$outboundSchema.parse(customerMeter));
}
function customerMeterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerMeter' from JSON`);
}
