// @polar-sh/sdk/dist/commonjs/models/components/discountpercentageonceforeverduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountPercentageOnceForeverDuration$ = exports.DiscountPercentageOnceForeverDuration$outboundSchema = exports.DiscountPercentageOnceForeverDuration$inboundSchema = exports.DiscountPercentageOnceForeverDurationMetadata$ = exports.DiscountPercentageOnceForeverDurationMetadata$outboundSchema = exports.DiscountPercentageOnceForeverDurationMetadata$inboundSchema = void 0;
exports.discountPercentageOnceForeverDurationMetadataToJSON = discountPercentageOnceForeverDurationMetadataToJSON;
exports.discountPercentageOnceForeverDurationMetadataFromJSON = discountPercentageOnceForeverDurationMetadataFromJSON;
exports.discountPercentageOnceForeverDurationToJSON = discountPercentageOnceForeverDurationToJSON;
exports.discountPercentageOnceForeverDurationFromJSON = discountPercentageOnceForeverDurationFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discountproduct_js_1 = require("./discountproduct.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountPercentageOnceForeverDurationMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountPercentageOnceForeverDurationMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageOnceForeverDurationMetadata$;
(function (DiscountPercentageOnceForeverDurationMetadata$) {
    /** @deprecated use `DiscountPercentageOnceForeverDurationMetadata$inboundSchema` instead. */
    DiscountPercentageOnceForeverDurationMetadata$.inboundSchema = exports.DiscountPercentageOnceForeverDurationMetadata$inboundSchema;
    /** @deprecated use `DiscountPercentageOnceForeverDurationMetadata$outboundSchema` instead. */
    DiscountPercentageOnceForeverDurationMetadata$.outboundSchema = exports.DiscountPercentageOnceForeverDurationMetadata$outboundSchema;
})(DiscountPercentageOnceForeverDurationMetadata$ || (exports.DiscountPercentageOnceForeverDurationMetadata$ = DiscountPercentageOnceForeverDurationMetadata$ = {}));
function discountPercentageOnceForeverDurationMetadataToJSON(discountPercentageOnceForeverDurationMetadata) {
    return JSON.stringify(exports.DiscountPercentageOnceForeverDurationMetadata$outboundSchema.parse(discountPercentageOnceForeverDurationMetadata));
}
function discountPercentageOnceForeverDurationMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageOnceForeverDurationMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDurationMetadata' from JSON`);
}
/** @internal */
exports.DiscountPercentageOnceForeverDuration$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    type: discounttype_js_1.DiscountType$inboundSchema,
    basis_points: z.number().int(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "basis_points": "basisPoints",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountPercentageOnceForeverDuration$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    type: discounttype_js_1.DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        basisPoints: "basis_points",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageOnceForeverDuration$;
(function (DiscountPercentageOnceForeverDuration$) {
    /** @deprecated use `DiscountPercentageOnceForeverDuration$inboundSchema` instead. */
    DiscountPercentageOnceForeverDuration$.inboundSchema = exports.DiscountPercentageOnceForeverDuration$inboundSchema;
    /** @deprecated use `DiscountPercentageOnceForeverDuration$outboundSchema` instead. */
    DiscountPercentageOnceForeverDuration$.outboundSchema = exports.DiscountPercentageOnceForeverDuration$outboundSchema;
})(DiscountPercentageOnceForeverDuration$ || (exports.DiscountPercentageOnceForeverDuration$ = DiscountPercentageOnceForeverDuration$ = {}));
function discountPercentageOnceForeverDurationToJSON(discountPercentageOnceForeverDuration) {
    return JSON.stringify(exports.DiscountPercentageOnceForeverDuration$outboundSchema.parse(discountPercentageOnceForeverDuration));
}
function discountPercentageOnceForeverDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDuration' from JSON`);
}
