// @polar-sh/sdk/dist/commonjs/models/components/discountpercentageonceforeverdurationbase.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountPercentageOnceForeverDurationBase$ = exports.DiscountPercentageOnceForeverDurationBase$outboundSchema = exports.DiscountPercentageOnceForeverDurationBase$inboundSchema = exports.DiscountPercentageOnceForeverDurationBaseMetadata$ = exports.DiscountPercentageOnceForeverDurationBaseMetadata$outboundSchema = exports.DiscountPercentageOnceForeverDurationBaseMetadata$inboundSchema = void 0;
exports.discountPercentageOnceForeverDurationBaseMetadataToJSON = discountPercentageOnceForeverDurationBaseMetadataToJSON;
exports.discountPercentageOnceForeverDurationBaseMetadataFromJSON = discountPercentageOnceForeverDurationBaseMetadataFromJSON;
exports.discountPercentageOnceForeverDurationBaseToJSON = discountPercentageOnceForeverDurationBaseToJSON;
exports.discountPercentageOnceForeverDurationBaseFromJSON = discountPercentageOnceForeverDurationBaseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountPercentageOnceForeverDurationBaseMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountPercentageOnceForeverDurationBaseMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageOnceForeverDurationBaseMetadata$;
(function (DiscountPercentageOnceForeverDurationBaseMetadata$) {
    /** @deprecated use `DiscountPercentageOnceForeverDurationBaseMetadata$inboundSchema` instead. */
    DiscountPercentageOnceForeverDurationBaseMetadata$.inboundSchema = exports.DiscountPercentageOnceForeverDurationBaseMetadata$inboundSchema;
    /** @deprecated use `DiscountPercentageOnceForeverDurationBaseMetadata$outboundSchema` instead. */
    DiscountPercentageOnceForeverDurationBaseMetadata$.outboundSchema = exports.DiscountPercentageOnceForeverDurationBaseMetadata$outboundSchema;
})(DiscountPercentageOnceForeverDurationBaseMetadata$ || (exports.DiscountPercentageOnceForeverDurationBaseMetadata$ = DiscountPercentageOnceForeverDurationBaseMetadata$ = {}));
function discountPercentageOnceForeverDurationBaseMetadataToJSON(discountPercentageOnceForeverDurationBaseMetadata) {
    return JSON.stringify(exports.DiscountPercentageOnceForeverDurationBaseMetadata$outboundSchema.parse(discountPercentageOnceForeverDurationBaseMetadata));
}
function discountPercentageOnceForeverDurationBaseMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageOnceForeverDurationBaseMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDurationBaseMetadata' from JSON`);
}
/** @internal */
exports.DiscountPercentageOnceForeverDurationBase$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    type: discounttype_js_1.DiscountType$inboundSchema,
    basis_points: z.number().int(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    max_redemptions: z.nullable(z.number().int()),
    redemptions_count: z.number().int(),
    organization_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "basis_points": "basisPoints",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountPercentageOnceForeverDurationBase$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    type: discounttype_js_1.DiscountType$outboundSchema,
    basisPoints: z.number().int(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    maxRedemptions: z.nullable(z.number().int()),
    redemptionsCount: z.number().int(),
    organizationId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        basisPoints: "basis_points",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountPercentageOnceForeverDurationBase$;
(function (DiscountPercentageOnceForeverDurationBase$) {
    /** @deprecated use `DiscountPercentageOnceForeverDurationBase$inboundSchema` instead. */
    DiscountPercentageOnceForeverDurationBase$.inboundSchema = exports.DiscountPercentageOnceForeverDurationBase$inboundSchema;
    /** @deprecated use `DiscountPercentageOnceForeverDurationBase$outboundSchema` instead. */
    DiscountPercentageOnceForeverDurationBase$.outboundSchema = exports.DiscountPercentageOnceForeverDurationBase$outboundSchema;
})(DiscountPercentageOnceForeverDurationBase$ || (exports.DiscountPercentageOnceForeverDurationBase$ = DiscountPercentageOnceForeverDurationBase$ = {}));
function discountPercentageOnceForeverDurationBaseToJSON(discountPercentageOnceForeverDurationBase) {
    return JSON.stringify(exports.DiscountPercentageOnceForeverDurationBase$outboundSchema.parse(discountPercentageOnceForeverDurationBase));
}
function discountPercentageOnceForeverDurationBaseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageOnceForeverDurationBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDurationBase' from JSON`);
}
