// @polar-sh/sdk/dist/commonjs/models/components/discountfixedrepeatdurationcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountFixedRepeatDurationCreate$ = exports.DiscountFixedRepeatDurationCreate$outboundSchema = exports.DiscountFixedRepeatDurationCreate$inboundSchema = exports.DiscountFixedRepeatDurationCreateMetadata$ = exports.DiscountFixedRepeatDurationCreateMetadata$outboundSchema = exports.DiscountFixedRepeatDurationCreateMetadata$inboundSchema = void 0;
exports.discountFixedRepeatDurationCreateMetadataToJSON = discountFixedRepeatDurationCreateMetadataToJSON;
exports.discountFixedRepeatDurationCreateMetadataFromJSON = discountFixedRepeatDurationCreateMetadataFromJSON;
exports.discountFixedRepeatDurationCreateToJSON = discountFixedRepeatDurationCreateToJSON;
exports.discountFixedRepeatDurationCreateFromJSON = discountFixedRepeatDurationCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.DiscountFixedRepeatDurationCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.DiscountFixedRepeatDurationCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountFixedRepeatDurationCreateMetadata$;
(function (DiscountFixedRepeatDurationCreateMetadata$) {
    /** @deprecated use `DiscountFixedRepeatDurationCreateMetadata$inboundSchema` instead. */
    DiscountFixedRepeatDurationCreateMetadata$.inboundSchema = exports.DiscountFixedRepeatDurationCreateMetadata$inboundSchema;
    /** @deprecated use `DiscountFixedRepeatDurationCreateMetadata$outboundSchema` instead. */
    DiscountFixedRepeatDurationCreateMetadata$.outboundSchema = exports.DiscountFixedRepeatDurationCreateMetadata$outboundSchema;
})(DiscountFixedRepeatDurationCreateMetadata$ || (exports.DiscountFixedRepeatDurationCreateMetadata$ = DiscountFixedRepeatDurationCreateMetadata$ = {}));
function discountFixedRepeatDurationCreateMetadataToJSON(discountFixedRepeatDurationCreateMetadata) {
    return JSON.stringify(exports.DiscountFixedRepeatDurationCreateMetadata$outboundSchema.parse(discountFixedRepeatDurationCreateMetadata));
}
function discountFixedRepeatDurationCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountFixedRepeatDurationCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedRepeatDurationCreateMetadata' from JSON`);
}
/** @internal */
exports.DiscountFixedRepeatDurationCreate$inboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    duration_in_months: z.number().int(),
    type: discounttype_js_1.DiscountType$inboundSchema,
    amount: z.number().int(),
    currency: z.string().default("usd"),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    code: z.nullable(z.string()).optional(),
    starts_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    max_redemptions: z.nullable(z.number().int()).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "duration_in_months": "durationInMonths",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountFixedRepeatDurationCreate$outboundSchema = z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    durationInMonths: z.number().int(),
    type: discounttype_js_1.DiscountType$outboundSchema,
    amount: z.number().int(),
    currency: z.string().default("usd"),
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    code: z.nullable(z.string()).optional(),
    startsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    maxRedemptions: z.nullable(z.number().int()).optional(),
    products: z.nullable(z.array(z.string())).optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        durationInMonths: "duration_in_months",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountFixedRepeatDurationCreate$;
(function (DiscountFixedRepeatDurationCreate$) {
    /** @deprecated use `DiscountFixedRepeatDurationCreate$inboundSchema` instead. */
    DiscountFixedRepeatDurationCreate$.inboundSchema = exports.DiscountFixedRepeatDurationCreate$inboundSchema;
    /** @deprecated use `DiscountFixedRepeatDurationCreate$outboundSchema` instead. */
    DiscountFixedRepeatDurationCreate$.outboundSchema = exports.DiscountFixedRepeatDurationCreate$outboundSchema;
})(DiscountFixedRepeatDurationCreate$ || (exports.DiscountFixedRepeatDurationCreate$ = DiscountFixedRepeatDurationCreate$ = {}));
function discountFixedRepeatDurationCreateToJSON(discountFixedRepeatDurationCreate) {
    return JSON.stringify(exports.DiscountFixedRepeatDurationCreate$outboundSchema.parse(discountFixedRepeatDurationCreate));
}
function discountFixedRepeatDurationCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountFixedRepeatDurationCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountFixedRepeatDurationCreate' from JSON`);
}
