// @polar-sh/sdk/dist/commonjs/models/components/eventcreatecustomer.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventCreateCustomer$ = exports.EventCreateCustomer$outboundSchema = exports.EventCreateCustomer$inboundSchema = exports.EventCreateCustomerMetadata$ = exports.EventCreateCustomerMetadata$outboundSchema = exports.EventCreateCustomerMetadata$inboundSchema = void 0;
exports.eventCreateCustomerMetadataToJSON = eventCreateCustomerMetadataToJSON;
exports.eventCreateCustomerMetadataFromJSON = eventCreateCustomerMetadataFromJSON;
exports.eventCreateCustomerToJSON = eventCreateCustomerToJSON;
exports.eventCreateCustomerFromJSON = eventCreateCustomerFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.EventCreateCustomerMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.EventCreateCustomerMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventCreateCustomerMetadata$;
(function (EventCreateCustomerMetadata$) {
    /** @deprecated use `EventCreateCustomerMetadata$inboundSchema` instead. */
    EventCreateCustomerMetadata$.inboundSchema = exports.EventCreateCustomerMetadata$inboundSchema;
    /** @deprecated use `EventCreateCustomerMetadata$outboundSchema` instead. */
    EventCreateCustomerMetadata$.outboundSchema = exports.EventCreateCustomerMetadata$outboundSchema;
})(EventCreateCustomerMetadata$ || (exports.EventCreateCustomerMetadata$ = EventCreateCustomerMetadata$ = {}));
function eventCreateCustomerMetadataToJSON(eventCreateCustomerMetadata) {
    return JSON.stringify(exports.EventCreateCustomerMetadata$outboundSchema.parse(eventCreateCustomerMetadata));
}
function eventCreateCustomerMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventCreateCustomerMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventCreateCustomerMetadata' from JSON`);
}
/** @internal */
exports.EventCreateCustomer$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    name: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    customer_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
    });
});
/** @internal */
exports.EventCreateCustomer$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    timestamp: z.date().transform(v => v.toISOString()).optional(),
    name: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    customerId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventCreateCustomer$;
(function (EventCreateCustomer$) {
    /** @deprecated use `EventCreateCustomer$inboundSchema` instead. */
    EventCreateCustomer$.inboundSchema = exports.EventCreateCustomer$inboundSchema;
    /** @deprecated use `EventCreateCustomer$outboundSchema` instead. */
    EventCreateCustomer$.outboundSchema = exports.EventCreateCustomer$outboundSchema;
})(EventCreateCustomer$ || (exports.EventCreateCustomer$ = EventCreateCustomer$ = {}));
function eventCreateCustomerToJSON(eventCreateCustomer) {
    return JSON.stringify(exports.EventCreateCustomer$outboundSchema.parse(eventCreateCustomer));
}
function eventCreateCustomerFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventCreateCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventCreateCustomer' from JSON`);
}
