// @polar-sh/sdk/dist/commonjs/models/components/fileuploadcompleted.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileUploadCompleted$ = exports.FileUploadCompleted$outboundSchema = exports.FileUploadCompleted$inboundSchema = void 0;
exports.fileUploadCompletedToJSON = fileUploadCompletedToJSON;
exports.fileUploadCompletedFromJSON = fileUploadCompletedFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const s3fileuploadcompletedpart_js_1 = require("./s3fileuploadcompletedpart.js");
/** @internal */
exports.FileUploadCompleted$inboundSchema = z.object({
    id: z.string(),
    path: z.string(),
    parts: z.array(s3fileuploadcompletedpart_js_1.S3FileUploadCompletedPart$inboundSchema),
});
/** @internal */
exports.FileUploadCompleted$outboundSchema = z.object({
    id: z.string(),
    path: z.string(),
    parts: z.array(s3fileuploadcompletedpart_js_1.S3FileUploadCompletedPart$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileUploadCompleted$;
(function (FileUploadCompleted$) {
    /** @deprecated use `FileUploadCompleted$inboundSchema` instead. */
    FileUploadCompleted$.inboundSchema = exports.FileUploadCompleted$inboundSchema;
    /** @deprecated use `FileUploadCompleted$outboundSchema` instead. */
    FileUploadCompleted$.outboundSchema = exports.FileUploadCompleted$outboundSchema;
})(FileUploadCompleted$ || (exports.FileUploadCompleted$ = FileUploadCompleted$ = {}));
function fileUploadCompletedToJSON(fileUploadCompleted) {
    return JSON.stringify(exports.FileUploadCompleted$outboundSchema.parse(fileUploadCompleted));
}
function fileUploadCompletedFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FileUploadCompleted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileUploadCompleted' from JSON`);
}
