// @polar-sh/sdk/dist/commonjs/models/components/filter.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Clauses$ = exports.Clauses$outboundSchema = exports.Clauses$inboundSchema = exports.Filter$ = exports.Filter$outboundSchema = exports.Filter$inboundSchema = void 0;
exports.filterToJSON = filterToJSON;
exports.filterFromJSON = filterFromJSON;
exports.clausesToJSON = clausesToJSON;
exports.clausesFromJSON = clausesFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const filterclause_js_1 = require("./filterclause.js");
const filterconjunction_js_1 = require("./filterconjunction.js");
/** @internal */
exports.Filter$inboundSchema = z
    .object({
    conjunction: filterconjunction_js_1.FilterConjunction$inboundSchema,
    clauses: z.array(z.union([filterclause_js_1.FilterClause$inboundSchema, z.lazy(() => exports.Filter$inboundSchema)])),
});
/** @internal */
exports.Filter$outboundSchema = z.object({
    conjunction: filterconjunction_js_1.FilterConjunction$outboundSchema,
    clauses: z.array(z.union([filterclause_js_1.FilterClause$outboundSchema, z.lazy(() => exports.Filter$outboundSchema)])),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Filter$;
(function (Filter$) {
    /** @deprecated use `Filter$inboundSchema` instead. */
    Filter$.inboundSchema = exports.Filter$inboundSchema;
    /** @deprecated use `Filter$outboundSchema` instead. */
    Filter$.outboundSchema = exports.Filter$outboundSchema;
})(Filter$ || (exports.Filter$ = Filter$ = {}));
function filterToJSON(filter) {
    return JSON.stringify(exports.Filter$outboundSchema.parse(filter));
}
function filterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Filter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Filter' from JSON`);
}
/** @internal */
exports.Clauses$inboundSchema = z.union([filterclause_js_1.FilterClause$inboundSchema, z.lazy(() => exports.Filter$inboundSchema)]);
/** @internal */
exports.Clauses$outboundSchema = z.union([filterclause_js_1.FilterClause$outboundSchema, z.lazy(() => exports.Filter$outboundSchema)]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Clauses$;
(function (Clauses$) {
    /** @deprecated use `Clauses$inboundSchema` instead. */
    Clauses$.inboundSchema = exports.Clauses$inboundSchema;
    /** @deprecated use `Clauses$outboundSchema` instead. */
    Clauses$.outboundSchema = exports.Clauses$outboundSchema;
})(Clauses$ || (exports.Clauses$ = Clauses$ = {}));
function clausesToJSON(clauses) {
    return JSON.stringify(exports.Clauses$outboundSchema.parse(clauses));
}
function clausesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Clauses$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Clauses' from JSON`);
}
