// @polar-sh/sdk/dist/commonjs/models/components/genericpayment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericPayment$ = exports.GenericPayment$outboundSchema = exports.GenericPayment$inboundSchema = void 0;
exports.genericPaymentToJSON = genericPaymentToJSON;
exports.genericPaymentFromJSON = genericPaymentFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const paymentprocessor_js_1 = require("./paymentprocessor.js");
const paymentstatus_js_1 = require("./paymentstatus.js");
/** @internal */
exports.GenericPayment$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    processor: paymentprocessor_js_1.PaymentProcessor$inboundSchema,
    status: paymentstatus_js_1.PaymentStatus$inboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    method: z.string(),
    decline_reason: z.nullable(z.string()),
    decline_message: z.nullable(z.string()),
    organization_id: z.string(),
    checkout_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    processor_metadata: z.record(z.any()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "decline_reason": "declineReason",
        "decline_message": "declineMessage",
        "organization_id": "organizationId",
        "checkout_id": "checkoutId",
        "order_id": "orderId",
        "processor_metadata": "processorMetadata",
    });
});
/** @internal */
exports.GenericPayment$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    processor: paymentprocessor_js_1.PaymentProcessor$outboundSchema,
    status: paymentstatus_js_1.PaymentStatus$outboundSchema,
    amount: z.number().int(),
    currency: z.string(),
    method: z.string(),
    declineReason: z.nullable(z.string()),
    declineMessage: z.nullable(z.string()),
    organizationId: z.string(),
    checkoutId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    processorMetadata: z.record(z.any()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        declineReason: "decline_reason",
        declineMessage: "decline_message",
        organizationId: "organization_id",
        checkoutId: "checkout_id",
        orderId: "order_id",
        processorMetadata: "processor_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenericPayment$;
(function (GenericPayment$) {
    /** @deprecated use `GenericPayment$inboundSchema` instead. */
    GenericPayment$.inboundSchema = exports.GenericPayment$inboundSchema;
    /** @deprecated use `GenericPayment$outboundSchema` instead. */
    GenericPayment$.outboundSchema = exports.GenericPayment$outboundSchema;
})(GenericPayment$ || (exports.GenericPayment$ = GenericPayment$ = {}));
function genericPaymentToJSON(genericPayment) {
    return JSON.stringify(exports.GenericPayment$outboundSchema.parse(genericPayment));
}
function genericPaymentFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenericPayment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenericPayment' from JSON`);
}
