// @polar-sh/sdk/dist/commonjs/models/components/introspecttokenrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntrospectTokenRequest$ = exports.IntrospectTokenRequest$outboundSchema = exports.IntrospectTokenRequest$inboundSchema = exports.IntrospectTokenRequestTokenTypeHint$ = exports.IntrospectTokenRequestTokenTypeHint$outboundSchema = exports.IntrospectTokenRequestTokenTypeHint$inboundSchema = exports.IntrospectTokenRequestTokenTypeHint = void 0;
exports.introspectTokenRequestToJSON = introspectTokenRequestToJSON;
exports.introspectTokenRequestFromJSON = introspectTokenRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
exports.IntrospectTokenRequestTokenTypeHint = {
    AccessToken: "access_token",
    RefreshToken: "refresh_token",
};
/** @internal */
exports.IntrospectTokenRequestTokenTypeHint$inboundSchema = z.nativeEnum(exports.IntrospectTokenRequestTokenTypeHint);
/** @internal */
exports.IntrospectTokenRequestTokenTypeHint$outboundSchema = exports.IntrospectTokenRequestTokenTypeHint$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IntrospectTokenRequestTokenTypeHint$;
(function (IntrospectTokenRequestTokenTypeHint$) {
    /** @deprecated use `IntrospectTokenRequestTokenTypeHint$inboundSchema` instead. */
    IntrospectTokenRequestTokenTypeHint$.inboundSchema = exports.IntrospectTokenRequestTokenTypeHint$inboundSchema;
    /** @deprecated use `IntrospectTokenRequestTokenTypeHint$outboundSchema` instead. */
    IntrospectTokenRequestTokenTypeHint$.outboundSchema = exports.IntrospectTokenRequestTokenTypeHint$outboundSchema;
})(IntrospectTokenRequestTokenTypeHint$ || (exports.IntrospectTokenRequestTokenTypeHint$ = IntrospectTokenRequestTokenTypeHint$ = {}));
/** @internal */
exports.IntrospectTokenRequest$inboundSchema = z.object({
    token: z.string(),
    token_type_hint: z.nullable(exports.IntrospectTokenRequestTokenTypeHint$inboundSchema)
        .optional(),
    client_id: z.string(),
    client_secret: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "token_type_hint": "tokenTypeHint",
        "client_id": "clientId",
        "client_secret": "clientSecret",
    });
});
/** @internal */
exports.IntrospectTokenRequest$outboundSchema = z.object({
    token: z.string(),
    tokenTypeHint: z.nullable(exports.IntrospectTokenRequestTokenTypeHint$outboundSchema)
        .optional(),
    clientId: z.string(),
    clientSecret: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        tokenTypeHint: "token_type_hint",
        clientId: "client_id",
        clientSecret: "client_secret",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IntrospectTokenRequest$;
(function (IntrospectTokenRequest$) {
    /** @deprecated use `IntrospectTokenRequest$inboundSchema` instead. */
    IntrospectTokenRequest$.inboundSchema = exports.IntrospectTokenRequest$inboundSchema;
    /** @deprecated use `IntrospectTokenRequest$outboundSchema` instead. */
    IntrospectTokenRequest$.outboundSchema = exports.IntrospectTokenRequest$outboundSchema;
})(IntrospectTokenRequest$ || (exports.IntrospectTokenRequest$ = IntrospectTokenRequest$ = {}));
function introspectTokenRequestToJSON(introspectTokenRequest) {
    return JSON.stringify(exports.IntrospectTokenRequest$outboundSchema.parse(introspectTokenRequest));
}
function introspectTokenRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.IntrospectTokenRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'IntrospectTokenRequest' from JSON`);
}
