// @polar-sh/sdk/dist/commonjs/models/components/licensekeyactivationread.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeyActivationRead$ = exports.LicenseKeyActivationRead$outboundSchema = exports.LicenseKeyActivationRead$inboundSchema = exports.LicenseKeyActivationReadMeta$ = exports.LicenseKeyActivationReadMeta$outboundSchema = exports.LicenseKeyActivationReadMeta$inboundSchema = void 0;
exports.licenseKeyActivationReadMetaToJSON = licenseKeyActivationReadMetaToJSON;
exports.licenseKeyActivationReadMetaFromJSON = licenseKeyActivationReadMetaFromJSON;
exports.licenseKeyActivationReadToJSON = licenseKeyActivationReadToJSON;
exports.licenseKeyActivationReadFromJSON = licenseKeyActivationReadFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const licensekeyread_js_1 = require("./licensekeyread.js");
/** @internal */
exports.LicenseKeyActivationReadMeta$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.LicenseKeyActivationReadMeta$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyActivationReadMeta$;
(function (LicenseKeyActivationReadMeta$) {
    /** @deprecated use `LicenseKeyActivationReadMeta$inboundSchema` instead. */
    LicenseKeyActivationReadMeta$.inboundSchema = exports.LicenseKeyActivationReadMeta$inboundSchema;
    /** @deprecated use `LicenseKeyActivationReadMeta$outboundSchema` instead. */
    LicenseKeyActivationReadMeta$.outboundSchema = exports.LicenseKeyActivationReadMeta$outboundSchema;
})(LicenseKeyActivationReadMeta$ || (exports.LicenseKeyActivationReadMeta$ = LicenseKeyActivationReadMeta$ = {}));
function licenseKeyActivationReadMetaToJSON(licenseKeyActivationReadMeta) {
    return JSON.stringify(exports.LicenseKeyActivationReadMeta$outboundSchema.parse(licenseKeyActivationReadMeta));
}
function licenseKeyActivationReadMetaFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyActivationReadMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyActivationReadMeta' from JSON`);
}
/** @internal */
exports.LicenseKeyActivationRead$inboundSchema = z.object({
    id: z.string(),
    license_key_id: z.string(),
    label: z.string(),
    meta: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    license_key: licensekeyread_js_1.LicenseKeyRead$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "license_key_id": "licenseKeyId",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "license_key": "licenseKey",
    });
});
/** @internal */
exports.LicenseKeyActivationRead$outboundSchema = z.object({
    id: z.string(),
    licenseKeyId: z.string(),
    label: z.string(),
    meta: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    licenseKey: licensekeyread_js_1.LicenseKeyRead$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        licenseKeyId: "license_key_id",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        licenseKey: "license_key",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyActivationRead$;
(function (LicenseKeyActivationRead$) {
    /** @deprecated use `LicenseKeyActivationRead$inboundSchema` instead. */
    LicenseKeyActivationRead$.inboundSchema = exports.LicenseKeyActivationRead$inboundSchema;
    /** @deprecated use `LicenseKeyActivationRead$outboundSchema` instead. */
    LicenseKeyActivationRead$.outboundSchema = exports.LicenseKeyActivationRead$outboundSchema;
})(LicenseKeyActivationRead$ || (exports.LicenseKeyActivationRead$ = LicenseKeyActivationRead$ = {}));
function licenseKeyActivationReadToJSON(licenseKeyActivationRead) {
    return JSON.stringify(exports.LicenseKeyActivationRead$outboundSchema.parse(licenseKeyActivationRead));
}
function licenseKeyActivationReadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyActivationRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyActivationRead' from JSON`);
}
