// @polar-sh/sdk/dist/commonjs/models/components/licensekeyupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeyUpdate$ = exports.LicenseKeyUpdate$outboundSchema = exports.LicenseKeyUpdate$inboundSchema = void 0;
exports.licenseKeyUpdateToJSON = licenseKeyUpdateToJSON;
exports.licenseKeyUpdateFromJSON = licenseKeyUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const licensekeystatus_js_1 = require("./licensekeystatus.js");
/** @internal */
exports.LicenseKeyUpdate$inboundSchema = z.object({
    status: z.nullable(licensekeystatus_js_1.LicenseKeyStatus$inboundSchema).optional(),
    usage: z.number().int().default(0),
    limit_activations: z.nullable(z.number().int()).optional(),
    limit_usage: z.nullable(z.number().int()).optional(),
    expires_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "limit_activations": "limitActivations",
        "limit_usage": "limitUsage",
        "expires_at": "expiresAt",
    });
});
/** @internal */
exports.LicenseKeyUpdate$outboundSchema = z.object({
    status: z.nullable(licensekeystatus_js_1.LicenseKeyStatus$outboundSchema).optional(),
    usage: z.number().int().default(0),
    limitActivations: z.nullable(z.number().int()).optional(),
    limitUsage: z.nullable(z.number().int()).optional(),
    expiresAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        limitActivations: "limit_activations",
        limitUsage: "limit_usage",
        expiresAt: "expires_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyUpdate$;
(function (LicenseKeyUpdate$) {
    /** @deprecated use `LicenseKeyUpdate$inboundSchema` instead. */
    LicenseKeyUpdate$.inboundSchema = exports.LicenseKeyUpdate$inboundSchema;
    /** @deprecated use `LicenseKeyUpdate$outboundSchema` instead. */
    LicenseKeyUpdate$.outboundSchema = exports.LicenseKeyUpdate$outboundSchema;
})(LicenseKeyUpdate$ || (exports.LicenseKeyUpdate$ = LicenseKeyUpdate$ = {}));
function licenseKeyUpdateToJSON(licenseKeyUpdate) {
    return JSON.stringify(exports.LicenseKeyUpdate$outboundSchema.parse(licenseKeyUpdate));
}
function licenseKeyUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyUpdate' from JSON`);
}
