// @polar-sh/sdk/dist/commonjs/models/components/listresourcefileread.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceFileRead$ = exports.ListResourceFileRead$outboundSchema = exports.ListResourceFileRead$inboundSchema = exports.FileRead$ = exports.FileRead$outboundSchema = exports.FileRead$inboundSchema = void 0;
exports.fileReadToJSON = fileReadToJSON;
exports.fileReadFromJSON = fileReadFromJSON;
exports.listResourceFileReadToJSON = listResourceFileReadToJSON;
exports.listResourceFileReadFromJSON = listResourceFileReadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const downloadablefileread_js_1 = require("./downloadablefileread.js");
const organizationavatarfileread_js_1 = require("./organizationavatarfileread.js");
const pagination_js_1 = require("./pagination.js");
const productmediafileread_js_1 = require("./productmediafileread.js");
/** @internal */
exports.FileRead$inboundSchema = z.union([
    organizationavatarfileread_js_1.OrganizationAvatarFileRead$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    productmediafileread_js_1.ProductMediaFileRead$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    downloadablefileread_js_1.DownloadableFileRead$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
]);
/** @internal */
exports.FileRead$outboundSchema = z.union([
    organizationavatarfileread_js_1.OrganizationAvatarFileRead$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    productmediafileread_js_1.ProductMediaFileRead$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    downloadablefileread_js_1.DownloadableFileRead$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileRead$;
(function (FileRead$) {
    /** @deprecated use `FileRead$inboundSchema` instead. */
    FileRead$.inboundSchema = exports.FileRead$inboundSchema;
    /** @deprecated use `FileRead$outboundSchema` instead. */
    FileRead$.outboundSchema = exports.FileRead$outboundSchema;
})(FileRead$ || (exports.FileRead$ = FileRead$ = {}));
function fileReadToJSON(fileRead) {
    return JSON.stringify(exports.FileRead$outboundSchema.parse(fileRead));
}
function fileReadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FileRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileRead' from JSON`);
}
/** @internal */
exports.ListResourceFileRead$inboundSchema = z.object({
    items: z.array(z.union([
        organizationavatarfileread_js_1.OrganizationAvatarFileRead$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({ service: v.service }))),
        productmediafileread_js_1.ProductMediaFileRead$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
            service: v.service,
        }))),
        downloadablefileread_js_1.DownloadableFileRead$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
            service: v.service,
        }))),
    ])),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceFileRead$outboundSchema = z.object({
    items: z.array(z.union([
        organizationavatarfileread_js_1.OrganizationAvatarFileRead$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({ service: v.service }))),
        productmediafileread_js_1.ProductMediaFileRead$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
            service: v.service,
        }))),
        downloadablefileread_js_1.DownloadableFileRead$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
            service: v.service,
        }))),
    ])),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceFileRead$;
(function (ListResourceFileRead$) {
    /** @deprecated use `ListResourceFileRead$inboundSchema` instead. */
    ListResourceFileRead$.inboundSchema = exports.ListResourceFileRead$inboundSchema;
    /** @deprecated use `ListResourceFileRead$outboundSchema` instead. */
    ListResourceFileRead$.outboundSchema = exports.ListResourceFileRead$outboundSchema;
})(ListResourceFileRead$ || (exports.ListResourceFileRead$ = ListResourceFileRead$ = {}));
function listResourceFileReadToJSON(listResourceFileRead) {
    return JSON.stringify(exports.ListResourceFileRead$outboundSchema.parse(listResourceFileRead));
}
function listResourceFileReadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListResourceFileRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceFileRead' from JSON`);
}
