// @polar-sh/sdk/dist/commonjs/models/components/listresourceorganization.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceOrganization$ = exports.ListResourceOrganization$outboundSchema = exports.ListResourceOrganization$inboundSchema = void 0;
exports.listResourceOrganizationToJSON = listResourceOrganizationToJSON;
exports.listResourceOrganizationFromJSON = listResourceOrganizationFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const organization_js_1 = require("./organization.js");
const pagination_js_1 = require("./pagination.js");
/** @internal */
exports.ListResourceOrganization$inboundSchema = z.object({
    items: z.array(organization_js_1.Organization$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceOrganization$outboundSchema = z.object({
    items: z.array(organization_js_1.Organization$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceOrganization$;
(function (ListResourceOrganization$) {
    /** @deprecated use `ListResourceOrganization$inboundSchema` instead. */
    ListResourceOrganization$.inboundSchema = exports.ListResourceOrganization$inboundSchema;
    /** @deprecated use `ListResourceOrganization$outboundSchema` instead. */
    ListResourceOrganization$.outboundSchema = exports.ListResourceOrganization$outboundSchema;
})(ListResourceOrganization$ || (exports.ListResourceOrganization$ = ListResourceOrganization$ = {}));
function listResourceOrganizationToJSON(listResourceOrganization) {
    return JSON.stringify(exports.ListResourceOrganization$outboundSchema.parse(listResourceOrganization));
}
function listResourceOrganizationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListResourceOrganization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceOrganization' from JSON`);
}
