// @polar-sh/sdk/dist/commonjs/models/components/meterquantities.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeterQuantities$ = exports.MeterQuantities$outboundSchema = exports.MeterQuantities$inboundSchema = void 0;
exports.meterQuantitiesToJSON = meterQuantitiesToJSON;
exports.meterQuantitiesFromJSON = meterQuantitiesFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const meterquantity_js_1 = require("./meterquantity.js");
/** @internal */
exports.MeterQuantities$inboundSchema = z.object({
    quantities: z.array(meterquantity_js_1.MeterQuantity$inboundSchema),
    total: z.number(),
});
/** @internal */
exports.MeterQuantities$outboundSchema = z.object({
    quantities: z.array(meterquantity_js_1.MeterQuantity$outboundSchema),
    total: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MeterQuantities$;
(function (MeterQuantities$) {
    /** @deprecated use `MeterQuantities$inboundSchema` instead. */
    MeterQuantities$.inboundSchema = exports.MeterQuantities$inboundSchema;
    /** @deprecated use `MeterQuantities$outboundSchema` instead. */
    MeterQuantities$.outboundSchema = exports.MeterQuantities$outboundSchema;
})(MeterQuantities$ || (exports.MeterQuantities$ = MeterQuantities$ = {}));
function meterQuantitiesToJSON(meterQuantities) {
    return JSON.stringify(exports.MeterQuantities$outboundSchema.parse(meterQuantities));
}
function meterQuantitiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MeterQuantities$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeterQuantities' from JSON`);
}
