// @polar-sh/sdk/dist/commonjs/models/components/metricsresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricsResponse$ = exports.MetricsResponse$outboundSchema = exports.MetricsResponse$inboundSchema = void 0;
exports.metricsResponseToJSON = metricsResponseToJSON;
exports.metricsResponseFromJSON = metricsResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const metricperiod_js_1 = require("./metricperiod.js");
const metrics_js_1 = require("./metrics.js");
const metricstotals_js_1 = require("./metricstotals.js");
/** @internal */
exports.MetricsResponse$inboundSchema = z.object({
    periods: z.array(metricperiod_js_1.MetricPeriod$inboundSchema),
    totals: metricstotals_js_1.MetricsTotals$inboundSchema,
    metrics: metrics_js_1.Metrics$inboundSchema,
});
/** @internal */
exports.MetricsResponse$outboundSchema = z.object({
    periods: z.array(metricperiod_js_1.MetricPeriod$outboundSchema),
    totals: metricstotals_js_1.MetricsTotals$outboundSchema,
    metrics: metrics_js_1.Metrics$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsResponse$;
(function (MetricsResponse$) {
    /** @deprecated use `MetricsResponse$inboundSchema` instead. */
    MetricsResponse$.inboundSchema = exports.MetricsResponse$inboundSchema;
    /** @deprecated use `MetricsResponse$outboundSchema` instead. */
    MetricsResponse$.outboundSchema = exports.MetricsResponse$outboundSchema;
})(MetricsResponse$ || (exports.MetricsResponse$ = MetricsResponse$ = {}));
function metricsResponseToJSON(metricsResponse) {
    return JSON.stringify(exports.MetricsResponse$outboundSchema.parse(metricsResponse));
}
function metricsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsResponse' from JSON`);
}
