// @polar-sh/sdk/dist/commonjs/models/components/orderupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderUpdate$ = exports.OrderUpdate$outboundSchema = exports.OrderUpdate$inboundSchema = void 0;
exports.orderUpdateToJSON = orderUpdateToJSON;
exports.orderUpdateFromJSON = orderUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
/** @internal */
exports.OrderUpdate$inboundSchema = z.object({
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "billing_name": "billingName",
        "billing_address": "billingAddress",
    });
});
/** @internal */
exports.OrderUpdate$outboundSchema = z.object({
    billingName: z.nullable(z.string()),
    billingAddress: z.nullable(address_js_1.Address$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        billingName: "billing_name",
        billingAddress: "billing_address",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderUpdate$;
(function (OrderUpdate$) {
    /** @deprecated use `OrderUpdate$inboundSchema` instead. */
    OrderUpdate$.inboundSchema = exports.OrderUpdate$inboundSchema;
    /** @deprecated use `OrderUpdate$outboundSchema` instead. */
    OrderUpdate$.outboundSchema = exports.OrderUpdate$outboundSchema;
})(OrderUpdate$ || (exports.OrderUpdate$ = OrderUpdate$ = {}));
function orderUpdateToJSON(orderUpdate) {
    return JSON.stringify(exports.OrderUpdate$outboundSchema.parse(orderUpdate));
}
function orderUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrderUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderUpdate' from JSON`);
}
