// @polar-sh/sdk/dist/commonjs/models/components/orderitemschema.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderItemSchema$ = exports.OrderItemSchema$outboundSchema = exports.OrderItemSchema$inboundSchema = void 0;
exports.orderItemSchemaToJSON = orderItemSchemaToJSON;
exports.orderItemSchemaFromJSON = orderItemSchemaFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.OrderItemSchema$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    label: z.string(),
    amount: z.number().int(),
    tax_amount: z.number().int(),
    proration: z.boolean(),
    product_price_id: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "tax_amount": "taxAmount",
        "product_price_id": "productPriceId",
    });
});
/** @internal */
exports.OrderItemSchema$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    label: z.string(),
    amount: z.number().int(),
    taxAmount: z.number().int(),
    proration: z.boolean(),
    productPriceId: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        taxAmount: "tax_amount",
        productPriceId: "product_price_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderItemSchema$;
(function (OrderItemSchema$) {
    /** @deprecated use `OrderItemSchema$inboundSchema` instead. */
    OrderItemSchema$.inboundSchema = exports.OrderItemSchema$inboundSchema;
    /** @deprecated use `OrderItemSchema$outboundSchema` instead. */
    OrderItemSchema$.outboundSchema = exports.OrderItemSchema$outboundSchema;
})(OrderItemSchema$ || (exports.OrderItemSchema$ = OrderItemSchema$ = {}));
function orderItemSchemaToJSON(orderItemSchema) {
    return JSON.stringify(exports.OrderItemSchema$outboundSchema.parse(orderItemSchema));
}
function orderItemSchemaFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrderItemSchema$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderItemSchema' from JSON`);
}
