// @polar-sh/sdk/dist/commonjs/models/components/organizationupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUpdate$ = exports.OrganizationUpdate$outboundSchema = exports.OrganizationUpdate$inboundSchema = void 0;
exports.organizationUpdateToJSON = organizationUpdateToJSON;
exports.organizationUpdateFromJSON = organizationUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const organizationdetails_js_1 = require("./organizationdetails.js");
const organizationfeaturesettings_js_1 = require("./organizationfeaturesettings.js");
const organizationnotificationsettings_js_1 = require("./organizationnotificationsettings.js");
const organizationsociallink_js_1 = require("./organizationsociallink.js");
const organizationsubscriptionsettings_js_1 = require("./organizationsubscriptionsettings.js");
/** @internal */
exports.OrganizationUpdate$inboundSchema = z.object({
    name: z.nullable(z.string()).optional(),
    avatar_url: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    website: z.nullable(z.string()).optional(),
    socials: z.nullable(z.array(organizationsociallink_js_1.OrganizationSocialLink$inboundSchema)).optional(),
    details: z.nullable(organizationdetails_js_1.OrganizationDetails$inboundSchema).optional(),
    feature_settings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$inboundSchema)
        .optional(),
    subscription_settings: z.nullable(organizationsubscriptionsettings_js_1.OrganizationSubscriptionSettings$inboundSchema).optional(),
    notification_settings: z.nullable(organizationnotificationsettings_js_1.OrganizationNotificationSettings$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "avatar_url": "avatarUrl",
        "feature_settings": "featureSettings",
        "subscription_settings": "subscriptionSettings",
        "notification_settings": "notificationSettings",
    });
});
/** @internal */
exports.OrganizationUpdate$outboundSchema = z.object({
    name: z.nullable(z.string()).optional(),
    avatarUrl: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    website: z.nullable(z.string()).optional(),
    socials: z.nullable(z.array(organizationsociallink_js_1.OrganizationSocialLink$outboundSchema))
        .optional(),
    details: z.nullable(organizationdetails_js_1.OrganizationDetails$outboundSchema).optional(),
    featureSettings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$outboundSchema)
        .optional(),
    subscriptionSettings: z.nullable(organizationsubscriptionsettings_js_1.OrganizationSubscriptionSettings$outboundSchema).optional(),
    notificationSettings: z.nullable(organizationnotificationsettings_js_1.OrganizationNotificationSettings$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        avatarUrl: "avatar_url",
        featureSettings: "feature_settings",
        subscriptionSettings: "subscription_settings",
        notificationSettings: "notification_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationUpdate$;
(function (OrganizationUpdate$) {
    /** @deprecated use `OrganizationUpdate$inboundSchema` instead. */
    OrganizationUpdate$.inboundSchema = exports.OrganizationUpdate$inboundSchema;
    /** @deprecated use `OrganizationUpdate$outboundSchema` instead. */
    OrganizationUpdate$.outboundSchema = exports.OrganizationUpdate$outboundSchema;
})(OrganizationUpdate$ || (exports.OrganizationUpdate$ = OrganizationUpdate$ = {}));
function organizationUpdateToJSON(organizationUpdate) {
    return JSON.stringify(exports.OrganizationUpdate$outboundSchema.parse(organizationUpdate));
}
function organizationUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationUpdate' from JSON`);
}
