// @polar-sh/sdk/dist/commonjs/models/components/paymentmethodcardmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodCardMetadata$ = exports.PaymentMethodCardMetadata$outboundSchema = exports.PaymentMethodCardMetadata$inboundSchema = void 0;
exports.paymentMethodCardMetadataToJSON = paymentMethodCardMetadataToJSON;
exports.paymentMethodCardMetadataFromJSON = paymentMethodCardMetadataFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.PaymentMethodCardMetadata$inboundSchema = z.object({
    brand: z.string(),
    last4: z.string(),
    exp_month: z.number().int(),
    exp_year: z.number().int(),
    wallet: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "exp_month": "expMonth",
        "exp_year": "expYear",
    });
});
/** @internal */
exports.PaymentMethodCardMetadata$outboundSchema = z.object({
    brand: z.string(),
    last4: z.string(),
    expMonth: z.number().int(),
    expYear: z.number().int(),
    wallet: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        expMonth: "exp_month",
        expYear: "exp_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PaymentMethodCardMetadata$;
(function (PaymentMethodCardMetadata$) {
    /** @deprecated use `PaymentMethodCardMetadata$inboundSchema` instead. */
    PaymentMethodCardMetadata$.inboundSchema = exports.PaymentMethodCardMetadata$inboundSchema;
    /** @deprecated use `PaymentMethodCardMetadata$outboundSchema` instead. */
    PaymentMethodCardMetadata$.outboundSchema = exports.PaymentMethodCardMetadata$outboundSchema;
})(PaymentMethodCardMetadata$ || (exports.PaymentMethodCardMetadata$ = PaymentMethodCardMetadata$ = {}));
function paymentMethodCardMetadataToJSON(paymentMethodCardMetadata) {
    return JSON.stringify(exports.PaymentMethodCardMetadata$outboundSchema.parse(paymentMethodCardMetadata));
}
function paymentMethodCardMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PaymentMethodCardMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PaymentMethodCardMetadata' from JSON`);
}
