// @polar-sh/sdk/dist/commonjs/models/components/paymentmethodgeneric.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodGeneric$ = exports.PaymentMethodGeneric$outboundSchema = exports.PaymentMethodGeneric$inboundSchema = void 0;
exports.paymentMethodGenericToJSON = paymentMethodGenericToJSON;
exports.paymentMethodGenericFromJSON = paymentMethodGenericFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const paymentprocessor_js_1 = require("./paymentprocessor.js");
/** @internal */
exports.PaymentMethodGeneric$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    processor: paymentprocessor_js_1.PaymentProcessor$inboundSchema,
    customer_id: z.string(),
    type: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "customer_id": "customerId",
    });
});
/** @internal */
exports.PaymentMethodGeneric$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    processor: paymentprocessor_js_1.PaymentProcessor$outboundSchema,
    customerId: z.string(),
    type: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customerId: "customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PaymentMethodGeneric$;
(function (PaymentMethodGeneric$) {
    /** @deprecated use `PaymentMethodGeneric$inboundSchema` instead. */
    PaymentMethodGeneric$.inboundSchema = exports.PaymentMethodGeneric$inboundSchema;
    /** @deprecated use `PaymentMethodGeneric$outboundSchema` instead. */
    PaymentMethodGeneric$.outboundSchema = exports.PaymentMethodGeneric$outboundSchema;
})(PaymentMethodGeneric$ || (exports.PaymentMethodGeneric$ = PaymentMethodGeneric$ = {}));
function paymentMethodGenericToJSON(paymentMethodGeneric) {
    return JSON.stringify(exports.PaymentMethodGeneric$outboundSchema.parse(paymentMethodGeneric));
}
function paymentMethodGenericFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PaymentMethodGeneric$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PaymentMethodGeneric' from JSON`);
}
