// @polar-sh/sdk/dist/commonjs/models/components/productcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCreate$ = exports.ProductCreate$outboundSchema = exports.ProductCreate$inboundSchema = exports.ProductCreatePrices$ = exports.ProductCreatePrices$outboundSchema = exports.ProductCreatePrices$inboundSchema = exports.ProductCreateMetadata$ = exports.ProductCreateMetadata$outboundSchema = exports.ProductCreateMetadata$inboundSchema = void 0;
exports.productCreateMetadataToJSON = productCreateMetadataToJSON;
exports.productCreateMetadataFromJSON = productCreateMetadataFromJSON;
exports.productCreatePricesToJSON = productCreatePricesToJSON;
exports.productCreatePricesFromJSON = productCreatePricesFromJSON;
exports.productCreateToJSON = productCreateToJSON;
exports.productCreateFromJSON = productCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const productpricemeteredunitcreate_js_1 = require("./productpricemeteredunitcreate.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
/** @internal */
exports.ProductCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/** @internal */
exports.ProductCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.number(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreateMetadata$;
(function (ProductCreateMetadata$) {
    /** @deprecated use `ProductCreateMetadata$inboundSchema` instead. */
    ProductCreateMetadata$.inboundSchema = exports.ProductCreateMetadata$inboundSchema;
    /** @deprecated use `ProductCreateMetadata$outboundSchema` instead. */
    ProductCreateMetadata$.outboundSchema = exports.ProductCreateMetadata$outboundSchema;
})(ProductCreateMetadata$ || (exports.ProductCreateMetadata$ = ProductCreateMetadata$ = {}));
function productCreateMetadataToJSON(productCreateMetadata) {
    return JSON.stringify(exports.ProductCreateMetadata$outboundSchema.parse(productCreateMetadata));
}
function productCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreateMetadata' from JSON`);
}
/** @internal */
exports.ProductCreatePrices$inboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema,
    productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema,
    productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema,
    productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema,
]);
/** @internal */
exports.ProductCreatePrices$outboundSchema = z.union([
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema,
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreatePrices$;
(function (ProductCreatePrices$) {
    /** @deprecated use `ProductCreatePrices$inboundSchema` instead. */
    ProductCreatePrices$.inboundSchema = exports.ProductCreatePrices$inboundSchema;
    /** @deprecated use `ProductCreatePrices$outboundSchema` instead. */
    ProductCreatePrices$.outboundSchema = exports.ProductCreatePrices$outboundSchema;
})(ProductCreatePrices$ || (exports.ProductCreatePrices$ = ProductCreatePrices$ = {}));
function productCreatePricesToJSON(productCreatePrices) {
    return JSON.stringify(exports.ProductCreatePrices$outboundSchema.parse(productCreatePrices));
}
function productCreatePricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreatePrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreatePrices' from JSON`);
}
/** @internal */
exports.ProductCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    recurring_interval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema),
    prices: z.array(z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$inboundSchema,
        productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema,
        productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema,
        productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "recurring_interval": "recurringInterval",
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ProductCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.number(), z.boolean()])).optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    recurringInterval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema),
    prices: z.array(z.union([
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema,
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        recurringInterval: "recurring_interval",
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreate$;
(function (ProductCreate$) {
    /** @deprecated use `ProductCreate$inboundSchema` instead. */
    ProductCreate$.inboundSchema = exports.ProductCreate$inboundSchema;
    /** @deprecated use `ProductCreate$outboundSchema` instead. */
    ProductCreate$.outboundSchema = exports.ProductCreate$outboundSchema;
})(ProductCreate$ || (exports.ProductCreate$ = ProductCreate$ = {}));
function productCreateToJSON(productCreate) {
    return JSON.stringify(exports.ProductCreate$outboundSchema.parse(productCreate));
}
function productCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreate' from JSON`);
}
