// @polar-sh/sdk/dist/commonjs/models/components/productprice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPrice$ = exports.ProductPrice$outboundSchema = exports.ProductPrice$inboundSchema = void 0;
exports.productPriceToJSON = productPriceToJSON;
exports.productPriceFromJSON = productPriceFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const productpricecustom_js_1 = require("./productpricecustom.js");
const productpricefixed_js_1 = require("./productpricefixed.js");
const productpricefree_js_1 = require("./productpricefree.js");
const productpricemeteredunit_js_1 = require("./productpricemeteredunit.js");
/** @internal */
exports.ProductPrice$inboundSchema = z.union([
    productpricemeteredunit_js_1.ProductPriceMeteredUnit$inboundSchema.and(z.object({ amount_type: z.literal("metered_unit") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricecustom_js_1.ProductPriceCustom$inboundSchema.and(z.object({ amount_type: z.literal("custom") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefixed_js_1.ProductPriceFixed$inboundSchema.and(z.object({ amount_type: z.literal("fixed") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
    productpricefree_js_1.ProductPriceFree$inboundSchema.and(z.object({ amount_type: z.literal("free") }).transform((v) => ({
        amountType: v.amount_type,
    }))),
]);
/** @internal */
exports.ProductPrice$outboundSchema = z.union([
    productpricemeteredunit_js_1.ProductPriceMeteredUnit$outboundSchema.and(z.object({ amountType: z.literal("metered_unit") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricecustom_js_1.ProductPriceCustom$outboundSchema.and(z.object({ amountType: z.literal("custom") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefixed_js_1.ProductPriceFixed$outboundSchema.and(z.object({ amountType: z.literal("fixed") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
    productpricefree_js_1.ProductPriceFree$outboundSchema.and(z.object({ amountType: z.literal("free") }).transform((v) => ({
        amount_type: v.amountType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPrice$;
(function (ProductPrice$) {
    /** @deprecated use `ProductPrice$inboundSchema` instead. */
    ProductPrice$.inboundSchema = exports.ProductPrice$inboundSchema;
    /** @deprecated use `ProductPrice$outboundSchema` instead. */
    ProductPrice$.outboundSchema = exports.ProductPrice$outboundSchema;
})(ProductPrice$ || (exports.ProductPrice$ = ProductPrice$ = {}));
function productPriceToJSON(productPrice) {
    return JSON.stringify(exports.ProductPrice$outboundSchema.parse(productPrice));
}
function productPriceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPrice' from JSON`);
}
