// @polar-sh/sdk/dist/commonjs/models/components/productpricemeter.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceMeter$ = exports.ProductPriceMeter$outboundSchema = exports.ProductPriceMeter$inboundSchema = void 0;
exports.productPriceMeterToJSON = productPriceMeterToJSON;
exports.productPriceMeterFromJSON = productPriceMeterFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ProductPriceMeter$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
});
/** @internal */
exports.ProductPriceMeter$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceMeter$;
(function (ProductPriceMeter$) {
    /** @deprecated use `ProductPriceMeter$inboundSchema` instead. */
    ProductPriceMeter$.inboundSchema = exports.ProductPriceMeter$inboundSchema;
    /** @deprecated use `ProductPriceMeter$outboundSchema` instead. */
    ProductPriceMeter$.outboundSchema = exports.ProductPriceMeter$outboundSchema;
})(ProductPriceMeter$ || (exports.ProductPriceMeter$ = ProductPriceMeter$ = {}));
function productPriceMeterToJSON(productPriceMeter) {
    return JSON.stringify(exports.ProductPriceMeter$outboundSchema.parse(productPriceMeter));
}
function productPriceMeterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceMeter' from JSON`);
}
