// @polar-sh/sdk/dist/commonjs/models/components/productpricecustom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceCustom$ = exports.ProductPriceCustom$outboundSchema = exports.ProductPriceCustom$inboundSchema = void 0;
exports.productPriceCustomToJSON = productPriceCustomToJSON;
exports.productPriceCustomFromJSON = productPriceCustomFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const productpricetype_js_1 = require("./productpricetype.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
/** @internal */
exports.ProductPriceCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("custom"),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: productpricetype_js_1.ProductPriceType$inboundSchema,
    recurring_interval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema),
    price_currency: z.string(),
    minimum_amount: z.nullable(z.number().int()),
    maximum_amount: z.nullable(z.number().int()),
    preset_amount: z.nullable(z.number().int()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
exports.ProductPriceCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("custom"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: productpricetype_js_1.ProductPriceType$outboundSchema,
    recurringInterval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema),
    priceCurrency: z.string(),
    minimumAmount: z.nullable(z.number().int()),
    maximumAmount: z.nullable(z.number().int()),
    presetAmount: z.nullable(z.number().int()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceCustom$;
(function (ProductPriceCustom$) {
    /** @deprecated use `ProductPriceCustom$inboundSchema` instead. */
    ProductPriceCustom$.inboundSchema = exports.ProductPriceCustom$inboundSchema;
    /** @deprecated use `ProductPriceCustom$outboundSchema` instead. */
    ProductPriceCustom$.outboundSchema = exports.ProductPriceCustom$outboundSchema;
})(ProductPriceCustom$ || (exports.ProductPriceCustom$ = ProductPriceCustom$ = {}));
function productPriceCustomToJSON(productPriceCustom) {
    return JSON.stringify(exports.ProductPriceCustom$outboundSchema.parse(productPriceCustom));
}
function productPriceCustomFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceCustom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceCustom' from JSON`);
}
