// @polar-sh/sdk/dist/commonjs/models/components/productmediafileread.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductMediaFileRead$ = exports.ProductMediaFileRead$outboundSchema = exports.ProductMediaFileRead$inboundSchema = void 0;
exports.productMediaFileReadToJSON = productMediaFileReadToJSON;
exports.productMediaFileReadFromJSON = productMediaFileReadFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ProductMediaFileRead$inboundSchema = z.object({
    id: z.string(),
    organization_id: z.string(),
    name: z.string(),
    path: z.string(),
    mime_type: z.string(),
    size: z.number().int(),
    storage_version: z.nullable(z.string()),
    checksum_etag: z.nullable(z.string()),
    checksum_sha256_base64: z.nullable(z.string()),
    checksum_sha256_hex: z.nullable(z.string()),
    last_modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    version: z.nullable(z.string()),
    service: z.literal("product_media"),
    is_uploaded: z.boolean(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    size_readable: z.string(),
    public_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "mime_type": "mimeType",
        "storage_version": "storageVersion",
        "checksum_etag": "checksumEtag",
        "checksum_sha256_base64": "checksumSha256Base64",
        "checksum_sha256_hex": "checksumSha256Hex",
        "last_modified_at": "lastModifiedAt",
        "is_uploaded": "isUploaded",
        "created_at": "createdAt",
        "size_readable": "sizeReadable",
        "public_url": "publicUrl",
    });
});
/** @internal */
exports.ProductMediaFileRead$outboundSchema = z.object({
    id: z.string(),
    organizationId: z.string(),
    name: z.string(),
    path: z.string(),
    mimeType: z.string(),
    size: z.number().int(),
    storageVersion: z.nullable(z.string()),
    checksumEtag: z.nullable(z.string()),
    checksumSha256Base64: z.nullable(z.string()),
    checksumSha256Hex: z.nullable(z.string()),
    lastModifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    version: z.nullable(z.string()),
    service: z.literal("product_media"),
    isUploaded: z.boolean(),
    createdAt: z.date().transform(v => v.toISOString()),
    sizeReadable: z.string(),
    publicUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        mimeType: "mime_type",
        storageVersion: "storage_version",
        checksumEtag: "checksum_etag",
        checksumSha256Base64: "checksum_sha256_base64",
        checksumSha256Hex: "checksum_sha256_hex",
        lastModifiedAt: "last_modified_at",
        isUploaded: "is_uploaded",
        createdAt: "created_at",
        sizeReadable: "size_readable",
        publicUrl: "public_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductMediaFileRead$;
(function (ProductMediaFileRead$) {
    /** @deprecated use `ProductMediaFileRead$inboundSchema` instead. */
    ProductMediaFileRead$.inboundSchema = exports.ProductMediaFileRead$inboundSchema;
    /** @deprecated use `ProductMediaFileRead$outboundSchema` instead. */
    ProductMediaFileRead$.outboundSchema = exports.ProductMediaFileRead$outboundSchema;
})(ProductMediaFileRead$ || (exports.ProductMediaFileRead$ = ProductMediaFileRead$ = {}));
function productMediaFileReadToJSON(productMediaFileRead) {
    return JSON.stringify(exports.ProductMediaFileRead$outboundSchema.parse(productMediaFileRead));
}
function productMediaFileReadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductMediaFileRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductMediaFileRead' from JSON`);
}
