// @polar-sh/sdk/dist/commonjs/models/components/propertyaggregation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyAggregation$ = exports.PropertyAggregation$outboundSchema = exports.PropertyAggregation$inboundSchema = exports.Func$ = exports.Func$outboundSchema = exports.Func$inboundSchema = exports.Func = void 0;
exports.propertyAggregationToJSON = propertyAggregationToJSON;
exports.propertyAggregationFromJSON = propertyAggregationFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.Func = {
    Sum: "sum",
    Max: "max",
    Min: "min",
    Avg: "avg",
};
/** @internal */
exports.Func$inboundSchema = z.nativeEnum(exports.Func);
/** @internal */
exports.Func$outboundSchema = exports.Func$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Func$;
(function (Func$) {
    /** @deprecated use `Func$inboundSchema` instead. */
    Func$.inboundSchema = exports.Func$inboundSchema;
    /** @deprecated use `Func$outboundSchema` instead. */
    Func$.outboundSchema = exports.Func$outboundSchema;
})(Func$ || (exports.Func$ = Func$ = {}));
/** @internal */
exports.PropertyAggregation$inboundSchema = z.object({
    func: exports.Func$inboundSchema,
    property: z.string(),
});
/** @internal */
exports.PropertyAggregation$outboundSchema = z.object({
    func: exports.Func$outboundSchema,
    property: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PropertyAggregation$;
(function (PropertyAggregation$) {
    /** @deprecated use `PropertyAggregation$inboundSchema` instead. */
    PropertyAggregation$.inboundSchema = exports.PropertyAggregation$inboundSchema;
    /** @deprecated use `PropertyAggregation$outboundSchema` instead. */
    PropertyAggregation$.outboundSchema = exports.PropertyAggregation$outboundSchema;
})(PropertyAggregation$ || (exports.PropertyAggregation$ = PropertyAggregation$ = {}));
function propertyAggregationToJSON(propertyAggregation) {
    return JSON.stringify(exports.PropertyAggregation$outboundSchema.parse(propertyAggregation));
}
function propertyAggregationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PropertyAggregation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PropertyAggregation' from JSON`);
}
