// @polar-sh/sdk/dist/commonjs/models/components/s3filecreatepart.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3FileCreatePart$ = exports.S3FileCreatePart$outboundSchema = exports.S3FileCreatePart$inboundSchema = void 0;
exports.s3FileCreatePartToJSON = s3FileCreatePartToJSON;
exports.s3FileCreatePartFromJSON = s3FileCreatePartFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.S3FileCreatePart$inboundSchema = z.object({
    number: z.number().int(),
    chunk_start: z.number().int(),
    chunk_end: z.number().int(),
    checksum_sha256_base64: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "chunk_start": "chunkStart",
        "chunk_end": "chunkEnd",
        "checksum_sha256_base64": "checksumSha256Base64",
    });
});
/** @internal */
exports.S3FileCreatePart$outboundSchema = z.object({
    number: z.number().int(),
    chunkStart: z.number().int(),
    chunkEnd: z.number().int(),
    checksumSha256Base64: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        chunkStart: "chunk_start",
        chunkEnd: "chunk_end",
        checksumSha256Base64: "checksum_sha256_base64",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var S3FileCreatePart$;
(function (S3FileCreatePart$) {
    /** @deprecated use `S3FileCreatePart$inboundSchema` instead. */
    S3FileCreatePart$.inboundSchema = exports.S3FileCreatePart$inboundSchema;
    /** @deprecated use `S3FileCreatePart$outboundSchema` instead. */
    S3FileCreatePart$.outboundSchema = exports.S3FileCreatePart$outboundSchema;
})(S3FileCreatePart$ || (exports.S3FileCreatePart$ = S3FileCreatePart$ = {}));
function s3FileCreatePartToJSON(s3FileCreatePart) {
    return JSON.stringify(exports.S3FileCreatePart$outboundSchema.parse(s3FileCreatePart));
}
function s3FileCreatePartFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.S3FileCreatePart$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'S3FileCreatePart' from JSON`);
}
