// @polar-sh/sdk/dist/commonjs/models/components/subscriptionrevoke.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionRevoke$ = exports.SubscriptionRevoke$outboundSchema = exports.SubscriptionRevoke$inboundSchema = void 0;
exports.subscriptionRevokeToJSON = subscriptionRevokeToJSON;
exports.subscriptionRevokeFromJSON = subscriptionRevokeFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customercancellationreason_js_1 = require("./customercancellationreason.js");
/** @internal */
exports.SubscriptionRevoke$inboundSchema = z.object({
    customer_cancellation_reason: z.nullable(customercancellationreason_js_1.CustomerCancellationReason$inboundSchema).optional(),
    customer_cancellation_comment: z.nullable(z.string()).optional(),
    revoke: z.literal(true),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
    });
});
/** @internal */
exports.SubscriptionRevoke$outboundSchema = z.object({
    customerCancellationReason: z.nullable(customercancellationreason_js_1.CustomerCancellationReason$outboundSchema).optional(),
    customerCancellationComment: z.nullable(z.string()).optional(),
    revoke: z.literal(true),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerCancellationReason: "customer_cancellation_reason",
        customerCancellationComment: "customer_cancellation_comment",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionRevoke$;
(function (SubscriptionRevoke$) {
    /** @deprecated use `SubscriptionRevoke$inboundSchema` instead. */
    SubscriptionRevoke$.inboundSchema = exports.SubscriptionRevoke$inboundSchema;
    /** @deprecated use `SubscriptionRevoke$outboundSchema` instead. */
    SubscriptionRevoke$.outboundSchema = exports.SubscriptionRevoke$outboundSchema;
})(SubscriptionRevoke$ || (exports.SubscriptionRevoke$ = SubscriptionRevoke$ = {}));
function subscriptionRevokeToJSON(subscriptionRevoke) {
    return JSON.stringify(exports.SubscriptionRevoke$outboundSchema.parse(subscriptionRevoke));
}
function subscriptionRevokeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionRevoke$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionRevoke' from JSON`);
}
