// @polar-sh/sdk/dist/commonjs/models/components/subscriptionrevokedevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionRevokedEvent$ = exports.SubscriptionRevokedEvent$outboundSchema = exports.SubscriptionRevokedEvent$inboundSchema = void 0;
exports.subscriptionRevokedEventToJSON = subscriptionRevokedEventToJSON;
exports.subscriptionRevokedEventFromJSON = subscriptionRevokedEventFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customer_js_1 = require("./customer.js");
const subscriptionrevokedmetadata_js_1 = require("./subscriptionrevokedmetadata.js");
/** @internal */
exports.SubscriptionRevokedEvent$inboundSchema = z.object({
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(customer_js_1.Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("subscription.revoked"),
    metadata: subscriptionrevokedmetadata_js_1.SubscriptionRevokedMetadata$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
exports.SubscriptionRevokedEvent$outboundSchema = z.object({
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(customer_js_1.Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    source: z.literal("system"),
    name: z.literal("subscription.revoked"),
    metadata: subscriptionrevokedmetadata_js_1.SubscriptionRevokedMetadata$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionRevokedEvent$;
(function (SubscriptionRevokedEvent$) {
    /** @deprecated use `SubscriptionRevokedEvent$inboundSchema` instead. */
    SubscriptionRevokedEvent$.inboundSchema = exports.SubscriptionRevokedEvent$inboundSchema;
    /** @deprecated use `SubscriptionRevokedEvent$outboundSchema` instead. */
    SubscriptionRevokedEvent$.outboundSchema = exports.SubscriptionRevokedEvent$outboundSchema;
})(SubscriptionRevokedEvent$ || (exports.SubscriptionRevokedEvent$ = SubscriptionRevokedEvent$ = {}));
function subscriptionRevokedEventToJSON(subscriptionRevokedEvent) {
    return JSON.stringify(exports.SubscriptionRevokedEvent$outboundSchema.parse(subscriptionRevokedEvent));
}
function subscriptionRevokedEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionRevokedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionRevokedEvent' from JSON`);
}
