// @polar-sh/sdk/dist/commonjs/models/components/systemevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemEvent$ = exports.SystemEvent$outboundSchema = exports.SystemEvent$inboundSchema = void 0;
exports.systemEventToJSON = systemEventToJSON;
exports.systemEventFromJSON = systemEventFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitcycledevent_js_1 = require("./benefitcycledevent.js");
const benefitgrantedevent_js_1 = require("./benefitgrantedevent.js");
const benefitrevokedevent_js_1 = require("./benefitrevokedevent.js");
const benefitupdatedevent_js_1 = require("./benefitupdatedevent.js");
const metercreditevent_js_1 = require("./metercreditevent.js");
const meterresetevent_js_1 = require("./meterresetevent.js");
const subscriptioncycledevent_js_1 = require("./subscriptioncycledevent.js");
const subscriptionproductupdatedevent_js_1 = require("./subscriptionproductupdatedevent.js");
const subscriptionrevokedevent_js_1 = require("./subscriptionrevokedevent.js");
/** @internal */
exports.SystemEvent$inboundSchema = z.union([
    benefitcycledevent_js_1.BenefitCycledEvent$inboundSchema.and(z.object({ name: z.literal("benefit.cycled") }).transform((v) => ({
        name: v.name,
    }))),
    benefitgrantedevent_js_1.BenefitGrantedEvent$inboundSchema.and(z.object({ name: z.literal("benefit.granted") }).transform((v) => ({
        name: v.name,
    }))),
    benefitrevokedevent_js_1.BenefitRevokedEvent$inboundSchema.and(z.object({ name: z.literal("benefit.revoked") }).transform((v) => ({
        name: v.name,
    }))),
    benefitupdatedevent_js_1.BenefitUpdatedEvent$inboundSchema.and(z.object({ name: z.literal("benefit.updated") }).transform((v) => ({
        name: v.name,
    }))),
    metercreditevent_js_1.MeterCreditEvent$inboundSchema.and(z.object({ name: z.literal("meter.credited") }).transform((v) => ({
        name: v.name,
    }))),
    meterresetevent_js_1.MeterResetEvent$inboundSchema.and(z.object({ name: z.literal("meter.reset") }).transform((v) => ({
        name: v.name,
    }))),
    subscriptioncycledevent_js_1.SubscriptionCycledEvent$inboundSchema.and(z.object({ name: z.literal("subscription.cycled") }).transform((v) => ({
        name: v.name,
    }))),
    subscriptionproductupdatedevent_js_1.SubscriptionProductUpdatedEvent$inboundSchema.and(z.object({ name: z.literal("subscription.product_updated") }).transform((v) => ({ name: v.name }))),
    subscriptionrevokedevent_js_1.SubscriptionRevokedEvent$inboundSchema.and(z.object({ name: z.literal("subscription.revoked") }).transform((v) => ({
        name: v.name,
    }))),
]);
/** @internal */
exports.SystemEvent$outboundSchema = z.union([
    benefitcycledevent_js_1.BenefitCycledEvent$outboundSchema.and(z.object({ name: z.literal("benefit.cycled") }).transform((v) => ({
        name: v.name,
    }))),
    benefitgrantedevent_js_1.BenefitGrantedEvent$outboundSchema.and(z.object({ name: z.literal("benefit.granted") }).transform((v) => ({
        name: v.name,
    }))),
    benefitrevokedevent_js_1.BenefitRevokedEvent$outboundSchema.and(z.object({ name: z.literal("benefit.revoked") }).transform((v) => ({
        name: v.name,
    }))),
    benefitupdatedevent_js_1.BenefitUpdatedEvent$outboundSchema.and(z.object({ name: z.literal("benefit.updated") }).transform((v) => ({
        name: v.name,
    }))),
    metercreditevent_js_1.MeterCreditEvent$outboundSchema.and(z.object({ name: z.literal("meter.credited") }).transform((v) => ({
        name: v.name,
    }))),
    meterresetevent_js_1.MeterResetEvent$outboundSchema.and(z.object({ name: z.literal("meter.reset") }).transform((v) => ({
        name: v.name,
    }))),
    subscriptioncycledevent_js_1.SubscriptionCycledEvent$outboundSchema.and(z.object({ name: z.literal("subscription.cycled") }).transform((v) => ({
        name: v.name,
    }))),
    subscriptionproductupdatedevent_js_1.SubscriptionProductUpdatedEvent$outboundSchema.and(z.object({ name: z.literal("subscription.product_updated") }).transform((v) => ({ name: v.name }))),
    subscriptionrevokedevent_js_1.SubscriptionRevokedEvent$outboundSchema.and(z.object({ name: z.literal("subscription.revoked") }).transform((v) => ({
        name: v.name,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SystemEvent$;
(function (SystemEvent$) {
    /** @deprecated use `SystemEvent$inboundSchema` instead. */
    SystemEvent$.inboundSchema = exports.SystemEvent$inboundSchema;
    /** @deprecated use `SystemEvent$outboundSchema` instead. */
    SystemEvent$.outboundSchema = exports.SystemEvent$outboundSchema;
})(SystemEvent$ || (exports.SystemEvent$ = SystemEvent$ = {}));
function systemEventToJSON(systemEvent) {
    return JSON.stringify(exports.SystemEvent$outboundSchema.parse(systemEvent));
}
function systemEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SystemEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SystemEvent' from JSON`);
}
