// @polar-sh/sdk/dist/commonjs/models/components/tokenresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenResponse$ = exports.TokenResponse$outboundSchema = exports.TokenResponse$inboundSchema = void 0;
exports.tokenResponseToJSON = tokenResponseToJSON;
exports.tokenResponseFromJSON = tokenResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.TokenResponse$inboundSchema = z.object({
    access_token: z.string(),
    token_type: z.literal("Bearer"),
    expires_in: z.number().int(),
    refresh_token: z.nullable(z.string()),
    scope: z.string(),
    id_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "access_token": "accessToken",
        "token_type": "tokenType",
        "expires_in": "expiresIn",
        "refresh_token": "refreshToken",
        "id_token": "idToken",
    });
});
/** @internal */
exports.TokenResponse$outboundSchema = z.object({
    accessToken: z.string(),
    tokenType: z.literal("Bearer"),
    expiresIn: z.number().int(),
    refreshToken: z.nullable(z.string()),
    scope: z.string(),
    idToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accessToken: "access_token",
        tokenType: "token_type",
        expiresIn: "expires_in",
        refreshToken: "refresh_token",
        idToken: "id_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TokenResponse$;
(function (TokenResponse$) {
    /** @deprecated use `TokenResponse$inboundSchema` instead. */
    TokenResponse$.inboundSchema = exports.TokenResponse$inboundSchema;
    /** @deprecated use `TokenResponse$outboundSchema` instead. */
    TokenResponse$.outboundSchema = exports.TokenResponse$outboundSchema;
})(TokenResponse$ || (exports.TokenResponse$ = TokenResponse$ = {}));
function tokenResponseToJSON(tokenResponse) {
    return JSON.stringify(exports.TokenResponse$outboundSchema.parse(tokenResponse));
}
function tokenResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TokenResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TokenResponse' from JSON`);
}
