// @polar-sh/sdk/dist/commonjs/models/components/webhookevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEvent$ = exports.WebhookEvent$outboundSchema = exports.WebhookEvent$inboundSchema = void 0;
exports.webhookEventToJSON = webhookEventToJSON;
exports.webhookEventFromJSON = webhookEventFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const webhookeventtype_js_1 = require("./webhookeventtype.js");
/** @internal */
exports.WebhookEvent$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    last_http_code: z.nullable(z.number().int()).optional(),
    succeeded: z.nullable(z.boolean()).optional(),
    payload: z.nullable(z.string()),
    type: webhookeventtype_js_1.WebhookEventType$inboundSchema,
    is_archived: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "last_http_code": "lastHttpCode",
        "is_archived": "isArchived",
    });
});
/** @internal */
exports.WebhookEvent$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    lastHttpCode: z.nullable(z.number().int()).optional(),
    succeeded: z.nullable(z.boolean()).optional(),
    payload: z.nullable(z.string()),
    type: webhookeventtype_js_1.WebhookEventType$outboundSchema,
    isArchived: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        lastHttpCode: "last_http_code",
        isArchived: "is_archived",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookEvent$;
(function (WebhookEvent$) {
    /** @deprecated use `WebhookEvent$inboundSchema` instead. */
    WebhookEvent$.inboundSchema = exports.WebhookEvent$inboundSchema;
    /** @deprecated use `WebhookEvent$outboundSchema` instead. */
    WebhookEvent$.outboundSchema = exports.WebhookEvent$outboundSchema;
})(WebhookEvent$ || (exports.WebhookEvent$ = WebhookEvent$ = {}));
function webhookEventToJSON(webhookEvent) {
    return JSON.stringify(exports.WebhookEvent$outboundSchema.parse(webhookEvent));
}
function webhookEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEvent' from JSON`);
}
