// @polar-sh/sdk/dist/commonjs/models/components/webhookendpointcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEndpointCreate$ = exports.WebhookEndpointCreate$outboundSchema = exports.WebhookEndpointCreate$inboundSchema = void 0;
exports.webhookEndpointCreateToJSON = webhookEndpointCreateToJSON;
exports.webhookEndpointCreateFromJSON = webhookEndpointCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const webhookeventtype_js_1 = require("./webhookeventtype.js");
const webhookformat_js_1 = require("./webhookformat.js");
/** @internal */
exports.WebhookEndpointCreate$inboundSchema = z.object({
    url: z.string(),
    secret: z.nullable(z.string()).optional(),
    format: webhookformat_js_1.WebhookFormat$inboundSchema,
    events: z.array(webhookeventtype_js_1.WebhookEventType$inboundSchema),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.WebhookEndpointCreate$outboundSchema = z.object({
    url: z.string(),
    secret: z.nullable(z.string()).optional(),
    format: webhookformat_js_1.WebhookFormat$outboundSchema,
    events: z.array(webhookeventtype_js_1.WebhookEventType$outboundSchema),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookEndpointCreate$;
(function (WebhookEndpointCreate$) {
    /** @deprecated use `WebhookEndpointCreate$inboundSchema` instead. */
    WebhookEndpointCreate$.inboundSchema = exports.WebhookEndpointCreate$inboundSchema;
    /** @deprecated use `WebhookEndpointCreate$outboundSchema` instead. */
    WebhookEndpointCreate$.outboundSchema = exports.WebhookEndpointCreate$outboundSchema;
})(WebhookEndpointCreate$ || (exports.WebhookEndpointCreate$ = WebhookEndpointCreate$ = {}));
function webhookEndpointCreateToJSON(webhookEndpointCreate) {
    return JSON.stringify(exports.WebhookEndpointCreate$outboundSchema.parse(webhookEndpointCreate));
}
function webhookEndpointCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookEndpointCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEndpointCreate' from JSON`);
}
