// @polar-sh/sdk/dist/commonjs/models/components/webhookeventtype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEventType$ = exports.WebhookEventType$outboundSchema = exports.WebhookEventType$inboundSchema = exports.WebhookEventType = void 0;
const z = __importStar(require("zod"));
exports.WebhookEventType = {
    CheckoutCreated: "checkout.created",
    CheckoutUpdated: "checkout.updated",
    CustomerCreated: "customer.created",
    CustomerUpdated: "customer.updated",
    CustomerDeleted: "customer.deleted",
    CustomerStateChanged: "customer.state_changed",
    OrderCreated: "order.created",
    OrderUpdated: "order.updated",
    OrderPaid: "order.paid",
    OrderRefunded: "order.refunded",
    SubscriptionCreated: "subscription.created",
    SubscriptionUpdated: "subscription.updated",
    SubscriptionActive: "subscription.active",
    SubscriptionCanceled: "subscription.canceled",
    SubscriptionUncanceled: "subscription.uncanceled",
    SubscriptionRevoked: "subscription.revoked",
    RefundCreated: "refund.created",
    RefundUpdated: "refund.updated",
    ProductCreated: "product.created",
    ProductUpdated: "product.updated",
    BenefitCreated: "benefit.created",
    BenefitUpdated: "benefit.updated",
    BenefitGrantCreated: "benefit_grant.created",
    BenefitGrantCycled: "benefit_grant.cycled",
    BenefitGrantUpdated: "benefit_grant.updated",
    BenefitGrantRevoked: "benefit_grant.revoked",
    OrganizationUpdated: "organization.updated",
};
/** @internal */
exports.WebhookEventType$inboundSchema = z.nativeEnum(exports.WebhookEventType);
/** @internal */
exports.WebhookEventType$outboundSchema = exports.WebhookEventType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookEventType$;
(function (WebhookEventType$) {
    /** @deprecated use `WebhookEventType$inboundSchema` instead. */
    WebhookEventType$.inboundSchema = exports.WebhookEventType$inboundSchema;
    /** @deprecated use `WebhookEventType$outboundSchema` instead. */
    WebhookEventType$.outboundSchema = exports.WebhookEventType$outboundSchema;
})(WebhookEventType$ || (exports.WebhookEventType$ = WebhookEventType$ = {}));
