// @polar-sh/sdk/dist/commonjs/models/components/webhookendpoint.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEndpoint$ = exports.WebhookEndpoint$outboundSchema = exports.WebhookEndpoint$inboundSchema = void 0;
exports.webhookEndpointToJSON = webhookEndpointToJSON;
exports.webhookEndpointFromJSON = webhookEndpointFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const webhookeventtype_js_1 = require("./webhookeventtype.js");
const webhookformat_js_1 = require("./webhookformat.js");
/** @internal */
exports.WebhookEndpoint$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    url: z.string(),
    format: webhookformat_js_1.WebhookFormat$inboundSchema,
    secret: z.string(),
    organization_id: z.string(),
    events: z.array(webhookeventtype_js_1.WebhookEventType$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.WebhookEndpoint$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    url: z.string(),
    format: webhookformat_js_1.WebhookFormat$outboundSchema,
    secret: z.string(),
    organizationId: z.string(),
    events: z.array(webhookeventtype_js_1.WebhookEventType$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookEndpoint$;
(function (WebhookEndpoint$) {
    /** @deprecated use `WebhookEndpoint$inboundSchema` instead. */
    WebhookEndpoint$.inboundSchema = exports.WebhookEndpoint$inboundSchema;
    /** @deprecated use `WebhookEndpoint$outboundSchema` instead. */
    WebhookEndpoint$.outboundSchema = exports.WebhookEndpoint$outboundSchema;
})(WebhookEndpoint$ || (exports.WebhookEndpoint$ = WebhookEndpoint$ = {}));
function webhookEndpointToJSON(webhookEndpoint) {
    return JSON.stringify(exports.WebhookEndpoint$outboundSchema.parse(webhookEndpoint));
}
function webhookEndpointFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEndpoint' from JSON`);
}
