// @polar-sh/sdk/dist/esm/models/operations/checkoutsclientupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CheckoutUpdatePublic$inboundSchema, CheckoutUpdatePublic$outboundSchema, } from "../components/checkoutupdatepublic.js";
/** @internal */
export const CheckoutsClientUpdateRequest$inboundSchema = z.object({
    client_secret: z.string(),
    CheckoutUpdatePublic: CheckoutUpdatePublic$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "client_secret": "clientSecret",
        "CheckoutUpdatePublic": "checkoutUpdatePublic",
    });
});
/** @internal */
export const CheckoutsClientUpdateRequest$outboundSchema = z.object({
    clientSecret: z.string(),
    checkoutUpdatePublic: CheckoutUpdatePublic$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        clientSecret: "client_secret",
        checkoutUpdatePublic: "CheckoutUpdatePublic",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutsClientUpdateRequest$;
(function (CheckoutsClientUpdateRequest$) {
    /** @deprecated use `CheckoutsClientUpdateRequest$inboundSchema` instead. */
    CheckoutsClientUpdateRequest$.inboundSchema = CheckoutsClientUpdateRequest$inboundSchema;
    /** @deprecated use `CheckoutsClientUpdateRequest$outboundSchema` instead. */
    CheckoutsClientUpdateRequest$.outboundSchema = CheckoutsClientUpdateRequest$outboundSchema;
})(CheckoutsClientUpdateRequest$ || (CheckoutsClientUpdateRequest$ = {}));
export function checkoutsClientUpdateRequestToJSON(checkoutsClientUpdateRequest) {
    return JSON.stringify(CheckoutsClientUpdateRequest$outboundSchema.parse(checkoutsClientUpdateRequest));
}
export function checkoutsClientUpdateRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutsClientUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsClientUpdateRequest' from JSON`);
}
