// @polar-sh/sdk/dist/esm/models/operations/checkoutsclientconfirm.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CheckoutConfirmStripe$inboundSchema, CheckoutConfirmStripe$outboundSchema, } from "../components/checkoutconfirmstripe.js";
/** @internal */
export const CheckoutsClientConfirmRequest$inboundSchema = z.object({
    client_secret: z.string(),
    CheckoutConfirmStripe: CheckoutConfirmStripe$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "client_secret": "clientSecret",
        "CheckoutConfirmStripe": "checkoutConfirmStripe",
    });
});
/** @internal */
export const CheckoutsClientConfirmRequest$outboundSchema = z.object({
    clientSecret: z.string(),
    checkoutConfirmStripe: CheckoutConfirmStripe$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        clientSecret: "client_secret",
        checkoutConfirmStripe: "CheckoutConfirmStripe",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutsClientConfirmRequest$;
(function (CheckoutsClientConfirmRequest$) {
    /** @deprecated use `CheckoutsClientConfirmRequest$inboundSchema` instead. */
    CheckoutsClientConfirmRequest$.inboundSchema = CheckoutsClientConfirmRequest$inboundSchema;
    /** @deprecated use `CheckoutsClientConfirmRequest$outboundSchema` instead. */
    CheckoutsClientConfirmRequest$.outboundSchema = CheckoutsClientConfirmRequest$outboundSchema;
})(CheckoutsClientConfirmRequest$ || (CheckoutsClientConfirmRequest$ = {}));
export function checkoutsClientConfirmRequestToJSON(checkoutsClientConfirmRequest) {
    return JSON.stringify(CheckoutsClientConfirmRequest$outboundSchema.parse(checkoutsClientConfirmRequest));
}
export function checkoutsClientConfirmRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutsClientConfirmRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsClientConfirmRequest' from JSON`);
}
