// @polar-sh/sdk/dist/esm/models/operations/checkoutsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CheckoutsGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CheckoutsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutsGetRequest$;
(function (CheckoutsGetRequest$) {
    /** @deprecated use `CheckoutsGetRequest$inboundSchema` instead. */
    CheckoutsGetRequest$.inboundSchema = CheckoutsGetRequest$inboundSchema;
    /** @deprecated use `CheckoutsGetRequest$outboundSchema` instead. */
    CheckoutsGetRequest$.outboundSchema = CheckoutsGetRequest$outboundSchema;
})(CheckoutsGetRequest$ || (CheckoutsGetRequest$ = {}));
export function checkoutsGetRequestToJSON(checkoutsGetRequest) {
    return JSON.stringify(CheckoutsGetRequest$outboundSchema.parse(checkoutsGetRequest));
}
export function checkoutsGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsGetRequest' from JSON`);
}
