// @polar-sh/sdk/dist/esm/models/operations/checkoutlinksdelete.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CheckoutLinksDeleteRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CheckoutLinksDeleteRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinksDeleteRequest$;
(function (CheckoutLinksDeleteRequest$) {
    /** @deprecated use `CheckoutLinksDeleteRequest$inboundSchema` instead. */
    CheckoutLinksDeleteRequest$.inboundSchema = CheckoutLinksDeleteRequest$inboundSchema;
    /** @deprecated use `CheckoutLinksDeleteRequest$outboundSchema` instead. */
    CheckoutLinksDeleteRequest$.outboundSchema = CheckoutLinksDeleteRequest$outboundSchema;
})(CheckoutLinksDeleteRequest$ || (CheckoutLinksDeleteRequest$ = {}));
export function checkoutLinksDeleteRequestToJSON(checkoutLinksDeleteRequest) {
    return JSON.stringify(CheckoutLinksDeleteRequest$outboundSchema.parse(checkoutLinksDeleteRequest));
}
export function checkoutLinksDeleteRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinksDeleteRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksDeleteRequest' from JSON`);
}
