// @polar-sh/sdk/dist/esm/models/operations/customerportalordersgetpaymentstatus.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersGetPaymentStatusSecurity$;
(function (CustomerPortalOrdersGetPaymentStatusSecurity$) {
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusSecurity$.inboundSchema = CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusSecurity$.outboundSchema = CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema;
})(CustomerPortalOrdersGetPaymentStatusSecurity$ || (CustomerPortalOrdersGetPaymentStatusSecurity$ = {}));
export function customerPortalOrdersGetPaymentStatusSecurityToJSON(customerPortalOrdersGetPaymentStatusSecurity) {
    return JSON.stringify(CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema.parse(customerPortalOrdersGetPaymentStatusSecurity));
}
export function customerPortalOrdersGetPaymentStatusSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGetPaymentStatusSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersGetPaymentStatusRequest$;
(function (CustomerPortalOrdersGetPaymentStatusRequest$) {
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusRequest$.inboundSchema = CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusRequest$.outboundSchema = CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema;
})(CustomerPortalOrdersGetPaymentStatusRequest$ || (CustomerPortalOrdersGetPaymentStatusRequest$ = {}));
export function customerPortalOrdersGetPaymentStatusRequestToJSON(customerPortalOrdersGetPaymentStatusRequest) {
    return JSON.stringify(CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema.parse(customerPortalOrdersGetPaymentStatusRequest));
}
export function customerPortalOrdersGetPaymentStatusRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGetPaymentStatusRequest' from JSON`);
}
