// @polar-sh/sdk/dist/esm/models/operations/customerportalbenefitgrantsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalBenefitGrantsGetSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalBenefitGrantsGetSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalBenefitGrantsGetSecurity$;
(function (CustomerPortalBenefitGrantsGetSecurity$) {
    /** @deprecated use `CustomerPortalBenefitGrantsGetSecurity$inboundSchema` instead. */
    CustomerPortalBenefitGrantsGetSecurity$.inboundSchema = CustomerPortalBenefitGrantsGetSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsGetSecurity$outboundSchema` instead. */
    CustomerPortalBenefitGrantsGetSecurity$.outboundSchema = CustomerPortalBenefitGrantsGetSecurity$outboundSchema;
})(CustomerPortalBenefitGrantsGetSecurity$ || (CustomerPortalBenefitGrantsGetSecurity$ = {}));
export function customerPortalBenefitGrantsGetSecurityToJSON(customerPortalBenefitGrantsGetSecurity) {
    return JSON.stringify(CustomerPortalBenefitGrantsGetSecurity$outboundSchema.parse(customerPortalBenefitGrantsGetSecurity));
}
export function customerPortalBenefitGrantsGetSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalBenefitGrantsGetSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsGetSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalBenefitGrantsGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalBenefitGrantsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalBenefitGrantsGetRequest$;
(function (CustomerPortalBenefitGrantsGetRequest$) {
    /** @deprecated use `CustomerPortalBenefitGrantsGetRequest$inboundSchema` instead. */
    CustomerPortalBenefitGrantsGetRequest$.inboundSchema = CustomerPortalBenefitGrantsGetRequest$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsGetRequest$outboundSchema` instead. */
    CustomerPortalBenefitGrantsGetRequest$.outboundSchema = CustomerPortalBenefitGrantsGetRequest$outboundSchema;
})(CustomerPortalBenefitGrantsGetRequest$ || (CustomerPortalBenefitGrantsGetRequest$ = {}));
export function customerPortalBenefitGrantsGetRequestToJSON(customerPortalBenefitGrantsGetRequest) {
    return JSON.stringify(CustomerPortalBenefitGrantsGetRequest$outboundSchema.parse(customerPortalBenefitGrantsGetRequest));
}
export function customerPortalBenefitGrantsGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalBenefitGrantsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsGetRequest' from JSON`);
}
