// @polar-sh/sdk/dist/esm/models/operations/customerportalsubscriptionsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSubscriptionsGetSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalSubscriptionsGetSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSubscriptionsGetSecurity$;
(function (CustomerPortalSubscriptionsGetSecurity$) {
    /** @deprecated use `CustomerPortalSubscriptionsGetSecurity$inboundSchema` instead. */
    CustomerPortalSubscriptionsGetSecurity$.inboundSchema = CustomerPortalSubscriptionsGetSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalSubscriptionsGetSecurity$outboundSchema` instead. */
    CustomerPortalSubscriptionsGetSecurity$.outboundSchema = CustomerPortalSubscriptionsGetSecurity$outboundSchema;
})(CustomerPortalSubscriptionsGetSecurity$ || (CustomerPortalSubscriptionsGetSecurity$ = {}));
export function customerPortalSubscriptionsGetSecurityToJSON(customerPortalSubscriptionsGetSecurity) {
    return JSON.stringify(CustomerPortalSubscriptionsGetSecurity$outboundSchema.parse(customerPortalSubscriptionsGetSecurity));
}
export function customerPortalSubscriptionsGetSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSubscriptionsGetSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsGetSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalSubscriptionsGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalSubscriptionsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalSubscriptionsGetRequest$;
(function (CustomerPortalSubscriptionsGetRequest$) {
    /** @deprecated use `CustomerPortalSubscriptionsGetRequest$inboundSchema` instead. */
    CustomerPortalSubscriptionsGetRequest$.inboundSchema = CustomerPortalSubscriptionsGetRequest$inboundSchema;
    /** @deprecated use `CustomerPortalSubscriptionsGetRequest$outboundSchema` instead. */
    CustomerPortalSubscriptionsGetRequest$.outboundSchema = CustomerPortalSubscriptionsGetRequest$outboundSchema;
})(CustomerPortalSubscriptionsGetRequest$ || (CustomerPortalSubscriptionsGetRequest$ = {}));
export function customerPortalSubscriptionsGetRequestToJSON(customerPortalSubscriptionsGetRequest) {
    return JSON.stringify(CustomerPortalSubscriptionsGetRequest$outboundSchema.parse(customerPortalSubscriptionsGetRequest));
}
export function customerPortalSubscriptionsGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSubscriptionsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsGetRequest' from JSON`);
}
