// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomersget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalCustomersGetSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalCustomersGetSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalCustomersGetSecurity$;
(function (CustomerPortalCustomersGetSecurity$) {
    /** @deprecated use `CustomerPortalCustomersGetSecurity$inboundSchema` instead. */
    CustomerPortalCustomersGetSecurity$.inboundSchema = CustomerPortalCustomersGetSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersGetSecurity$outboundSchema` instead. */
    CustomerPortalCustomersGetSecurity$.outboundSchema = CustomerPortalCustomersGetSecurity$outboundSchema;
})(CustomerPortalCustomersGetSecurity$ || (CustomerPortalCustomersGetSecurity$ = {}));
export function customerPortalCustomersGetSecurityToJSON(customerPortalCustomersGetSecurity) {
    return JSON.stringify(CustomerPortalCustomersGetSecurity$outboundSchema.parse(customerPortalCustomersGetSecurity));
}
export function customerPortalCustomersGetSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomersGetSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersGetSecurity' from JSON`);
}
