// @polar-sh/sdk/dist/esm/models/operations/customersgetexternal.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomersGetExternalRequest$inboundSchema = z.object({
    external_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "external_id": "externalId",
    });
});
/** @internal */
export const CustomersGetExternalRequest$outboundSchema = z.object({
    externalId: z.string(),
}).transform((v) => {
    return remap$(v, {
        externalId: "external_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomersGetExternalRequest$;
(function (CustomersGetExternalRequest$) {
    /** @deprecated use `CustomersGetExternalRequest$inboundSchema` instead. */
    CustomersGetExternalRequest$.inboundSchema = CustomersGetExternalRequest$inboundSchema;
    /** @deprecated use `CustomersGetExternalRequest$outboundSchema` instead. */
    CustomersGetExternalRequest$.outboundSchema = CustomersGetExternalRequest$outboundSchema;
})(CustomersGetExternalRequest$ || (CustomersGetExternalRequest$ = {}));
export function customersGetExternalRequestToJSON(customersGetExternalRequest) {
    return JSON.stringify(CustomersGetExternalRequest$outboundSchema.parse(customersGetExternalRequest));
}
export function customersGetExternalRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomersGetExternalRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersGetExternalRequest' from JSON`);
}
