// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomersdeletepaymentmethod.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalCustomersDeletePaymentMethodSecurity$;
(function (CustomerPortalCustomersDeletePaymentMethodSecurity$) {
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodSecurity$.inboundSchema = CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodSecurity$.outboundSchema = CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema;
})(CustomerPortalCustomersDeletePaymentMethodSecurity$ || (CustomerPortalCustomersDeletePaymentMethodSecurity$ = {}));
export function customerPortalCustomersDeletePaymentMethodSecurityToJSON(customerPortalCustomersDeletePaymentMethodSecurity) {
    return JSON.stringify(CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema.parse(customerPortalCustomersDeletePaymentMethodSecurity));
}
export function customerPortalCustomersDeletePaymentMethodSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersDeletePaymentMethodSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalCustomersDeletePaymentMethodRequest$;
(function (CustomerPortalCustomersDeletePaymentMethodRequest$) {
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodRequest$.inboundSchema = CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodRequest$.outboundSchema = CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema;
})(CustomerPortalCustomersDeletePaymentMethodRequest$ || (CustomerPortalCustomersDeletePaymentMethodRequest$ = {}));
export function customerPortalCustomersDeletePaymentMethodRequestToJSON(customerPortalCustomersDeletePaymentMethodRequest) {
    return JSON.stringify(CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema.parse(customerPortalCustomersDeletePaymentMethodRequest));
}
export function customerPortalCustomersDeletePaymentMethodRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersDeletePaymentMethodRequest' from JSON`);
}
