// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomersaddpaymentmethod.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalCustomersAddPaymentMethodSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalCustomersAddPaymentMethodSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalCustomersAddPaymentMethodSecurity$;
(function (CustomerPortalCustomersAddPaymentMethodSecurity$) {
    /** @deprecated use `CustomerPortalCustomersAddPaymentMethodSecurity$inboundSchema` instead. */
    CustomerPortalCustomersAddPaymentMethodSecurity$.inboundSchema = CustomerPortalCustomersAddPaymentMethodSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersAddPaymentMethodSecurity$outboundSchema` instead. */
    CustomerPortalCustomersAddPaymentMethodSecurity$.outboundSchema = CustomerPortalCustomersAddPaymentMethodSecurity$outboundSchema;
})(CustomerPortalCustomersAddPaymentMethodSecurity$ || (CustomerPortalCustomersAddPaymentMethodSecurity$ = {}));
export function customerPortalCustomersAddPaymentMethodSecurityToJSON(customerPortalCustomersAddPaymentMethodSecurity) {
    return JSON.stringify(CustomerPortalCustomersAddPaymentMethodSecurity$outboundSchema.parse(customerPortalCustomersAddPaymentMethodSecurity));
}
export function customerPortalCustomersAddPaymentMethodSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomersAddPaymentMethodSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersAddPaymentMethodSecurity' from JSON`);
}
