// @polar-sh/sdk/dist/esm/models/operations/discountsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const DiscountsGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const DiscountsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DiscountsGetRequest$;
(function (DiscountsGetRequest$) {
    /** @deprecated use `DiscountsGetRequest$inboundSchema` instead. */
    DiscountsGetRequest$.inboundSchema = DiscountsGetRequest$inboundSchema;
    /** @deprecated use `DiscountsGetRequest$outboundSchema` instead. */
    DiscountsGetRequest$.outboundSchema = DiscountsGetRequest$outboundSchema;
})(DiscountsGetRequest$ || (DiscountsGetRequest$ = {}));
export function discountsGetRequestToJSON(discountsGetRequest) {
    return JSON.stringify(DiscountsGetRequest$outboundSchema.parse(discountsGetRequest));
}
export function discountsGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountsGetRequest' from JSON`);
}
