// @polar-sh/sdk/dist/esm/models/operations/fileslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceFileRead$inboundSchema, ListResourceFileRead$outboundSchema, } from "../components/listresourcefileread.js";
/** @internal */
export const FilesListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const FilesListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FilesListQueryParamOrganizationIDFilter$;
(function (FilesListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `FilesListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    FilesListQueryParamOrganizationIDFilter$.inboundSchema = FilesListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `FilesListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    FilesListQueryParamOrganizationIDFilter$.outboundSchema = FilesListQueryParamOrganizationIDFilter$outboundSchema;
})(FilesListQueryParamOrganizationIDFilter$ || (FilesListQueryParamOrganizationIDFilter$ = {}));
export function filesListQueryParamOrganizationIDFilterToJSON(filesListQueryParamOrganizationIDFilter) {
    return JSON.stringify(FilesListQueryParamOrganizationIDFilter$outboundSchema.parse(filesListQueryParamOrganizationIDFilter));
}
export function filesListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilesListQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
export const FileIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const FileIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FileIDFilter$;
(function (FileIDFilter$) {
    /** @deprecated use `FileIDFilter$inboundSchema` instead. */
    FileIDFilter$.inboundSchema = FileIDFilter$inboundSchema;
    /** @deprecated use `FileIDFilter$outboundSchema` instead. */
    FileIDFilter$.outboundSchema = FileIDFilter$outboundSchema;
})(FileIDFilter$ || (FileIDFilter$ = {}));
export function fileIDFilterToJSON(fileIDFilter) {
    return JSON.stringify(FileIDFilter$outboundSchema.parse(fileIDFilter));
}
export function fileIDFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FileIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileIDFilter' from JSON`);
}
/** @internal */
export const FilesListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    ids: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
export const FilesListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    ids: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FilesListRequest$;
(function (FilesListRequest$) {
    /** @deprecated use `FilesListRequest$inboundSchema` instead. */
    FilesListRequest$.inboundSchema = FilesListRequest$inboundSchema;
    /** @deprecated use `FilesListRequest$outboundSchema` instead. */
    FilesListRequest$.outboundSchema = FilesListRequest$outboundSchema;
})(FilesListRequest$ || (FilesListRequest$ = {}));
export function filesListRequestToJSON(filesListRequest) {
    return JSON.stringify(FilesListRequest$outboundSchema.parse(filesListRequest));
}
export function filesListRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilesListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesListRequest' from JSON`);
}
/** @internal */
export const FilesListResponse$inboundSchema = z.object({
    Result: ListResourceFileRead$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
/** @internal */
export const FilesListResponse$outboundSchema = z.object({
    result: ListResourceFileRead$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FilesListResponse$;
(function (FilesListResponse$) {
    /** @deprecated use `FilesListResponse$inboundSchema` instead. */
    FilesListResponse$.inboundSchema = FilesListResponse$inboundSchema;
    /** @deprecated use `FilesListResponse$outboundSchema` instead. */
    FilesListResponse$.outboundSchema = FilesListResponse$outboundSchema;
})(FilesListResponse$ || (FilesListResponse$ = {}));
export function filesListResponseToJSON(filesListResponse) {
    return JSON.stringify(FilesListResponse$outboundSchema.parse(filesListResponse));
}
export function filesListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesListResponse' from JSON`);
}
