// @polar-sh/sdk/dist/esm/models/operations/webhooksgetwebhookendpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const WebhooksGetWebhookEndpointRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const WebhooksGetWebhookEndpointRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhooksGetWebhookEndpointRequest$;
(function (WebhooksGetWebhookEndpointRequest$) {
    /** @deprecated use `WebhooksGetWebhookEndpointRequest$inboundSchema` instead. */
    WebhooksGetWebhookEndpointRequest$.inboundSchema = WebhooksGetWebhookEndpointRequest$inboundSchema;
    /** @deprecated use `WebhooksGetWebhookEndpointRequest$outboundSchema` instead. */
    WebhooksGetWebhookEndpointRequest$.outboundSchema = WebhooksGetWebhookEndpointRequest$outboundSchema;
})(WebhooksGetWebhookEndpointRequest$ || (WebhooksGetWebhookEndpointRequest$ = {}));
export function webhooksGetWebhookEndpointRequestToJSON(webhooksGetWebhookEndpointRequest) {
    return JSON.stringify(WebhooksGetWebhookEndpointRequest$outboundSchema.parse(webhooksGetWebhookEndpointRequest));
}
export function webhooksGetWebhookEndpointRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhooksGetWebhookEndpointRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksGetWebhookEndpointRequest' from JSON`);
}
