// @polar-sh/sdk/dist/commonjs/sdk/customerportal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortal = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const benefitgrants_js_1 = require("./benefitgrants.js");
const downloadables_js_1 = require("./downloadables.js");
const polarcustomermeters_js_1 = require("./polarcustomermeters.js");
const polarcustomers_js_1 = require("./polarcustomers.js");
const polarlicensekeys_js_1 = require("./polarlicensekeys.js");
const polarorders_js_1 = require("./polarorders.js");
const polarorganizations_js_1 = require("./polarorganizations.js");
const polarsubscriptions_js_1 = require("./polarsubscriptions.js");
class CustomerPortal extends sdks_js_1.ClientSDK {
    get benefitGrants() {
        return (this._benefitGrants ?? (this._benefitGrants = new benefitgrants_js_1.BenefitGrants(this._options)));
    }
    get customers() {
        return (this._customers ?? (this._customers = new polarcustomers_js_1.PolarCustomers(this._options)));
    }
    get customerMeters() {
        return (this._customerMeters ?? (this._customerMeters = new polarcustomermeters_js_1.PolarCustomerMeters(this._options)));
    }
    get downloadables() {
        return (this._downloadables ?? (this._downloadables = new downloadables_js_1.Downloadables(this._options)));
    }
    get licenseKeys() {
        return (this._licenseKeys ?? (this._licenseKeys = new polarlicensekeys_js_1.PolarLicenseKeys(this._options)));
    }
    get orders() {
        return (this._orders ?? (this._orders = new polarorders_js_1.PolarOrders(this._options)));
    }
    get organizations() {
        return (this._organizations ?? (this._organizations = new polarorganizations_js_1.PolarOrganizations(this._options)));
    }
    get subscriptions() {
        return (this._subscriptions ?? (this._subscriptions = new polarsubscriptions_js_1.PolarSubscriptions(this._options)));
    }
}
exports.CustomerPortal = CustomerPortal;
