// @polar-sh/sdk/dist/commonjs/sdk/licensekeys.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeys = void 0;
const licenseKeysActivate_js_1 = require("../funcs/licenseKeysActivate.js");
const licenseKeysDeactivate_js_1 = require("../funcs/licenseKeysDeactivate.js");
const licenseKeysGet_js_1 = require("../funcs/licenseKeysGet.js");
const licenseKeysGetActivation_js_1 = require("../funcs/licenseKeysGetActivation.js");
const licenseKeysList_js_1 = require("../funcs/licenseKeysList.js");
const licenseKeysUpdate_js_1 = require("../funcs/licenseKeysUpdate.js");
const licenseKeysValidate_js_1 = require("../funcs/licenseKeysValidate.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const operations_js_1 = require("../types/operations.js");
class LicenseKeys extends sdks_js_1.ClientSDK {
    /**
     * List License Keys
     *
     * @remarks
     * Get license keys connected to the given organization & filters.
     *
     * **Scopes**: `license_keys:read` `license_keys:write`
     */
    async list(request, options) {
        return (0, operations_js_1.unwrapResultIterator)((0, licenseKeysList_js_1.licenseKeysList)(this, request, options));
    }
    /**
     * Get License Key
     *
     * @remarks
     * Get a license key.
     *
     * **Scopes**: `license_keys:read` `license_keys:write`
     */
    async get(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, licenseKeysGet_js_1.licenseKeysGet)(this, request, options));
    }
    /**
     * Update License Key
     *
     * @remarks
     * Update a license key.
     *
     * **Scopes**: `license_keys:write`
     */
    async update(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, licenseKeysUpdate_js_1.licenseKeysUpdate)(this, request, options));
    }
    /**
     * Get Activation
     *
     * @remarks
     * Get a license key activation.
     *
     * **Scopes**: `license_keys:read` `license_keys:write`
     */
    async getActivation(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, licenseKeysGetActivation_js_1.licenseKeysGetActivation)(this, request, options));
    }
    /**
     * Validate License Key
     *
     * @remarks
     * Validate a license key.
     *
     * **Scopes**: `license_keys:write`
     */
    async validate(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, licenseKeysValidate_js_1.licenseKeysValidate)(this, request, options));
    }
    /**
     * Activate License Key
     *
     * @remarks
     * Activate a license key instance.
     *
     * **Scopes**: `license_keys:write`
     */
    async activate(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, licenseKeysActivate_js_1.licenseKeysActivate)(this, request, options));
    }
    /**
     * Deactivate License Key
     *
     * @remarks
     * Deactivate a license key instance.
     *
     * **Scopes**: `license_keys:write`
     */
    async deactivate(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, licenseKeysDeactivate_js_1.licenseKeysDeactivate)(this, request, options));
    }
}
exports.LicenseKeys = LicenseKeys;
