// @polar-sh/sdk/dist/commonjs/models/operations/checkoutslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutsListResponse$ = exports.CheckoutsListResponse$outboundSchema = exports.CheckoutsListResponse$inboundSchema = exports.CheckoutsListRequest$ = exports.CheckoutsListRequest$outboundSchema = exports.CheckoutsListRequest$inboundSchema = exports.StatusFilter$ = exports.StatusFilter$outboundSchema = exports.StatusFilter$inboundSchema = exports.CheckoutsListQueryParamCustomerIDFilter$ = exports.CheckoutsListQueryParamCustomerIDFilter$outboundSchema = exports.CheckoutsListQueryParamCustomerIDFilter$inboundSchema = exports.CheckoutsListQueryParamProductIDFilter$ = exports.CheckoutsListQueryParamProductIDFilter$outboundSchema = exports.CheckoutsListQueryParamProductIDFilter$inboundSchema = exports.CheckoutsListQueryParamOrganizationIDFilter$ = exports.CheckoutsListQueryParamOrganizationIDFilter$outboundSchema = exports.CheckoutsListQueryParamOrganizationIDFilter$inboundSchema = void 0;
exports.checkoutsListQueryParamOrganizationIDFilterToJSON = checkoutsListQueryParamOrganizationIDFilterToJSON;
exports.checkoutsListQueryParamOrganizationIDFilterFromJSON = checkoutsListQueryParamOrganizationIDFilterFromJSON;
exports.checkoutsListQueryParamProductIDFilterToJSON = checkoutsListQueryParamProductIDFilterToJSON;
exports.checkoutsListQueryParamProductIDFilterFromJSON = checkoutsListQueryParamProductIDFilterFromJSON;
exports.checkoutsListQueryParamCustomerIDFilterToJSON = checkoutsListQueryParamCustomerIDFilterToJSON;
exports.checkoutsListQueryParamCustomerIDFilterFromJSON = checkoutsListQueryParamCustomerIDFilterFromJSON;
exports.statusFilterToJSON = statusFilterToJSON;
exports.statusFilterFromJSON = statusFilterFromJSON;
exports.checkoutsListRequestToJSON = checkoutsListRequestToJSON;
exports.checkoutsListRequestFromJSON = checkoutsListRequestFromJSON;
exports.checkoutsListResponseToJSON = checkoutsListResponseToJSON;
exports.checkoutsListResponseFromJSON = checkoutsListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const checkoutsortproperty_js_1 = require("../components/checkoutsortproperty.js");
const checkoutstatus_js_1 = require("../components/checkoutstatus.js");
const listresourcecheckout_js_1 = require("../components/listresourcecheckout.js");
/** @internal */
exports.CheckoutsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsListQueryParamOrganizationIDFilter$;
(function (CheckoutsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `CheckoutsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    CheckoutsListQueryParamOrganizationIDFilter$.inboundSchema = exports.CheckoutsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `CheckoutsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    CheckoutsListQueryParamOrganizationIDFilter$.outboundSchema = exports.CheckoutsListQueryParamOrganizationIDFilter$outboundSchema;
})(CheckoutsListQueryParamOrganizationIDFilter$ || (exports.CheckoutsListQueryParamOrganizationIDFilter$ = CheckoutsListQueryParamOrganizationIDFilter$ = {}));
function checkoutsListQueryParamOrganizationIDFilterToJSON(checkoutsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.CheckoutsListQueryParamOrganizationIDFilter$outboundSchema.parse(checkoutsListQueryParamOrganizationIDFilter));
}
function checkoutsListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsListQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.CheckoutsListQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutsListQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsListQueryParamProductIDFilter$;
(function (CheckoutsListQueryParamProductIDFilter$) {
    /** @deprecated use `CheckoutsListQueryParamProductIDFilter$inboundSchema` instead. */
    CheckoutsListQueryParamProductIDFilter$.inboundSchema = exports.CheckoutsListQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `CheckoutsListQueryParamProductIDFilter$outboundSchema` instead. */
    CheckoutsListQueryParamProductIDFilter$.outboundSchema = exports.CheckoutsListQueryParamProductIDFilter$outboundSchema;
})(CheckoutsListQueryParamProductIDFilter$ || (exports.CheckoutsListQueryParamProductIDFilter$ = CheckoutsListQueryParamProductIDFilter$ = {}));
function checkoutsListQueryParamProductIDFilterToJSON(checkoutsListQueryParamProductIDFilter) {
    return JSON.stringify(exports.CheckoutsListQueryParamProductIDFilter$outboundSchema.parse(checkoutsListQueryParamProductIDFilter));
}
function checkoutsListQueryParamProductIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsListQueryParamProductIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsListQueryParamProductIDFilter' from JSON`);
}
/** @internal */
exports.CheckoutsListQueryParamCustomerIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsListQueryParamCustomerIDFilter$;
(function (CheckoutsListQueryParamCustomerIDFilter$) {
    /** @deprecated use `CheckoutsListQueryParamCustomerIDFilter$inboundSchema` instead. */
    CheckoutsListQueryParamCustomerIDFilter$.inboundSchema = exports.CheckoutsListQueryParamCustomerIDFilter$inboundSchema;
    /** @deprecated use `CheckoutsListQueryParamCustomerIDFilter$outboundSchema` instead. */
    CheckoutsListQueryParamCustomerIDFilter$.outboundSchema = exports.CheckoutsListQueryParamCustomerIDFilter$outboundSchema;
})(CheckoutsListQueryParamCustomerIDFilter$ || (exports.CheckoutsListQueryParamCustomerIDFilter$ = CheckoutsListQueryParamCustomerIDFilter$ = {}));
function checkoutsListQueryParamCustomerIDFilterToJSON(checkoutsListQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.CheckoutsListQueryParamCustomerIDFilter$outboundSchema.parse(checkoutsListQueryParamCustomerIDFilter));
}
function checkoutsListQueryParamCustomerIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsListQueryParamCustomerIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsListQueryParamCustomerIDFilter' from JSON`);
}
/** @internal */
exports.StatusFilter$inboundSchema = z.union([
    checkoutstatus_js_1.CheckoutStatus$inboundSchema,
    z.array(checkoutstatus_js_1.CheckoutStatus$inboundSchema),
]);
/** @internal */
exports.StatusFilter$outboundSchema = z.union([
    checkoutstatus_js_1.CheckoutStatus$outboundSchema,
    z.array(checkoutstatus_js_1.CheckoutStatus$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StatusFilter$;
(function (StatusFilter$) {
    /** @deprecated use `StatusFilter$inboundSchema` instead. */
    StatusFilter$.inboundSchema = exports.StatusFilter$inboundSchema;
    /** @deprecated use `StatusFilter$outboundSchema` instead. */
    StatusFilter$.outboundSchema = exports.StatusFilter$outboundSchema;
})(StatusFilter$ || (exports.StatusFilter$ = StatusFilter$ = {}));
function statusFilterToJSON(statusFilter) {
    return JSON.stringify(exports.StatusFilter$outboundSchema.parse(statusFilter));
}
function statusFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StatusFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StatusFilter' from JSON`);
}
/** @internal */
exports.CheckoutsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    customer_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    status: z.nullable(z.union([
        checkoutstatus_js_1.CheckoutStatus$inboundSchema,
        z.array(checkoutstatus_js_1.CheckoutStatus$inboundSchema),
    ])).optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(checkoutsortproperty_js_1.CheckoutSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
        "customer_id": "customerId",
    });
});
/** @internal */
exports.CheckoutsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    customerId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    status: z.nullable(z.union([
        checkoutstatus_js_1.CheckoutStatus$outboundSchema,
        z.array(checkoutstatus_js_1.CheckoutStatus$outboundSchema),
    ])).optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(checkoutsortproperty_js_1.CheckoutSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
        customerId: "customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsListRequest$;
(function (CheckoutsListRequest$) {
    /** @deprecated use `CheckoutsListRequest$inboundSchema` instead. */
    CheckoutsListRequest$.inboundSchema = exports.CheckoutsListRequest$inboundSchema;
    /** @deprecated use `CheckoutsListRequest$outboundSchema` instead. */
    CheckoutsListRequest$.outboundSchema = exports.CheckoutsListRequest$outboundSchema;
})(CheckoutsListRequest$ || (exports.CheckoutsListRequest$ = CheckoutsListRequest$ = {}));
function checkoutsListRequestToJSON(checkoutsListRequest) {
    return JSON.stringify(exports.CheckoutsListRequest$outboundSchema.parse(checkoutsListRequest));
}
function checkoutsListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsListRequest' from JSON`);
}
/** @internal */
exports.CheckoutsListResponse$inboundSchema = z.object({
    Result: listresourcecheckout_js_1.ListResourceCheckout$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CheckoutsListResponse$outboundSchema = z.object({
    result: listresourcecheckout_js_1.ListResourceCheckout$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsListResponse$;
(function (CheckoutsListResponse$) {
    /** @deprecated use `CheckoutsListResponse$inboundSchema` instead. */
    CheckoutsListResponse$.inboundSchema = exports.CheckoutsListResponse$inboundSchema;
    /** @deprecated use `CheckoutsListResponse$outboundSchema` instead. */
    CheckoutsListResponse$.outboundSchema = exports.CheckoutsListResponse$outboundSchema;
})(CheckoutsListResponse$ || (exports.CheckoutsListResponse$ = CheckoutsListResponse$ = {}));
function checkoutsListResponseToJSON(checkoutsListResponse) {
    return JSON.stringify(exports.CheckoutsListResponse$outboundSchema.parse(checkoutsListResponse));
}
function checkoutsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsListResponse' from JSON`);
}
