// @polar-sh/sdk/dist/commonjs/models/operations/checkoutsclientupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutsClientUpdateRequest$ = exports.CheckoutsClientUpdateRequest$outboundSchema = exports.CheckoutsClientUpdateRequest$inboundSchema = void 0;
exports.checkoutsClientUpdateRequestToJSON = checkoutsClientUpdateRequestToJSON;
exports.checkoutsClientUpdateRequestFromJSON = checkoutsClientUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const checkoutupdatepublic_js_1 = require("../components/checkoutupdatepublic.js");
/** @internal */
exports.CheckoutsClientUpdateRequest$inboundSchema = z.object({
    client_secret: z.string(),
    CheckoutUpdatePublic: checkoutupdatepublic_js_1.CheckoutUpdatePublic$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_secret": "clientSecret",
        "CheckoutUpdatePublic": "checkoutUpdatePublic",
    });
});
/** @internal */
exports.CheckoutsClientUpdateRequest$outboundSchema = z.object({
    clientSecret: z.string(),
    checkoutUpdatePublic: checkoutupdatepublic_js_1.CheckoutUpdatePublic$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientSecret: "client_secret",
        checkoutUpdatePublic: "CheckoutUpdatePublic",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsClientUpdateRequest$;
(function (CheckoutsClientUpdateRequest$) {
    /** @deprecated use `CheckoutsClientUpdateRequest$inboundSchema` instead. */
    CheckoutsClientUpdateRequest$.inboundSchema = exports.CheckoutsClientUpdateRequest$inboundSchema;
    /** @deprecated use `CheckoutsClientUpdateRequest$outboundSchema` instead. */
    CheckoutsClientUpdateRequest$.outboundSchema = exports.CheckoutsClientUpdateRequest$outboundSchema;
})(CheckoutsClientUpdateRequest$ || (exports.CheckoutsClientUpdateRequest$ = CheckoutsClientUpdateRequest$ = {}));
function checkoutsClientUpdateRequestToJSON(checkoutsClientUpdateRequest) {
    return JSON.stringify(exports.CheckoutsClientUpdateRequest$outboundSchema.parse(checkoutsClientUpdateRequest));
}
function checkoutsClientUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsClientUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsClientUpdateRequest' from JSON`);
}
