// @polar-sh/sdk/dist/commonjs/models/operations/checkoutlinkslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinksListResponse$ = exports.CheckoutLinksListResponse$outboundSchema = exports.CheckoutLinksListResponse$inboundSchema = exports.CheckoutLinksListRequest$ = exports.CheckoutLinksListRequest$outboundSchema = exports.CheckoutLinksListRequest$inboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$ = exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$ = exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema = void 0;
exports.checkoutLinksListQueryParamOrganizationIDFilterToJSON = checkoutLinksListQueryParamOrganizationIDFilterToJSON;
exports.checkoutLinksListQueryParamOrganizationIDFilterFromJSON = checkoutLinksListQueryParamOrganizationIDFilterFromJSON;
exports.checkoutLinksListQueryParamProductIDFilterToJSON = checkoutLinksListQueryParamProductIDFilterToJSON;
exports.checkoutLinksListQueryParamProductIDFilterFromJSON = checkoutLinksListQueryParamProductIDFilterFromJSON;
exports.checkoutLinksListRequestToJSON = checkoutLinksListRequestToJSON;
exports.checkoutLinksListRequestFromJSON = checkoutLinksListRequestFromJSON;
exports.checkoutLinksListResponseToJSON = checkoutLinksListResponseToJSON;
exports.checkoutLinksListResponseFromJSON = checkoutLinksListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const checkoutlinksortproperty_js_1 = require("../components/checkoutlinksortproperty.js");
const listresourcecheckoutlink_js_1 = require("../components/listresourcecheckoutlink.js");
/** @internal */
exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListQueryParamOrganizationIDFilter$;
(function (CheckoutLinksListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    CheckoutLinksListQueryParamOrganizationIDFilter$.inboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    CheckoutLinksListQueryParamOrganizationIDFilter$.outboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema;
})(CheckoutLinksListQueryParamOrganizationIDFilter$ || (exports.CheckoutLinksListQueryParamOrganizationIDFilter$ = CheckoutLinksListQueryParamOrganizationIDFilter$ = {}));
function checkoutLinksListQueryParamOrganizationIDFilterToJSON(checkoutLinksListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema.parse(checkoutLinksListQueryParamOrganizationIDFilter));
}
function checkoutLinksListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListQueryParamProductIDFilter$;
(function (CheckoutLinksListQueryParamProductIDFilter$) {
    /** @deprecated use `CheckoutLinksListQueryParamProductIDFilter$inboundSchema` instead. */
    CheckoutLinksListQueryParamProductIDFilter$.inboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `CheckoutLinksListQueryParamProductIDFilter$outboundSchema` instead. */
    CheckoutLinksListQueryParamProductIDFilter$.outboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema;
})(CheckoutLinksListQueryParamProductIDFilter$ || (exports.CheckoutLinksListQueryParamProductIDFilter$ = CheckoutLinksListQueryParamProductIDFilter$ = {}));
function checkoutLinksListQueryParamProductIDFilterToJSON(checkoutLinksListQueryParamProductIDFilter) {
    return JSON.stringify(exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema.parse(checkoutLinksListQueryParamProductIDFilter));
}
function checkoutLinksListQueryParamProductIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksListQueryParamProductIDFilter' from JSON`);
}
/** @internal */
exports.CheckoutLinksListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(checkoutlinksortproperty_js_1.CheckoutLinkSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
    });
});
/** @internal */
exports.CheckoutLinksListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(checkoutlinksortproperty_js_1.CheckoutLinkSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListRequest$;
(function (CheckoutLinksListRequest$) {
    /** @deprecated use `CheckoutLinksListRequest$inboundSchema` instead. */
    CheckoutLinksListRequest$.inboundSchema = exports.CheckoutLinksListRequest$inboundSchema;
    /** @deprecated use `CheckoutLinksListRequest$outboundSchema` instead. */
    CheckoutLinksListRequest$.outboundSchema = exports.CheckoutLinksListRequest$outboundSchema;
})(CheckoutLinksListRequest$ || (exports.CheckoutLinksListRequest$ = CheckoutLinksListRequest$ = {}));
function checkoutLinksListRequestToJSON(checkoutLinksListRequest) {
    return JSON.stringify(exports.CheckoutLinksListRequest$outboundSchema.parse(checkoutLinksListRequest));
}
function checkoutLinksListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinksListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksListRequest' from JSON`);
}
/** @internal */
exports.CheckoutLinksListResponse$inboundSchema = z.object({
    Result: listresourcecheckoutlink_js_1.ListResourceCheckoutLink$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CheckoutLinksListResponse$outboundSchema = z.object({
    result: listresourcecheckoutlink_js_1.ListResourceCheckoutLink$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListResponse$;
(function (CheckoutLinksListResponse$) {
    /** @deprecated use `CheckoutLinksListResponse$inboundSchema` instead. */
    CheckoutLinksListResponse$.inboundSchema = exports.CheckoutLinksListResponse$inboundSchema;
    /** @deprecated use `CheckoutLinksListResponse$outboundSchema` instead. */
    CheckoutLinksListResponse$.outboundSchema = exports.CheckoutLinksListResponse$outboundSchema;
})(CheckoutLinksListResponse$ || (exports.CheckoutLinksListResponse$ = CheckoutLinksListResponse$ = {}));
function checkoutLinksListResponseToJSON(checkoutLinksListResponse) {
    return JSON.stringify(exports.CheckoutLinksListResponse$outboundSchema.parse(checkoutLinksListResponse));
}
function checkoutLinksListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinksListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksListResponse' from JSON`);
}
