// @polar-sh/sdk/dist/commonjs/models/operations/checkoutlinksupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinksUpdateRequest$ = exports.CheckoutLinksUpdateRequest$outboundSchema = exports.CheckoutLinksUpdateRequest$inboundSchema = void 0;
exports.checkoutLinksUpdateRequestToJSON = checkoutLinksUpdateRequestToJSON;
exports.checkoutLinksUpdateRequestFromJSON = checkoutLinksUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const checkoutlinkupdate_js_1 = require("../components/checkoutlinkupdate.js");
/** @internal */
exports.CheckoutLinksUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    CheckoutLinkUpdate: checkoutlinkupdate_js_1.CheckoutLinkUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CheckoutLinkUpdate": "checkoutLinkUpdate",
    });
});
/** @internal */
exports.CheckoutLinksUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    checkoutLinkUpdate: checkoutlinkupdate_js_1.CheckoutLinkUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        checkoutLinkUpdate: "CheckoutLinkUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksUpdateRequest$;
(function (CheckoutLinksUpdateRequest$) {
    /** @deprecated use `CheckoutLinksUpdateRequest$inboundSchema` instead. */
    CheckoutLinksUpdateRequest$.inboundSchema = exports.CheckoutLinksUpdateRequest$inboundSchema;
    /** @deprecated use `CheckoutLinksUpdateRequest$outboundSchema` instead. */
    CheckoutLinksUpdateRequest$.outboundSchema = exports.CheckoutLinksUpdateRequest$outboundSchema;
})(CheckoutLinksUpdateRequest$ || (exports.CheckoutLinksUpdateRequest$ = CheckoutLinksUpdateRequest$ = {}));
function checkoutLinksUpdateRequestToJSON(checkoutLinksUpdateRequest) {
    return JSON.stringify(exports.CheckoutLinksUpdateRequest$outboundSchema.parse(checkoutLinksUpdateRequest));
}
function checkoutLinksUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinksUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksUpdateRequest' from JSON`);
}
