// @polar-sh/sdk/dist/commonjs/models/operations/customerportaldownloadableslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalDownloadablesListResponse$ = exports.CustomerPortalDownloadablesListResponse$outboundSchema = exports.CustomerPortalDownloadablesListResponse$inboundSchema = exports.CustomerPortalDownloadablesListRequest$ = exports.CustomerPortalDownloadablesListRequest$outboundSchema = exports.CustomerPortalDownloadablesListRequest$inboundSchema = exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$ = exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema = exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$inboundSchema = exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$ = exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$outboundSchema = exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$inboundSchema = exports.CustomerPortalDownloadablesListSecurity$ = exports.CustomerPortalDownloadablesListSecurity$outboundSchema = exports.CustomerPortalDownloadablesListSecurity$inboundSchema = void 0;
exports.customerPortalDownloadablesListSecurityToJSON = customerPortalDownloadablesListSecurityToJSON;
exports.customerPortalDownloadablesListSecurityFromJSON = customerPortalDownloadablesListSecurityFromJSON;
exports.customerPortalDownloadablesListQueryParamOrganizationIDFilterToJSON = customerPortalDownloadablesListQueryParamOrganizationIDFilterToJSON;
exports.customerPortalDownloadablesListQueryParamOrganizationIDFilterFromJSON = customerPortalDownloadablesListQueryParamOrganizationIDFilterFromJSON;
exports.customerPortalDownloadablesListQueryParamBenefitIDFilterToJSON = customerPortalDownloadablesListQueryParamBenefitIDFilterToJSON;
exports.customerPortalDownloadablesListQueryParamBenefitIDFilterFromJSON = customerPortalDownloadablesListQueryParamBenefitIDFilterFromJSON;
exports.customerPortalDownloadablesListRequestToJSON = customerPortalDownloadablesListRequestToJSON;
exports.customerPortalDownloadablesListRequestFromJSON = customerPortalDownloadablesListRequestFromJSON;
exports.customerPortalDownloadablesListResponseToJSON = customerPortalDownloadablesListResponseToJSON;
exports.customerPortalDownloadablesListResponseFromJSON = customerPortalDownloadablesListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourcedownloadableread_js_1 = require("../components/listresourcedownloadableread.js");
/** @internal */
exports.CustomerPortalDownloadablesListSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalDownloadablesListSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalDownloadablesListSecurity$;
(function (CustomerPortalDownloadablesListSecurity$) {
    /** @deprecated use `CustomerPortalDownloadablesListSecurity$inboundSchema` instead. */
    CustomerPortalDownloadablesListSecurity$.inboundSchema = exports.CustomerPortalDownloadablesListSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalDownloadablesListSecurity$outboundSchema` instead. */
    CustomerPortalDownloadablesListSecurity$.outboundSchema = exports.CustomerPortalDownloadablesListSecurity$outboundSchema;
})(CustomerPortalDownloadablesListSecurity$ || (exports.CustomerPortalDownloadablesListSecurity$ = CustomerPortalDownloadablesListSecurity$ = {}));
function customerPortalDownloadablesListSecurityToJSON(customerPortalDownloadablesListSecurity) {
    return JSON.stringify(exports.CustomerPortalDownloadablesListSecurity$outboundSchema.parse(customerPortalDownloadablesListSecurity));
}
function customerPortalDownloadablesListSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalDownloadablesListSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalDownloadablesListSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$;
(function (CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$.inboundSchema = exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$.outboundSchema = exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$outboundSchema;
})(CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$ || (exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$ = CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$ = {}));
function customerPortalDownloadablesListQueryParamOrganizationIDFilterToJSON(customerPortalDownloadablesListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$outboundSchema
        .parse(customerPortalDownloadablesListQueryParamOrganizationIDFilter));
}
function customerPortalDownloadablesListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalDownloadablesListQueryParamOrganizationIDFilter$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'CustomerPortalDownloadablesListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalDownloadablesListQueryParamBenefitIDFilter$;
(function (CustomerPortalDownloadablesListQueryParamBenefitIDFilter$) {
    /** @deprecated use `CustomerPortalDownloadablesListQueryParamBenefitIDFilter$inboundSchema` instead. */
    CustomerPortalDownloadablesListQueryParamBenefitIDFilter$.inboundSchema = exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$inboundSchema;
    /** @deprecated use `CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema` instead. */
    CustomerPortalDownloadablesListQueryParamBenefitIDFilter$.outboundSchema = exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema;
})(CustomerPortalDownloadablesListQueryParamBenefitIDFilter$ || (exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$ = CustomerPortalDownloadablesListQueryParamBenefitIDFilter$ = {}));
function customerPortalDownloadablesListQueryParamBenefitIDFilterToJSON(customerPortalDownloadablesListQueryParamBenefitIDFilter) {
    return JSON.stringify(exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema
        .parse(customerPortalDownloadablesListQueryParamBenefitIDFilter));
}
function customerPortalDownloadablesListQueryParamBenefitIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalDownloadablesListQueryParamBenefitIDFilter$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'CustomerPortalDownloadablesListQueryParamBenefitIDFilter' from JSON`);
}
/** @internal */
exports.CustomerPortalDownloadablesListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    benefit_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "benefit_id": "benefitId",
    });
});
/** @internal */
exports.CustomerPortalDownloadablesListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    benefitId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        benefitId: "benefit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalDownloadablesListRequest$;
(function (CustomerPortalDownloadablesListRequest$) {
    /** @deprecated use `CustomerPortalDownloadablesListRequest$inboundSchema` instead. */
    CustomerPortalDownloadablesListRequest$.inboundSchema = exports.CustomerPortalDownloadablesListRequest$inboundSchema;
    /** @deprecated use `CustomerPortalDownloadablesListRequest$outboundSchema` instead. */
    CustomerPortalDownloadablesListRequest$.outboundSchema = exports.CustomerPortalDownloadablesListRequest$outboundSchema;
})(CustomerPortalDownloadablesListRequest$ || (exports.CustomerPortalDownloadablesListRequest$ = CustomerPortalDownloadablesListRequest$ = {}));
function customerPortalDownloadablesListRequestToJSON(customerPortalDownloadablesListRequest) {
    return JSON.stringify(exports.CustomerPortalDownloadablesListRequest$outboundSchema.parse(customerPortalDownloadablesListRequest));
}
function customerPortalDownloadablesListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalDownloadablesListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalDownloadablesListRequest' from JSON`);
}
/** @internal */
exports.CustomerPortalDownloadablesListResponse$inboundSchema = z.object({
    Result: listresourcedownloadableread_js_1.ListResourceDownloadableRead$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CustomerPortalDownloadablesListResponse$outboundSchema = z.object({
    result: listresourcedownloadableread_js_1.ListResourceDownloadableRead$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalDownloadablesListResponse$;
(function (CustomerPortalDownloadablesListResponse$) {
    /** @deprecated use `CustomerPortalDownloadablesListResponse$inboundSchema` instead. */
    CustomerPortalDownloadablesListResponse$.inboundSchema = exports.CustomerPortalDownloadablesListResponse$inboundSchema;
    /** @deprecated use `CustomerPortalDownloadablesListResponse$outboundSchema` instead. */
    CustomerPortalDownloadablesListResponse$.outboundSchema = exports.CustomerPortalDownloadablesListResponse$outboundSchema;
})(CustomerPortalDownloadablesListResponse$ || (exports.CustomerPortalDownloadablesListResponse$ = CustomerPortalDownloadablesListResponse$ = {}));
function customerPortalDownloadablesListResponseToJSON(customerPortalDownloadablesListResponse) {
    return JSON.stringify(exports.CustomerPortalDownloadablesListResponse$outboundSchema.parse(customerPortalDownloadablesListResponse));
}
function customerPortalDownloadablesListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalDownloadablesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalDownloadablesListResponse' from JSON`);
}
