// @polar-sh/sdk/dist/commonjs/models/operations/customerportalbenefitgrantsget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalBenefitGrantsGetRequest$ = exports.CustomerPortalBenefitGrantsGetRequest$outboundSchema = exports.CustomerPortalBenefitGrantsGetRequest$inboundSchema = exports.CustomerPortalBenefitGrantsGetSecurity$ = exports.CustomerPortalBenefitGrantsGetSecurity$outboundSchema = exports.CustomerPortalBenefitGrantsGetSecurity$inboundSchema = void 0;
exports.customerPortalBenefitGrantsGetSecurityToJSON = customerPortalBenefitGrantsGetSecurityToJSON;
exports.customerPortalBenefitGrantsGetSecurityFromJSON = customerPortalBenefitGrantsGetSecurityFromJSON;
exports.customerPortalBenefitGrantsGetRequestToJSON = customerPortalBenefitGrantsGetRequestToJSON;
exports.customerPortalBenefitGrantsGetRequestFromJSON = customerPortalBenefitGrantsGetRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalBenefitGrantsGetSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalBenefitGrantsGetSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalBenefitGrantsGetSecurity$;
(function (CustomerPortalBenefitGrantsGetSecurity$) {
    /** @deprecated use `CustomerPortalBenefitGrantsGetSecurity$inboundSchema` instead. */
    CustomerPortalBenefitGrantsGetSecurity$.inboundSchema = exports.CustomerPortalBenefitGrantsGetSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsGetSecurity$outboundSchema` instead. */
    CustomerPortalBenefitGrantsGetSecurity$.outboundSchema = exports.CustomerPortalBenefitGrantsGetSecurity$outboundSchema;
})(CustomerPortalBenefitGrantsGetSecurity$ || (exports.CustomerPortalBenefitGrantsGetSecurity$ = CustomerPortalBenefitGrantsGetSecurity$ = {}));
function customerPortalBenefitGrantsGetSecurityToJSON(customerPortalBenefitGrantsGetSecurity) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsGetSecurity$outboundSchema.parse(customerPortalBenefitGrantsGetSecurity));
}
function customerPortalBenefitGrantsGetSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalBenefitGrantsGetSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsGetSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalBenefitGrantsGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalBenefitGrantsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalBenefitGrantsGetRequest$;
(function (CustomerPortalBenefitGrantsGetRequest$) {
    /** @deprecated use `CustomerPortalBenefitGrantsGetRequest$inboundSchema` instead. */
    CustomerPortalBenefitGrantsGetRequest$.inboundSchema = exports.CustomerPortalBenefitGrantsGetRequest$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsGetRequest$outboundSchema` instead. */
    CustomerPortalBenefitGrantsGetRequest$.outboundSchema = exports.CustomerPortalBenefitGrantsGetRequest$outboundSchema;
})(CustomerPortalBenefitGrantsGetRequest$ || (exports.CustomerPortalBenefitGrantsGetRequest$ = CustomerPortalBenefitGrantsGetRequest$ = {}));
function customerPortalBenefitGrantsGetRequestToJSON(customerPortalBenefitGrantsGetRequest) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsGetRequest$outboundSchema.parse(customerPortalBenefitGrantsGetRequest));
}
function customerPortalBenefitGrantsGetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalBenefitGrantsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsGetRequest' from JSON`);
}
