// @polar-sh/sdk/dist/commonjs/models/operations/customerportalcustomersdeletepaymentmethod.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalCustomersDeletePaymentMethodRequest$ = exports.CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodSecurity$ = exports.CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema = void 0;
exports.customerPortalCustomersDeletePaymentMethodSecurityToJSON = customerPortalCustomersDeletePaymentMethodSecurityToJSON;
exports.customerPortalCustomersDeletePaymentMethodSecurityFromJSON = customerPortalCustomersDeletePaymentMethodSecurityFromJSON;
exports.customerPortalCustomersDeletePaymentMethodRequestToJSON = customerPortalCustomersDeletePaymentMethodRequestToJSON;
exports.customerPortalCustomersDeletePaymentMethodRequestFromJSON = customerPortalCustomersDeletePaymentMethodRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomersDeletePaymentMethodSecurity$;
(function (CustomerPortalCustomersDeletePaymentMethodSecurity$) {
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodSecurity$.inboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodSecurity$.outboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema;
})(CustomerPortalCustomersDeletePaymentMethodSecurity$ || (exports.CustomerPortalCustomersDeletePaymentMethodSecurity$ = CustomerPortalCustomersDeletePaymentMethodSecurity$ = {}));
function customerPortalCustomersDeletePaymentMethodSecurityToJSON(customerPortalCustomersDeletePaymentMethodSecurity) {
    return JSON.stringify(exports.CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema.parse(customerPortalCustomersDeletePaymentMethodSecurity));
}
function customerPortalCustomersDeletePaymentMethodSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomersDeletePaymentMethodSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersDeletePaymentMethodSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomersDeletePaymentMethodRequest$;
(function (CustomerPortalCustomersDeletePaymentMethodRequest$) {
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodRequest$.inboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema` instead. */
    CustomerPortalCustomersDeletePaymentMethodRequest$.outboundSchema = exports.CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema;
})(CustomerPortalCustomersDeletePaymentMethodRequest$ || (exports.CustomerPortalCustomersDeletePaymentMethodRequest$ = CustomerPortalCustomersDeletePaymentMethodRequest$ = {}));
function customerPortalCustomersDeletePaymentMethodRequestToJSON(customerPortalCustomersDeletePaymentMethodRequest) {
    return JSON.stringify(exports.CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema.parse(customerPortalCustomersDeletePaymentMethodRequest));
}
function customerPortalCustomersDeletePaymentMethodRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomersDeletePaymentMethodRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersDeletePaymentMethodRequest' from JSON`);
}
