// @polar-sh/sdk/dist/commonjs/models/operations/customerportalordersgetpaymentstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalOrdersGetPaymentStatusRequest$ = exports.CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema = exports.CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema = exports.CustomerPortalOrdersGetPaymentStatusSecurity$ = exports.CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema = exports.CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema = void 0;
exports.customerPortalOrdersGetPaymentStatusSecurityToJSON = customerPortalOrdersGetPaymentStatusSecurityToJSON;
exports.customerPortalOrdersGetPaymentStatusSecurityFromJSON = customerPortalOrdersGetPaymentStatusSecurityFromJSON;
exports.customerPortalOrdersGetPaymentStatusRequestToJSON = customerPortalOrdersGetPaymentStatusRequestToJSON;
exports.customerPortalOrdersGetPaymentStatusRequestFromJSON = customerPortalOrdersGetPaymentStatusRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersGetPaymentStatusSecurity$;
(function (CustomerPortalOrdersGetPaymentStatusSecurity$) {
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusSecurity$.inboundSchema = exports.CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusSecurity$.outboundSchema = exports.CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema;
})(CustomerPortalOrdersGetPaymentStatusSecurity$ || (exports.CustomerPortalOrdersGetPaymentStatusSecurity$ = CustomerPortalOrdersGetPaymentStatusSecurity$ = {}));
function customerPortalOrdersGetPaymentStatusSecurityToJSON(customerPortalOrdersGetPaymentStatusSecurity) {
    return JSON.stringify(exports.CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema.parse(customerPortalOrdersGetPaymentStatusSecurity));
}
function customerPortalOrdersGetPaymentStatusSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersGetPaymentStatusSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGetPaymentStatusSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersGetPaymentStatusRequest$;
(function (CustomerPortalOrdersGetPaymentStatusRequest$) {
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusRequest$.inboundSchema = exports.CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema` instead. */
    CustomerPortalOrdersGetPaymentStatusRequest$.outboundSchema = exports.CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema;
})(CustomerPortalOrdersGetPaymentStatusRequest$ || (exports.CustomerPortalOrdersGetPaymentStatusRequest$ = CustomerPortalOrdersGetPaymentStatusRequest$ = {}));
function customerPortalOrdersGetPaymentStatusRequestToJSON(customerPortalOrdersGetPaymentStatusRequest) {
    return JSON.stringify(exports.CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema.parse(customerPortalOrdersGetPaymentStatusRequest));
}
function customerPortalOrdersGetPaymentStatusRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersGetPaymentStatusRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGetPaymentStatusRequest' from JSON`);
}
