// @polar-sh/sdk/dist/commonjs/models/operations/customersupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomersUpdateRequest$ = exports.CustomersUpdateRequest$outboundSchema = exports.CustomersUpdateRequest$inboundSchema = void 0;
exports.customersUpdateRequestToJSON = customersUpdateRequestToJSON;
exports.customersUpdateRequestFromJSON = customersUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerupdate_js_1 = require("../components/customerupdate.js");
/** @internal */
exports.CustomersUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    CustomerUpdate: customerupdate_js_1.CustomerUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CustomerUpdate": "customerUpdate",
    });
});
/** @internal */
exports.CustomersUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    customerUpdate: customerupdate_js_1.CustomerUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerUpdate: "CustomerUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomersUpdateRequest$;
(function (CustomersUpdateRequest$) {
    /** @deprecated use `CustomersUpdateRequest$inboundSchema` instead. */
    CustomersUpdateRequest$.inboundSchema = exports.CustomersUpdateRequest$inboundSchema;
    /** @deprecated use `CustomersUpdateRequest$outboundSchema` instead. */
    CustomersUpdateRequest$.outboundSchema = exports.CustomersUpdateRequest$outboundSchema;
})(CustomersUpdateRequest$ || (exports.CustomersUpdateRequest$ = CustomersUpdateRequest$ = {}));
function customersUpdateRequestToJSON(customersUpdateRequest) {
    return JSON.stringify(exports.CustomersUpdateRequest$outboundSchema.parse(customersUpdateRequest));
}
function customersUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomersUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersUpdateRequest' from JSON`);
}
