// @polar-sh/sdk/dist/commonjs/models/operations/customerportalcustomerslistpaymentmethods.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalCustomersListPaymentMethodsResponse$ = exports.CustomerPortalCustomersListPaymentMethodsResponse$outboundSchema = exports.CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema = exports.CustomerPortalCustomersListPaymentMethodsRequest$ = exports.CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema = exports.CustomerPortalCustomersListPaymentMethodsRequest$inboundSchema = exports.CustomerPortalCustomersListPaymentMethodsSecurity$ = exports.CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema = exports.CustomerPortalCustomersListPaymentMethodsSecurity$inboundSchema = void 0;
exports.customerPortalCustomersListPaymentMethodsSecurityToJSON = customerPortalCustomersListPaymentMethodsSecurityToJSON;
exports.customerPortalCustomersListPaymentMethodsSecurityFromJSON = customerPortalCustomersListPaymentMethodsSecurityFromJSON;
exports.customerPortalCustomersListPaymentMethodsRequestToJSON = customerPortalCustomersListPaymentMethodsRequestToJSON;
exports.customerPortalCustomersListPaymentMethodsRequestFromJSON = customerPortalCustomersListPaymentMethodsRequestFromJSON;
exports.customerPortalCustomersListPaymentMethodsResponseToJSON = customerPortalCustomersListPaymentMethodsResponseToJSON;
exports.customerPortalCustomersListPaymentMethodsResponseFromJSON = customerPortalCustomersListPaymentMethodsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourcecustomerpaymentmethod_js_1 = require("../components/listresourcecustomerpaymentmethod.js");
/** @internal */
exports.CustomerPortalCustomersListPaymentMethodsSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomersListPaymentMethodsSecurity$;
(function (CustomerPortalCustomersListPaymentMethodsSecurity$) {
    /** @deprecated use `CustomerPortalCustomersListPaymentMethodsSecurity$inboundSchema` instead. */
    CustomerPortalCustomersListPaymentMethodsSecurity$.inboundSchema = exports.CustomerPortalCustomersListPaymentMethodsSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema` instead. */
    CustomerPortalCustomersListPaymentMethodsSecurity$.outboundSchema = exports.CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema;
})(CustomerPortalCustomersListPaymentMethodsSecurity$ || (exports.CustomerPortalCustomersListPaymentMethodsSecurity$ = CustomerPortalCustomersListPaymentMethodsSecurity$ = {}));
function customerPortalCustomersListPaymentMethodsSecurityToJSON(customerPortalCustomersListPaymentMethodsSecurity) {
    return JSON.stringify(exports.CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema.parse(customerPortalCustomersListPaymentMethodsSecurity));
}
function customerPortalCustomersListPaymentMethodsSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomersListPaymentMethodsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersListPaymentMethodsSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalCustomersListPaymentMethodsRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
});
/** @internal */
exports.CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomersListPaymentMethodsRequest$;
(function (CustomerPortalCustomersListPaymentMethodsRequest$) {
    /** @deprecated use `CustomerPortalCustomersListPaymentMethodsRequest$inboundSchema` instead. */
    CustomerPortalCustomersListPaymentMethodsRequest$.inboundSchema = exports.CustomerPortalCustomersListPaymentMethodsRequest$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema` instead. */
    CustomerPortalCustomersListPaymentMethodsRequest$.outboundSchema = exports.CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema;
})(CustomerPortalCustomersListPaymentMethodsRequest$ || (exports.CustomerPortalCustomersListPaymentMethodsRequest$ = CustomerPortalCustomersListPaymentMethodsRequest$ = {}));
function customerPortalCustomersListPaymentMethodsRequestToJSON(customerPortalCustomersListPaymentMethodsRequest) {
    return JSON.stringify(exports.CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema.parse(customerPortalCustomersListPaymentMethodsRequest));
}
function customerPortalCustomersListPaymentMethodsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomersListPaymentMethodsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersListPaymentMethodsRequest' from JSON`);
}
/** @internal */
exports.CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema = z.object({
    Result: listresourcecustomerpaymentmethod_js_1.ListResourceCustomerPaymentMethod$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CustomerPortalCustomersListPaymentMethodsResponse$outboundSchema = z.object({
    result: listresourcecustomerpaymentmethod_js_1.ListResourceCustomerPaymentMethod$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomersListPaymentMethodsResponse$;
(function (CustomerPortalCustomersListPaymentMethodsResponse$) {
    /** @deprecated use `CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema` instead. */
    CustomerPortalCustomersListPaymentMethodsResponse$.inboundSchema = exports.CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema;
    /** @deprecated use `CustomerPortalCustomersListPaymentMethodsResponse$outboundSchema` instead. */
    CustomerPortalCustomersListPaymentMethodsResponse$.outboundSchema = exports.CustomerPortalCustomersListPaymentMethodsResponse$outboundSchema;
})(CustomerPortalCustomersListPaymentMethodsResponse$ || (exports.CustomerPortalCustomersListPaymentMethodsResponse$ = CustomerPortalCustomersListPaymentMethodsResponse$ = {}));
function customerPortalCustomersListPaymentMethodsResponseToJSON(customerPortalCustomersListPaymentMethodsResponse) {
    return JSON.stringify(exports.CustomerPortalCustomersListPaymentMethodsResponse$outboundSchema.parse(customerPortalCustomersListPaymentMethodsResponse));
}
function customerPortalCustomersListPaymentMethodsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersListPaymentMethodsResponse' from JSON`);
}
