// @polar-sh/sdk/dist/commonjs/models/operations/customerportalbenefitgrantsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalBenefitGrantsUpdateRequest$ = exports.CustomerPortalBenefitGrantsUpdateRequest$outboundSchema = exports.CustomerPortalBenefitGrantsUpdateRequest$inboundSchema = exports.CustomerPortalBenefitGrantsUpdateSecurity$ = exports.CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema = exports.CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema = void 0;
exports.customerPortalBenefitGrantsUpdateSecurityToJSON = customerPortalBenefitGrantsUpdateSecurityToJSON;
exports.customerPortalBenefitGrantsUpdateSecurityFromJSON = customerPortalBenefitGrantsUpdateSecurityFromJSON;
exports.customerPortalBenefitGrantsUpdateRequestToJSON = customerPortalBenefitGrantsUpdateRequestToJSON;
exports.customerPortalBenefitGrantsUpdateRequestFromJSON = customerPortalBenefitGrantsUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerbenefitgrantupdate_js_1 = require("../components/customerbenefitgrantupdate.js");
/** @internal */
exports.CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalBenefitGrantsUpdateSecurity$;
(function (CustomerPortalBenefitGrantsUpdateSecurity$) {
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateSecurity$.inboundSchema = exports.CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateSecurity$.outboundSchema = exports.CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema;
})(CustomerPortalBenefitGrantsUpdateSecurity$ || (exports.CustomerPortalBenefitGrantsUpdateSecurity$ = CustomerPortalBenefitGrantsUpdateSecurity$ = {}));
function customerPortalBenefitGrantsUpdateSecurityToJSON(customerPortalBenefitGrantsUpdateSecurity) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema.parse(customerPortalBenefitGrantsUpdateSecurity));
}
function customerPortalBenefitGrantsUpdateSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsUpdateSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalBenefitGrantsUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    CustomerBenefitGrantUpdate: customerbenefitgrantupdate_js_1.CustomerBenefitGrantUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CustomerBenefitGrantUpdate": "customerBenefitGrantUpdate",
    });
});
/** @internal */
exports.CustomerPortalBenefitGrantsUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    customerBenefitGrantUpdate: customerbenefitgrantupdate_js_1.CustomerBenefitGrantUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerBenefitGrantUpdate: "CustomerBenefitGrantUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalBenefitGrantsUpdateRequest$;
(function (CustomerPortalBenefitGrantsUpdateRequest$) {
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateRequest$inboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateRequest$.inboundSchema = exports.CustomerPortalBenefitGrantsUpdateRequest$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateRequest$outboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateRequest$.outboundSchema = exports.CustomerPortalBenefitGrantsUpdateRequest$outboundSchema;
})(CustomerPortalBenefitGrantsUpdateRequest$ || (exports.CustomerPortalBenefitGrantsUpdateRequest$ = CustomerPortalBenefitGrantsUpdateRequest$ = {}));
function customerPortalBenefitGrantsUpdateRequestToJSON(customerPortalBenefitGrantsUpdateRequest) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsUpdateRequest$outboundSchema.parse(customerPortalBenefitGrantsUpdateRequest));
}
function customerPortalBenefitGrantsUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalBenefitGrantsUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsUpdateRequest' from JSON`);
}
