// @polar-sh/sdk/dist/commonjs/models/operations/customersupdateexternal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomersUpdateExternalRequest$ = exports.CustomersUpdateExternalRequest$outboundSchema = exports.CustomersUpdateExternalRequest$inboundSchema = void 0;
exports.customersUpdateExternalRequestToJSON = customersUpdateExternalRequestToJSON;
exports.customersUpdateExternalRequestFromJSON = customersUpdateExternalRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerupdateexternalid_js_1 = require("../components/customerupdateexternalid.js");
/** @internal */
exports.CustomersUpdateExternalRequest$inboundSchema = z.object({
    external_id: z.string(),
    CustomerUpdateExternalID: customerupdateexternalid_js_1.CustomerUpdateExternalID$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "external_id": "externalId",
        "CustomerUpdateExternalID": "customerUpdateExternalID",
    });
});
/** @internal */
exports.CustomersUpdateExternalRequest$outboundSchema = z.object({
    externalId: z.string(),
    customerUpdateExternalID: customerupdateexternalid_js_1.CustomerUpdateExternalID$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        externalId: "external_id",
        customerUpdateExternalID: "CustomerUpdateExternalID",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomersUpdateExternalRequest$;
(function (CustomersUpdateExternalRequest$) {
    /** @deprecated use `CustomersUpdateExternalRequest$inboundSchema` instead. */
    CustomersUpdateExternalRequest$.inboundSchema = exports.CustomersUpdateExternalRequest$inboundSchema;
    /** @deprecated use `CustomersUpdateExternalRequest$outboundSchema` instead. */
    CustomersUpdateExternalRequest$.outboundSchema = exports.CustomersUpdateExternalRequest$outboundSchema;
})(CustomersUpdateExternalRequest$ || (exports.CustomersUpdateExternalRequest$ = CustomersUpdateExternalRequest$ = {}));
function customersUpdateExternalRequestToJSON(customersUpdateExternalRequest) {
    return JSON.stringify(exports.CustomersUpdateExternalRequest$outboundSchema.parse(customersUpdateExternalRequest));
}
function customersUpdateExternalRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomersUpdateExternalRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersUpdateExternalRequest' from JSON`);
}
