// @polar-sh/sdk/dist/commonjs/models/operations/customerportalcustomermetersget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalCustomerMetersGetRequest$ = exports.CustomerPortalCustomerMetersGetRequest$outboundSchema = exports.CustomerPortalCustomerMetersGetRequest$inboundSchema = exports.CustomerPortalCustomerMetersGetSecurity$ = exports.CustomerPortalCustomerMetersGetSecurity$outboundSchema = exports.CustomerPortalCustomerMetersGetSecurity$inboundSchema = void 0;
exports.customerPortalCustomerMetersGetSecurityToJSON = customerPortalCustomerMetersGetSecurityToJSON;
exports.customerPortalCustomerMetersGetSecurityFromJSON = customerPortalCustomerMetersGetSecurityFromJSON;
exports.customerPortalCustomerMetersGetRequestToJSON = customerPortalCustomerMetersGetRequestToJSON;
exports.customerPortalCustomerMetersGetRequestFromJSON = customerPortalCustomerMetersGetRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalCustomerMetersGetSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalCustomerMetersGetSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomerMetersGetSecurity$;
(function (CustomerPortalCustomerMetersGetSecurity$) {
    /** @deprecated use `CustomerPortalCustomerMetersGetSecurity$inboundSchema` instead. */
    CustomerPortalCustomerMetersGetSecurity$.inboundSchema = exports.CustomerPortalCustomerMetersGetSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalCustomerMetersGetSecurity$outboundSchema` instead. */
    CustomerPortalCustomerMetersGetSecurity$.outboundSchema = exports.CustomerPortalCustomerMetersGetSecurity$outboundSchema;
})(CustomerPortalCustomerMetersGetSecurity$ || (exports.CustomerPortalCustomerMetersGetSecurity$ = CustomerPortalCustomerMetersGetSecurity$ = {}));
function customerPortalCustomerMetersGetSecurityToJSON(customerPortalCustomerMetersGetSecurity) {
    return JSON.stringify(exports.CustomerPortalCustomerMetersGetSecurity$outboundSchema.parse(customerPortalCustomerMetersGetSecurity));
}
function customerPortalCustomerMetersGetSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomerMetersGetSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomerMetersGetSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalCustomerMetersGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalCustomerMetersGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalCustomerMetersGetRequest$;
(function (CustomerPortalCustomerMetersGetRequest$) {
    /** @deprecated use `CustomerPortalCustomerMetersGetRequest$inboundSchema` instead. */
    CustomerPortalCustomerMetersGetRequest$.inboundSchema = exports.CustomerPortalCustomerMetersGetRequest$inboundSchema;
    /** @deprecated use `CustomerPortalCustomerMetersGetRequest$outboundSchema` instead. */
    CustomerPortalCustomerMetersGetRequest$.outboundSchema = exports.CustomerPortalCustomerMetersGetRequest$outboundSchema;
})(CustomerPortalCustomerMetersGetRequest$ || (exports.CustomerPortalCustomerMetersGetRequest$ = CustomerPortalCustomerMetersGetRequest$ = {}));
function customerPortalCustomerMetersGetRequestToJSON(customerPortalCustomerMetersGetRequest) {
    return JSON.stringify(exports.CustomerPortalCustomerMetersGetRequest$outboundSchema.parse(customerPortalCustomerMetersGetRequest));
}
function customerPortalCustomerMetersGetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalCustomerMetersGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomerMetersGetRequest' from JSON`);
}
