// @polar-sh/sdk/dist/commonjs/models/operations/customerportalorderslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalOrdersListResponse$ = exports.CustomerPortalOrdersListResponse$outboundSchema = exports.CustomerPortalOrdersListResponse$inboundSchema = exports.CustomerPortalOrdersListRequest$ = exports.CustomerPortalOrdersListRequest$outboundSchema = exports.CustomerPortalOrdersListRequest$inboundSchema = exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$ = exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema = exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$inboundSchema = exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$ = exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema = exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$inboundSchema = exports.CustomerPortalOrdersListQueryParamProductIDFilter$ = exports.CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema = exports.CustomerPortalOrdersListQueryParamProductIDFilter$inboundSchema = exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$ = exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$outboundSchema = exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$inboundSchema = exports.CustomerPortalOrdersListSecurity$ = exports.CustomerPortalOrdersListSecurity$outboundSchema = exports.CustomerPortalOrdersListSecurity$inboundSchema = void 0;
exports.customerPortalOrdersListSecurityToJSON = customerPortalOrdersListSecurityToJSON;
exports.customerPortalOrdersListSecurityFromJSON = customerPortalOrdersListSecurityFromJSON;
exports.customerPortalOrdersListQueryParamOrganizationIDFilterToJSON = customerPortalOrdersListQueryParamOrganizationIDFilterToJSON;
exports.customerPortalOrdersListQueryParamOrganizationIDFilterFromJSON = customerPortalOrdersListQueryParamOrganizationIDFilterFromJSON;
exports.customerPortalOrdersListQueryParamProductIDFilterToJSON = customerPortalOrdersListQueryParamProductIDFilterToJSON;
exports.customerPortalOrdersListQueryParamProductIDFilterFromJSON = customerPortalOrdersListQueryParamProductIDFilterFromJSON;
exports.customerPortalOrdersListQueryParamProductBillingTypeFilterToJSON = customerPortalOrdersListQueryParamProductBillingTypeFilterToJSON;
exports.customerPortalOrdersListQueryParamProductBillingTypeFilterFromJSON = customerPortalOrdersListQueryParamProductBillingTypeFilterFromJSON;
exports.customerPortalOrdersListQueryParamSubscriptionIDFilterToJSON = customerPortalOrdersListQueryParamSubscriptionIDFilterToJSON;
exports.customerPortalOrdersListQueryParamSubscriptionIDFilterFromJSON = customerPortalOrdersListQueryParamSubscriptionIDFilterFromJSON;
exports.customerPortalOrdersListRequestToJSON = customerPortalOrdersListRequestToJSON;
exports.customerPortalOrdersListRequestFromJSON = customerPortalOrdersListRequestFromJSON;
exports.customerPortalOrdersListResponseToJSON = customerPortalOrdersListResponseToJSON;
exports.customerPortalOrdersListResponseFromJSON = customerPortalOrdersListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerordersortproperty_js_1 = require("../components/customerordersortproperty.js");
const listresourcecustomerorder_js_1 = require("../components/listresourcecustomerorder.js");
const productbillingtype_js_1 = require("../components/productbillingtype.js");
/** @internal */
exports.CustomerPortalOrdersListSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalOrdersListSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListSecurity$;
(function (CustomerPortalOrdersListSecurity$) {
    /** @deprecated use `CustomerPortalOrdersListSecurity$inboundSchema` instead. */
    CustomerPortalOrdersListSecurity$.inboundSchema = exports.CustomerPortalOrdersListSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListSecurity$outboundSchema` instead. */
    CustomerPortalOrdersListSecurity$.outboundSchema = exports.CustomerPortalOrdersListSecurity$outboundSchema;
})(CustomerPortalOrdersListSecurity$ || (exports.CustomerPortalOrdersListSecurity$ = CustomerPortalOrdersListSecurity$ = {}));
function customerPortalOrdersListSecurityToJSON(customerPortalOrdersListSecurity) {
    return JSON.stringify(exports.CustomerPortalOrdersListSecurity$outboundSchema.parse(customerPortalOrdersListSecurity));
}
function customerPortalOrdersListSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListQueryParamOrganizationIDFilter$;
(function (CustomerPortalOrdersListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `CustomerPortalOrdersListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    CustomerPortalOrdersListQueryParamOrganizationIDFilter$.inboundSchema = exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    CustomerPortalOrdersListQueryParamOrganizationIDFilter$.outboundSchema = exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$outboundSchema;
})(CustomerPortalOrdersListQueryParamOrganizationIDFilter$ || (exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$ = CustomerPortalOrdersListQueryParamOrganizationIDFilter$ = {}));
function customerPortalOrdersListQueryParamOrganizationIDFilterToJSON(customerPortalOrdersListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$outboundSchema.parse(customerPortalOrdersListQueryParamOrganizationIDFilter));
}
function customerPortalOrdersListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListQueryParamOrganizationIDFilter$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersListQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListQueryParamProductIDFilter$;
(function (CustomerPortalOrdersListQueryParamProductIDFilter$) {
    /** @deprecated use `CustomerPortalOrdersListQueryParamProductIDFilter$inboundSchema` instead. */
    CustomerPortalOrdersListQueryParamProductIDFilter$.inboundSchema = exports.CustomerPortalOrdersListQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema` instead. */
    CustomerPortalOrdersListQueryParamProductIDFilter$.outboundSchema = exports.CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema;
})(CustomerPortalOrdersListQueryParamProductIDFilter$ || (exports.CustomerPortalOrdersListQueryParamProductIDFilter$ = CustomerPortalOrdersListQueryParamProductIDFilter$ = {}));
function customerPortalOrdersListQueryParamProductIDFilterToJSON(customerPortalOrdersListQueryParamProductIDFilter) {
    return JSON.stringify(exports.CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema.parse(customerPortalOrdersListQueryParamProductIDFilter));
}
function customerPortalOrdersListQueryParamProductIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListQueryParamProductIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListQueryParamProductIDFilter' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$inboundSchema = z.union([
    productbillingtype_js_1.ProductBillingType$inboundSchema,
    z.array(productbillingtype_js_1.ProductBillingType$inboundSchema),
]);
/** @internal */
exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema = z.union([
    productbillingtype_js_1.ProductBillingType$outboundSchema,
    z.array(productbillingtype_js_1.ProductBillingType$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListQueryParamProductBillingTypeFilter$;
(function (CustomerPortalOrdersListQueryParamProductBillingTypeFilter$) {
    /** @deprecated use `CustomerPortalOrdersListQueryParamProductBillingTypeFilter$inboundSchema` instead. */
    CustomerPortalOrdersListQueryParamProductBillingTypeFilter$.inboundSchema = exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema` instead. */
    CustomerPortalOrdersListQueryParamProductBillingTypeFilter$.outboundSchema = exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema;
})(CustomerPortalOrdersListQueryParamProductBillingTypeFilter$ || (exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$ = CustomerPortalOrdersListQueryParamProductBillingTypeFilter$ = {}));
function customerPortalOrdersListQueryParamProductBillingTypeFilterToJSON(customerPortalOrdersListQueryParamProductBillingTypeFilter) {
    return JSON.stringify(exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema
        .parse(customerPortalOrdersListQueryParamProductBillingTypeFilter));
}
function customerPortalOrdersListQueryParamProductBillingTypeFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListQueryParamProductBillingTypeFilter$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListQueryParamProductBillingTypeFilter' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListQueryParamSubscriptionIDFilter$;
(function (CustomerPortalOrdersListQueryParamSubscriptionIDFilter$) {
    /** @deprecated use `CustomerPortalOrdersListQueryParamSubscriptionIDFilter$inboundSchema` instead. */
    CustomerPortalOrdersListQueryParamSubscriptionIDFilter$.inboundSchema = exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema` instead. */
    CustomerPortalOrdersListQueryParamSubscriptionIDFilter$.outboundSchema = exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema;
})(CustomerPortalOrdersListQueryParamSubscriptionIDFilter$ || (exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$ = CustomerPortalOrdersListQueryParamSubscriptionIDFilter$ = {}));
function customerPortalOrdersListQueryParamSubscriptionIDFilterToJSON(customerPortalOrdersListQueryParamSubscriptionIDFilter) {
    return JSON.stringify(exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema.parse(customerPortalOrdersListQueryParamSubscriptionIDFilter));
}
function customerPortalOrdersListQueryParamSubscriptionIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListQueryParamSubscriptionIDFilter$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListQueryParamSubscriptionIDFilter' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    product_billing_type: z.nullable(z.union([
        productbillingtype_js_1.ProductBillingType$inboundSchema,
        z.array(productbillingtype_js_1.ProductBillingType$inboundSchema),
    ])).optional(),
    subscription_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(customerordersortproperty_js_1.CustomerOrderSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_billing_type": "productBillingType",
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
exports.CustomerPortalOrdersListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    productBillingType: z.nullable(z.union([
        productbillingtype_js_1.ProductBillingType$outboundSchema,
        z.array(productbillingtype_js_1.ProductBillingType$outboundSchema),
    ])).optional(),
    subscriptionId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(customerordersortproperty_js_1.CustomerOrderSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
        productBillingType: "product_billing_type",
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListRequest$;
(function (CustomerPortalOrdersListRequest$) {
    /** @deprecated use `CustomerPortalOrdersListRequest$inboundSchema` instead. */
    CustomerPortalOrdersListRequest$.inboundSchema = exports.CustomerPortalOrdersListRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListRequest$outboundSchema` instead. */
    CustomerPortalOrdersListRequest$.outboundSchema = exports.CustomerPortalOrdersListRequest$outboundSchema;
})(CustomerPortalOrdersListRequest$ || (exports.CustomerPortalOrdersListRequest$ = CustomerPortalOrdersListRequest$ = {}));
function customerPortalOrdersListRequestToJSON(customerPortalOrdersListRequest) {
    return JSON.stringify(exports.CustomerPortalOrdersListRequest$outboundSchema.parse(customerPortalOrdersListRequest));
}
function customerPortalOrdersListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListRequest' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersListResponse$inboundSchema = z.object({
    Result: listresourcecustomerorder_js_1.ListResourceCustomerOrder$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CustomerPortalOrdersListResponse$outboundSchema = z.object({
    result: listresourcecustomerorder_js_1.ListResourceCustomerOrder$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersListResponse$;
(function (CustomerPortalOrdersListResponse$) {
    /** @deprecated use `CustomerPortalOrdersListResponse$inboundSchema` instead. */
    CustomerPortalOrdersListResponse$.inboundSchema = exports.CustomerPortalOrdersListResponse$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersListResponse$outboundSchema` instead. */
    CustomerPortalOrdersListResponse$.outboundSchema = exports.CustomerPortalOrdersListResponse$outboundSchema;
})(CustomerPortalOrdersListResponse$ || (exports.CustomerPortalOrdersListResponse$ = CustomerPortalOrdersListResponse$ = {}));
function customerPortalOrdersListResponseToJSON(customerPortalOrdersListResponse) {
    return JSON.stringify(exports.CustomerPortalOrdersListResponse$outboundSchema.parse(customerPortalOrdersListResponse));
}
function customerPortalOrdersListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListResponse' from JSON`);
}
