// @polar-sh/sdk/dist/commonjs/models/operations/customerportalordersgenerateinvoice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalOrdersGenerateInvoiceRequest$ = exports.CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema = exports.CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema = exports.CustomerPortalOrdersGenerateInvoiceSecurity$ = exports.CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema = exports.CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema = void 0;
exports.customerPortalOrdersGenerateInvoiceSecurityToJSON = customerPortalOrdersGenerateInvoiceSecurityToJSON;
exports.customerPortalOrdersGenerateInvoiceSecurityFromJSON = customerPortalOrdersGenerateInvoiceSecurityFromJSON;
exports.customerPortalOrdersGenerateInvoiceRequestToJSON = customerPortalOrdersGenerateInvoiceRequestToJSON;
exports.customerPortalOrdersGenerateInvoiceRequestFromJSON = customerPortalOrdersGenerateInvoiceRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersGenerateInvoiceSecurity$;
(function (CustomerPortalOrdersGenerateInvoiceSecurity$) {
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceSecurity$.inboundSchema = exports.CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceSecurity$.outboundSchema = exports.CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema;
})(CustomerPortalOrdersGenerateInvoiceSecurity$ || (exports.CustomerPortalOrdersGenerateInvoiceSecurity$ = CustomerPortalOrdersGenerateInvoiceSecurity$ = {}));
function customerPortalOrdersGenerateInvoiceSecurityToJSON(customerPortalOrdersGenerateInvoiceSecurity) {
    return JSON.stringify(exports.CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema.parse(customerPortalOrdersGenerateInvoiceSecurity));
}
function customerPortalOrdersGenerateInvoiceSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGenerateInvoiceSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersGenerateInvoiceRequest$;
(function (CustomerPortalOrdersGenerateInvoiceRequest$) {
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceRequest$.inboundSchema = exports.CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceRequest$.outboundSchema = exports.CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema;
})(CustomerPortalOrdersGenerateInvoiceRequest$ || (exports.CustomerPortalOrdersGenerateInvoiceRequest$ = CustomerPortalOrdersGenerateInvoiceRequest$ = {}));
function customerPortalOrdersGenerateInvoiceRequestToJSON(customerPortalOrdersGenerateInvoiceRequest) {
    return JSON.stringify(exports.CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema.parse(customerPortalOrdersGenerateInvoiceRequest));
}
function customerPortalOrdersGenerateInvoiceRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGenerateInvoiceRequest' from JSON`);
}
