// @polar-sh/sdk/dist/commonjs/models/operations/discountslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountsListResponse$ = exports.DiscountsListResponse$outboundSchema = exports.DiscountsListResponse$inboundSchema = exports.DiscountsListRequest$ = exports.DiscountsListRequest$outboundSchema = exports.DiscountsListRequest$inboundSchema = exports.DiscountsListQueryParamOrganizationIDFilter$ = exports.DiscountsListQueryParamOrganizationIDFilter$outboundSchema = exports.DiscountsListQueryParamOrganizationIDFilter$inboundSchema = void 0;
exports.discountsListQueryParamOrganizationIDFilterToJSON = discountsListQueryParamOrganizationIDFilterToJSON;
exports.discountsListQueryParamOrganizationIDFilterFromJSON = discountsListQueryParamOrganizationIDFilterFromJSON;
exports.discountsListRequestToJSON = discountsListRequestToJSON;
exports.discountsListRequestFromJSON = discountsListRequestFromJSON;
exports.discountsListResponseToJSON = discountsListResponseToJSON;
exports.discountsListResponseFromJSON = discountsListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountsortproperty_js_1 = require("../components/discountsortproperty.js");
const listresourcediscount_js_1 = require("../components/listresourcediscount.js");
/** @internal */
exports.DiscountsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.DiscountsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountsListQueryParamOrganizationIDFilter$;
(function (DiscountsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `DiscountsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    DiscountsListQueryParamOrganizationIDFilter$.inboundSchema = exports.DiscountsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `DiscountsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    DiscountsListQueryParamOrganizationIDFilter$.outboundSchema = exports.DiscountsListQueryParamOrganizationIDFilter$outboundSchema;
})(DiscountsListQueryParamOrganizationIDFilter$ || (exports.DiscountsListQueryParamOrganizationIDFilter$ = DiscountsListQueryParamOrganizationIDFilter$ = {}));
function discountsListQueryParamOrganizationIDFilterToJSON(discountsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.DiscountsListQueryParamOrganizationIDFilter$outboundSchema.parse(discountsListQueryParamOrganizationIDFilter));
}
function discountsListQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountsListQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountsListQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.DiscountsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(discountsortproperty_js_1.DiscountSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.DiscountsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(discountsortproperty_js_1.DiscountSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountsListRequest$;
(function (DiscountsListRequest$) {
    /** @deprecated use `DiscountsListRequest$inboundSchema` instead. */
    DiscountsListRequest$.inboundSchema = exports.DiscountsListRequest$inboundSchema;
    /** @deprecated use `DiscountsListRequest$outboundSchema` instead. */
    DiscountsListRequest$.outboundSchema = exports.DiscountsListRequest$outboundSchema;
})(DiscountsListRequest$ || (exports.DiscountsListRequest$ = DiscountsListRequest$ = {}));
function discountsListRequestToJSON(discountsListRequest) {
    return JSON.stringify(exports.DiscountsListRequest$outboundSchema.parse(discountsListRequest));
}
function discountsListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountsListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountsListRequest' from JSON`);
}
/** @internal */
exports.DiscountsListResponse$inboundSchema = z.object({
    Result: listresourcediscount_js_1.ListResourceDiscount$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.DiscountsListResponse$outboundSchema = z.object({
    result: listresourcediscount_js_1.ListResourceDiscount$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountsListResponse$;
(function (DiscountsListResponse$) {
    /** @deprecated use `DiscountsListResponse$inboundSchema` instead. */
    DiscountsListResponse$.inboundSchema = exports.DiscountsListResponse$inboundSchema;
    /** @deprecated use `DiscountsListResponse$outboundSchema` instead. */
    DiscountsListResponse$.outboundSchema = exports.DiscountsListResponse$outboundSchema;
})(DiscountsListResponse$ || (exports.DiscountsListResponse$ = DiscountsListResponse$ = {}));
function discountsListResponseToJSON(discountsListResponse) {
    return JSON.stringify(exports.DiscountsListResponse$outboundSchema.parse(discountsListResponse));
}
function discountsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountsListResponse' from JSON`);
}
