// @polar-sh/sdk/dist/commonjs/models/operations/filesupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesUpdateResponseFilesUpdate$ = exports.FilesUpdateResponseFilesUpdate$outboundSchema = exports.FilesUpdateResponseFilesUpdate$inboundSchema = exports.FilesUpdateRequest$ = exports.FilesUpdateRequest$outboundSchema = exports.FilesUpdateRequest$inboundSchema = void 0;
exports.filesUpdateRequestToJSON = filesUpdateRequestToJSON;
exports.filesUpdateRequestFromJSON = filesUpdateRequestFromJSON;
exports.filesUpdateResponseFilesUpdateToJSON = filesUpdateResponseFilesUpdateToJSON;
exports.filesUpdateResponseFilesUpdateFromJSON = filesUpdateResponseFilesUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const downloadablefileread_js_1 = require("../components/downloadablefileread.js");
const filepatch_js_1 = require("../components/filepatch.js");
const organizationavatarfileread_js_1 = require("../components/organizationavatarfileread.js");
const productmediafileread_js_1 = require("../components/productmediafileread.js");
/** @internal */
exports.FilesUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    FilePatch: filepatch_js_1.FilePatch$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "FilePatch": "filePatch",
    });
});
/** @internal */
exports.FilesUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    filePatch: filepatch_js_1.FilePatch$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        filePatch: "FilePatch",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUpdateRequest$;
(function (FilesUpdateRequest$) {
    /** @deprecated use `FilesUpdateRequest$inboundSchema` instead. */
    FilesUpdateRequest$.inboundSchema = exports.FilesUpdateRequest$inboundSchema;
    /** @deprecated use `FilesUpdateRequest$outboundSchema` instead. */
    FilesUpdateRequest$.outboundSchema = exports.FilesUpdateRequest$outboundSchema;
})(FilesUpdateRequest$ || (exports.FilesUpdateRequest$ = FilesUpdateRequest$ = {}));
function filesUpdateRequestToJSON(filesUpdateRequest) {
    return JSON.stringify(exports.FilesUpdateRequest$outboundSchema.parse(filesUpdateRequest));
}
function filesUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FilesUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesUpdateRequest' from JSON`);
}
/** @internal */
exports.FilesUpdateResponseFilesUpdate$inboundSchema = z.union([
    productmediafileread_js_1.ProductMediaFileRead$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    organizationavatarfileread_js_1.OrganizationAvatarFileRead$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    downloadablefileread_js_1.DownloadableFileRead$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
]);
/** @internal */
exports.FilesUpdateResponseFilesUpdate$outboundSchema = z.union([
    productmediafileread_js_1.ProductMediaFileRead$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    organizationavatarfileread_js_1.OrganizationAvatarFileRead$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    downloadablefileread_js_1.DownloadableFileRead$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUpdateResponseFilesUpdate$;
(function (FilesUpdateResponseFilesUpdate$) {
    /** @deprecated use `FilesUpdateResponseFilesUpdate$inboundSchema` instead. */
    FilesUpdateResponseFilesUpdate$.inboundSchema = exports.FilesUpdateResponseFilesUpdate$inboundSchema;
    /** @deprecated use `FilesUpdateResponseFilesUpdate$outboundSchema` instead. */
    FilesUpdateResponseFilesUpdate$.outboundSchema = exports.FilesUpdateResponseFilesUpdate$outboundSchema;
})(FilesUpdateResponseFilesUpdate$ || (exports.FilesUpdateResponseFilesUpdate$ = FilesUpdateResponseFilesUpdate$ = {}));
function filesUpdateResponseFilesUpdateToJSON(filesUpdateResponseFilesUpdate) {
    return JSON.stringify(exports.FilesUpdateResponseFilesUpdate$outboundSchema.parse(filesUpdateResponseFilesUpdate));
}
function filesUpdateResponseFilesUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FilesUpdateResponseFilesUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesUpdateResponseFilesUpdate' from JSON`);
}
