// @polar-sh/sdk/dist/commonjs/models/operations/filesuploaded.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesUploadedResponseFilesUploaded$ = exports.FilesUploadedResponseFilesUploaded$outboundSchema = exports.FilesUploadedResponseFilesUploaded$inboundSchema = exports.FilesUploadedRequest$ = exports.FilesUploadedRequest$outboundSchema = exports.FilesUploadedRequest$inboundSchema = void 0;
exports.filesUploadedRequestToJSON = filesUploadedRequestToJSON;
exports.filesUploadedRequestFromJSON = filesUploadedRequestFromJSON;
exports.filesUploadedResponseFilesUploadedToJSON = filesUploadedResponseFilesUploadedToJSON;
exports.filesUploadedResponseFilesUploadedFromJSON = filesUploadedResponseFilesUploadedFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const downloadablefileread_js_1 = require("../components/downloadablefileread.js");
const fileuploadcompleted_js_1 = require("../components/fileuploadcompleted.js");
const organizationavatarfileread_js_1 = require("../components/organizationavatarfileread.js");
const productmediafileread_js_1 = require("../components/productmediafileread.js");
/** @internal */
exports.FilesUploadedRequest$inboundSchema = z.object({
    id: z.string(),
    FileUploadCompleted: fileuploadcompleted_js_1.FileUploadCompleted$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "FileUploadCompleted": "fileUploadCompleted",
    });
});
/** @internal */
exports.FilesUploadedRequest$outboundSchema = z.object({
    id: z.string(),
    fileUploadCompleted: fileuploadcompleted_js_1.FileUploadCompleted$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fileUploadCompleted: "FileUploadCompleted",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUploadedRequest$;
(function (FilesUploadedRequest$) {
    /** @deprecated use `FilesUploadedRequest$inboundSchema` instead. */
    FilesUploadedRequest$.inboundSchema = exports.FilesUploadedRequest$inboundSchema;
    /** @deprecated use `FilesUploadedRequest$outboundSchema` instead. */
    FilesUploadedRequest$.outboundSchema = exports.FilesUploadedRequest$outboundSchema;
})(FilesUploadedRequest$ || (exports.FilesUploadedRequest$ = FilesUploadedRequest$ = {}));
function filesUploadedRequestToJSON(filesUploadedRequest) {
    return JSON.stringify(exports.FilesUploadedRequest$outboundSchema.parse(filesUploadedRequest));
}
function filesUploadedRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FilesUploadedRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesUploadedRequest' from JSON`);
}
/** @internal */
exports.FilesUploadedResponseFilesUploaded$inboundSchema = z.union([
    productmediafileread_js_1.ProductMediaFileRead$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    organizationavatarfileread_js_1.OrganizationAvatarFileRead$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    downloadablefileread_js_1.DownloadableFileRead$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
]);
/** @internal */
exports.FilesUploadedResponseFilesUploaded$outboundSchema = z.union([
    productmediafileread_js_1.ProductMediaFileRead$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    organizationavatarfileread_js_1.OrganizationAvatarFileRead$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
    downloadablefileread_js_1.DownloadableFileRead$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUploadedResponseFilesUploaded$;
(function (FilesUploadedResponseFilesUploaded$) {
    /** @deprecated use `FilesUploadedResponseFilesUploaded$inboundSchema` instead. */
    FilesUploadedResponseFilesUploaded$.inboundSchema = exports.FilesUploadedResponseFilesUploaded$inboundSchema;
    /** @deprecated use `FilesUploadedResponseFilesUploaded$outboundSchema` instead. */
    FilesUploadedResponseFilesUploaded$.outboundSchema = exports.FilesUploadedResponseFilesUploaded$outboundSchema;
})(FilesUploadedResponseFilesUploaded$ || (exports.FilesUploadedResponseFilesUploaded$ = FilesUploadedResponseFilesUploaded$ = {}));
function filesUploadedResponseFilesUploadedToJSON(filesUploadedResponseFilesUploaded) {
    return JSON.stringify(exports.FilesUploadedResponseFilesUploaded$outboundSchema.parse(filesUploadedResponseFilesUploaded));
}
function filesUploadedResponseFilesUploadedFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FilesUploadedResponseFilesUploaded$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesUploadedResponseFilesUploaded' from JSON`);
}
