// @polar-sh/sdk/dist/commonjs/models/operations/oauth2authorize.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Oauth2AuthorizeResponseOauth2Authorize$ = exports.Oauth2AuthorizeResponseOauth2Authorize$outboundSchema = exports.Oauth2AuthorizeResponseOauth2Authorize$inboundSchema = void 0;
exports.oauth2AuthorizeResponseOauth2AuthorizeToJSON = oauth2AuthorizeResponseOauth2AuthorizeToJSON;
exports.oauth2AuthorizeResponseOauth2AuthorizeFromJSON = oauth2AuthorizeResponseOauth2AuthorizeFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const authorizeresponseorganization_js_1 = require("../components/authorizeresponseorganization.js");
const authorizeresponseuser_js_1 = require("../components/authorizeresponseuser.js");
/** @internal */
exports.Oauth2AuthorizeResponseOauth2Authorize$inboundSchema = z.union([
    authorizeresponseorganization_js_1.AuthorizeResponseOrganization$inboundSchema.and(z.object({ sub_type: z.literal("organization") }).transform((v) => ({
        subType: v.sub_type,
    }))),
    authorizeresponseuser_js_1.AuthorizeResponseUser$inboundSchema.and(z.object({ sub_type: z.literal("user") }).transform((v) => ({
        subType: v.sub_type,
    }))),
]);
/** @internal */
exports.Oauth2AuthorizeResponseOauth2Authorize$outboundSchema = z.union([
    authorizeresponseorganization_js_1.AuthorizeResponseOrganization$outboundSchema.and(z.object({ subType: z.literal("organization") }).transform((v) => ({
        sub_type: v.subType,
    }))),
    authorizeresponseuser_js_1.AuthorizeResponseUser$outboundSchema.and(z.object({ subType: z.literal("user") }).transform((v) => ({
        sub_type: v.subType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Oauth2AuthorizeResponseOauth2Authorize$;
(function (Oauth2AuthorizeResponseOauth2Authorize$) {
    /** @deprecated use `Oauth2AuthorizeResponseOauth2Authorize$inboundSchema` instead. */
    Oauth2AuthorizeResponseOauth2Authorize$.inboundSchema = exports.Oauth2AuthorizeResponseOauth2Authorize$inboundSchema;
    /** @deprecated use `Oauth2AuthorizeResponseOauth2Authorize$outboundSchema` instead. */
    Oauth2AuthorizeResponseOauth2Authorize$.outboundSchema = exports.Oauth2AuthorizeResponseOauth2Authorize$outboundSchema;
})(Oauth2AuthorizeResponseOauth2Authorize$ || (exports.Oauth2AuthorizeResponseOauth2Authorize$ = Oauth2AuthorizeResponseOauth2Authorize$ = {}));
function oauth2AuthorizeResponseOauth2AuthorizeToJSON(oauth2AuthorizeResponseOauth2Authorize) {
    return JSON.stringify(exports.Oauth2AuthorizeResponseOauth2Authorize$outboundSchema.parse(oauth2AuthorizeResponseOauth2Authorize));
}
function oauth2AuthorizeResponseOauth2AuthorizeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Oauth2AuthorizeResponseOauth2Authorize$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Oauth2AuthorizeResponseOauth2Authorize' from JSON`);
}
