// @polar-sh/sdk/dist/commonjs/models/operations/webhooksupdatewebhookendpoint.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksUpdateWebhookEndpointRequest$ = exports.WebhooksUpdateWebhookEndpointRequest$outboundSchema = exports.WebhooksUpdateWebhookEndpointRequest$inboundSchema = void 0;
exports.webhooksUpdateWebhookEndpointRequestToJSON = webhooksUpdateWebhookEndpointRequestToJSON;
exports.webhooksUpdateWebhookEndpointRequestFromJSON = webhooksUpdateWebhookEndpointRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const webhookendpointupdate_js_1 = require("../components/webhookendpointupdate.js");
/** @internal */
exports.WebhooksUpdateWebhookEndpointRequest$inboundSchema = z.object({
    id: z.string(),
    WebhookEndpointUpdate: webhookendpointupdate_js_1.WebhookEndpointUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "WebhookEndpointUpdate": "webhookEndpointUpdate",
    });
});
/** @internal */
exports.WebhooksUpdateWebhookEndpointRequest$outboundSchema = z.object({
    id: z.string(),
    webhookEndpointUpdate: webhookendpointupdate_js_1.WebhookEndpointUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        webhookEndpointUpdate: "WebhookEndpointUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhooksUpdateWebhookEndpointRequest$;
(function (WebhooksUpdateWebhookEndpointRequest$) {
    /** @deprecated use `WebhooksUpdateWebhookEndpointRequest$inboundSchema` instead. */
    WebhooksUpdateWebhookEndpointRequest$.inboundSchema = exports.WebhooksUpdateWebhookEndpointRequest$inboundSchema;
    /** @deprecated use `WebhooksUpdateWebhookEndpointRequest$outboundSchema` instead. */
    WebhooksUpdateWebhookEndpointRequest$.outboundSchema = exports.WebhooksUpdateWebhookEndpointRequest$outboundSchema;
})(WebhooksUpdateWebhookEndpointRequest$ || (exports.WebhooksUpdateWebhookEndpointRequest$ = WebhooksUpdateWebhookEndpointRequest$ = {}));
function webhooksUpdateWebhookEndpointRequestToJSON(webhooksUpdateWebhookEndpointRequest) {
    return JSON.stringify(exports.WebhooksUpdateWebhookEndpointRequest$outboundSchema.parse(webhooksUpdateWebhookEndpointRequest));
}
function webhooksUpdateWebhookEndpointRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhooksUpdateWebhookEndpointRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksUpdateWebhookEndpointRequest' from JSON`);
}
