// @wasp.sh/lib-auth/dist/node.js
import { createJWT, validateJWT } from "oslo/jwt";
import { TimeSpan } from "oslo";
import { hash, verify } from "@node-rs/argon2";
import { parseCookies } from "oslo/cookie";

//#region src/node/jwt.ts
/**
* Creates helper functions for creating and validating JWT tokens.
*
* @param JWT_SECRET Secret used to sign the JWT token.
* @param JWT_ALGORITHM Algorithm used to sign the JWT token.
* @returns An object containing `createJWT` and `validateJWT` functions.
*/
function createJWTHelpers(JWT_SECRET, JWT_ALGORITHM) {
	return {
		createJWT: (data, options) => {
			return createJWT(JWT_ALGORITHM, JWT_SECRET, data, options);
		},
		validateJWT: async (token) => {
			const { payload } = await validateJWT(JWT_ALGORITHM, JWT_SECRET, token);
			return payload;
		}
	};
}

//#endregion
//#region src/node/password.ts
const hashingOptions = {
	memoryCost: 19456,
	timeCost: 2,
	outputLen: 32,
	parallelism: 1
};
/**
* Hashes a password using the Argon2 algorithm.
* - It normalizes the password password before hashing to ensure consistency.
* - It doesn't require a salt input as the argon2 library generates a random salt for each hash.
*
* @param password The plain-text password to hash.
* @returns A PHC string (e.g. $argon2id$v=19$m=19456,t=2,p=1$...) whichs contains the hashed password and the salt.
* @see https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md for more information about the PHC string format.
*/
async function hashPassword(password) {
	return hash(normalizePassword(password), hashingOptions);
}
/**
* Verifies if a given password matches the hashed password.
* @param hashedPassword A PHC string (e.g. $argon2id$v=19$m=19456,t=2,p=1$...) whichs contains the hashed password and the salt.
* @param password The plain-text password to verify against the hashed password.
* @throws An error if the password is invalid.
* @see https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md for more information about the PHC string format.
*/
async function verifyPassword(hashedPassword, password) {
	if (!await verify(hashedPassword, normalizePassword(password), hashingOptions)) throw new Error("Invalid password");
}
/**
* We are normalising the password to ensure that the password is always hashed in the same way.
* @param password The plain-text password to normalize.
* @returns The normalized password.
*/
function normalizePassword(password) {
	return password.normalize("NFKC");
}

//#endregion
export { TimeSpan, createJWTHelpers, hashPassword, parseCookies, verifyPassword };
