// @polar-sh/sdk/dist/esm/models/operations/checkoutsclientget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CheckoutsClientGetRequest$inboundSchema = z.object({
    client_secret: z.string(),
}).transform((v) => {
    return remap$(v, {
        "client_secret": "clientSecret",
    });
});
/** @internal */
export const CheckoutsClientGetRequest$outboundSchema = z.object({
    clientSecret: z.string(),
}).transform((v) => {
    return remap$(v, {
        clientSecret: "client_secret",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutsClientGetRequest$;
(function (CheckoutsClientGetRequest$) {
    /** @deprecated use `CheckoutsClientGetRequest$inboundSchema` instead. */
    CheckoutsClientGetRequest$.inboundSchema = CheckoutsClientGetRequest$inboundSchema;
    /** @deprecated use `CheckoutsClientGetRequest$outboundSchema` instead. */
    CheckoutsClientGetRequest$.outboundSchema = CheckoutsClientGetRequest$outboundSchema;
})(CheckoutsClientGetRequest$ || (CheckoutsClientGetRequest$ = {}));
export function checkoutsClientGetRequestToJSON(checkoutsClientGetRequest) {
    return JSON.stringify(CheckoutsClientGetRequest$outboundSchema.parse(checkoutsClientGetRequest));
}
export function checkoutsClientGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutsClientGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsClientGetRequest' from JSON`);
}
