// @polar-sh/sdk/dist/esm/models/operations/checkoutlinksget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CheckoutLinksGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CheckoutLinksGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CheckoutLinksGetRequest$;
(function (CheckoutLinksGetRequest$) {
    /** @deprecated use `CheckoutLinksGetRequest$inboundSchema` instead. */
    CheckoutLinksGetRequest$.inboundSchema = CheckoutLinksGetRequest$inboundSchema;
    /** @deprecated use `CheckoutLinksGetRequest$outboundSchema` instead. */
    CheckoutLinksGetRequest$.outboundSchema = CheckoutLinksGetRequest$outboundSchema;
})(CheckoutLinksGetRequest$ || (CheckoutLinksGetRequest$ = {}));
export function checkoutLinksGetRequestToJSON(checkoutLinksGetRequest) {
    return JSON.stringify(CheckoutLinksGetRequest$outboundSchema.parse(checkoutLinksGetRequest));
}
export function checkoutLinksGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinksGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinksGetRequest' from JSON`);
}
