// @polar-sh/sdk/dist/esm/models/operations/customerportalordersinvoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalOrdersInvoiceSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalOrdersInvoiceSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersInvoiceSecurity$;
(function (CustomerPortalOrdersInvoiceSecurity$) {
    /** @deprecated use `CustomerPortalOrdersInvoiceSecurity$inboundSchema` instead. */
    CustomerPortalOrdersInvoiceSecurity$.inboundSchema = CustomerPortalOrdersInvoiceSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersInvoiceSecurity$outboundSchema` instead. */
    CustomerPortalOrdersInvoiceSecurity$.outboundSchema = CustomerPortalOrdersInvoiceSecurity$outboundSchema;
})(CustomerPortalOrdersInvoiceSecurity$ || (CustomerPortalOrdersInvoiceSecurity$ = {}));
export function customerPortalOrdersInvoiceSecurityToJSON(customerPortalOrdersInvoiceSecurity) {
    return JSON.stringify(CustomerPortalOrdersInvoiceSecurity$outboundSchema.parse(customerPortalOrdersInvoiceSecurity));
}
export function customerPortalOrdersInvoiceSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersInvoiceSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersInvoiceSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalOrdersInvoiceRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalOrdersInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersInvoiceRequest$;
(function (CustomerPortalOrdersInvoiceRequest$) {
    /** @deprecated use `CustomerPortalOrdersInvoiceRequest$inboundSchema` instead. */
    CustomerPortalOrdersInvoiceRequest$.inboundSchema = CustomerPortalOrdersInvoiceRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersInvoiceRequest$outboundSchema` instead. */
    CustomerPortalOrdersInvoiceRequest$.outboundSchema = CustomerPortalOrdersInvoiceRequest$outboundSchema;
})(CustomerPortalOrdersInvoiceRequest$ || (CustomerPortalOrdersInvoiceRequest$ = {}));
export function customerPortalOrdersInvoiceRequestToJSON(customerPortalOrdersInvoiceRequest) {
    return JSON.stringify(CustomerPortalOrdersInvoiceRequest$outboundSchema.parse(customerPortalOrdersInvoiceRequest));
}
export function customerPortalOrdersInvoiceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersInvoiceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersInvoiceRequest' from JSON`);
}
