// @polar-sh/sdk/dist/esm/models/operations/customerportalbenefitgrantsupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerBenefitGrantUpdate$inboundSchema, CustomerBenefitGrantUpdate$outboundSchema, } from "../components/customerbenefitgrantupdate.js";
/** @internal */
export const CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalBenefitGrantsUpdateSecurity$;
(function (CustomerPortalBenefitGrantsUpdateSecurity$) {
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateSecurity$.inboundSchema = CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateSecurity$.outboundSchema = CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema;
})(CustomerPortalBenefitGrantsUpdateSecurity$ || (CustomerPortalBenefitGrantsUpdateSecurity$ = {}));
export function customerPortalBenefitGrantsUpdateSecurityToJSON(customerPortalBenefitGrantsUpdateSecurity) {
    return JSON.stringify(CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema.parse(customerPortalBenefitGrantsUpdateSecurity));
}
export function customerPortalBenefitGrantsUpdateSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalBenefitGrantsUpdateSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsUpdateSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalBenefitGrantsUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    CustomerBenefitGrantUpdate: CustomerBenefitGrantUpdate$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "CustomerBenefitGrantUpdate": "customerBenefitGrantUpdate",
    });
});
/** @internal */
export const CustomerPortalBenefitGrantsUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    customerBenefitGrantUpdate: CustomerBenefitGrantUpdate$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        customerBenefitGrantUpdate: "CustomerBenefitGrantUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalBenefitGrantsUpdateRequest$;
(function (CustomerPortalBenefitGrantsUpdateRequest$) {
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateRequest$inboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateRequest$.inboundSchema = CustomerPortalBenefitGrantsUpdateRequest$inboundSchema;
    /** @deprecated use `CustomerPortalBenefitGrantsUpdateRequest$outboundSchema` instead. */
    CustomerPortalBenefitGrantsUpdateRequest$.outboundSchema = CustomerPortalBenefitGrantsUpdateRequest$outboundSchema;
})(CustomerPortalBenefitGrantsUpdateRequest$ || (CustomerPortalBenefitGrantsUpdateRequest$ = {}));
export function customerPortalBenefitGrantsUpdateRequestToJSON(customerPortalBenefitGrantsUpdateRequest) {
    return JSON.stringify(CustomerPortalBenefitGrantsUpdateRequest$outboundSchema.parse(customerPortalBenefitGrantsUpdateRequest));
}
export function customerPortalBenefitGrantsUpdateRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalBenefitGrantsUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsUpdateRequest' from JSON`);
}
