// @polar-sh/sdk/dist/esm/models/operations/customerportalordersgenerateinvoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersGenerateInvoiceSecurity$;
(function (CustomerPortalOrdersGenerateInvoiceSecurity$) {
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceSecurity$.inboundSchema = CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceSecurity$.outboundSchema = CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema;
})(CustomerPortalOrdersGenerateInvoiceSecurity$ || (CustomerPortalOrdersGenerateInvoiceSecurity$ = {}));
export function customerPortalOrdersGenerateInvoiceSecurityToJSON(customerPortalOrdersGenerateInvoiceSecurity) {
    return JSON.stringify(CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema.parse(customerPortalOrdersGenerateInvoiceSecurity));
}
export function customerPortalOrdersGenerateInvoiceSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersGenerateInvoiceSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGenerateInvoiceSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersGenerateInvoiceRequest$;
(function (CustomerPortalOrdersGenerateInvoiceRequest$) {
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceRequest$.inboundSchema = CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema` instead. */
    CustomerPortalOrdersGenerateInvoiceRequest$.outboundSchema = CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema;
})(CustomerPortalOrdersGenerateInvoiceRequest$ || (CustomerPortalOrdersGenerateInvoiceRequest$ = {}));
export function customerPortalOrdersGenerateInvoiceRequestToJSON(customerPortalOrdersGenerateInvoiceRequest) {
    return JSON.stringify(CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema.parse(customerPortalOrdersGenerateInvoiceRequest));
}
export function customerPortalOrdersGenerateInvoiceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersGenerateInvoiceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersGenerateInvoiceRequest' from JSON`);
}
