// @polar-sh/sdk/dist/esm/models/operations/customfieldsdelete.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomFieldsDeleteRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomFieldsDeleteRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomFieldsDeleteRequest$;
(function (CustomFieldsDeleteRequest$) {
    /** @deprecated use `CustomFieldsDeleteRequest$inboundSchema` instead. */
    CustomFieldsDeleteRequest$.inboundSchema = CustomFieldsDeleteRequest$inboundSchema;
    /** @deprecated use `CustomFieldsDeleteRequest$outboundSchema` instead. */
    CustomFieldsDeleteRequest$.outboundSchema = CustomFieldsDeleteRequest$outboundSchema;
})(CustomFieldsDeleteRequest$ || (CustomFieldsDeleteRequest$ = {}));
export function customFieldsDeleteRequestToJSON(customFieldsDeleteRequest) {
    return JSON.stringify(CustomFieldsDeleteRequest$outboundSchema.parse(customFieldsDeleteRequest));
}
export function customFieldsDeleteRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldsDeleteRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldsDeleteRequest' from JSON`);
}
