// @polar-sh/sdk/dist/esm/models/operations/customerportalorganizationsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalOrganizationsGetRequest$inboundSchema = z.object({
    slug: z.string(),
});
/** @internal */
export const CustomerPortalOrganizationsGetRequest$outboundSchema = z.object({
    slug: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrganizationsGetRequest$;
(function (CustomerPortalOrganizationsGetRequest$) {
    /** @deprecated use `CustomerPortalOrganizationsGetRequest$inboundSchema` instead. */
    CustomerPortalOrganizationsGetRequest$.inboundSchema = CustomerPortalOrganizationsGetRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrganizationsGetRequest$outboundSchema` instead. */
    CustomerPortalOrganizationsGetRequest$.outboundSchema = CustomerPortalOrganizationsGetRequest$outboundSchema;
})(CustomerPortalOrganizationsGetRequest$ || (CustomerPortalOrganizationsGetRequest$ = {}));
export function customerPortalOrganizationsGetRequestToJSON(customerPortalOrganizationsGetRequest) {
    return JSON.stringify(CustomerPortalOrganizationsGetRequest$outboundSchema.parse(customerPortalOrganizationsGetRequest));
}
export function customerPortalOrganizationsGetRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrganizationsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrganizationsGetRequest' from JSON`);
}
