// @polar-sh/sdk/dist/esm/models/operations/ordersgenerateinvoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrdersGenerateInvoiceRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const OrdersGenerateInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrdersGenerateInvoiceRequest$;
(function (OrdersGenerateInvoiceRequest$) {
    /** @deprecated use `OrdersGenerateInvoiceRequest$inboundSchema` instead. */
    OrdersGenerateInvoiceRequest$.inboundSchema = OrdersGenerateInvoiceRequest$inboundSchema;
    /** @deprecated use `OrdersGenerateInvoiceRequest$outboundSchema` instead. */
    OrdersGenerateInvoiceRequest$.outboundSchema = OrdersGenerateInvoiceRequest$outboundSchema;
})(OrdersGenerateInvoiceRequest$ || (OrdersGenerateInvoiceRequest$ = {}));
export function ordersGenerateInvoiceRequestToJSON(ordersGenerateInvoiceRequest) {
    return JSON.stringify(OrdersGenerateInvoiceRequest$outboundSchema.parse(ordersGenerateInvoiceRequest));
}
export function ordersGenerateInvoiceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrdersGenerateInvoiceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersGenerateInvoiceRequest' from JSON`);
}
