// @polar-sh/sdk/dist/esm/models/operations/ordersinvoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrdersInvoiceRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const OrdersInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrdersInvoiceRequest$;
(function (OrdersInvoiceRequest$) {
    /** @deprecated use `OrdersInvoiceRequest$inboundSchema` instead. */
    OrdersInvoiceRequest$.inboundSchema = OrdersInvoiceRequest$inboundSchema;
    /** @deprecated use `OrdersInvoiceRequest$outboundSchema` instead. */
    OrdersInvoiceRequest$.outboundSchema = OrdersInvoiceRequest$outboundSchema;
})(OrdersInvoiceRequest$ || (OrdersInvoiceRequest$ = {}));
export function ordersInvoiceRequestToJSON(ordersInvoiceRequest) {
    return JSON.stringify(OrdersInvoiceRequest$outboundSchema.parse(ordersInvoiceRequest));
}
export function ordersInvoiceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrdersInvoiceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersInvoiceRequest' from JSON`);
}
