// @polar-sh/sdk/dist/esm/models/operations/productsupdatebenefits.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductBenefitsUpdate$inboundSchema, ProductBenefitsUpdate$outboundSchema, } from "../components/productbenefitsupdate.js";
/** @internal */
export const ProductsUpdateBenefitsRequest$inboundSchema = z.object({
    id: z.string(),
    ProductBenefitsUpdate: ProductBenefitsUpdate$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "ProductBenefitsUpdate": "productBenefitsUpdate",
    });
});
/** @internal */
export const ProductsUpdateBenefitsRequest$outboundSchema = z.object({
    id: z.string(),
    productBenefitsUpdate: ProductBenefitsUpdate$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        productBenefitsUpdate: "ProductBenefitsUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ProductsUpdateBenefitsRequest$;
(function (ProductsUpdateBenefitsRequest$) {
    /** @deprecated use `ProductsUpdateBenefitsRequest$inboundSchema` instead. */
    ProductsUpdateBenefitsRequest$.inboundSchema = ProductsUpdateBenefitsRequest$inboundSchema;
    /** @deprecated use `ProductsUpdateBenefitsRequest$outboundSchema` instead. */
    ProductsUpdateBenefitsRequest$.outboundSchema = ProductsUpdateBenefitsRequest$outboundSchema;
})(ProductsUpdateBenefitsRequest$ || (ProductsUpdateBenefitsRequest$ = {}));
export function productsUpdateBenefitsRequestToJSON(productsUpdateBenefitsRequest) {
    return JSON.stringify(ProductsUpdateBenefitsRequest$outboundSchema.parse(productsUpdateBenefitsRequest));
}
export function productsUpdateBenefitsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductsUpdateBenefitsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductsUpdateBenefitsRequest' from JSON`);
}
