// @polar-sh/sdk/dist/esm/models/operations/webhooksupdatewebhookendpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { WebhookEndpointUpdate$inboundSchema, WebhookEndpointUpdate$outboundSchema, } from "../components/webhookendpointupdate.js";
/** @internal */
export const WebhooksUpdateWebhookEndpointRequest$inboundSchema = z.object({
    id: z.string(),
    WebhookEndpointUpdate: WebhookEndpointUpdate$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "WebhookEndpointUpdate": "webhookEndpointUpdate",
    });
});
/** @internal */
export const WebhooksUpdateWebhookEndpointRequest$outboundSchema = z.object({
    id: z.string(),
    webhookEndpointUpdate: WebhookEndpointUpdate$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        webhookEndpointUpdate: "WebhookEndpointUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhooksUpdateWebhookEndpointRequest$;
(function (WebhooksUpdateWebhookEndpointRequest$) {
    /** @deprecated use `WebhooksUpdateWebhookEndpointRequest$inboundSchema` instead. */
    WebhooksUpdateWebhookEndpointRequest$.inboundSchema = WebhooksUpdateWebhookEndpointRequest$inboundSchema;
    /** @deprecated use `WebhooksUpdateWebhookEndpointRequest$outboundSchema` instead. */
    WebhooksUpdateWebhookEndpointRequest$.outboundSchema = WebhooksUpdateWebhookEndpointRequest$outboundSchema;
})(WebhooksUpdateWebhookEndpointRequest$ || (WebhooksUpdateWebhookEndpointRequest$ = {}));
export function webhooksUpdateWebhookEndpointRequestToJSON(webhooksUpdateWebhookEndpointRequest) {
    return JSON.stringify(WebhooksUpdateWebhookEndpointRequest$outboundSchema.parse(webhooksUpdateWebhookEndpointRequest));
}
export function webhooksUpdateWebhookEndpointRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhooksUpdateWebhookEndpointRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksUpdateWebhookEndpointRequest' from JSON`);
}
