// @polar-sh/sdk/dist/esm/models/operations/webhooksresetwebhookendpointsecret.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const WebhooksResetWebhookEndpointSecretRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const WebhooksResetWebhookEndpointSecretRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhooksResetWebhookEndpointSecretRequest$;
(function (WebhooksResetWebhookEndpointSecretRequest$) {
    /** @deprecated use `WebhooksResetWebhookEndpointSecretRequest$inboundSchema` instead. */
    WebhooksResetWebhookEndpointSecretRequest$.inboundSchema = WebhooksResetWebhookEndpointSecretRequest$inboundSchema;
    /** @deprecated use `WebhooksResetWebhookEndpointSecretRequest$outboundSchema` instead. */
    WebhooksResetWebhookEndpointSecretRequest$.outboundSchema = WebhooksResetWebhookEndpointSecretRequest$outboundSchema;
})(WebhooksResetWebhookEndpointSecretRequest$ || (WebhooksResetWebhookEndpointSecretRequest$ = {}));
export function webhooksResetWebhookEndpointSecretRequestToJSON(webhooksResetWebhookEndpointSecretRequest) {
    return JSON.stringify(WebhooksResetWebhookEndpointSecretRequest$outboundSchema.parse(webhooksResetWebhookEndpointSecretRequest));
}
export function webhooksResetWebhookEndpointSecretRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhooksResetWebhookEndpointSecretRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksResetWebhookEndpointSecretRequest' from JSON`);
}
