// @polar-sh/sdk/dist/esm/models/operations/webhooksdeletewebhookendpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const WebhooksDeleteWebhookEndpointRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const WebhooksDeleteWebhookEndpointRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var WebhooksDeleteWebhookEndpointRequest$;
(function (WebhooksDeleteWebhookEndpointRequest$) {
    /** @deprecated use `WebhooksDeleteWebhookEndpointRequest$inboundSchema` instead. */
    WebhooksDeleteWebhookEndpointRequest$.inboundSchema = WebhooksDeleteWebhookEndpointRequest$inboundSchema;
    /** @deprecated use `WebhooksDeleteWebhookEndpointRequest$outboundSchema` instead. */
    WebhooksDeleteWebhookEndpointRequest$.outboundSchema = WebhooksDeleteWebhookEndpointRequest$outboundSchema;
})(WebhooksDeleteWebhookEndpointRequest$ || (WebhooksDeleteWebhookEndpointRequest$ = {}));
export function webhooksDeleteWebhookEndpointRequestToJSON(webhooksDeleteWebhookEndpointRequest) {
    return JSON.stringify(WebhooksDeleteWebhookEndpointRequest$outboundSchema.parse(webhooksDeleteWebhookEndpointRequest));
}
export function webhooksDeleteWebhookEndpointRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhooksDeleteWebhookEndpointRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksDeleteWebhookEndpointRequest' from JSON`);
}
