// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersGetState.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersGetState = void 0;
const customersGetState_js_1 = require("../../funcs/customersGetState.js");
const customersgetstate_js_1 = require("../../models/operations/customersgetstate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersgetstate_js_1.CustomersGetStateRequest$inboundSchema,
};
exports.tool$customersGetState = {
    name: "customers-get-state",
    description: `Get Customer State

Get a customer state by ID.

The customer state includes information about
the customer's active subscriptions and benefits.

It's the ideal endpoint to use when you need to get a full overview
of a customer's status.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersGetState_js_1.customersGetState)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
