// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersUpdate = void 0;
const customersUpdate_js_1 = require("../../funcs/customersUpdate.js");
const customersupdate_js_1 = require("../../models/operations/customersupdate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersupdate_js_1.CustomersUpdateRequest$inboundSchema,
};
exports.tool$customersUpdate = {
    name: "customers-update",
    description: `Update Customer

Update a customer.

**Scopes**: \`customers:write\``,
    scopes: ["write", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersUpdate_js_1.customersUpdate)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
