// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersCreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersCreate = void 0;
const customersCreate_js_1 = require("../../funcs/customersCreate.js");
const customercreate_js_1 = require("../../models/components/customercreate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customercreate_js_1.CustomerCreate$inboundSchema,
};
exports.tool$customersCreate = {
    name: "customers-create",
    description: `Create Customer

Create a customer.

**Scopes**: \`customers:write\``,
    scopes: ["write", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersCreate_js_1.customersCreate)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
