// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersDeleteExternal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersDeleteExternal = void 0;
const customersDeleteExternal_js_1 = require("../../funcs/customersDeleteExternal.js");
const customersdeleteexternal_js_1 = require("../../models/operations/customersdeleteexternal.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersdeleteexternal_js_1.CustomersDeleteExternalRequest$inboundSchema,
};
exports.tool$customersDeleteExternal = {
    name: "customers-delete-external",
    description: `Delete Customer by External ID

Delete a customer by external ID.

Immediately cancels any active subscriptions and revokes any active benefits.

**Scopes**: \`customers:write\``,
    scopes: ["write", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersDeleteExternal_js_1.customersDeleteExternal)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return (0, tools_js_1.formatResult)(void 0, apiCall);
    },
};
