// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersList = void 0;
const customersList_js_1 = require("../../funcs/customersList.js");
const customerslist_js_1 = require("../../models/operations/customerslist.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customerslist_js_1.CustomersListRequest$inboundSchema,
};
exports.tool$customersList = {
    name: "customers-list",
    description: `List Customers

List customers.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersList_js_1.customersList)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value.result;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
