// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customerMetersGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customerMetersGet = void 0;
const customerMetersGet_js_1 = require("../../funcs/customerMetersGet.js");
const customermetersget_js_1 = require("../../models/operations/customermetersget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customermetersget_js_1.CustomerMetersGetRequest$inboundSchema,
};
exports.tool$customerMetersGet = {
    name: "customer-meters-get",
    description: `Get Customer Meter

Get a customer meter by ID.

**Scopes**: \`customer_meters:read\``,
    scopes: ["read", "customer_meters"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customerMetersGet_js_1.customerMetersGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
