// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersGet = void 0;
const customersGet_js_1 = require("../../funcs/customersGet.js");
const customersget_js_1 = require("../../models/operations/customersget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersget_js_1.CustomersGetRequest$inboundSchema,
};
exports.tool$customersGet = {
    name: "customers-get",
    description: `Get Customer

Get a customer by ID.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersGet_js_1.customersGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
