// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersUpdateExternal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersUpdateExternal = void 0;
const customersUpdateExternal_js_1 = require("../../funcs/customersUpdateExternal.js");
const customersupdateexternal_js_1 = require("../../models/operations/customersupdateexternal.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersupdateexternal_js_1.CustomersUpdateExternalRequest$inboundSchema,
};
exports.tool$customersUpdateExternal = {
    name: "customers-update-external",
    description: `Update Customer by External ID

Update a customer by external ID.

**Scopes**: \`customers:write\``,
    scopes: ["write", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersUpdateExternal_js_1.customersUpdateExternal)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
