// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersGetStateExternal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersGetStateExternal = void 0;
const customersGetStateExternal_js_1 = require("../../funcs/customersGetStateExternal.js");
const customersgetstateexternal_js_1 = require("../../models/operations/customersgetstateexternal.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersgetstateexternal_js_1.CustomersGetStateExternalRequest$inboundSchema,
};
exports.tool$customersGetStateExternal = {
    name: "customers-get-state-external",
    description: `Get Customer State by External ID

Get a customer state by external ID.

The customer state includes information about
the customer's active subscriptions and benefits.

It's the ideal endpoint to use when you need to get a full overview
of a customer's status.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersGetStateExternal_js_1.customersGetStateExternal)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
