// @polar-sh/sdk/dist/commonjs/mcp-server/tools/metricsGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$metricsGet = void 0;
const metricsGet_js_1 = require("../../funcs/metricsGet.js");
const metricsget_js_1 = require("../../models/operations/metricsget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: metricsget_js_1.MetricsGetRequest$inboundSchema,
};
exports.tool$metricsGet = {
    name: "metrics-get",
    description: `Get Metrics

Get metrics about your orders and subscriptions.

Currency values are output in cents.

**Scopes**: \`metrics:read\``,
    scopes: ["read", "metrics"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, metricsGet_js_1.metricsGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
