// @polar-sh/sdk/dist/commonjs/mcp-server/tools/ordersGenerateInvoice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$ordersGenerateInvoice = void 0;
const ordersGenerateInvoice_js_1 = require("../../funcs/ordersGenerateInvoice.js");
const ordersgenerateinvoice_js_1 = require("../../models/operations/ordersgenerateinvoice.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: ordersgenerateinvoice_js_1.OrdersGenerateInvoiceRequest$inboundSchema,
};
exports.tool$ordersGenerateInvoice = {
    name: "orders-generate-invoice",
    description: `Generate Order Invoice

Trigger generation of an order's invoice.

**Scopes**: \`orders:read\``,
    scopes: ["write", "orders"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, ordersGenerateInvoice_js_1.ordersGenerateInvoice)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
