// @polar-sh/sdk/dist/commonjs/mcp-server/tools/ordersInvoice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$ordersInvoice = void 0;
const ordersInvoice_js_1 = require("../../funcs/ordersInvoice.js");
const ordersinvoice_js_1 = require("../../models/operations/ordersinvoice.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: ordersinvoice_js_1.OrdersInvoiceRequest$inboundSchema,
};
exports.tool$ordersInvoice = {
    name: "orders-invoice",
    description: `Get Order Invoice

Get an order's invoice data.

**Scopes**: \`orders:read\``,
    scopes: ["read", "orders"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, ordersInvoice_js_1.ordersInvoice)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
