// @polar-sh/sdk/dist/commonjs/mcp-server/tools/ordersUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$ordersUpdate = void 0;
const ordersUpdate_js_1 = require("../../funcs/ordersUpdate.js");
const ordersupdate_js_1 = require("../../models/operations/ordersupdate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: ordersupdate_js_1.OrdersUpdateRequest$inboundSchema,
};
exports.tool$ordersUpdate = {
    name: "orders-update",
    description: `Update Order

Update an order.

**Scopes**: \`orders:write\``,
    scopes: ["write", "orders"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, ordersUpdate_js_1.ordersUpdate)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
