// @polar-sh/sdk/dist/commonjs/mcp-server/tools/ordersGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$ordersGet = void 0;
const ordersGet_js_1 = require("../../funcs/ordersGet.js");
const ordersget_js_1 = require("../../models/operations/ordersget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: ordersget_js_1.OrdersGetRequest$inboundSchema,
};
exports.tool$ordersGet = {
    name: "orders-get",
    description: `Get Order

Get an order by ID.

**Scopes**: \`orders:read\``,
    scopes: ["read", "orders"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, ordersGet_js_1.ordersGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
