// @polar-sh/sdk/dist/commonjs/mcp-server/tools/paymentsList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$paymentsList = void 0;
const paymentsList_js_1 = require("../../funcs/paymentsList.js");
const paymentslist_js_1 = require("../../models/operations/paymentslist.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: paymentslist_js_1.PaymentsListRequest$inboundSchema,
};
exports.tool$paymentsList = {
    name: "payments-list",
    description: `List Payments

List payments.

**Scopes**: \`payments:read\``,
    scopes: ["read", "payments"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, paymentsList_js_1.paymentsList)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value.result;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
