// @polar-sh/sdk/dist/commonjs/mcp-server/tools/productsCreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$productsCreate = void 0;
const productsCreate_js_1 = require("../../funcs/productsCreate.js");
const productcreate_js_1 = require("../../models/components/productcreate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: productcreate_js_1.ProductCreate$inboundSchema,
};
exports.tool$productsCreate = {
    name: "products-create",
    description: `Create Product

Create a product.

**Scopes**: \`products:write\``,
    scopes: ["write", "products"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, productsCreate_js_1.productsCreate)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
