// @polar-sh/sdk/dist/commonjs/mcp-server/tools/productsList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$productsList = void 0;
const productsList_js_1 = require("../../funcs/productsList.js");
const productslist_js_1 = require("../../models/operations/productslist.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: productslist_js_1.ProductsListRequest$inboundSchema,
};
exports.tool$productsList = {
    name: "products-list",
    description: `List Products

List products.

**Scopes**: \`products:read\` \`products:write\``,
    scopes: ["read", "products"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, productsList_js_1.productsList)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value.result;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
