// @polar-sh/sdk/dist/commonjs/mcp-server/tools/subscriptionsRevoke.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$subscriptionsRevoke = void 0;
const subscriptionsRevoke_js_1 = require("../../funcs/subscriptionsRevoke.js");
const subscriptionsrevoke_js_1 = require("../../models/operations/subscriptionsrevoke.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: subscriptionsrevoke_js_1.SubscriptionsRevokeRequest$inboundSchema,
};
exports.tool$subscriptionsRevoke = {
    name: "subscriptions-revoke",
    description: `Revoke Subscription

Revoke a subscription, i.e cancel immediately.

**Scopes**: \`subscriptions:write\``,
    scopes: ["write", "subscriptions"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, subscriptionsRevoke_js_1.subscriptionsRevoke)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
