// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersGetExternal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersGetExternal = void 0;
const customersGetExternal_js_1 = require("../../funcs/customersGetExternal.js");
const customersgetexternal_js_1 = require("../../models/operations/customersgetexternal.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersgetexternal_js_1.CustomersGetExternalRequest$inboundSchema,
};
exports.tool$customersGetExternal = {
    name: "customers-get-external",
    description: `Get Customer by External ID

Get a customer by external ID.

**Scopes**: \`customers:read\` \`customers:write\``,
    scopes: ["read", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersGetExternal_js_1.customersGetExternal)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
