// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customerMetersList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customerMetersList = void 0;
const customerMetersList_js_1 = require("../../funcs/customerMetersList.js");
const customermeterslist_js_1 = require("../../models/operations/customermeterslist.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customermeterslist_js_1.CustomerMetersListRequest$inboundSchema,
};
exports.tool$customerMetersList = {
    name: "customer-meters-list",
    description: `List Customer Meters

List customer meters.

**Scopes**: \`customer_meters:read\``,
    scopes: ["read", "customer_meters"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customerMetersList_js_1.customerMetersList)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value.result;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
