// @polar-sh/sdk/dist/commonjs/mcp-server/tools/ordersList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$ordersList = void 0;
const ordersList_js_1 = require("../../funcs/ordersList.js");
const orderslist_js_1 = require("../../models/operations/orderslist.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: orderslist_js_1.OrdersListRequest$inboundSchema,
};
exports.tool$ordersList = {
    name: "orders-list",
    description: `List Orders

List orders.

**Scopes**: \`orders:read\``,
    scopes: ["read", "orders"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, ordersList_js_1.ordersList)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value.result;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
