// @polar-sh/sdk/dist/commonjs/mcp-server/tools/paymentsGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$paymentsGet = void 0;
const paymentsGet_js_1 = require("../../funcs/paymentsGet.js");
const paymentsget_js_1 = require("../../models/operations/paymentsget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: paymentsget_js_1.PaymentsGetRequest$inboundSchema,
};
exports.tool$paymentsGet = {
    name: "payments-get",
    description: `Get Payment

Get a payment by ID.

**Scopes**: \`payments:read\``,
    scopes: ["read", "payments"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, paymentsGet_js_1.paymentsGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
