// @polar-sh/sdk/dist/commonjs/mcp-server/tools/productsUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$productsUpdate = void 0;
const productsUpdate_js_1 = require("../../funcs/productsUpdate.js");
const productsupdate_js_1 = require("../../models/operations/productsupdate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: productsupdate_js_1.ProductsUpdateRequest$inboundSchema,
};
exports.tool$productsUpdate = {
    name: "products-update",
    description: `Update Product

Update a product.

**Scopes**: \`products:write\``,
    scopes: ["write", "products"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, productsUpdate_js_1.productsUpdate)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
