// @polar-sh/sdk/dist/commonjs/mcp-server/tools/productsUpdateBenefits.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$productsUpdateBenefits = void 0;
const productsUpdateBenefits_js_1 = require("../../funcs/productsUpdateBenefits.js");
const productsupdatebenefits_js_1 = require("../../models/operations/productsupdatebenefits.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: productsupdatebenefits_js_1.ProductsUpdateBenefitsRequest$inboundSchema,
};
exports.tool$productsUpdateBenefits = {
    name: "products-update-benefits",
    description: `Update Product Benefits

Update benefits granted by a product.

**Scopes**: \`products:write\``,
    scopes: ["write", "products"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, productsUpdateBenefits_js_1.productsUpdateBenefits)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
