// @polar-sh/sdk/dist/commonjs/mcp-server/tools/productsGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$productsGet = void 0;
const productsGet_js_1 = require("../../funcs/productsGet.js");
const productsget_js_1 = require("../../models/operations/productsget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: productsget_js_1.ProductsGetRequest$inboundSchema,
};
exports.tool$productsGet = {
    name: "products-get",
    description: `Get Product

Get a product by ID.

**Scopes**: \`products:read\` \`products:write\``,
    scopes: ["read", "products"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, productsGet_js_1.productsGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
