// @polar-sh/sdk/dist/commonjs/mcp-server/tools/subscriptionsGet.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$subscriptionsGet = void 0;
const subscriptionsGet_js_1 = require("../../funcs/subscriptionsGet.js");
const subscriptionsget_js_1 = require("../../models/operations/subscriptionsget.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: subscriptionsget_js_1.SubscriptionsGetRequest$inboundSchema,
};
exports.tool$subscriptionsGet = {
    name: "subscriptions-get",
    description: `Get Subscription

Get a subscription by ID.

**Scopes**: \`subscriptions:read\` \`subscriptions:write\``,
    scopes: ["read", "subscriptions"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, subscriptionsGet_js_1.subscriptionsGet)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
