// @polar-sh/sdk/dist/commonjs/mcp-server/tools/subscriptionsUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$subscriptionsUpdate = void 0;
const subscriptionsUpdate_js_1 = require("../../funcs/subscriptionsUpdate.js");
const subscriptionsupdate_js_1 = require("../../models/operations/subscriptionsupdate.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: subscriptionsupdate_js_1.SubscriptionsUpdateRequest$inboundSchema,
};
exports.tool$subscriptionsUpdate = {
    name: "subscriptions-update",
    description: `Update Subscription

Update a subscription.

**Scopes**: \`subscriptions:write\``,
    scopes: ["write", "subscriptions"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, subscriptionsUpdate_js_1.subscriptionsUpdate)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
