// @polar-sh/sdk/dist/commonjs/mcp-server/tools/subscriptionsExport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$subscriptionsExport = void 0;
const subscriptionsExport_js_1 = require("../../funcs/subscriptionsExport.js");
const subscriptionsexport_js_1 = require("../../models/operations/subscriptionsexport.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: subscriptionsexport_js_1.SubscriptionsExportRequest$inboundSchema,
};
exports.tool$subscriptionsExport = {
    name: "subscriptions-export",
    description: `Export Subscriptions

Export subscriptions as a CSV file.

**Scopes**: \`subscriptions:read\` \`subscriptions:write\``,
    scopes: ["read", "subscriptions"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, subscriptionsExport_js_1.subscriptionsExport)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
