// @polar-sh/sdk/dist/commonjs/mcp-server/tools/subscriptionsList.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$subscriptionsList = void 0;
const subscriptionsList_js_1 = require("../../funcs/subscriptionsList.js");
const subscriptionslist_js_1 = require("../../models/operations/subscriptionslist.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: subscriptionslist_js_1.SubscriptionsListRequest$inboundSchema,
};
exports.tool$subscriptionsList = {
    name: "subscriptions-list",
    description: `List Subscriptions

List subscriptions.

**Scopes**: \`subscriptions:read\` \`subscriptions:write\``,
    scopes: ["read", "subscriptions"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, subscriptionsList_js_1.subscriptionsList)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        const value = result.value.result;
        return (0, tools_js_1.formatResult)(value, apiCall);
    },
};
