// @polar-sh/sdk/dist/commonjs/funcs/checkoutsClientUpdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.checkoutsClientUpdate = checkoutsClientUpdate;
const encodings_js_1 = require("../lib/encodings.js");
const M = __importStar(require("../lib/matchers.js"));
const primitives_js_1 = require("../lib/primitives.js");
const schemas_js_1 = require("../lib/schemas.js");
const url_js_1 = require("../lib/url.js");
const checkoutpublic_js_1 = require("../models/components/checkoutpublic.js");
const checkoutforbiddenerror_js_1 = require("../models/errors/checkoutforbiddenerror.js");
const expiredcheckouterror_js_1 = require("../models/errors/expiredcheckouterror.js");
const httpvalidationerror_js_1 = require("../models/errors/httpvalidationerror.js");
const resourcenotfound_js_1 = require("../models/errors/resourcenotfound.js");
const checkoutsclientupdate_js_1 = require("../models/operations/checkoutsclientupdate.js");
const async_js_1 = require("../types/async.js");
/**
 * Update Checkout Session from Client
 *
 * @remarks
 * Update a checkout session by client secret.
 */
function checkoutsClientUpdate(client, request, options) {
    return new async_js_1.APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = (0, schemas_js_1.safeParse)(request, (value) => checkoutsclientupdate_js_1.CheckoutsClientUpdateRequest$outboundSchema.parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [parsed, { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = (0, encodings_js_1.encodeJSON)("body", payload.CheckoutUpdatePublic, {
        explode: true,
    });
    const pathParams = {
        client_secret: (0, encodings_js_1.encodeSimple)("client_secret", payload.client_secret, {
            explode: false,
            charEncoding: "percent",
        }),
    };
    const path = (0, url_js_1.pathToFunc)("/v1/checkouts/client/{client_secret}")(pathParams);
    const headers = new Headers((0, primitives_js_1.compactMap)({
        "Content-Type": "application/json",
        Accept: "application/json",
    }));
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "checkouts:client_update",
        oAuth2Scopes: [],
        resolvedSecurity: null,
        securitySource: null,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        method: "PATCH",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["403", "404", "410", "422", "4XX", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result] = await M.match(M.json(200, checkoutpublic_js_1.CheckoutPublic$inboundSchema), M.jsonErr(403, checkoutforbiddenerror_js_1.CheckoutForbiddenError$inboundSchema), M.jsonErr(404, resourcenotfound_js_1.ResourceNotFound$inboundSchema), M.jsonErr(410, expiredcheckouterror_js_1.ExpiredCheckoutError$inboundSchema), M.jsonErr(422, httpvalidationerror_js_1.HTTPValidationError$inboundSchema), M.fail("4XX"), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
