// @polar-sh/sdk/dist/commonjs/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.ServerList = exports.ServerSandbox = exports.ServerProduction = void 0;
exports.serverURLFromOptions = serverURLFromOptions;
const url_js_1 = require("./url.js");
/**
 * Production environment
 */
exports.ServerProduction = "production";
/**
 * Sandbox environment
 */
exports.ServerSandbox = "sandbox";
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = {
    [exports.ServerProduction]: "https://api.polar.sh",
    [exports.ServerSandbox]: "https://sandbox-api.polar.sh",
};
function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = options.server ?? exports.ServerProduction;
        serverURL = exports.ServerList[server] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "0.1.0",
    sdkVersion: "0.34.17",
    genVersion: "2.698.4",
    userAgent: "speakeasy-sdk/typescript 0.34.17 2.698.4 0.1.0 @polar-sh/sdk",
};
