// @polar-sh/sdk/dist/commonjs/models/errors/ordernoteligibleforretry.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderNotEligibleForRetry$ = exports.OrderNotEligibleForRetry$outboundSchema = exports.OrderNotEligibleForRetry$inboundSchema = exports.OrderNotEligibleForRetry = void 0;
const z = __importStar(require("zod"));
const polarerror_js_1 = require("./polarerror.js");
class OrderNotEligibleForRetry extends polarerror_js_1.PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "OrderNotEligibleForRetry";
    }
}
exports.OrderNotEligibleForRetry = OrderNotEligibleForRetry;
/** @internal */
exports.OrderNotEligibleForRetry$inboundSchema = z.object({
    error: z.literal("OrderNotEligibleForRetry"),
    detail: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new OrderNotEligibleForRetry(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
exports.OrderNotEligibleForRetry$outboundSchema = z.instanceof(OrderNotEligibleForRetry)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.literal("OrderNotEligibleForRetry"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderNotEligibleForRetry$;
(function (OrderNotEligibleForRetry$) {
    /** @deprecated use `OrderNotEligibleForRetry$inboundSchema` instead. */
    OrderNotEligibleForRetry$.inboundSchema = exports.OrderNotEligibleForRetry$inboundSchema;
    /** @deprecated use `OrderNotEligibleForRetry$outboundSchema` instead. */
    OrderNotEligibleForRetry$.outboundSchema = exports.OrderNotEligibleForRetry$outboundSchema;
})(OrderNotEligibleForRetry$ || (exports.OrderNotEligibleForRetry$ = OrderNotEligibleForRetry$ = {}));
